{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import numpy as np\n",
    "import io\n",
    "import matplotlib.pyplot as plt\n",
    "pd.options.display.max_rows = 8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_vcf(path):\n",
    "    with open(path, 'r') as f:\n",
    "        lines = [l for l in f if not l.startswith('##')]\n",
    "    return pd.read_csv(io.StringIO(''.join(lines)), dtype={'#CHROM': str, 'POS':int, 'ID':str, 'REF': str, 'ALT': str, 'QUAL': str, 'FILTER': str, 'INFO': str}, sep='\\t').rename(columns={'#CHROM': 'CHROM'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### All CADD Annotations were manipulated for annotation in the other Jupyter Notebook machine entitled CADD Annotations.ipynb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Steps for Annovar to get variants annotated: \n",
    "\n",
    "1. Upload the necessary files into Jupyter Notebook for data parsing\n",
    "2. Create 3 more columns since it needs 8 column format\n",
    "3. Make sure to keep title first row, like below\n",
    "4. Create header and export to vcf\n",
    "5. Put on WinSCP \n",
    "6. Cut header until only data\n",
    "7. Run the two commands used to annotate the files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Clinvar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the file manipulation for clinvar pathogenic coding data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic_coding_for_annovar = read_vcf('clinvar_pathogenic_coding.vcf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10</th>\n",
       "      <th>100154922</th>\n",
       "      <th>226426</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <th>..</th>\n",
       "      <th>...</th>\n",
       "      <th>....</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>226427</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100246864</td>\n",
       "      <td>504028</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253422</td>\n",
       "      <td>419610</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>253212</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16770</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85981796</td>\n",
       "      <td>279771</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16771</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759332</td>\n",
       "      <td>10517</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16772</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>10516</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16773</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>10519</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16774 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr10  100154922  226426   G  A .. ... ....\n",
       "0      chr10  100183802  226427   C  A  .   .    .\n",
       "1      chr10  100246864  504028  AT  A  .   .    .\n",
       "2      chr10  100253422  419610   G  A  .   .    .\n",
       "3      chr10  100256298  253212  TG  T  .   .    .\n",
       "...      ...        ...     ...  .. .. ..  ..  ...\n",
       "16770   chrX   85981796  279771   C  A  .   .    .\n",
       "16771   chrX    9759332   10517   C  T  .   .    .\n",
       "16772   chrX    9759390   10516   A  G  .   .    .\n",
       "16773   chrX    9759390   10519   A  T  .   .    .\n",
       "\n",
       "[16774 rows x 8 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_coding_for_annovar['..'] = '.'\n",
    "pathogenic_coding_for_annovar['...'] = '.'\n",
    "pathogenic_coding_for_annovar['....'] = '.'\n",
    "pathogenic_coding_for_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>22646</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>226427</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100246864</td>\n",
       "      <td>504028</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253422</td>\n",
       "      <td>419610</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16771</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85981796</td>\n",
       "      <td>279771</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16772</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759332</td>\n",
       "      <td>10517</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16773</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>10516</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16774</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>10519</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16775 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               \n",
       "0      chr10  100154922   22646   G  A  .  .  .\n",
       "1      chr10  100183802  226427   C  A  .  .  .\n",
       "2      chr10  100246864  504028  AT  A  .  .  .\n",
       "3      chr10  100253422  419610   G  A  .  .  .\n",
       "...      ...        ...     ...  .. .. .. .. ..\n",
       "16771   chrX   85981796  279771   C  A  .  .  .\n",
       "16772   chrX    9759332   10517   C  T  .  .  .\n",
       "16773   chrX    9759390   10516   A  G  .  .  .\n",
       "16774   chrX    9759390   10519   A  T  .  .  .\n",
       "\n",
       "[16775 rows x 8 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_coding_for_annovar.loc[-1] = ['chr10', '100154922', '22646', 'G', 'A', '.', '.', '.']\n",
    "pathogenic_coding_for_annovar.index = pathogenic_coding_for_annovar.index + 1  # shifting index\n",
    "pathogenic_coding_for_annovar = pathogenic_coding_for_annovar.sort_index()\n",
    "pathogenic_coding_for_annovar.columns = ['', '', '', '', '', '', '', '']\n",
    "pathogenic_coding_for_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "output_VCF = \"clinvar_pathogenic_coding_for_annovar.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "pathogenic_coding_for_annovar.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for clinvar pathogenic coding data from ANNOVAR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>100154922</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ERLIN1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>ERLIN1:NM_001347858:exon9:c.C283T:p.R95X,ERLIN...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>100183802</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ERLIN1</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>ERLIN1:NM_006459:exon2:c.G149T:p.G50V,ERLIN1:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5619750829</td>\n",
       "      <td>0.984</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100246865</td>\n",
       "      <td>100246865</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>CWF19L1:NM_001303406:exon5:c.368delA:p.D123Vfs...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253422</td>\n",
       "      <td>100253422</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>CWF19L1:NM_001303406:exon3:c.C211T:p.R71X,CWF1...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16771</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85981796</td>\n",
       "      <td>85981796</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>CHM:NM_000390:exon3:c.G130T:p.G44X,CHM:NM_0011...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16772</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759332</td>\n",
       "      <td>9759332</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GPR143:NM_000273:exon3:c.G455A:p.S152N</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.676591847234</td>\n",
       "      <td>0.695</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16773</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>9759390</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GPR143:NM_000273:exon3:c.T397C:p.W133R</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.790856144973</td>\n",
       "      <td>0.400</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16774</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>9759390</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GPR143:NM_000273:exon3:c.T397A:p.W133R</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.790215722738</td>\n",
       "      <td>0.386</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16775 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0      chr10  100154922  100154922   G   A       exonic       ERLIN1   \n",
       "1      chr10  100183802  100183802   C   A       exonic       ERLIN1   \n",
       "2      chr10  100246865  100246865   T   -       exonic      CWF19L1   \n",
       "3      chr10  100253422  100253422   G   A       exonic      CWF19L1   \n",
       "...      ...        ...        ...  ..  ..          ...          ...   \n",
       "16771   chrX   85981796   85981796   C   A       exonic          CHM   \n",
       "16772   chrX    9759332    9759332   C   T       exonic       GPR143   \n",
       "16773   chrX    9759390    9759390   A   G       exonic       GPR143   \n",
       "16774   chrX    9759390    9759390   A   T       exonic       GPR143   \n",
       "\n",
       "      GeneDetail.refGene   ExonicFunc.refGene  \\\n",
       "0                      .             stopgain   \n",
       "1                      .    nonsynonymous SNV   \n",
       "2                      .  frameshift deletion   \n",
       "3                      .             stopgain   \n",
       "...                  ...                  ...   \n",
       "16771                  .             stopgain   \n",
       "16772                  .    nonsynonymous SNV   \n",
       "16773                  .    nonsynonymous SNV   \n",
       "16774                  .    nonsynonymous SNV   \n",
       "\n",
       "                                        AAChange.refGene  ... BP4 BP5 BP6 BP7  \\\n",
       "0      ERLIN1:NM_001347858:exon9:c.C283T:p.R95X,ERLIN...  ...   0   0   0   0   \n",
       "1      ERLIN1:NM_006459:exon2:c.G149T:p.G50V,ERLIN1:N...  ...   .   .   .   .   \n",
       "2      CWF19L1:NM_001303406:exon5:c.368delA:p.D123Vfs...  ...   .   .   .   .   \n",
       "3      CWF19L1:NM_001303406:exon3:c.C211T:p.R71X,CWF1...  ...   0   0   0   0   \n",
       "...                                                  ...  ...  ..  ..  ..  ..   \n",
       "16771  CHM:NM_000390:exon3:c.G130T:p.G44X,CHM:NM_0011...  ...   0   0   0   0   \n",
       "16772             GPR143:NM_000273:exon3:c.G455A:p.S152N  ...   0   0   0   0   \n",
       "16773             GPR143:NM_000273:exon3:c.T397C:p.W133R  ...   0   0   0   0   \n",
       "16774             GPR143:NM_000273:exon3:c.T397A:p.W133R  ...   0   0   0   0   \n",
       "\n",
       "      cosmic70            MCAP  REVEL regsnp_fpr regsnp_disease  \\\n",
       "0            .               .      .          .              .   \n",
       "1            .    0.5619750829  0.984          .              .   \n",
       "2            .               .      .          .              .   \n",
       "3            .               .      .          .              .   \n",
       "...        ...             ...    ...        ...            ...   \n",
       "16771        .               .      .          .              .   \n",
       "16772        .  0.676591847234  0.695          .              .   \n",
       "16773        .  0.790856144973  0.400          .              .   \n",
       "16774        .  0.790215722738  0.386          .              .   \n",
       "\n",
       "      regsnp_splicing_site  \n",
       "0                        .  \n",
       "1                        .  \n",
       "2                        .  \n",
       "3                        .  \n",
       "...                    ...  \n",
       "16771                    .  \n",
       "16772                    .  \n",
       "16773                    .  \n",
       "16774                    .  \n",
       "\n",
       "[16775 rows x 149 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_coding_annotated_with_annovar = pd.read_csv('clinvar_pathogenic_coding_from_cadd.csv')\n",
    "clinvar_pathogenic_coding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for clinvar pathogenic coding data from CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8.0</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1.0</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31074</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7.0</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>33</td>\n",
       "      <td>827</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.198603</td>\n",
       "      <td>13.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31075</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>33</td>\n",
       "      <td>827</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.198603</td>\n",
       "      <td>13.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31076</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7.0</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>39</td>\n",
       "      <td>652</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.018377</td>\n",
       "      <td>25.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31077</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>39</td>\n",
       "      <td>652</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.018377</td>\n",
       "      <td>25.80</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31078 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos  Ref Alt Type  Length           AnnoType  \\\n",
       "0          1    1014143    C   T  SNV       0   CodingTranscript   \n",
       "1          1    1014143    C   T  SNV       0         Intergenic   \n",
       "2          1    1014143    C   T  SNV       0  RegulatoryFeature   \n",
       "3          1    1014143    C   T  SNV       0  RegulatoryFeature   \n",
       "...      ...        ...  ...  ..  ...     ...                ...   \n",
       "31074      X  155280059    G   C  SNV       0   CodingTranscript   \n",
       "31075      X  155280059    G   C  SNV       0  RegulatoryFeature   \n",
       "31076      X  155506930  GAT   G  DEL       2   CodingTranscript   \n",
       "31077      X  155506930  GAT   G  DEL       2         Transcript   \n",
       "\n",
       "          Consequence  ConsScore         ConsDetail  ...  Freq10000bp  \\\n",
       "0         STOP_GAINED        8.0        stop_gained  ...           22   \n",
       "1            UPSTREAM        1.0           upstream  ...           22   \n",
       "2          REGULATORY        4.0         regulatory  ...           22   \n",
       "3          REGULATORY        4.0         regulatory  ...           22   \n",
       "...               ...        ...                ...  ...          ...   \n",
       "31074  NON_SYNONYMOUS        7.0           missense  ...           11   \n",
       "31075      REGULATORY        4.0         regulatory  ...           11   \n",
       "31076     FRAME_SHIFT        7.0         frameshift  ...            6   \n",
       "31077        INTRONIC        2.0  intron,non_coding  ...            6   \n",
       "\n",
       "       Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0               87         1622                  Promoter                NaN   \n",
       "1               87         1622                  Promoter                NaN   \n",
       "2               87         1622                  Promoter                NaN   \n",
       "3               87         1622                  Promoter                NaN   \n",
       "...            ...          ...                       ...                ...   \n",
       "31074           33          827  Promoter Flanking Region                NaN   \n",
       "31075           33          827  Promoter Flanking Region                NaN   \n",
       "31076           39          652                       NaN                NaN   \n",
       "31077           39          652                       NaN                NaN   \n",
       "\n",
       "       dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore  PHRED  \n",
       "0                   NaN           75.0          107.0  5.410558  35.00  \n",
       "1                   NaN           75.0          107.0  5.410558  35.00  \n",
       "2                   NaN           75.0          107.0  5.410558  35.00  \n",
       "3                   NaN           75.0          107.0  5.410558  35.00  \n",
       "...                 ...            ...            ...       ...    ...  \n",
       "31074               NaN            NaN            NaN  1.198603  13.85  \n",
       "31075               NaN            NaN            NaN  1.198603  13.85  \n",
       "31076               NaN            1.0            1.0  4.018377  25.80  \n",
       "31077               NaN            1.0            1.0  4.018377  25.80  \n",
       "\n",
       "[31078 rows x 125 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_coding_annotated_with_cadd = pd.read_table('clinvar_coding_pathogenic_cadd_annotations_noheader.tsv')\n",
    "clinvar_pathogenic_coding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the file manipulation for clinvar benign noncoding data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10</th>\n",
       "      <th>100989312</th>\n",
       "      <th>136588</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100990864</td>\n",
       "      <td>136589</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100990866</td>\n",
       "      <td>136590</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100991026</td>\n",
       "      <td>136591</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100991027</td>\n",
       "      <td>136592</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1555</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>213614</td>\n",
       "      <td>CCTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1556</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>95249</td>\n",
       "      <td>CCTCTT</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1557</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>95251</td>\n",
       "      <td>CCTCTTCTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1558</th>\n",
       "      <td>chrX</td>\n",
       "      <td>78118027</td>\n",
       "      <td>558817</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1559 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr10  100989312  136588                                G  A\n",
       "0     chr10  100990864  136589                                C  T\n",
       "1     chr10  100990866  136590                                T  C\n",
       "2     chr10  100991026  136591                                C  A\n",
       "3     chr10  100991027  136592                                G  A\n",
       "...     ...        ...     ...                              ... ..\n",
       "1555   chrX   71132767  213614       CCTCTTCTCTTCTCTTCTCTTCTCTT  C\n",
       "1556   chrX   71132767   95249                           CCTCTT  C\n",
       "1557   chrX   71132767   95251  CCTCTTCTCTTCTCTTCTCTTCTCTTCTCTT  C\n",
       "1558   chrX   78118027  558817                                C  T\n",
       "\n",
       "[1559 rows x 5 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_noncoding_for_annovar = read_vcf('clinvar_noncoding_benign_for_cadd_noheader_with_chr_final_for_annovar.vcf')\n",
    "clinvar_benign_noncoding_for_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989312</td>\n",
       "      <td>136588</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100990864</td>\n",
       "      <td>136589</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100990866</td>\n",
       "      <td>136590</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100991026</td>\n",
       "      <td>136591</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1556</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>213614</td>\n",
       "      <td>CCTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1557</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>95249</td>\n",
       "      <td>CCTCTT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1558</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>95251</td>\n",
       "      <td>CCTCTTCTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1559</th>\n",
       "      <td>chrX</td>\n",
       "      <td>78118027</td>\n",
       "      <td>558817</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1560 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                           \n",
       "0     chr10  100989312  136588                                G  A  .  .  .\n",
       "1     chr10  100990864  136589                                C  T  .  .  .\n",
       "2     chr10  100990866  136590                                T  C  .  .  .\n",
       "3     chr10  100991026  136591                                C  A  .  .  .\n",
       "...     ...        ...     ...                              ... .. .. .. ..\n",
       "1556   chrX   71132767  213614       CCTCTTCTCTTCTCTTCTCTTCTCTT  C  .  .  .\n",
       "1557   chrX   71132767   95249                           CCTCTT  C  .  .  .\n",
       "1558   chrX   71132767   95251  CCTCTTCTCTTCTCTTCTCTTCTCTTCTCTT  C  .  .  .\n",
       "1559   chrX   78118027  558817                                C  T  .  .  .\n",
       "\n",
       "[1560 rows x 8 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_noncoding_for_annovar['..'] = '.'\n",
    "clinvar_benign_noncoding_for_annovar['...'] = '.'\n",
    "clinvar_benign_noncoding_for_annovar['....'] = '.'\n",
    "clinvar_benign_noncoding_for_annovar.loc[-1] = ['chr10', '100989312', '136588', 'G', 'A', '.', '.', '.']\n",
    "clinvar_benign_noncoding_for_annovar.index = clinvar_benign_noncoding_for_annovar.index + 1  # shifting index\n",
    "clinvar_benign_noncoding_for_annovar = clinvar_benign_noncoding_for_annovar.sort_index()\n",
    "clinvar_benign_noncoding_for_annovar.columns = ['', '', '', '', '', '', '', '']\n",
    "clinvar_benign_noncoding_for_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "output_VCF = \"clinvar_benign_noncoding_for_annovar.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "clinvar_benign_noncoding_for_annovar.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for clinvar benign noncoding data from ANNOVAR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>Func.knownGene</th>\n",
       "      <th>Gene.knownGene</th>\n",
       "      <th>GeneDetail.knownGene</th>\n",
       "      <th>ExonicFunc.knownGene</th>\n",
       "      <th>AAChange.knownGene</th>\n",
       "      <th>Func.ensGene</th>\n",
       "      <th>Gene.ensGene</th>\n",
       "      <th>GeneDetail.ensGene</th>\n",
       "      <th>ExonicFunc.ensGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>SIFT_pred</th>\n",
       "      <th>Polyphen2_HDIV_score</th>\n",
       "      <th>Polyphen2_HDIV_pred</th>\n",
       "      <th>Polyphen2_HVAR_score</th>\n",
       "      <th>Polyphen2_HVAR_pred</th>\n",
       "      <th>LRT_score</th>\n",
       "      <th>LRT_pred</th>\n",
       "      <th>MutationTaster_score</th>\n",
       "      <th>MutationTaster_pred</th>\n",
       "      <th>MutationAssessor_score</th>\n",
       "      <th>MutationAssessor_pred</th>\n",
       "      <th>FATHMM_score</th>\n",
       "      <th>FATHMM_pred</th>\n",
       "      <th>RadialSVM_score</th>\n",
       "      <th>RadialSVM_pred</th>\n",
       "      <th>LR_score</th>\n",
       "      <th>LR_pred</th>\n",
       "      <th>VEST3_score</th>\n",
       "      <th>CADD_raw</th>\n",
       "      <th>CADD_phred</th>\n",
       "      <th>GERP++_RS</th>\n",
       "      <th>phyloP46way_placental</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>Interpro_domain</th>\n",
       "      <th>SIFT_score.1</th>\n",
       "      <th>SIFT_converted_rankscore</th>\n",
       "      <th>SIFT_pred.1</th>\n",
       "      <th>Polyphen2_HDIV_score.1</th>\n",
       "      <th>Polyphen2_HDIV_rankscore</th>\n",
       "      <th>Polyphen2_HDIV_pred.1</th>\n",
       "      <th>Polyphen2_HVAR_score.1</th>\n",
       "      <th>Polyphen2_HVAR_rankscore</th>\n",
       "      <th>Polyphen2_HVAR_pred.1</th>\n",
       "      <th>LRT_score.1</th>\n",
       "      <th>LRT_converted_rankscore</th>\n",
       "      <th>LRT_pred.1</th>\n",
       "      <th>MutationTaster_score.1</th>\n",
       "      <th>MutationTaster_converted_rankscore</th>\n",
       "      <th>MutationTaster_pred.1</th>\n",
       "      <th>MutationAssessor_score.1</th>\n",
       "      <th>MutationAssessor_score_rankscore</th>\n",
       "      <th>MutationAssessor_pred.1</th>\n",
       "      <th>FATHMM_score.1</th>\n",
       "      <th>FATHMM_converted_rankscore</th>\n",
       "      <th>FATHMM_pred.1</th>\n",
       "      <th>PROVEAN_score</th>\n",
       "      <th>PROVEAN_converted_rankscore</th>\n",
       "      <th>PROVEAN_pred</th>\n",
       "      <th>VEST3_score.1</th>\n",
       "      <th>VEST3_rankscore</th>\n",
       "      <th>MetaSVM_score</th>\n",
       "      <th>MetaSVM_rankscore</th>\n",
       "      <th>MetaSVM_pred</th>\n",
       "      <th>MetaLR_score</th>\n",
       "      <th>MetaLR_rankscore</th>\n",
       "      <th>MetaLR_pred</th>\n",
       "      <th>M-CAP_score</th>\n",
       "      <th>M-CAP_rankscore</th>\n",
       "      <th>M-CAP_pred</th>\n",
       "      <th>CADD_raw.1</th>\n",
       "      <th>CADD_raw_rankscore</th>\n",
       "      <th>CADD_phred.1</th>\n",
       "      <th>DANN_score</th>\n",
       "      <th>DANN_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_score</th>\n",
       "      <th>fathmm-MKL_coding_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_pred</th>\n",
       "      <th>Eigen_coding_or_noncoding</th>\n",
       "      <th>Eigen-raw</th>\n",
       "      <th>Eigen-PC-raw</th>\n",
       "      <th>GenoCanyon_score</th>\n",
       "      <th>GenoCanyon_score_rankscore</th>\n",
       "      <th>integrated_fitCons_score</th>\n",
       "      <th>integrated_fitCons_score_rankscore</th>\n",
       "      <th>integrated_confidence_value</th>\n",
       "      <th>GERP++_RS.1</th>\n",
       "      <th>GERP++_RS_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate.1</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP20way_mammalian</th>\n",
       "      <th>phyloP20way_mammalian_rankscore</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons20way_mammalian</th>\n",
       "      <th>phastCons20way_mammalian_rankscore</th>\n",
       "      <th>SiPhy_29way_logOdds.1</th>\n",
       "      <th>SiPhy_29way_logOdds_rankscore</th>\n",
       "      <th>Interpro_domain.1</th>\n",
       "      <th>GTEx_V6_gene</th>\n",
       "      <th>GTEx_V6_tissue</th>\n",
       "      <th>dbscSNV_ADA_SCORE</th>\n",
       "      <th>dbscSNV_RF_SCORE</th>\n",
       "      <th>InterVar_automated</th>\n",
       "      <th>PVS1</th>\n",
       "      <th>PS1</th>\n",
       "      <th>PS2</th>\n",
       "      <th>PS3</th>\n",
       "      <th>PS4</th>\n",
       "      <th>PM1</th>\n",
       "      <th>PM2</th>\n",
       "      <th>PM3</th>\n",
       "      <th>PM4</th>\n",
       "      <th>PM5</th>\n",
       "      <th>PM6</th>\n",
       "      <th>PP1</th>\n",
       "      <th>PP2</th>\n",
       "      <th>PP3</th>\n",
       "      <th>PP4</th>\n",
       "      <th>PP5</th>\n",
       "      <th>BA1</th>\n",
       "      <th>BS1</th>\n",
       "      <th>BS2</th>\n",
       "      <th>BS3</th>\n",
       "      <th>BS4</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>BP3</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989312</td>\n",
       "      <td>100989312</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.G1102A:p.V368I,TWNK:...</td>\n",
       "      <td>exonic</td>\n",
       "      <td>C10orf2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>C10orf2:uc001ksf.3:exon1:c.G1102A:p.V368I,C10o...</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:ENST00000311916.6:exon1:c.G1102A:p.V368I,...</td>\n",
       "      <td>0.45</td>\n",
       "      <td>T</td>\n",
       "      <td>0.528</td>\n",
       "      <td>P</td>\n",
       "      <td>0.046</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>D</td>\n",
       "      <td>0.895</td>\n",
       "      <td>D</td>\n",
       "      <td>0.945</td>\n",
       "      <td>L</td>\n",
       "      <td>-3.32</td>\n",
       "      <td>D</td>\n",
       "      <td>-0.752</td>\n",
       "      <td>T</td>\n",
       "      <td>0.016</td>\n",
       "      <td>T</td>\n",
       "      <td>0.057</td>\n",
       "      <td>1.265</td>\n",
       "      <td>10.13</td>\n",
       "      <td>5.13</td>\n",
       "      <td>2.869</td>\n",
       "      <td>2.482</td>\n",
       "      <td>7.064</td>\n",
       "      <td>.</td>\n",
       "      <td>0.583</td>\n",
       "      <td>0.067</td>\n",
       "      <td>T</td>\n",
       "      <td>0.341</td>\n",
       "      <td>0.358</td>\n",
       "      <td>B</td>\n",
       "      <td>0.029</td>\n",
       "      <td>0.233</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.559</td>\n",
       "      <td>D</td>\n",
       "      <td>0.895</td>\n",
       "      <td>0.360</td>\n",
       "      <td>D</td>\n",
       "      <td>0.915</td>\n",
       "      <td>0.234</td>\n",
       "      <td>L</td>\n",
       "      <td>-3.32</td>\n",
       "      <td>0.951</td>\n",
       "      <td>D</td>\n",
       "      <td>-0.26</td>\n",
       "      <td>0.112</td>\n",
       "      <td>N</td>\n",
       "      <td>0.057</td>\n",
       "      <td>0.040</td>\n",
       "      <td>-0.752</td>\n",
       "      <td>0.579</td>\n",
       "      <td>T</td>\n",
       "      <td>0.016</td>\n",
       "      <td>0.065</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.707</td>\n",
       "      <td>0.266</td>\n",
       "      <td>14.45</td>\n",
       "      <td>0.925</td>\n",
       "      <td>0.213</td>\n",
       "      <td>0.856</td>\n",
       "      <td>0.444</td>\n",
       "      <td>D</td>\n",
       "      <td>c</td>\n",
       "      <td>0.028</td>\n",
       "      <td>0.207</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.747</td>\n",
       "      <td>0.672</td>\n",
       "      <td>0.522</td>\n",
       "      <td>0</td>\n",
       "      <td>5.13</td>\n",
       "      <td>0.696</td>\n",
       "      <td>2.572</td>\n",
       "      <td>0.454</td>\n",
       "      <td>1.047</td>\n",
       "      <td>0.674</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.411</td>\n",
       "      <td>0.991</td>\n",
       "      <td>0.552</td>\n",
       "      <td>7.064</td>\n",
       "      <td>0.242</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.353</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100990864</td>\n",
       "      <td>100990864</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>C10orf2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.624215246637</td>\n",
       "      <td>B</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100990866</td>\n",
       "      <td>100990866</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>C10orf2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>0.052</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0717488789238</td>\n",
       "      <td>PD</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100991026</td>\n",
       "      <td>100991026</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>NM_001163812:c.*1C&gt;A;NM_001163814:c.*1C&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>C10orf2</td>\n",
       "      <td>uc001ksg.3:c.*1C&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>ENST00000370228.1:c.*1C&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1556</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132768</td>\n",
       "      <td>71132792</td>\n",
       "      <td>CTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1557</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132768</td>\n",
       "      <td>71132772</td>\n",
       "      <td>CTCTT</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1558</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132768</td>\n",
       "      <td>71132797</td>\n",
       "      <td>CTCTTCTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1559</th>\n",
       "      <td>chrX</td>\n",
       "      <td>78118027</td>\n",
       "      <td>78118027</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PGK1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PGK1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PGK1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.61301369863</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1560 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr      Start        End                             Ref Alt  \\\n",
       "0     chr10  100989312  100989312                               G   A   \n",
       "1     chr10  100990864  100990864                               C   T   \n",
       "2     chr10  100990866  100990866                               T   C   \n",
       "3     chr10  100991026  100991026                               C   A   \n",
       "...     ...        ...        ...                             ...  ..   \n",
       "1556   chrX   71132768   71132792       CTCTTCTCTTCTCTTCTCTTCTCTT   -   \n",
       "1557   chrX   71132768   71132772                           CTCTT   -   \n",
       "1558   chrX   71132768   71132797  CTCTTCTCTTCTCTTCTCTTCTCTTCTCTT   -   \n",
       "1559   chrX   78118027   78118027                               C   T   \n",
       "\n",
       "     Func.refGene Gene.refGene                         GeneDetail.refGene  \\\n",
       "0          exonic         TWNK                                          .   \n",
       "1        intronic         TWNK                                          .   \n",
       "2        intronic         TWNK                                          .   \n",
       "3            UTR3         TWNK  NM_001163812:c.*1C>A;NM_001163814:c.*1C>A   \n",
       "...           ...          ...                                        ...   \n",
       "1556     intronic        MED12                                          .   \n",
       "1557     intronic        MED12                                          .   \n",
       "1558     intronic        MED12                                          .   \n",
       "1559     intronic         PGK1                                          .   \n",
       "\n",
       "     ExonicFunc.refGene                                   AAChange.refGene  \\\n",
       "0     nonsynonymous SNV  TWNK:NM_001163812:exon1:c.G1102A:p.V368I,TWNK:...   \n",
       "1                     .                                                  .   \n",
       "2                     .                                                  .   \n",
       "3                     .                                                  .   \n",
       "...                 ...                                                ...   \n",
       "1556                  .                                                  .   \n",
       "1557                  .                                                  .   \n",
       "1558                  .                                                  .   \n",
       "1559                  .                                                  .   \n",
       "\n",
       "     Func.knownGene Gene.knownGene GeneDetail.knownGene ExonicFunc.knownGene  \\\n",
       "0            exonic        C10orf2                    .    nonsynonymous SNV   \n",
       "1          intronic        C10orf2                    .                    .   \n",
       "2          intronic        C10orf2                    .                    .   \n",
       "3              UTR3        C10orf2   uc001ksg.3:c.*1C>A                    .   \n",
       "...             ...            ...                  ...                  ...   \n",
       "1556       intronic          MED12                    .                    .   \n",
       "1557       intronic          MED12                    .                    .   \n",
       "1558       intronic          MED12                    .                    .   \n",
       "1559       intronic           PGK1                    .                    .   \n",
       "\n",
       "                                     AAChange.knownGene Func.ensGene  \\\n",
       "0     C10orf2:uc001ksf.3:exon1:c.G1102A:p.V368I,C10o...       exonic   \n",
       "1                                                     .     intronic   \n",
       "2                                                     .     intronic   \n",
       "3                                                     .         UTR3   \n",
       "...                                                 ...          ...   \n",
       "1556                                                  .     intronic   \n",
       "1557                                                  .     intronic   \n",
       "1558                                                  .     intronic   \n",
       "1559                                                  .     intronic   \n",
       "\n",
       "     Gene.ensGene         GeneDetail.ensGene ExonicFunc.ensGene  \\\n",
       "0            TWNK                          .  nonsynonymous SNV   \n",
       "1            TWNK                          .                  .   \n",
       "2            TWNK                          .                  .   \n",
       "3            TWNK  ENST00000370228.1:c.*1C>A                  .   \n",
       "...           ...                        ...                ...   \n",
       "1556        MED12                          .                  .   \n",
       "1557        MED12                          .                  .   \n",
       "1558        MED12                          .                  .   \n",
       "1559         PGK1                          .                  .   \n",
       "\n",
       "                                       AAChange.ensGene SIFT_score SIFT_pred  \\\n",
       "0     TWNK:ENST00000311916.6:exon1:c.G1102A:p.V368I,...       0.45         T   \n",
       "1                                                     .          .         .   \n",
       "2                                                     .          .         .   \n",
       "3                                                     .          .         .   \n",
       "...                                                 ...        ...       ...   \n",
       "1556                                                  .          .         .   \n",
       "1557                                                  .          .         .   \n",
       "1558                                                  .          .         .   \n",
       "1559                                                  .          .         .   \n",
       "\n",
       "     Polyphen2_HDIV_score Polyphen2_HDIV_pred Polyphen2_HVAR_score  \\\n",
       "0                   0.528                   P                0.046   \n",
       "1                       .                   .                    .   \n",
       "2                       .                   .                    .   \n",
       "3                       .                   .                    .   \n",
       "...                   ...                 ...                  ...   \n",
       "1556                    .                   .                    .   \n",
       "1557                    .                   .                    .   \n",
       "1558                    .                   .                    .   \n",
       "1559                    .                   .                    .   \n",
       "\n",
       "     Polyphen2_HVAR_pred LRT_score LRT_pred MutationTaster_score  \\\n",
       "0                      B     0.000        D                0.895   \n",
       "1                      .         .        .                    .   \n",
       "2                      .         .        .                    .   \n",
       "3                      .         .        .                    .   \n",
       "...                  ...       ...      ...                  ...   \n",
       "1556                   .         .        .                    .   \n",
       "1557                   .         .        .                    .   \n",
       "1558                   .         .        .                    .   \n",
       "1559                   .         .        .                    .   \n",
       "\n",
       "     MutationTaster_pred MutationAssessor_score MutationAssessor_pred  \\\n",
       "0                      D                  0.945                     L   \n",
       "1                      .                      .                     .   \n",
       "2                      .                      .                     .   \n",
       "3                      .                      .                     .   \n",
       "...                  ...                    ...                   ...   \n",
       "1556                   .                      .                     .   \n",
       "1557                   .                      .                     .   \n",
       "1558                   .                      .                     .   \n",
       "1559                   .                      .                     .   \n",
       "\n",
       "     FATHMM_score FATHMM_pred RadialSVM_score RadialSVM_pred LR_score LR_pred  \\\n",
       "0           -3.32           D          -0.752              T    0.016       T   \n",
       "1               .           .               .              .        .       .   \n",
       "2               .           .               .              .        .       .   \n",
       "3               .           .               .              .        .       .   \n",
       "...           ...         ...             ...            ...      ...     ...   \n",
       "1556            .           .               .              .        .       .   \n",
       "1557            .           .               .              .        .       .   \n",
       "1558            .           .               .              .        .       .   \n",
       "1559            .           .               .              .        .       .   \n",
       "\n",
       "     VEST3_score CADD_raw CADD_phred GERP++_RS phyloP46way_placental  \\\n",
       "0          0.057    1.265      10.13      5.13                 2.869   \n",
       "1              .        .          .         .                     .   \n",
       "2              .        .          .         .                     .   \n",
       "3              .        .          .         .                     .   \n",
       "...          ...      ...        ...       ...                   ...   \n",
       "1556           .        .          .         .                     .   \n",
       "1557           .        .          .         .                     .   \n",
       "1558           .        .          .         .                     .   \n",
       "1559           .        .          .         .                     .   \n",
       "\n",
       "     phyloP100way_vertebrate SiPhy_29way_logOdds Interpro_domain SIFT_score.1  \\\n",
       "0                      2.482               7.064               .        0.583   \n",
       "1                          .                   .               .            .   \n",
       "2                          .                   .               .            .   \n",
       "3                          .                   .               .            .   \n",
       "...                      ...                 ...             ...          ...   \n",
       "1556                       .                   .               .            .   \n",
       "1557                       .                   .               .            .   \n",
       "1558                       .                   .               .            .   \n",
       "1559                       .                   .               .            .   \n",
       "\n",
       "     SIFT_converted_rankscore SIFT_pred.1 Polyphen2_HDIV_score.1  \\\n",
       "0                       0.067           T                  0.341   \n",
       "1                           .           .                      .   \n",
       "2                           .           .                      .   \n",
       "3                           .           .                      .   \n",
       "...                       ...         ...                    ...   \n",
       "1556                        .           .                      .   \n",
       "1557                        .           .                      .   \n",
       "1558                        .           .                      .   \n",
       "1559                        .           .                      .   \n",
       "\n",
       "     Polyphen2_HDIV_rankscore Polyphen2_HDIV_pred.1 Polyphen2_HVAR_score.1  \\\n",
       "0                       0.358                     B                  0.029   \n",
       "1                           .                     .                      .   \n",
       "2                           .                     .                      .   \n",
       "3                           .                     .                      .   \n",
       "...                       ...                   ...                    ...   \n",
       "1556                        .                     .                      .   \n",
       "1557                        .                     .                      .   \n",
       "1558                        .                     .                      .   \n",
       "1559                        .                     .                      .   \n",
       "\n",
       "     Polyphen2_HVAR_rankscore Polyphen2_HVAR_pred.1 LRT_score.1  \\\n",
       "0                       0.233                     B       0.000   \n",
       "1                           .                     .           .   \n",
       "2                           .                     .           .   \n",
       "3                           .                     .           .   \n",
       "...                       ...                   ...         ...   \n",
       "1556                        .                     .           .   \n",
       "1557                        .                     .           .   \n",
       "1558                        .                     .           .   \n",
       "1559                        .                     .           .   \n",
       "\n",
       "     LRT_converted_rankscore LRT_pred.1 MutationTaster_score.1  \\\n",
       "0                      0.559          D                  0.895   \n",
       "1                          .          .                      .   \n",
       "2                          .          .                      .   \n",
       "3                          .          .                      .   \n",
       "...                      ...        ...                    ...   \n",
       "1556                       .          .                      .   \n",
       "1557                       .          .                      .   \n",
       "1558                       .          .                      .   \n",
       "1559                       .          .                      .   \n",
       "\n",
       "     MutationTaster_converted_rankscore MutationTaster_pred.1  \\\n",
       "0                                 0.360                     D   \n",
       "1                                     .                     .   \n",
       "2                                     .                     .   \n",
       "3                                     .                     .   \n",
       "...                                 ...                   ...   \n",
       "1556                                  .                     .   \n",
       "1557                                  .                     .   \n",
       "1558                                  .                     .   \n",
       "1559                                  .                     .   \n",
       "\n",
       "     MutationAssessor_score.1 MutationAssessor_score_rankscore  \\\n",
       "0                       0.915                            0.234   \n",
       "1                           .                                .   \n",
       "2                           .                                .   \n",
       "3                           .                                .   \n",
       "...                       ...                              ...   \n",
       "1556                        .                                .   \n",
       "1557                        .                                .   \n",
       "1558                        .                                .   \n",
       "1559                        .                                .   \n",
       "\n",
       "     MutationAssessor_pred.1 FATHMM_score.1 FATHMM_converted_rankscore  \\\n",
       "0                          L          -3.32                      0.951   \n",
       "1                          .              .                          .   \n",
       "2                          .              .                          .   \n",
       "3                          .              .                          .   \n",
       "...                      ...            ...                        ...   \n",
       "1556                       .              .                          .   \n",
       "1557                       .              .                          .   \n",
       "1558                       .              .                          .   \n",
       "1559                       .              .                          .   \n",
       "\n",
       "     FATHMM_pred.1 PROVEAN_score PROVEAN_converted_rankscore PROVEAN_pred  \\\n",
       "0                D         -0.26                       0.112            N   \n",
       "1                .             .                           .            .   \n",
       "2                .             .                           .            .   \n",
       "3                .             .                           .            .   \n",
       "...            ...           ...                         ...          ...   \n",
       "1556             .             .                           .            .   \n",
       "1557             .             .                           .            .   \n",
       "1558             .             .                           .            .   \n",
       "1559             .             .                           .            .   \n",
       "\n",
       "     VEST3_score.1 VEST3_rankscore MetaSVM_score MetaSVM_rankscore  \\\n",
       "0            0.057           0.040        -0.752             0.579   \n",
       "1                .               .             .                 .   \n",
       "2                .               .             .                 .   \n",
       "3                .               .             .                 .   \n",
       "...            ...             ...           ...               ...   \n",
       "1556             .               .             .                 .   \n",
       "1557             .               .             .                 .   \n",
       "1558             .               .             .                 .   \n",
       "1559             .               .             .                 .   \n",
       "\n",
       "     MetaSVM_pred MetaLR_score MetaLR_rankscore MetaLR_pred M-CAP_score  \\\n",
       "0               T        0.016            0.065           T           .   \n",
       "1               .            .                .           .           .   \n",
       "2               .            .                .           .           .   \n",
       "3               .            .                .           .           .   \n",
       "...           ...          ...              ...         ...         ...   \n",
       "1556            .            .                .           .           .   \n",
       "1557            .            .                .           .           .   \n",
       "1558            .            .                .           .           .   \n",
       "1559            .            .                .           .           .   \n",
       "\n",
       "     M-CAP_rankscore M-CAP_pred CADD_raw.1 CADD_raw_rankscore CADD_phred.1  \\\n",
       "0                  .          .      1.707              0.266        14.45   \n",
       "1                  .          .          .                  .            .   \n",
       "2                  .          .          .                  .            .   \n",
       "3                  .          .          .                  .            .   \n",
       "...              ...        ...        ...                ...          ...   \n",
       "1556               .          .          .                  .            .   \n",
       "1557               .          .          .                  .            .   \n",
       "1558               .          .          .                  .            .   \n",
       "1559               .          .          .                  .            .   \n",
       "\n",
       "     DANN_score DANN_rankscore fathmm-MKL_coding_score  \\\n",
       "0         0.925          0.213                   0.856   \n",
       "1             .              .                       .   \n",
       "2             .              .                       .   \n",
       "3             .              .                       .   \n",
       "...         ...            ...                     ...   \n",
       "1556          .              .                       .   \n",
       "1557          .              .                       .   \n",
       "1558          .              .                       .   \n",
       "1559          .              .                       .   \n",
       "\n",
       "     fathmm-MKL_coding_rankscore fathmm-MKL_coding_pred  \\\n",
       "0                          0.444                      D   \n",
       "1                              .                      .   \n",
       "2                              .                      .   \n",
       "3                              .                      .   \n",
       "...                          ...                    ...   \n",
       "1556                           .                      .   \n",
       "1557                           .                      .   \n",
       "1558                           .                      .   \n",
       "1559                           .                      .   \n",
       "\n",
       "     Eigen_coding_or_noncoding Eigen-raw Eigen-PC-raw GenoCanyon_score  \\\n",
       "0                            c     0.028        0.207            1.000   \n",
       "1                            .         .            .                .   \n",
       "2                            .         .            .                .   \n",
       "3                            .         .            .                .   \n",
       "...                        ...       ...          ...              ...   \n",
       "1556                         .         .            .                .   \n",
       "1557                         .         .            .                .   \n",
       "1558                         .         .            .                .   \n",
       "1559                         .         .            .                .   \n",
       "\n",
       "     GenoCanyon_score_rankscore integrated_fitCons_score  \\\n",
       "0                         0.747                    0.672   \n",
       "1                             .                        .   \n",
       "2                             .                        .   \n",
       "3                             .                        .   \n",
       "...                         ...                      ...   \n",
       "1556                          .                        .   \n",
       "1557                          .                        .   \n",
       "1558                          .                        .   \n",
       "1559                          .                        .   \n",
       "\n",
       "     integrated_fitCons_score_rankscore integrated_confidence_value  \\\n",
       "0                                 0.522                           0   \n",
       "1                                     .                           .   \n",
       "2                                     .                           .   \n",
       "3                                     .                           .   \n",
       "...                                 ...                         ...   \n",
       "1556                                  .                           .   \n",
       "1557                                  .                           .   \n",
       "1558                                  .                           .   \n",
       "1559                                  .                           .   \n",
       "\n",
       "     GERP++_RS.1 GERP++_RS_rankscore phyloP100way_vertebrate.1  \\\n",
       "0           5.13               0.696                     2.572   \n",
       "1              .                   .                         .   \n",
       "2              .                   .                         .   \n",
       "3              .                   .                         .   \n",
       "...          ...                 ...                       ...   \n",
       "1556           .                   .                         .   \n",
       "1557           .                   .                         .   \n",
       "1558           .                   .                         .   \n",
       "1559           .                   .                         .   \n",
       "\n",
       "     phyloP100way_vertebrate_rankscore phyloP20way_mammalian  \\\n",
       "0                                0.454                 1.047   \n",
       "1                                    .                     .   \n",
       "2                                    .                     .   \n",
       "3                                    .                     .   \n",
       "...                                ...                   ...   \n",
       "1556                                 .                     .   \n",
       "1557                                 .                     .   \n",
       "1558                                 .                     .   \n",
       "1559                                 .                     .   \n",
       "\n",
       "     phyloP20way_mammalian_rankscore phastCons100way_vertebrate  \\\n",
       "0                              0.674                      0.998   \n",
       "1                                  .                          .   \n",
       "2                                  .                          .   \n",
       "3                                  .                          .   \n",
       "...                              ...                        ...   \n",
       "1556                               .                          .   \n",
       "1557                               .                          .   \n",
       "1558                               .                          .   \n",
       "1559                               .                          .   \n",
       "\n",
       "     phastCons100way_vertebrate_rankscore phastCons20way_mammalian  \\\n",
       "0                                   0.411                    0.991   \n",
       "1                                       .                        .   \n",
       "2                                       .                        .   \n",
       "3                                       .                        .   \n",
       "...                                   ...                      ...   \n",
       "1556                                    .                        .   \n",
       "1557                                    .                        .   \n",
       "1558                                    .                        .   \n",
       "1559                                    .                        .   \n",
       "\n",
       "     phastCons20way_mammalian_rankscore SiPhy_29way_logOdds.1  \\\n",
       "0                                 0.552                 7.064   \n",
       "1                                     .                     .   \n",
       "2                                     .                     .   \n",
       "3                                     .                     .   \n",
       "...                                 ...                   ...   \n",
       "1556                                  .                     .   \n",
       "1557                                  .                     .   \n",
       "1558                                  .                     .   \n",
       "1559                                  .                     .   \n",
       "\n",
       "     SiPhy_29way_logOdds_rankscore Interpro_domain.1 GTEx_V6_gene  \\\n",
       "0                            0.242                 .            .   \n",
       "1                                .                 .            .   \n",
       "2                                .                 .            .   \n",
       "3                                .                 .            .   \n",
       "...                            ...               ...          ...   \n",
       "1556                             .                 .            .   \n",
       "1557                             .                 .            .   \n",
       "1558                             .                 .            .   \n",
       "1559                             .                 .            .   \n",
       "\n",
       "     GTEx_V6_tissue dbscSNV_ADA_SCORE dbscSNV_RF_SCORE InterVar_automated  \\\n",
       "0                 .                 .                .             Benign   \n",
       "1                 .                 0                0                  .   \n",
       "2                 .                 0            0.052                  .   \n",
       "3                 .                 .                .                  .   \n",
       "...             ...               ...              ...                ...   \n",
       "1556              .                 .                .                  .   \n",
       "1557              .                 .                .                  .   \n",
       "1558              .                 .                .                  .   \n",
       "1559              .                 .                .                  .   \n",
       "\n",
       "     PVS1 PS1 PS2 PS3 PS4 PM1 PM2 PM3 PM4 PM5 PM6 PP1 PP2 PP3 PP4 PP5 BA1 BS1  \\\n",
       "0       0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   1   \n",
       "1       .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "2       .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "3       .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "...   ...  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..   \n",
       "1556    .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "1557    .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "1558    .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "1559    .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "\n",
       "     BS2 BS3 BS4 BP1 BP2 BP3 BP4 BP5 BP6 BP7 cosmic70 MCAP  REVEL  \\\n",
       "0      0   0   0   0   0   0   0   0   1   0        .    .  0.353   \n",
       "1      .   .   .   .   .   .   .   .   .   .        .    .      .   \n",
       "2      .   .   .   .   .   .   .   .   .   .        .    .      .   \n",
       "3      .   .   .   .   .   .   .   .   .   .        .    .      .   \n",
       "...   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..      ...  ...    ...   \n",
       "1556   .   .   .   .   .   .   .   .   .   .        .    .      .   \n",
       "1557   .   .   .   .   .   .   .   .   .   .        .    .      .   \n",
       "1558   .   .   .   .   .   .   .   .   .   .        .    .      .   \n",
       "1559   .   .   .   .   .   .   .   .   .   .        .    .      .   \n",
       "\n",
       "           regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0                   .              .                    .  \n",
       "1      0.624215246637              B                   on  \n",
       "2     0.0717488789238             PD                   on  \n",
       "3                   .              .                    .  \n",
       "...               ...            ...                  ...  \n",
       "1556                .              .                    .  \n",
       "1557                .              .                    .  \n",
       "1558                .              .                    .  \n",
       "1559    0.61301369863              B                  off  \n",
       "\n",
       "[1560 rows x 149 columns]"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_noncoding_annotated_with_annovar = pd.read_csv('clinvar_benign_noncoding.hg38_multianno.csv')\n",
    "clinvar_benign_noncoding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for clinvar benign noncoding data from CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEName</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>motifEScoreChng</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>FeatureID</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>Intron</th>\n",
       "      <th>Exon</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>relcDNApos</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>relCDSpos</th>\n",
       "      <th>protPos</th>\n",
       "      <th>relProtPos</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Dst2Splice</th>\n",
       "      <th>Dst2SplType</th>\n",
       "      <th>minDistTSS</th>\n",
       "      <th>minDistTSE</th>\n",
       "      <th>SIFTcat</th>\n",
       "      <th>SIFTval</th>\n",
       "      <th>PolyPhenCat</th>\n",
       "      <th>PolyPhenVal</th>\n",
       "      <th>priPhCons</th>\n",
       "      <th>mamPhCons</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>priPhyloP</th>\n",
       "      <th>mamPhyloP</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>bStatistic</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>mirSVR-Score</th>\n",
       "      <th>mirSVR-E</th>\n",
       "      <th>mirSVR-Aln</th>\n",
       "      <th>cHmm_E1</th>\n",
       "      <th>cHmm_E2</th>\n",
       "      <th>cHmm_E3</th>\n",
       "      <th>cHmm_E4</th>\n",
       "      <th>cHmm_E5</th>\n",
       "      <th>cHmm_E6</th>\n",
       "      <th>cHmm_E7</th>\n",
       "      <th>cHmm_E8</th>\n",
       "      <th>cHmm_E9</th>\n",
       "      <th>cHmm_E10</th>\n",
       "      <th>cHmm_E11</th>\n",
       "      <th>cHmm_E12</th>\n",
       "      <th>cHmm_E13</th>\n",
       "      <th>cHmm_E14</th>\n",
       "      <th>cHmm_E15</th>\n",
       "      <th>cHmm_E16</th>\n",
       "      <th>cHmm_E17</th>\n",
       "      <th>cHmm_E18</th>\n",
       "      <th>cHmm_E19</th>\n",
       "      <th>cHmm_E20</th>\n",
       "      <th>cHmm_E21</th>\n",
       "      <th>cHmm_E22</th>\n",
       "      <th>cHmm_E23</th>\n",
       "      <th>cHmm_E24</th>\n",
       "      <th>cHmm_E25</th>\n",
       "      <th>GerpRS</th>\n",
       "      <th>GerpRSpval</th>\n",
       "      <th>GerpN</th>\n",
       "      <th>GerpS</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>motifDist</th>\n",
       "      <th>EncodeH3K4me1-sum</th>\n",
       "      <th>EncodeH3K4me1-max</th>\n",
       "      <th>EncodeH3K4me2-sum</th>\n",
       "      <th>EncodeH3K4me2-max</th>\n",
       "      <th>EncodeH3K4me3-sum</th>\n",
       "      <th>EncodeH3K4me3-max</th>\n",
       "      <th>EncodeH3K9ac-sum</th>\n",
       "      <th>EncodeH3K9ac-max</th>\n",
       "      <th>EncodeH3K9me3-sum</th>\n",
       "      <th>EncodeH3K9me3-max</th>\n",
       "      <th>EncodeH3K27ac-sum</th>\n",
       "      <th>EncodeH3K27ac-max</th>\n",
       "      <th>EncodeH3K27me3-sum</th>\n",
       "      <th>EncodeH3K27me3-max</th>\n",
       "      <th>EncodeH3K36me3-sum</th>\n",
       "      <th>EncodeH3K36me3-max</th>\n",
       "      <th>EncodeH3K79me2-sum</th>\n",
       "      <th>EncodeH3K79me2-max</th>\n",
       "      <th>EncodeH4K20me1-sum</th>\n",
       "      <th>EncodeH4K20me1-max</th>\n",
       "      <th>EncodeH2AFZ-sum</th>\n",
       "      <th>EncodeH2AFZ-max</th>\n",
       "      <th>EncodeDNase-sum</th>\n",
       "      <th>EncodeDNase-max</th>\n",
       "      <th>EncodetotalRNA-sum</th>\n",
       "      <th>EncodetotalRNA-max</th>\n",
       "      <th>Grantham</th>\n",
       "      <th>Dist2Mutation</th>\n",
       "      <th>Freq100bp</th>\n",
       "      <th>Rare100bp</th>\n",
       "      <th>Sngl100bp</th>\n",
       "      <th>Freq1000bp</th>\n",
       "      <th>Rare1000bp</th>\n",
       "      <th>Sngl1000bp</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1041950</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>CCDS30551.1</td>\n",
       "      <td>6/35</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-6.0</td>\n",
       "      <td>ACCEPTOR</td>\n",
       "      <td>7083</td>\n",
       "      <td>553</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.398</td>\n",
       "      <td>-0.498</td>\n",
       "      <td>-0.484</td>\n",
       "      <td>940.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.03</td>\n",
       "      <td>4.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.16</td>\n",
       "      <td>1.40</td>\n",
       "      <td>6.10</td>\n",
       "      <td>2.06</td>\n",
       "      <td>6.40</td>\n",
       "      <td>0.99</td>\n",
       "      <td>2.91</td>\n",
       "      <td>0.78</td>\n",
       "      <td>5.47</td>\n",
       "      <td>1.33</td>\n",
       "      <td>2.72</td>\n",
       "      <td>1.32</td>\n",
       "      <td>3.20</td>\n",
       "      <td>1.44</td>\n",
       "      <td>13.29</td>\n",
       "      <td>3.23</td>\n",
       "      <td>15.60</td>\n",
       "      <td>3.09</td>\n",
       "      <td>21.17</td>\n",
       "      <td>3.99</td>\n",
       "      <td>5.78</td>\n",
       "      <td>4.03</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>200</td>\n",
       "      <td>23</td>\n",
       "      <td>100</td>\n",
       "      <td>1931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>0.002</td>\n",
       "      <td>10.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.092155</td>\n",
       "      <td>2.700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1042190</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>0.669</td>\n",
       "      <td>0.093</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>CCDS30551.1</td>\n",
       "      <td>7/35</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7323</td>\n",
       "      <td>793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.011</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.346</td>\n",
       "      <td>-1.186</td>\n",
       "      <td>-1.417</td>\n",
       "      <td>940.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.30</td>\n",
       "      <td>-20.60</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.13</td>\n",
       "      <td>8.05</td>\n",
       "      <td>1.59</td>\n",
       "      <td>9.22</td>\n",
       "      <td>2.78</td>\n",
       "      <td>5.16</td>\n",
       "      <td>1.25</td>\n",
       "      <td>4.23</td>\n",
       "      <td>2.12</td>\n",
       "      <td>8.84</td>\n",
       "      <td>2.66</td>\n",
       "      <td>6.76</td>\n",
       "      <td>1.20</td>\n",
       "      <td>9.21</td>\n",
       "      <td>2.85</td>\n",
       "      <td>26.09</td>\n",
       "      <td>9.04</td>\n",
       "      <td>20.55</td>\n",
       "      <td>4.96</td>\n",
       "      <td>28.70</td>\n",
       "      <td>5.25</td>\n",
       "      <td>4.82</td>\n",
       "      <td>1.26</td>\n",
       "      <td>0.80</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.54</td>\n",
       "      <td>0.19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>213</td>\n",
       "      <td>25</td>\n",
       "      <td>109</td>\n",
       "      <td>1955</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>-0.401276</td>\n",
       "      <td>0.118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>0.678</td>\n",
       "      <td>0.079</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>CCDS30551.1</td>\n",
       "      <td>7/35</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-14.0</td>\n",
       "      <td>ACCEPTOR</td>\n",
       "      <td>23101</td>\n",
       "      <td>12892</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.418</td>\n",
       "      <td>-0.047</td>\n",
       "      <td>-0.165</td>\n",
       "      <td>940.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.30</td>\n",
       "      <td>1.42</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.96</td>\n",
       "      <td>2.89</td>\n",
       "      <td>0.74</td>\n",
       "      <td>3.29</td>\n",
       "      <td>1.03</td>\n",
       "      <td>6.55</td>\n",
       "      <td>2.05</td>\n",
       "      <td>6.62</td>\n",
       "      <td>1.17</td>\n",
       "      <td>6.56</td>\n",
       "      <td>2.21</td>\n",
       "      <td>5.19</td>\n",
       "      <td>1.22</td>\n",
       "      <td>5.39</td>\n",
       "      <td>2.43</td>\n",
       "      <td>19.45</td>\n",
       "      <td>4.05</td>\n",
       "      <td>12.22</td>\n",
       "      <td>3.37</td>\n",
       "      <td>18.53</td>\n",
       "      <td>4.93</td>\n",
       "      <td>5.36</td>\n",
       "      <td>2.41</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>216</td>\n",
       "      <td>24</td>\n",
       "      <td>117</td>\n",
       "      <td>2004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-0.012669</td>\n",
       "      <td>1.640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1045707</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>0.656</td>\n",
       "      <td>0.093</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000188157</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>CCDS30551.1</td>\n",
       "      <td>14/35</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>308</td>\n",
       "      <td>643</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.076</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.475</td>\n",
       "      <td>-0.843</td>\n",
       "      <td>-0.993</td>\n",
       "      <td>940.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.94</td>\n",
       "      <td>-13.90</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.50</td>\n",
       "      <td>0.82</td>\n",
       "      <td>1.41</td>\n",
       "      <td>0.67</td>\n",
       "      <td>3.61</td>\n",
       "      <td>0.94</td>\n",
       "      <td>3.39</td>\n",
       "      <td>0.95</td>\n",
       "      <td>1.25</td>\n",
       "      <td>0.55</td>\n",
       "      <td>2.68</td>\n",
       "      <td>0.85</td>\n",
       "      <td>1.59</td>\n",
       "      <td>0.94</td>\n",
       "      <td>35.93</td>\n",
       "      <td>7.26</td>\n",
       "      <td>8.03</td>\n",
       "      <td>3.06</td>\n",
       "      <td>17.59</td>\n",
       "      <td>3.50</td>\n",
       "      <td>3.05</td>\n",
       "      <td>1.14</td>\n",
       "      <td>0.84</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>208</td>\n",
       "      <td>22</td>\n",
       "      <td>131</td>\n",
       "      <td>2126</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-0.057159</td>\n",
       "      <td>1.287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3200</th>\n",
       "      <td>X</td>\n",
       "      <td>154419685</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>0.589</td>\n",
       "      <td>0.067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000102125</td>\n",
       "      <td>ENST00000601016</td>\n",
       "      <td>TAZ</td>\n",
       "      <td>CCDS14748.1</td>\n",
       "      <td>6/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-20.0</td>\n",
       "      <td>ACCEPTOR</td>\n",
       "      <td>307</td>\n",
       "      <td>46</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.282</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.367</td>\n",
       "      <td>-1.410</td>\n",
       "      <td>-2.587</td>\n",
       "      <td>84.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.59</td>\n",
       "      <td>-12.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.10</td>\n",
       "      <td>5.44</td>\n",
       "      <td>9.06</td>\n",
       "      <td>3.54</td>\n",
       "      <td>10.70</td>\n",
       "      <td>3.56</td>\n",
       "      <td>12.16</td>\n",
       "      <td>2.72</td>\n",
       "      <td>16.61</td>\n",
       "      <td>3.38</td>\n",
       "      <td>19.98</td>\n",
       "      <td>9.16</td>\n",
       "      <td>8.35</td>\n",
       "      <td>2.58</td>\n",
       "      <td>23.41</td>\n",
       "      <td>3.23</td>\n",
       "      <td>10.09</td>\n",
       "      <td>1.87</td>\n",
       "      <td>12.96</td>\n",
       "      <td>2.68</td>\n",
       "      <td>7.48</td>\n",
       "      <td>1.84</td>\n",
       "      <td>0.55</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>105</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.033909</td>\n",
       "      <td>1.464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>X</td>\n",
       "      <td>154420108</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>0.656</td>\n",
       "      <td>0.053</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000197180</td>\n",
       "      <td>ENST00000360656</td>\n",
       "      <td>CH17-340M24.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>117</td>\n",
       "      <td>14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.998</td>\n",
       "      <td>-2.121</td>\n",
       "      <td>-2.419</td>\n",
       "      <td>84.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.27</td>\n",
       "      <td>-10.50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.24</td>\n",
       "      <td>5.14</td>\n",
       "      <td>6.47</td>\n",
       "      <td>1.41</td>\n",
       "      <td>5.91</td>\n",
       "      <td>1.49</td>\n",
       "      <td>13.65</td>\n",
       "      <td>3.71</td>\n",
       "      <td>11.55</td>\n",
       "      <td>4.22</td>\n",
       "      <td>5.02</td>\n",
       "      <td>1.10</td>\n",
       "      <td>2.45</td>\n",
       "      <td>1.14</td>\n",
       "      <td>19.29</td>\n",
       "      <td>4.51</td>\n",
       "      <td>4.87</td>\n",
       "      <td>2.06</td>\n",
       "      <td>14.04</td>\n",
       "      <td>2.63</td>\n",
       "      <td>7.68</td>\n",
       "      <td>1.94</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>101</td>\n",
       "      <td>12</td>\n",
       "      <td>31</td>\n",
       "      <td>826</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>-0.549155</td>\n",
       "      <td>0.036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3202</th>\n",
       "      <td>X</td>\n",
       "      <td>154420108</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>0.656</td>\n",
       "      <td>0.053</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000102125</td>\n",
       "      <td>ENST00000601016</td>\n",
       "      <td>TAZ</td>\n",
       "      <td>CCDS14748.1</td>\n",
       "      <td>8/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-14.0</td>\n",
       "      <td>DONOR</td>\n",
       "      <td>117</td>\n",
       "      <td>14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.998</td>\n",
       "      <td>-2.121</td>\n",
       "      <td>-2.419</td>\n",
       "      <td>84.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.27</td>\n",
       "      <td>-10.50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.24</td>\n",
       "      <td>5.14</td>\n",
       "      <td>6.47</td>\n",
       "      <td>1.41</td>\n",
       "      <td>5.91</td>\n",
       "      <td>1.49</td>\n",
       "      <td>13.65</td>\n",
       "      <td>3.71</td>\n",
       "      <td>11.55</td>\n",
       "      <td>4.22</td>\n",
       "      <td>5.02</td>\n",
       "      <td>1.10</td>\n",
       "      <td>2.45</td>\n",
       "      <td>1.14</td>\n",
       "      <td>19.29</td>\n",
       "      <td>4.51</td>\n",
       "      <td>4.87</td>\n",
       "      <td>2.06</td>\n",
       "      <td>14.04</td>\n",
       "      <td>2.63</td>\n",
       "      <td>7.68</td>\n",
       "      <td>1.94</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>101</td>\n",
       "      <td>12</td>\n",
       "      <td>31</td>\n",
       "      <td>826</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>-0.549155</td>\n",
       "      <td>0.036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3203</th>\n",
       "      <td>X</td>\n",
       "      <td>154961190</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>0.358</td>\n",
       "      <td>0.040</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000185010</td>\n",
       "      <td>ENST00000360256</td>\n",
       "      <td>F8</td>\n",
       "      <td>CCDS35457.1</td>\n",
       "      <td>9/25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>61526</td>\n",
       "      <td>31037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.202</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.404</td>\n",
       "      <td>0.046</td>\n",
       "      <td>0.041</td>\n",
       "      <td>212.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.90</td>\n",
       "      <td>-1.23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.99</td>\n",
       "      <td>0.77</td>\n",
       "      <td>2.39</td>\n",
       "      <td>0.83</td>\n",
       "      <td>6.68</td>\n",
       "      <td>1.78</td>\n",
       "      <td>5.65</td>\n",
       "      <td>2.53</td>\n",
       "      <td>9.39</td>\n",
       "      <td>1.31</td>\n",
       "      <td>2.07</td>\n",
       "      <td>0.82</td>\n",
       "      <td>16.78</td>\n",
       "      <td>3.18</td>\n",
       "      <td>2.89</td>\n",
       "      <td>1.19</td>\n",
       "      <td>1.80</td>\n",
       "      <td>1.09</td>\n",
       "      <td>7.22</td>\n",
       "      <td>3.13</td>\n",
       "      <td>6.76</td>\n",
       "      <td>1.16</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>75</td>\n",
       "      <td>7</td>\n",
       "      <td>45</td>\n",
       "      <td>823</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.351934</td>\n",
       "      <td>5.879</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3204 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos  Ref Alt Type  Length    AnnoType  Consequence  \\\n",
       "0         1    1041950    T   C  SNV       0  Transcript  SPLICE_SITE   \n",
       "1         1    1042190    G   A  SNV       0  Transcript     INTRONIC   \n",
       "2         1    1043223  CCT   C  DEL       2  Transcript     INTRONIC   \n",
       "3         1    1045707    A   G  SNV       0  Transcript     INTRONIC   \n",
       "...     ...        ...  ...  ..  ...     ...         ...          ...   \n",
       "3200      X  154419685    C   T  SNV       0  Transcript     INTRONIC   \n",
       "3201      X  154420108    C   T  SNV       0  Intergenic   DOWNSTREAM   \n",
       "3202      X  154420108    C   T  SNV       0  Transcript     INTRONIC   \n",
       "3203      X  154961190    A   G  SNV       0  Transcript     INTRONIC   \n",
       "\n",
       "      ConsScore     ConsDetail     GC    CpG  motifECount motifEName  \\\n",
       "0             5  splice,intron  0.702  0.120          NaN        NaN   \n",
       "1             2         intron  0.669  0.093          NaN        NaN   \n",
       "2             2         intron  0.678  0.079          NaN        NaN   \n",
       "3             2         intron  0.656  0.093          NaN        NaN   \n",
       "...         ...            ...    ...    ...          ...        ...   \n",
       "3200          2         intron  0.589  0.067          NaN        NaN   \n",
       "3201          1     downstream  0.656  0.053          NaN        NaN   \n",
       "3202          2         intron  0.656  0.053          NaN        NaN   \n",
       "3203          2         intron  0.358  0.040          NaN        NaN   \n",
       "\n",
       "      motifEHIPos  motifEScoreChng  oAA  nAA           GeneID  \\\n",
       "0             NaN              NaN  NaN  NaN  ENSG00000188157   \n",
       "1             NaN              NaN  NaN  NaN  ENSG00000188157   \n",
       "2             NaN              NaN  NaN  NaN  ENSG00000188157   \n",
       "3             NaN              NaN  NaN  NaN  ENSG00000188157   \n",
       "...           ...              ...  ...  ...              ...   \n",
       "3200          NaN              NaN  NaN  NaN  ENSG00000102125   \n",
       "3201          NaN              NaN  NaN  NaN  ENSG00000197180   \n",
       "3202          NaN              NaN  NaN  NaN  ENSG00000102125   \n",
       "3203          NaN              NaN  NaN  NaN  ENSG00000185010   \n",
       "\n",
       "            FeatureID       GeneName         CCDS Intron Exon  cDNApos  \\\n",
       "0     ENST00000379370           AGRN  CCDS30551.1   6/35  NaN      NaN   \n",
       "1     ENST00000379370           AGRN  CCDS30551.1   7/35  NaN      NaN   \n",
       "2     ENST00000379370           AGRN  CCDS30551.1   7/35  NaN      NaN   \n",
       "3     ENST00000379370           AGRN  CCDS30551.1  14/35  NaN      NaN   \n",
       "...               ...            ...          ...    ...  ...      ...   \n",
       "3200  ENST00000601016            TAZ  CCDS14748.1   6/10  NaN      NaN   \n",
       "3201  ENST00000360656  CH17-340M24.3          NaN    NaN  NaN      NaN   \n",
       "3202  ENST00000601016            TAZ  CCDS14748.1   8/10  NaN      NaN   \n",
       "3203  ENST00000360256             F8  CCDS35457.1   9/25  NaN      NaN   \n",
       "\n",
       "      relcDNApos  CDSpos  relCDSpos  protPos  relProtPos Domain  Dst2Splice  \\\n",
       "0            NaN     NaN        NaN      NaN         NaN    NaN        -6.0   \n",
       "1            NaN     NaN        NaN      NaN         NaN    NaN         NaN   \n",
       "2            NaN     NaN        NaN      NaN         NaN    NaN       -14.0   \n",
       "3            NaN     NaN        NaN      NaN         NaN    NaN         NaN   \n",
       "...          ...     ...        ...      ...         ...    ...         ...   \n",
       "3200         NaN     NaN        NaN      NaN         NaN    NaN       -20.0   \n",
       "3201         NaN     NaN        NaN      NaN         NaN    NaN         NaN   \n",
       "3202         NaN     NaN        NaN      NaN         NaN    NaN       -14.0   \n",
       "3203         NaN     NaN        NaN      NaN         NaN    NaN         NaN   \n",
       "\n",
       "     Dst2SplType  minDistTSS  minDistTSE SIFTcat  SIFTval PolyPhenCat  \\\n",
       "0       ACCEPTOR        7083         553     NaN      NaN         NaN   \n",
       "1            NaN        7323         793     NaN      NaN         NaN   \n",
       "2       ACCEPTOR       23101       12892     NaN      NaN         NaN   \n",
       "3            NaN         308         643     NaN      NaN         NaN   \n",
       "...          ...         ...         ...     ...      ...         ...   \n",
       "3200    ACCEPTOR         307          46     NaN      NaN         NaN   \n",
       "3201         NaN         117          14     NaN      NaN         NaN   \n",
       "3202       DONOR         117          14     NaN      NaN         NaN   \n",
       "3203         NaN       61526       31037     NaN      NaN         NaN   \n",
       "\n",
       "      PolyPhenVal  priPhCons  mamPhCons  verPhCons  priPhyloP  mamPhyloP  \\\n",
       "0             NaN      0.006        0.0        0.0      0.398     -0.498   \n",
       "1             NaN      0.011        0.0        0.0     -0.346     -1.186   \n",
       "2             NaN      0.001        0.0        0.0      0.418     -0.047   \n",
       "3             NaN      0.076        0.0        0.0      0.475     -0.843   \n",
       "...           ...        ...        ...        ...        ...        ...   \n",
       "3200          NaN      0.282        0.0        0.0      0.367     -1.410   \n",
       "3201          NaN      0.007        0.0        0.0     -1.998     -2.121   \n",
       "3202          NaN      0.007        0.0        0.0     -1.998     -2.121   \n",
       "3203          NaN      0.202        0.0        0.0      0.404      0.046   \n",
       "\n",
       "      verPhyloP  bStatistic  targetScan  mirSVR-Score  mirSVR-E  mirSVR-Aln  \\\n",
       "0        -0.484       940.0         NaN           NaN       NaN         NaN   \n",
       "1        -1.417       940.0         NaN           NaN       NaN         NaN   \n",
       "2        -0.165       940.0         NaN           NaN       NaN         NaN   \n",
       "3        -0.993       940.0         NaN           NaN       NaN         NaN   \n",
       "...         ...         ...         ...           ...       ...         ...   \n",
       "3200     -2.587        84.0         NaN           NaN       NaN         NaN   \n",
       "3201     -2.419        84.0         NaN           NaN       NaN         NaN   \n",
       "3202     -2.419        84.0         NaN           NaN       NaN         NaN   \n",
       "3203      0.041       212.0         NaN           NaN       NaN         NaN   \n",
       "\n",
       "      cHmm_E1  cHmm_E2  cHmm_E3  cHmm_E4  cHmm_E5  cHmm_E6  cHmm_E7  cHmm_E8  \\\n",
       "0           0        2        0        2        0        7        6        5   \n",
       "1           1        1        0        2        0       10        4        6   \n",
       "2           1        1        0        3        1        6        4       14   \n",
       "3           0        1        0        4        0        5        8       14   \n",
       "...       ...      ...      ...      ...      ...      ...      ...      ...   \n",
       "3200        0        1        0        0        0        1       15        1   \n",
       "3201        0        1        0        0        0        1       16        2   \n",
       "3202        0        1        0        0        0        1       16        2   \n",
       "3203        0        0        1       33        0        0        0        0   \n",
       "\n",
       "      cHmm_E9  cHmm_E10  cHmm_E11  cHmm_E12  cHmm_E13  cHmm_E14  cHmm_E15  \\\n",
       "0           1         2         2         0         0         0        12   \n",
       "1           1         2         2         0         0         0        11   \n",
       "2           1         0         0         0         0         0        12   \n",
       "3           1         1         0         0         0         0         9   \n",
       "...       ...       ...       ...       ...       ...       ...       ...   \n",
       "3200        0         0         2         3         2         4        16   \n",
       "3201        0         1         2         3         1         3        15   \n",
       "3202        0         1         2         3         1         3        15   \n",
       "3203        0         0         0         0         0         0        12   \n",
       "\n",
       "      cHmm_E16  cHmm_E17  cHmm_E18  cHmm_E19  cHmm_E20  cHmm_E21  cHmm_E22  \\\n",
       "0            0         0         0         0         0         2         4   \n",
       "1            0         0         0         0         0         2         4   \n",
       "2            0         0         0         0         0         1         2   \n",
       "3            0         0         0         0         0         2         0   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "3200         0         0         0         0         0         0         0   \n",
       "3201         0         0         0         0         0         0         0   \n",
       "3202         0         0         0         0         0         0         0   \n",
       "3203         0         0         0         0         0         1         1   \n",
       "\n",
       "      cHmm_E23  cHmm_E24  cHmm_E25  GerpRS  GerpRSpval  GerpN  GerpS  \\\n",
       "0            3         0         0     NaN         NaN   4.03   4.03   \n",
       "1            2         0         0     NaN         NaN  10.30 -20.60   \n",
       "2            2         0         0     NaN         NaN  10.30   1.42   \n",
       "3            3         0         0     NaN         NaN   6.94 -13.90   \n",
       "...        ...       ...       ...     ...         ...    ...    ...   \n",
       "3200         1         1         1     NaN         NaN   7.59 -12.20   \n",
       "3201         1         1         1     NaN         NaN   5.27 -10.50   \n",
       "3202         1         1         1     NaN         NaN   5.27 -10.50   \n",
       "3203         0         0         0     NaN         NaN  10.90  -1.23   \n",
       "\n",
       "      tOverlapMotifs  motifDist  EncodeH3K4me1-sum  EncodeH3K4me1-max  \\\n",
       "0                NaN        NaN               7.16               1.40   \n",
       "1                1.0       0.13               8.05               1.59   \n",
       "2                1.0      -0.96               2.89               0.74   \n",
       "3                NaN        NaN               4.50               0.82   \n",
       "...              ...        ...                ...                ...   \n",
       "3200             NaN        NaN              17.10               5.44   \n",
       "3201             NaN        NaN              15.24               5.14   \n",
       "3202             NaN        NaN              15.24               5.14   \n",
       "3203             NaN        NaN               1.99               0.77   \n",
       "\n",
       "      EncodeH3K4me2-sum  EncodeH3K4me2-max  EncodeH3K4me3-sum  \\\n",
       "0                  6.10               2.06               6.40   \n",
       "1                  9.22               2.78               5.16   \n",
       "2                  3.29               1.03               6.55   \n",
       "3                  1.41               0.67               3.61   \n",
       "...                 ...                ...                ...   \n",
       "3200               9.06               3.54              10.70   \n",
       "3201               6.47               1.41               5.91   \n",
       "3202               6.47               1.41               5.91   \n",
       "3203               2.39               0.83               6.68   \n",
       "\n",
       "      EncodeH3K4me3-max  EncodeH3K9ac-sum  EncodeH3K9ac-max  \\\n",
       "0                  0.99              2.91              0.78   \n",
       "1                  1.25              4.23              2.12   \n",
       "2                  2.05              6.62              1.17   \n",
       "3                  0.94              3.39              0.95   \n",
       "...                 ...               ...               ...   \n",
       "3200               3.56             12.16              2.72   \n",
       "3201               1.49             13.65              3.71   \n",
       "3202               1.49             13.65              3.71   \n",
       "3203               1.78              5.65              2.53   \n",
       "\n",
       "      EncodeH3K9me3-sum  EncodeH3K9me3-max  EncodeH3K27ac-sum  \\\n",
       "0                  5.47               1.33               2.72   \n",
       "1                  8.84               2.66               6.76   \n",
       "2                  6.56               2.21               5.19   \n",
       "3                  1.25               0.55               2.68   \n",
       "...                 ...                ...                ...   \n",
       "3200              16.61               3.38              19.98   \n",
       "3201              11.55               4.22               5.02   \n",
       "3202              11.55               4.22               5.02   \n",
       "3203               9.39               1.31               2.07   \n",
       "\n",
       "      EncodeH3K27ac-max  EncodeH3K27me3-sum  EncodeH3K27me3-max  \\\n",
       "0                  1.32                3.20                1.44   \n",
       "1                  1.20                9.21                2.85   \n",
       "2                  1.22                5.39                2.43   \n",
       "3                  0.85                1.59                0.94   \n",
       "...                 ...                 ...                 ...   \n",
       "3200               9.16                8.35                2.58   \n",
       "3201               1.10                2.45                1.14   \n",
       "3202               1.10                2.45                1.14   \n",
       "3203               0.82               16.78                3.18   \n",
       "\n",
       "      EncodeH3K36me3-sum  EncodeH3K36me3-max  EncodeH3K79me2-sum  \\\n",
       "0                  13.29                3.23               15.60   \n",
       "1                  26.09                9.04               20.55   \n",
       "2                  19.45                4.05               12.22   \n",
       "3                  35.93                7.26                8.03   \n",
       "...                  ...                 ...                 ...   \n",
       "3200               23.41                3.23               10.09   \n",
       "3201               19.29                4.51                4.87   \n",
       "3202               19.29                4.51                4.87   \n",
       "3203                2.89                1.19                1.80   \n",
       "\n",
       "      EncodeH3K79me2-max  EncodeH4K20me1-sum  EncodeH4K20me1-max  \\\n",
       "0                   3.09               21.17                3.99   \n",
       "1                   4.96               28.70                5.25   \n",
       "2                   3.37               18.53                4.93   \n",
       "3                   3.06               17.59                3.50   \n",
       "...                  ...                 ...                 ...   \n",
       "3200                1.87               12.96                2.68   \n",
       "3201                2.06               14.04                2.63   \n",
       "3202                2.06               14.04                2.63   \n",
       "3203                1.09                7.22                3.13   \n",
       "\n",
       "      EncodeH2AFZ-sum  EncodeH2AFZ-max  EncodeDNase-sum  EncodeDNase-max  \\\n",
       "0                5.78             4.03             0.90             0.57   \n",
       "1                4.82             1.26             0.80             0.38   \n",
       "2                5.36             2.41             0.63             0.30   \n",
       "3                3.05             1.14             0.84             0.37   \n",
       "...               ...              ...              ...              ...   \n",
       "3200             7.48             1.84             0.55             0.10   \n",
       "3201             7.68             1.94             0.43             0.17   \n",
       "3202             7.68             1.94             0.43             0.17   \n",
       "3203             6.76             1.16             0.15             0.05   \n",
       "\n",
       "      EncodetotalRNA-sum  EncodetotalRNA-max  Grantham  Dist2Mutation  \\\n",
       "0                   0.31                0.17       NaN              2   \n",
       "1                   0.54                0.19       NaN             12   \n",
       "2                   0.33                0.26       NaN              6   \n",
       "3                   0.43                0.17       NaN              2   \n",
       "...                  ...                 ...       ...            ...   \n",
       "3200                0.82                0.16       NaN             22   \n",
       "3201                0.19                0.09       NaN             16   \n",
       "3202                0.19                0.09       NaN             16   \n",
       "3203                 NaN                 NaN       NaN             11   \n",
       "\n",
       "      Freq100bp  Rare100bp  Sngl100bp  Freq1000bp  Rare1000bp  Sngl1000bp  \\\n",
       "0             1          2         21           3           8         200   \n",
       "1             1          0         20           3           7         213   \n",
       "2             0          1         18           1          10         216   \n",
       "3             1          1         17           1          13         208   \n",
       "...         ...        ...        ...         ...         ...         ...   \n",
       "3200          0          0         16           0           5         105   \n",
       "3201          0          1          7           0           4         101   \n",
       "3202          0          1          7           0           4         101   \n",
       "3203          0          1         10           0           3          75   \n",
       "\n",
       "      Freq10000bp  Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  \\\n",
       "0              23          100         1931                       NaN   \n",
       "1              25          109         1955                       NaN   \n",
       "2              24          117         2004                       NaN   \n",
       "3              22          131         2126                       NaN   \n",
       "...           ...          ...          ...                       ...   \n",
       "3200           11           29          820                       NaN   \n",
       "3201           12           31          826                       NaN   \n",
       "3202           12           31          826                       NaN   \n",
       "3203            7           45          823                       NaN   \n",
       "\n",
       "      dbscSNV-ada_score dbscSNV-rf_score  RemapOverlapTF  RemapOverlapCL  \\\n",
       "0               0.00001            0.002            10.0            13.0   \n",
       "1                   NaN              NaN            12.0            14.0   \n",
       "2                   NaN              NaN             8.0             9.0   \n",
       "3                   NaN              NaN             9.0            10.0   \n",
       "...                 ...              ...             ...             ...   \n",
       "3200                NaN              NaN             4.0             4.0   \n",
       "3201                NaN              NaN            36.0            47.0   \n",
       "3202                NaN              NaN            36.0            47.0   \n",
       "3203                NaN              NaN             1.0             1.0   \n",
       "\n",
       "      RawScore  PHRED  \n",
       "0     0.092155  2.700  \n",
       "1    -0.401276  0.118  \n",
       "2    -0.012669  1.640  \n",
       "3    -0.057159  1.287  \n",
       "...        ...    ...  \n",
       "3200 -0.033909  1.464  \n",
       "3201 -0.549155  0.036  \n",
       "3202 -0.549155  0.036  \n",
       "3203  0.351934  5.879  \n",
       "\n",
       "[3204 rows x 125 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_noncoding_annotated_with_cadd = pd.read_table('clinvar_noncoding_benign_cadd_annotations_noheader.tsv')\n",
    "clinvar_benign_noncoding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the file manipulation for clinvar noncoding benign data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10</th>\n",
       "      <th>100987606</th>\n",
       "      <th>136593</th>\n",
       "      <th>G</th>\n",
       "      <th>T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102065918</td>\n",
       "      <td>284200</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102396271</td>\n",
       "      <td>474786</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102399466</td>\n",
       "      <td>541631</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102401202</td>\n",
       "      <td>474781</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4654</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964016</td>\n",
       "      <td>255991</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4655</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85978770</td>\n",
       "      <td>497462</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4656</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85978816</td>\n",
       "      <td>377662</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4657</th>\n",
       "      <td>chrX</td>\n",
       "      <td>93671995</td>\n",
       "      <td>208906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4658 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr10  100987606  136593  G  T\n",
       "0     chr10  102065918  284200  C  G\n",
       "1     chr10  102396271  474786  G  A\n",
       "2     chr10  102399466  541631  C  T\n",
       "3     chr10  102401202  474781  C  T\n",
       "...     ...        ...     ... .. ..\n",
       "4654   chrX   85964016  255991  T  C\n",
       "4655   chrX   85978770  497462  G  C\n",
       "4656   chrX   85978816  377662  T  A\n",
       "4657   chrX   93671995  208906  G  C\n",
       "\n",
       "[4658 rows x 5 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_coding_for_annovar = read_vcf('clinvar_coding_benign_for_cadd_noheader_with_chr_final_for_annovar.vcf')\n",
    "clinvar_benign_coding_for_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100987606</td>\n",
       "      <td>136593</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102065918</td>\n",
       "      <td>284200</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102396271</td>\n",
       "      <td>474786</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102399466</td>\n",
       "      <td>541631</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4655</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964016</td>\n",
       "      <td>255991</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4656</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85978770</td>\n",
       "      <td>497462</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4657</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85978816</td>\n",
       "      <td>377662</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4658</th>\n",
       "      <td>chrX</td>\n",
       "      <td>93671995</td>\n",
       "      <td>208906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4659 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             \n",
       "0     chr10  100987606  136593  G  T  .  .  .\n",
       "1     chr10  102065918  284200  C  G  .  .  .\n",
       "2     chr10  102396271  474786  G  A  .  .  .\n",
       "3     chr10  102399466  541631  C  T  .  .  .\n",
       "...     ...        ...     ... .. .. .. .. ..\n",
       "4655   chrX   85964016  255991  T  C  .  .  .\n",
       "4656   chrX   85978770  497462  G  C  .  .  .\n",
       "4657   chrX   85978816  377662  T  A  .  .  .\n",
       "4658   chrX   93671995  208906  G  C  .  .  .\n",
       "\n",
       "[4659 rows x 8 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_coding_for_annovar['..'] = '.'\n",
    "clinvar_benign_coding_for_annovar['...'] = '.'\n",
    "clinvar_benign_coding_for_annovar['....'] = '.'\n",
    "clinvar_benign_coding_for_annovar.loc[-1] = ['chr10', '100987606', '136593', 'G', 'T', '.', '.', '.']\n",
    "clinvar_benign_coding_for_annovar.index = clinvar_benign_coding_for_annovar.index + 1  # shifting index\n",
    "clinvar_benign_coding_for_annovar = clinvar_benign_coding_for_annovar.sort_index()\n",
    "clinvar_benign_coding_for_annovar.columns = ['', '', '', '', '', '', '', '']\n",
    "clinvar_benign_coding_for_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'clinvar_benign_coding_for_annovar' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-72-b990ddc4dc0d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutput_VCF\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mvcf\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0mvcf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mheader\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m \u001b[0mclinvar_benign_coding_for_annovar\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutput_VCF\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"\\t\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'a'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'clinvar_benign_coding_for_annovar' is not defined"
     ]
    }
   ],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "output_VCF = \"clinvar_benign_coding_for_annovar.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "clinvar_benign_coding_for_annovar.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for clinvar benign coding data from ANNOVAR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100987606</td>\n",
       "      <td>100987606</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>NM_021830:c.-605G&gt;T;NM_001163813:c.-2157G&gt;T;NM...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102065918</td>\n",
       "      <td>102065918</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>HPS6</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>HPS6:NM_024747:exon1:c.C444G:p.A148A</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102396271</td>\n",
       "      <td>102396271</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>NFKB2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>NFKB2:NM_001261403:exon2:c.G40A:p.E14K,NFKB2:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.018</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102399466</td>\n",
       "      <td>102399466</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>NFKB2</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>NFKB2:NM_001261403:exon12:c.C1296T:p.C432C,NFK...</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4655</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964016</td>\n",
       "      <td>85964016</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>CHM:NM_000390:exon5:c.A351G:p.A117A</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4656</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85978770</td>\n",
       "      <td>85978770</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>CHM:NM_000390:exon4:c.C311G:p.A104G,CHM:NM_001...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0245874451416</td>\n",
       "      <td>0.248</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4657</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85978816</td>\n",
       "      <td>85978816</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>CHM:NM_000390:exon4:c.A265T:p.S89C,CHM:NM_0011...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.100</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4658</th>\n",
       "      <td>chrX</td>\n",
       "      <td>93671995</td>\n",
       "      <td>93671995</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>NAP1L3</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>NAP1L3:NM_004538:exon1:c.C1310G:p.A437G</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0059993776503</td>\n",
       "      <td>0.060</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4659 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0     chr10  100987606  100987606   G   T         UTR5         TWNK   \n",
       "1     chr10  102065918  102065918   C   G       exonic         HPS6   \n",
       "2     chr10  102396271  102396271   G   A       exonic        NFKB2   \n",
       "3     chr10  102399466  102399466   C   T       exonic        NFKB2   \n",
       "...     ...        ...        ...  ..  ..          ...          ...   \n",
       "4655   chrX   85964016   85964016   T   C       exonic          CHM   \n",
       "4656   chrX   85978770   85978770   G   C       exonic          CHM   \n",
       "4657   chrX   85978816   85978816   T   A       exonic          CHM   \n",
       "4658   chrX   93671995   93671995   G   C       exonic       NAP1L3   \n",
       "\n",
       "                                     GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0     NM_021830:c.-605G>T;NM_001163813:c.-2157G>T;NM...                  .   \n",
       "1                                                     .     synonymous SNV   \n",
       "2                                                     .  nonsynonymous SNV   \n",
       "3                                                     .     synonymous SNV   \n",
       "...                                                 ...                ...   \n",
       "4655                                                  .     synonymous SNV   \n",
       "4656                                                  .  nonsynonymous SNV   \n",
       "4657                                                  .  nonsynonymous SNV   \n",
       "4658                                                  .  nonsynonymous SNV   \n",
       "\n",
       "                                       AAChange.refGene  ... BP4 BP5 BP6 BP7  \\\n",
       "0                                                     .  ...   .   .   .   .   \n",
       "1                  HPS6:NM_024747:exon1:c.C444G:p.A148A  ...   1   0   1   1   \n",
       "2     NFKB2:NM_001261403:exon2:c.G40A:p.E14K,NFKB2:N...  ...   0   0   0   0   \n",
       "3     NFKB2:NM_001261403:exon12:c.C1296T:p.C432C,NFK...  ...   1   0   0   1   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "4655                CHM:NM_000390:exon5:c.A351G:p.A117A  ...   1   0   1   1   \n",
       "4656  CHM:NM_000390:exon4:c.C311G:p.A104G,CHM:NM_001...  ...   0   0   0   0   \n",
       "4657  CHM:NM_000390:exon4:c.A265T:p.S89C,CHM:NM_0011...  ...   0   0   1   0   \n",
       "4658            NAP1L3:NM_004538:exon1:c.C1310G:p.A437G  ...   1   0   1   0   \n",
       "\n",
       "     cosmic70             MCAP  REVEL regsnp_fpr regsnp_disease  \\\n",
       "0           .                .      .          .              .   \n",
       "1           .                .      .          .              .   \n",
       "2           .                .  0.018          .              .   \n",
       "3           .                .      .          .              .   \n",
       "...       ...              ...    ...        ...            ...   \n",
       "4655        .                .      .          .              .   \n",
       "4656        .  0.0245874451416  0.248          .              .   \n",
       "4657        .                .  0.100          .              .   \n",
       "4658        .  0.0059993776503  0.060          .              .   \n",
       "\n",
       "     regsnp_splicing_site  \n",
       "0                       .  \n",
       "1                       .  \n",
       "2                       .  \n",
       "3                       .  \n",
       "...                   ...  \n",
       "4655                    .  \n",
       "4656                    .  \n",
       "4657                    .  \n",
       "4658                    .  \n",
       "\n",
       "[4659 rows x 149 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_coding_annotated_with_annovar = pd.read_csv('clinvar_benign_coding_for_annovar.hg38_multianno.csv')\n",
    "clinvar_benign_coding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for clinvar benign coding data from CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1616</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1616</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1616</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1616</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1014217</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>5</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>87</td>\n",
       "      <td>1629</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>-0.293246</td>\n",
       "      <td>0.276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9133</th>\n",
       "      <td>X</td>\n",
       "      <td>154776309</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>970</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>-0.180506</td>\n",
       "      <td>0.611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9134</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>3</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>INFRAME</td>\n",
       "      <td>6</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.893695</td>\n",
       "      <td>10.880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9135</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>3</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.893695</td>\n",
       "      <td>10.880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9136</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>3</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.893695</td>\n",
       "      <td>10.880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9137</th>\n",
       "      <td>X</td>\n",
       "      <td>154929926</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>5</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>878</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.089486</td>\n",
       "      <td>2.669</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9138 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref   Alt Type  Length           AnnoType  \\\n",
       "0         1    1014042   G     A  SNV       0   CodingTranscript   \n",
       "1         1    1014042   G     A  SNV       0         Intergenic   \n",
       "2         1    1014042   G     A  SNV       0  RegulatoryFeature   \n",
       "3         1    1014042   G     A  SNV       0  RegulatoryFeature   \n",
       "4         1    1014217   C     T  SNV       0   CodingTranscript   \n",
       "...     ...        ...  ..   ...  ...     ...                ...   \n",
       "9133      X  154776309   C     T  SNV       0         Intergenic   \n",
       "9134      X  154776813   C  CAAG  INS       3   CodingTranscript   \n",
       "9135      X  154776813   C  CAAG  INS       3         Intergenic   \n",
       "9136      X  154776813   C  CAAG  INS       3         Intergenic   \n",
       "9137      X  154929926   T     G  SNV       0   CodingTranscript   \n",
       "\n",
       "         Consequence  ConsScore         ConsDetail  ...  Freq10000bp  \\\n",
       "0     NON_SYNONYMOUS          7           missense  ...           22   \n",
       "1           UPSTREAM          1           upstream  ...           22   \n",
       "2         REGULATORY          4         regulatory  ...           22   \n",
       "3         REGULATORY          4         regulatory  ...           22   \n",
       "4         SYNONYMOUS          5         synonymous  ...           21   \n",
       "...              ...        ...                ...  ...          ...   \n",
       "9133      DOWNSTREAM          1         downstream  ...            4   \n",
       "9134         INFRAME          6  inframe_insertion  ...            4   \n",
       "9135      DOWNSTREAM          1         downstream  ...            4   \n",
       "9136      DOWNSTREAM          1         downstream  ...            4   \n",
       "9137      SYNONYMOUS          5         synonymous  ...            3   \n",
       "\n",
       "      Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              87         1616                  Promoter                NaN   \n",
       "1              87         1616                  Promoter                NaN   \n",
       "2              87         1616                  Promoter                NaN   \n",
       "3              87         1616                  Promoter                NaN   \n",
       "4              87         1629                  Promoter                NaN   \n",
       "...           ...          ...                       ...                ...   \n",
       "9133           21          970                       NaN                NaN   \n",
       "9134           22          980                       NaN                NaN   \n",
       "9135           22          980                       NaN                NaN   \n",
       "9136           22          980                       NaN                NaN   \n",
       "9137           31          878                       NaN                NaN   \n",
       "\n",
       "      dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore   PHRED  \n",
       "0                  NaN           69.0           97.0  0.076262   2.519  \n",
       "1                  NaN           69.0           97.0  0.076262   2.519  \n",
       "2                  NaN           69.0           97.0  0.076262   2.519  \n",
       "3                  NaN           69.0           97.0  0.076262   2.519  \n",
       "4                  NaN           69.0           94.0 -0.293246   0.276  \n",
       "...                ...            ...            ...       ...     ...  \n",
       "9133               NaN           17.0           17.0 -0.180506   0.611  \n",
       "9134               NaN            7.0            7.0  0.893695  10.880  \n",
       "9135               NaN            7.0            7.0  0.893695  10.880  \n",
       "9136               NaN            7.0            7.0  0.893695  10.880  \n",
       "9137               NaN            NaN            NaN  0.089486   2.669  \n",
       "\n",
       "[9138 rows x 125 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_coding_annotated_with_cadd = pd.read_table('clinvar_coding_benign_cadd_annotations_noheader.tsv')\n",
    "clinvar_benign_coding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for clinvar pathogenic noncoding data from ANNOVAR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988541</td>\n",
       "      <td>100988541</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.331delT:p.L112Sfs*2,...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989084</td>\n",
       "      <td>100989084</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.C874A:p.P292T,TWNK:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.153055471878</td>\n",
       "      <td>0.729</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989118</td>\n",
       "      <td>100989118</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.G908A:p.R303Q,TWNK:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0960831061338</td>\n",
       "      <td>0.654</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989154</td>\n",
       "      <td>100989154</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.G944T:p.W315L,TWNK:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.328561992494</td>\n",
       "      <td>0.803</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1110</th>\n",
       "      <td>chrX</td>\n",
       "      <td>74529428</td>\n",
       "      <td>74529428</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SLC16A2</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift insertion</td>\n",
       "      <td>SLC16A2:NM_006517:exon5:c.1387dupC:p.I465Hfs*51</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1111</th>\n",
       "      <td>chrX</td>\n",
       "      <td>77902641</td>\n",
       "      <td>77902641</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>splicing</td>\n",
       "      <td>COX7B</td>\n",
       "      <td>NM_001866:exon2:c.41-2A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00358744394619</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1112</th>\n",
       "      <td>chrX</td>\n",
       "      <td>78003237</td>\n",
       "      <td>78003237</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>splicing</td>\n",
       "      <td>ATP7A</td>\n",
       "      <td>NM_001282224:exon6:c.1707+1G&gt;A;NM_000052:exon6...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00179372197309</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1113</th>\n",
       "      <td>chrX</td>\n",
       "      <td>78122954</td>\n",
       "      <td>78122954</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PGK1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1114 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0     chr10  100988541  100988541   T   -       exonic         TWNK   \n",
       "1     chr10  100989084  100989084   C   A       exonic         TWNK   \n",
       "2     chr10  100989118  100989118   G   A       exonic         TWNK   \n",
       "3     chr10  100989154  100989154   G   T       exonic         TWNK   \n",
       "...     ...        ...        ...  ..  ..          ...          ...   \n",
       "1110   chrX   74529428   74529428   -   C       exonic      SLC16A2   \n",
       "1111   chrX   77902641   77902641   A   G     splicing        COX7B   \n",
       "1112   chrX   78003237   78003237   G   A     splicing        ATP7A   \n",
       "1113   chrX   78122954   78122954   G   A     intronic         PGK1   \n",
       "\n",
       "                                     GeneDetail.refGene    ExonicFunc.refGene  \\\n",
       "0                                                     .   frameshift deletion   \n",
       "1                                                     .     nonsynonymous SNV   \n",
       "2                                                     .     nonsynonymous SNV   \n",
       "3                                                     .     nonsynonymous SNV   \n",
       "...                                                 ...                   ...   \n",
       "1110                                                  .  frameshift insertion   \n",
       "1111                          NM_001866:exon2:c.41-2A>G                     .   \n",
       "1112  NM_001282224:exon6:c.1707+1G>A;NM_000052:exon6...                     .   \n",
       "1113                                                  .                     .   \n",
       "\n",
       "                                       AAChange.refGene  ... BP4 BP5 BP6 BP7  \\\n",
       "0     TWNK:NM_001163812:exon1:c.331delT:p.L112Sfs*2,...  ...   .   .   .   .   \n",
       "1     TWNK:NM_001163812:exon1:c.C874A:p.P292T,TWNK:N...  ...   0   0   0   0   \n",
       "2     TWNK:NM_001163812:exon1:c.G908A:p.R303Q,TWNK:N...  ...   0   0   0   0   \n",
       "3     TWNK:NM_001163812:exon1:c.G944T:p.W315L,TWNK:N...  ...   0   0   0   0   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "1110    SLC16A2:NM_006517:exon5:c.1387dupC:p.I465Hfs*51  ...   .   .   .   .   \n",
       "1111                                                  .  ...   .   .   .   .   \n",
       "1112                                                  .  ...   .   .   .   .   \n",
       "1113                                                  .  ...   .   .   .   .   \n",
       "\n",
       "     cosmic70             MCAP  REVEL        regsnp_fpr regsnp_disease  \\\n",
       "0           .                .      .                 .              .   \n",
       "1           .   0.153055471878  0.729                 .              .   \n",
       "2           .  0.0960831061338  0.654                 .              .   \n",
       "3           .   0.328561992494  0.803                 .              .   \n",
       "...       ...              ...    ...               ...            ...   \n",
       "1110        .                .      .                 .              .   \n",
       "1111        .                .      .  0.00358744394619              D   \n",
       "1112        .                .      .  0.00179372197309              D   \n",
       "1113        .                .      .               0.0              D   \n",
       "\n",
       "     regsnp_splicing_site  \n",
       "0                       .  \n",
       "1                       .  \n",
       "2                       .  \n",
       "3                       .  \n",
       "...                   ...  \n",
       "1110                    .  \n",
       "1111                   on  \n",
       "1112                   on  \n",
       "1113                   on  \n",
       "\n",
       "[1114 rows x 149 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_noncoding_annotated_with_annovar = pd.read_csv('clinvar.hg38_multianno.csv')\n",
    "clinvar_pathogenic_noncoding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated result for pathogenic noncoding variants from CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>11960768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>90</td>\n",
       "      <td>1434</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>0.99998</td>\n",
       "      <td>0.942</td>\n",
       "      <td>27.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>5.268674</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11960768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_donor</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>90</td>\n",
       "      <td>1434</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>0.99998</td>\n",
       "      <td>0.942</td>\n",
       "      <td>27.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>5.268674</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>11964787</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_donor</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>99</td>\n",
       "      <td>1332</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99987</td>\n",
       "      <td>0.886</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>4.686023</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>11972977</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>95</td>\n",
       "      <td>1304</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.497634</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2641</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>splice,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>7.476349</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2642</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>7.476349</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2643</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "      <td>828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>0.944</td>\n",
       "      <td>38.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>4.640280</td>\n",
       "      <td>29.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2644</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_acceptor</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "      <td>828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>0.944</td>\n",
       "      <td>38.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>4.640280</td>\n",
       "      <td>29.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2645 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0         1   11960768   G   A  SNV       0  RegulatoryFeature   \n",
       "1         1   11960768   G   A  SNV       0         Transcript   \n",
       "2         1   11964787   T   C  SNV       0         Transcript   \n",
       "3         1   11972977   C   T  SNV       0   CodingTranscript   \n",
       "...     ...        ...  ..  ..  ...     ...                ...   \n",
       "2641      X  154419746   G   T  SNV       0   CodingTranscript   \n",
       "2642      X  154419746   G   T  SNV       0         Intergenic   \n",
       "2643      X  154420211   G   C  SNV       0         Intergenic   \n",
       "2644      X  154420211   G   C  SNV       0         Transcript   \n",
       "\n",
       "           Consequence  ConsScore          ConsDetail  ...  Freq10000bp  \\\n",
       "0           REGULATORY          4          regulatory  ...           21   \n",
       "1     CANONICAL_SPLICE          6        splice_donor  ...           21   \n",
       "2     CANONICAL_SPLICE          6        splice_donor  ...           19   \n",
       "3          STOP_GAINED          8         stop_gained  ...           20   \n",
       "...                ...        ...                 ...  ...          ...   \n",
       "2641       STOP_GAINED          8  splice,stop_gained  ...           11   \n",
       "2642        DOWNSTREAM          1          downstream  ...           11   \n",
       "2643        DOWNSTREAM          1          downstream  ...           12   \n",
       "2644  CANONICAL_SPLICE          6     splice_acceptor  ...           12   \n",
       "\n",
       "      Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              90         1434  Promoter Flanking Region            0.99998   \n",
       "1              90         1434  Promoter Flanking Region            0.99998   \n",
       "2              99         1332                       NaN            0.99987   \n",
       "3              95         1304  Promoter Flanking Region                NaN   \n",
       "...           ...          ...                       ...                ...   \n",
       "2641           29          822                       NaN            0.99999   \n",
       "2642           29          822                       NaN            0.99999   \n",
       "2643           32          828                       NaN            0.99999   \n",
       "2644           32          828                       NaN            0.99999   \n",
       "\n",
       "      dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore PHRED  \n",
       "0                0.942           27.0           36.0  5.268674  35.0  \n",
       "1                0.942           27.0           36.0  5.268674  35.0  \n",
       "2                0.886           10.0           10.0  4.686023  31.0  \n",
       "3                  NaN           19.0           23.0  7.497634  38.0  \n",
       "...                ...            ...            ...       ...   ...  \n",
       "2641               1.0            9.0           10.0  7.476349  38.0  \n",
       "2642               1.0            9.0           10.0  7.476349  38.0  \n",
       "2643             0.944           38.0           46.0  4.640280  29.8  \n",
       "2644             0.944           38.0           46.0  4.640280  29.8  \n",
       "\n",
       "[2645 rows x 125 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_noncoding_annotated_with_cadd = pd.read_table('clinvar_noncoding_pathogenic_cadd_annotations_noheader.tsv')\n",
    "clinvar_pathogenic_noncoding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HGMD"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the file manipulation for hgmd coding data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10</th>\n",
       "      <th>100154922</th>\n",
       "      <th>CM140970</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100262050</td>\n",
       "      <td>CM162834</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47640</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47641</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47642</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47643</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47644 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr10  100154922   CM140970   G   A\n",
       "0      chr10  100183802   CM140971   C   A\n",
       "1      chr10  100253438  CI1824020   A  AT\n",
       "2      chr10  100256298   CD162836  TG   T\n",
       "3      chr10  100262050   CM162834   C   G\n",
       "...      ...        ...        ...  ..  ..\n",
       "47640   chrX    9760731   CM981395   A   G\n",
       "47641   chrX    9760732   CI183806   G  GA\n",
       "47642   chrX    9760736   CD171619  GC   G\n",
       "47643   chrX    9760741   CI115195   A  AG\n",
       "\n",
       "[47644 rows x 5 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_for_annovar = read_vcf('hgmd_coding_for_cadd_noheader_for_annovar.vcf')\n",
    "hgmd_coding_for_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47641</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47642</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47643</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47644</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47645 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   \n",
       "0      chr10  100154922   CM140970   G   A  .  .  .\n",
       "1      chr10  100183802   CM140971   C   A  .  .  .\n",
       "2      chr10  100253438  CI1824020   A  AT  .  .  .\n",
       "3      chr10  100256298   CD162836  TG   T  .  .  .\n",
       "...      ...        ...        ...  ..  .. .. .. ..\n",
       "47641   chrX    9760731   CM981395   A   G  .  .  .\n",
       "47642   chrX    9760732   CI183806   G  GA  .  .  .\n",
       "47643   chrX    9760736   CD171619  GC   G  .  .  .\n",
       "47644   chrX    9760741   CI115195   A  AG  .  .  .\n",
       "\n",
       "[47645 rows x 8 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_for_annovar['..'] = '.'\n",
    "hgmd_coding_for_annovar['...'] = '.'\n",
    "hgmd_coding_for_annovar['....'] = '.'\n",
    "hgmd_coding_for_annovar.loc[-1] = ['chr10', '100154922', 'CM140970', 'G', 'A', '.', '.', '.']\n",
    "hgmd_coding_for_annovar.index = hgmd_coding_for_annovar.index + 1  # shifting index\n",
    "hgmd_coding_for_annovar = hgmd_coding_for_annovar.sort_index()\n",
    "hgmd_coding_for_annovar.columns = ['', '', '', '', '', '', '', '']\n",
    "hgmd_coding_for_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "output_VCF = \"hgmd_coding_for_annovar.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "hgmd_coding_for_annovar.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for hgmd coding data from ANNOVAR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>100154922</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ERLIN1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>ERLIN1:NM_001347858:exon9:c.C283T:p.R95X,ERLIN...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>100183802</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ERLIN1</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>ERLIN1:NM_006459:exon2:c.G149T:p.G50V,ERLIN1:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5619750829</td>\n",
       "      <td>0.984</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>100253438</td>\n",
       "      <td>-</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>CWF19L1:NM_001303406:exon3:c.194dupA:p.Y65fs*0...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256299</td>\n",
       "      <td>100256299</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>CWF19L1:NM_001303406:exon2:c.56delC:p.P19Hfs*3...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47641</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>9760731</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GPR143:NM_000273:exon2:c.T346C:p.C116R</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.172141444647</td>\n",
       "      <td>0.295</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47642</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>9760732</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift insertion</td>\n",
       "      <td>GPR143:NM_000273:exon2:c.344dupT:p.C116Lfs*69</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47643</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760737</td>\n",
       "      <td>9760737</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>GPR143:NM_000273:exon2:c.340delG:p.A114Lfs*30</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47644</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>9760741</td>\n",
       "      <td>-</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift insertion</td>\n",
       "      <td>GPR143:NM_000273:exon2:c.335dupC:p.A113Cfs*72</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47645 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0      chr10  100154922  100154922   G   A       exonic       ERLIN1   \n",
       "1      chr10  100183802  100183802   C   A       exonic       ERLIN1   \n",
       "2      chr10  100253438  100253438   -   T       exonic      CWF19L1   \n",
       "3      chr10  100256299  100256299   G   -       exonic      CWF19L1   \n",
       "...      ...        ...        ...  ..  ..          ...          ...   \n",
       "47641   chrX    9760731    9760731   A   G       exonic       GPR143   \n",
       "47642   chrX    9760732    9760732   -   A       exonic       GPR143   \n",
       "47643   chrX    9760737    9760737   C   -       exonic       GPR143   \n",
       "47644   chrX    9760741    9760741   -   G       exonic       GPR143   \n",
       "\n",
       "      GeneDetail.refGene    ExonicFunc.refGene  \\\n",
       "0                      .              stopgain   \n",
       "1                      .     nonsynonymous SNV   \n",
       "2                      .              stopgain   \n",
       "3                      .   frameshift deletion   \n",
       "...                  ...                   ...   \n",
       "47641                  .     nonsynonymous SNV   \n",
       "47642                  .  frameshift insertion   \n",
       "47643                  .   frameshift deletion   \n",
       "47644                  .  frameshift insertion   \n",
       "\n",
       "                                        AAChange.refGene  ... BP4 BP5 BP6 BP7  \\\n",
       "0      ERLIN1:NM_001347858:exon9:c.C283T:p.R95X,ERLIN...  ...   0   0   0   0   \n",
       "1      ERLIN1:NM_006459:exon2:c.G149T:p.G50V,ERLIN1:N...  ...   .   .   .   .   \n",
       "2      CWF19L1:NM_001303406:exon3:c.194dupA:p.Y65fs*0...  ...   .   .   .   .   \n",
       "3      CWF19L1:NM_001303406:exon2:c.56delC:p.P19Hfs*3...  ...   .   .   .   .   \n",
       "...                                                  ...  ...  ..  ..  ..  ..   \n",
       "47641             GPR143:NM_000273:exon2:c.T346C:p.C116R  ...   0   0   0   0   \n",
       "47642      GPR143:NM_000273:exon2:c.344dupT:p.C116Lfs*69  ...   .   .   .   .   \n",
       "47643      GPR143:NM_000273:exon2:c.340delG:p.A114Lfs*30  ...   .   .   .   .   \n",
       "47644      GPR143:NM_000273:exon2:c.335dupC:p.A113Cfs*72  ...   .   .   .   .   \n",
       "\n",
       "      cosmic70            MCAP  REVEL regsnp_fpr regsnp_disease  \\\n",
       "0            .               .      .          .              .   \n",
       "1            .    0.5619750829  0.984          .              .   \n",
       "2            .               .      .          .              .   \n",
       "3            .               .      .          .              .   \n",
       "...        ...             ...    ...        ...            ...   \n",
       "47641        .  0.172141444647  0.295          .              .   \n",
       "47642        .               .      .          .              .   \n",
       "47643        .               .      .          .              .   \n",
       "47644        .               .      .          .              .   \n",
       "\n",
       "      regsnp_splicing_site  \n",
       "0                        .  \n",
       "1                        .  \n",
       "2                        .  \n",
       "3                        .  \n",
       "...                    ...  \n",
       "47641                    .  \n",
       "47642                    .  \n",
       "47643                    .  \n",
       "47644                    .  \n",
       "\n",
       "[47645 rows x 149 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_annotated_with_annovar = pd.read_csv('hgmd_coding_annotated.hg38_multianno.csv')\n",
    "hgmd_coding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for hgmd coding data from CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89254</th>\n",
       "      <td>X</td>\n",
       "      <td>154966616</td>\n",
       "      <td>CTTCA</td>\n",
       "      <td>C</td>\n",
       "      <td>DEL</td>\n",
       "      <td>4</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>35</td>\n",
       "      <td>840</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.413846</td>\n",
       "      <td>21.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89255</th>\n",
       "      <td>X</td>\n",
       "      <td>154966617</td>\n",
       "      <td>TTC</td>\n",
       "      <td>T</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>35</td>\n",
       "      <td>840</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.392576</td>\n",
       "      <td>21.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89256</th>\n",
       "      <td>X</td>\n",
       "      <td>155524585</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>26</td>\n",
       "      <td>682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.231926</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89257</th>\n",
       "      <td>X</td>\n",
       "      <td>155524585</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>26</td>\n",
       "      <td>682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.231926</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89258 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos    Ref Alt Type  Length           AnnoType  \\\n",
       "0          1    1014143      C   T  SNV       0   CodingTranscript   \n",
       "1          1    1014143      C   T  SNV       0         Intergenic   \n",
       "2          1    1014143      C   T  SNV       0  RegulatoryFeature   \n",
       "3          1    1014143      C   T  SNV       0  RegulatoryFeature   \n",
       "...      ...        ...    ...  ..  ...     ...                ...   \n",
       "89254      X  154966616  CTTCA   C  DEL       4   CodingTranscript   \n",
       "89255      X  154966617    TTC   T  DEL       2   CodingTranscript   \n",
       "89256      X  155524585      G   A  SNV       0   CodingTranscript   \n",
       "89257      X  155524585      G   A  SNV       0         Transcript   \n",
       "\n",
       "       Consequence  ConsScore         ConsDetail  ...  Freq10000bp  \\\n",
       "0      STOP_GAINED          8        stop_gained  ...           22   \n",
       "1         UPSTREAM          1           upstream  ...           22   \n",
       "2       REGULATORY          4         regulatory  ...           22   \n",
       "3       REGULATORY          4         regulatory  ...           22   \n",
       "...            ...        ...                ...  ...          ...   \n",
       "89254  FRAME_SHIFT          7         frameshift  ...            7   \n",
       "89255  FRAME_SHIFT          7         frameshift  ...            7   \n",
       "89256  STOP_GAINED          8        stop_gained  ...            9   \n",
       "89257     INTRONIC          2  intron,non_coding  ...            9   \n",
       "\n",
       "       Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0               87         1622                  Promoter                NaN   \n",
       "1               87         1622                  Promoter                NaN   \n",
       "2               87         1622                  Promoter                NaN   \n",
       "3               87         1622                  Promoter                NaN   \n",
       "...            ...          ...                       ...                ...   \n",
       "89254           35          840                       NaN                NaN   \n",
       "89255           35          840                       NaN                NaN   \n",
       "89256           26          682                       NaN                NaN   \n",
       "89257           26          682                       NaN                NaN   \n",
       "\n",
       "       dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore PHRED  \n",
       "0                   NaN           75.0          107.0  5.410558  35.0  \n",
       "1                   NaN           75.0          107.0  5.410558  35.0  \n",
       "2                   NaN           75.0          107.0  5.410558  35.0  \n",
       "3                   NaN           75.0          107.0  5.410558  35.0  \n",
       "...                 ...            ...            ...       ...   ...  \n",
       "89254               NaN            NaN            NaN  2.413846  21.5  \n",
       "89255               NaN            NaN            NaN  2.392576  21.3  \n",
       "89256               NaN            4.0            4.0  6.231926  35.0  \n",
       "89257               NaN            4.0            4.0  6.231926  35.0  \n",
       "\n",
       "[89258 rows x 125 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_annotated_with_cadd = pd.read_table('hgmd_coding_cadd_annotations_noheader.tsv')\n",
    "hgmd_coding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the file manipulation for hgmd noncoding data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10</th>\n",
       "      <th>100988295</th>\n",
       "      <th>CM114899</th>\n",
       "      <th>C</th>\n",
       "      <th>T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988415</td>\n",
       "      <td>CM164756</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988457</td>\n",
       "      <td>CM127719</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988526</td>\n",
       "      <td>CM1610318</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988540</td>\n",
       "      <td>CD169898</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3802</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964053</td>\n",
       "      <td>CS1810957</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3803</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964054</td>\n",
       "      <td>CS1723659</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3804</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85965588</td>\n",
       "      <td>CS173873</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3805</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85968639</td>\n",
       "      <td>CS032064</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3806 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chr10  100988295   CM114899   C  T\n",
       "0     chr10  100988415   CM164756   A  T\n",
       "1     chr10  100988457   CM127719   C  T\n",
       "2     chr10  100988526  CM1610318   A  G\n",
       "3     chr10  100988540   CD169898  CT  C\n",
       "...     ...        ...        ...  .. ..\n",
       "3802   chrX   85964053  CS1810957   C  G\n",
       "3803   chrX   85964054  CS1723659   T  C\n",
       "3804   chrX   85965588   CS173873   T  C\n",
       "3805   chrX   85968639   CS032064   A  T\n",
       "\n",
       "[3806 rows x 5 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_for_annovar = read_vcf('hgmd_noncoding_for_cadd_noheader_for_annovar.vcf')\n",
    "hgmd_noncoding_for_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988295</td>\n",
       "      <td>CM114899</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988415</td>\n",
       "      <td>CM164756</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988457</td>\n",
       "      <td>CM127719</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988526</td>\n",
       "      <td>CM1610318</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3803</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964053</td>\n",
       "      <td>CS1810957</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3804</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964054</td>\n",
       "      <td>CS1723659</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3805</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85965588</td>\n",
       "      <td>CS173873</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3806</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85968639</td>\n",
       "      <td>CS032064</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3807 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                \n",
       "0     chr10  100988295   CM114899  C  T  .  .  .\n",
       "1     chr10  100988415   CM164756  A  T  .  .  .\n",
       "2     chr10  100988457   CM127719  C  T  .  .  .\n",
       "3     chr10  100988526  CM1610318  A  G  .  .  .\n",
       "...     ...        ...        ... .. .. .. .. ..\n",
       "3803   chrX   85964053  CS1810957  C  G  .  .  .\n",
       "3804   chrX   85964054  CS1723659  T  C  .  .  .\n",
       "3805   chrX   85965588   CS173873  T  C  .  .  .\n",
       "3806   chrX   85968639   CS032064  A  T  .  .  .\n",
       "\n",
       "[3807 rows x 8 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_for_annovar['..'] = '.'\n",
    "hgmd_noncoding_for_annovar['...'] = '.'\n",
    "hgmd_noncoding_for_annovar['....'] = '.'\n",
    "hgmd_noncoding_for_annovar.loc[-1] = ['chr10', '100988295', 'CM114899', 'C', 'T', '.', '.', '.']\n",
    "hgmd_noncoding_for_annovar.index = hgmd_noncoding_for_annovar.index + 1  # shifting index\n",
    "hgmd_noncoding_for_annovar = hgmd_noncoding_for_annovar.sort_index()\n",
    "hgmd_noncoding_for_annovar.columns = ['', '', '', '', '', '', '', '']\n",
    "hgmd_noncoding_for_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "output_VCF = \"hgmd_noncoding_for_annovar.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "hgmd_noncoding_for_annovar.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for hgmd noncoding data from ANNOVAR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>Func.knownGene</th>\n",
       "      <th>Gene.knownGene</th>\n",
       "      <th>GeneDetail.knownGene</th>\n",
       "      <th>ExonicFunc.knownGene</th>\n",
       "      <th>AAChange.knownGene</th>\n",
       "      <th>Func.ensGene</th>\n",
       "      <th>Gene.ensGene</th>\n",
       "      <th>GeneDetail.ensGene</th>\n",
       "      <th>ExonicFunc.ensGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>SIFT_pred</th>\n",
       "      <th>Polyphen2_HDIV_score</th>\n",
       "      <th>Polyphen2_HDIV_pred</th>\n",
       "      <th>Polyphen2_HVAR_score</th>\n",
       "      <th>Polyphen2_HVAR_pred</th>\n",
       "      <th>LRT_score</th>\n",
       "      <th>LRT_pred</th>\n",
       "      <th>MutationTaster_score</th>\n",
       "      <th>MutationTaster_pred</th>\n",
       "      <th>MutationAssessor_score</th>\n",
       "      <th>MutationAssessor_pred</th>\n",
       "      <th>FATHMM_score</th>\n",
       "      <th>FATHMM_pred</th>\n",
       "      <th>RadialSVM_score</th>\n",
       "      <th>RadialSVM_pred</th>\n",
       "      <th>LR_score</th>\n",
       "      <th>LR_pred</th>\n",
       "      <th>VEST3_score</th>\n",
       "      <th>CADD_raw</th>\n",
       "      <th>CADD_phred</th>\n",
       "      <th>GERP++_RS</th>\n",
       "      <th>phyloP46way_placental</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>Interpro_domain</th>\n",
       "      <th>SIFT_score.1</th>\n",
       "      <th>SIFT_converted_rankscore</th>\n",
       "      <th>SIFT_pred.1</th>\n",
       "      <th>Polyphen2_HDIV_score.1</th>\n",
       "      <th>Polyphen2_HDIV_rankscore</th>\n",
       "      <th>Polyphen2_HDIV_pred.1</th>\n",
       "      <th>Polyphen2_HVAR_score.1</th>\n",
       "      <th>Polyphen2_HVAR_rankscore</th>\n",
       "      <th>Polyphen2_HVAR_pred.1</th>\n",
       "      <th>LRT_score.1</th>\n",
       "      <th>LRT_converted_rankscore</th>\n",
       "      <th>LRT_pred.1</th>\n",
       "      <th>MutationTaster_score.1</th>\n",
       "      <th>MutationTaster_converted_rankscore</th>\n",
       "      <th>MutationTaster_pred.1</th>\n",
       "      <th>MutationAssessor_score.1</th>\n",
       "      <th>MutationAssessor_score_rankscore</th>\n",
       "      <th>MutationAssessor_pred.1</th>\n",
       "      <th>FATHMM_score.1</th>\n",
       "      <th>FATHMM_converted_rankscore</th>\n",
       "      <th>FATHMM_pred.1</th>\n",
       "      <th>PROVEAN_score</th>\n",
       "      <th>PROVEAN_converted_rankscore</th>\n",
       "      <th>PROVEAN_pred</th>\n",
       "      <th>VEST3_score.1</th>\n",
       "      <th>VEST3_rankscore</th>\n",
       "      <th>MetaSVM_score</th>\n",
       "      <th>MetaSVM_rankscore</th>\n",
       "      <th>MetaSVM_pred</th>\n",
       "      <th>MetaLR_score</th>\n",
       "      <th>MetaLR_rankscore</th>\n",
       "      <th>MetaLR_pred</th>\n",
       "      <th>M-CAP_score</th>\n",
       "      <th>M-CAP_rankscore</th>\n",
       "      <th>M-CAP_pred</th>\n",
       "      <th>CADD_raw.1</th>\n",
       "      <th>CADD_raw_rankscore</th>\n",
       "      <th>CADD_phred.1</th>\n",
       "      <th>DANN_score</th>\n",
       "      <th>DANN_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_score</th>\n",
       "      <th>fathmm-MKL_coding_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_pred</th>\n",
       "      <th>Eigen_coding_or_noncoding</th>\n",
       "      <th>Eigen-raw</th>\n",
       "      <th>Eigen-PC-raw</th>\n",
       "      <th>GenoCanyon_score</th>\n",
       "      <th>GenoCanyon_score_rankscore</th>\n",
       "      <th>integrated_fitCons_score</th>\n",
       "      <th>integrated_fitCons_score_rankscore</th>\n",
       "      <th>integrated_confidence_value</th>\n",
       "      <th>GERP++_RS.1</th>\n",
       "      <th>GERP++_RS_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate.1</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP20way_mammalian</th>\n",
       "      <th>phyloP20way_mammalian_rankscore</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons20way_mammalian</th>\n",
       "      <th>phastCons20way_mammalian_rankscore</th>\n",
       "      <th>SiPhy_29way_logOdds.1</th>\n",
       "      <th>SiPhy_29way_logOdds_rankscore</th>\n",
       "      <th>Interpro_domain.1</th>\n",
       "      <th>GTEx_V6_gene</th>\n",
       "      <th>GTEx_V6_tissue</th>\n",
       "      <th>dbscSNV_ADA_SCORE</th>\n",
       "      <th>dbscSNV_RF_SCORE</th>\n",
       "      <th>InterVar_automated</th>\n",
       "      <th>PVS1</th>\n",
       "      <th>PS1</th>\n",
       "      <th>PS2</th>\n",
       "      <th>PS3</th>\n",
       "      <th>PS4</th>\n",
       "      <th>PM1</th>\n",
       "      <th>PM2</th>\n",
       "      <th>PM3</th>\n",
       "      <th>PM4</th>\n",
       "      <th>PM5</th>\n",
       "      <th>PM6</th>\n",
       "      <th>PP1</th>\n",
       "      <th>PP2</th>\n",
       "      <th>PP3</th>\n",
       "      <th>PP4</th>\n",
       "      <th>PP5</th>\n",
       "      <th>BA1</th>\n",
       "      <th>BS1</th>\n",
       "      <th>BS2</th>\n",
       "      <th>BS3</th>\n",
       "      <th>BS4</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>BP3</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988295</td>\n",
       "      <td>100988295</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.C85T:p.R29X,TWNK:NM_...</td>\n",
       "      <td>exonic</td>\n",
       "      <td>C10orf2</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>C10orf2:uc001ksf.3:exon1:c.C85T:p.R29X,C10orf2...</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>TWNK:ENST00000311916.6:exon1:c.C85T:p.R29X,TWN...</td>\n",
       "      <td>0.76</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.018</td>\n",
       "      <td>N</td>\n",
       "      <td>1.000</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.804</td>\n",
       "      <td>15.34</td>\n",
       "      <td>-0.054</td>\n",
       "      <td>-0.047</td>\n",
       "      <td>-0.078</td>\n",
       "      <td>6.492</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.018</td>\n",
       "      <td>0.276</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>0.810</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>6.409</td>\n",
       "      <td>0.885</td>\n",
       "      <td>29.6</td>\n",
       "      <td>0.993</td>\n",
       "      <td>0.591</td>\n",
       "      <td>0.191</td>\n",
       "      <td>0.203</td>\n",
       "      <td>N</td>\n",
       "      <td>c</td>\n",
       "      <td>0.086</td>\n",
       "      <td>-0.217</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.454</td>\n",
       "      <td>0.543</td>\n",
       "      <td>0.218</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.054</td>\n",
       "      <td>0.131</td>\n",
       "      <td>0.224</td>\n",
       "      <td>0.174</td>\n",
       "      <td>-0.142</td>\n",
       "      <td>0.109</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.137</td>\n",
       "      <td>0.932</td>\n",
       "      <td>0.405</td>\n",
       "      <td>6.492</td>\n",
       "      <td>0.212</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988415</td>\n",
       "      <td>100988415</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.A205T:p.I69F,TWNK:NM...</td>\n",
       "      <td>exonic</td>\n",
       "      <td>C10orf2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>C10orf2:uc001ksf.3:exon1:c.A205T:p.I69F,C10orf...</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:ENST00000311916.6:exon1:c.A205T:p.I69F,TW...</td>\n",
       "      <td>0.25</td>\n",
       "      <td>T</td>\n",
       "      <td>0.996</td>\n",
       "      <td>D</td>\n",
       "      <td>0.9</td>\n",
       "      <td>P</td>\n",
       "      <td>0.000</td>\n",
       "      <td>D</td>\n",
       "      <td>0.991</td>\n",
       "      <td>D</td>\n",
       "      <td>1.845</td>\n",
       "      <td>L</td>\n",
       "      <td>-3.37</td>\n",
       "      <td>D</td>\n",
       "      <td>0.702</td>\n",
       "      <td>D</td>\n",
       "      <td>0.841</td>\n",
       "      <td>D</td>\n",
       "      <td>0.551</td>\n",
       "      <td>2.870</td>\n",
       "      <td>15.56</td>\n",
       "      <td>5.5</td>\n",
       "      <td>2.089</td>\n",
       "      <td>1.505</td>\n",
       "      <td>9.028</td>\n",
       "      <td>.</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.241</td>\n",
       "      <td>T</td>\n",
       "      <td>0.996</td>\n",
       "      <td>0.670</td>\n",
       "      <td>D</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.621</td>\n",
       "      <td>P</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.504</td>\n",
       "      <td>D</td>\n",
       "      <td>0.971</td>\n",
       "      <td>0.414</td>\n",
       "      <td>D</td>\n",
       "      <td>2.28</td>\n",
       "      <td>0.651</td>\n",
       "      <td>M</td>\n",
       "      <td>-3.37</td>\n",
       "      <td>0.953</td>\n",
       "      <td>D</td>\n",
       "      <td>-1.84</td>\n",
       "      <td>0.431</td>\n",
       "      <td>N</td>\n",
       "      <td>0.551</td>\n",
       "      <td>0.587</td>\n",
       "      <td>0.702</td>\n",
       "      <td>0.932</td>\n",
       "      <td>D</td>\n",
       "      <td>0.841</td>\n",
       "      <td>0.947</td>\n",
       "      <td>D</td>\n",
       "      <td>0.093</td>\n",
       "      <td>0.760</td>\n",
       "      <td>D</td>\n",
       "      <td>1.435</td>\n",
       "      <td>0.235</td>\n",
       "      <td>12.97</td>\n",
       "      <td>0.978</td>\n",
       "      <td>0.358</td>\n",
       "      <td>0.907</td>\n",
       "      <td>0.518</td>\n",
       "      <td>D</td>\n",
       "      <td>c</td>\n",
       "      <td>0.407</td>\n",
       "      <td>0.402</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.747</td>\n",
       "      <td>0.628</td>\n",
       "      <td>0.401</td>\n",
       "      <td>0</td>\n",
       "      <td>5.5</td>\n",
       "      <td>0.813</td>\n",
       "      <td>1.493</td>\n",
       "      <td>0.350</td>\n",
       "      <td>1.076</td>\n",
       "      <td>0.850</td>\n",
       "      <td>0.953</td>\n",
       "      <td>0.330</td>\n",
       "      <td>0.990</td>\n",
       "      <td>0.544</td>\n",
       "      <td>9.028</td>\n",
       "      <td>0.352</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0927775052736</td>\n",
       "      <td>0.592</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988457</td>\n",
       "      <td>100988457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.C247T:p.P83S,TWNK:NM...</td>\n",
       "      <td>exonic</td>\n",
       "      <td>C10orf2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>C10orf2:uc001ksf.3:exon1:c.C247T:p.P83S,C10orf...</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:ENST00000311916.6:exon1:c.C247T:p.P83S,TW...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>D</td>\n",
       "      <td>0.985</td>\n",
       "      <td>D</td>\n",
       "      <td>0.827</td>\n",
       "      <td>P</td>\n",
       "      <td>0.000</td>\n",
       "      <td>D</td>\n",
       "      <td>0.883</td>\n",
       "      <td>D</td>\n",
       "      <td>1.845</td>\n",
       "      <td>L</td>\n",
       "      <td>-3.33</td>\n",
       "      <td>D</td>\n",
       "      <td>1.027</td>\n",
       "      <td>D</td>\n",
       "      <td>0.868</td>\n",
       "      <td>D</td>\n",
       "      <td>0.695</td>\n",
       "      <td>1.917</td>\n",
       "      <td>12.37</td>\n",
       "      <td>5.59</td>\n",
       "      <td>2.631</td>\n",
       "      <td>2.411</td>\n",
       "      <td>18.175</td>\n",
       "      <td>.</td>\n",
       "      <td>0.208</td>\n",
       "      <td>0.416</td>\n",
       "      <td>T</td>\n",
       "      <td>0.985</td>\n",
       "      <td>0.592</td>\n",
       "      <td>D</td>\n",
       "      <td>0.827</td>\n",
       "      <td>0.576</td>\n",
       "      <td>P</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.504</td>\n",
       "      <td>D</td>\n",
       "      <td>0.685</td>\n",
       "      <td>0.358</td>\n",
       "      <td>D</td>\n",
       "      <td>2.28</td>\n",
       "      <td>0.651</td>\n",
       "      <td>M</td>\n",
       "      <td>-3.33</td>\n",
       "      <td>0.951</td>\n",
       "      <td>D</td>\n",
       "      <td>-1.84</td>\n",
       "      <td>0.431</td>\n",
       "      <td>N</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.700</td>\n",
       "      <td>1.027</td>\n",
       "      <td>0.977</td>\n",
       "      <td>D</td>\n",
       "      <td>0.868</td>\n",
       "      <td>0.956</td>\n",
       "      <td>D</td>\n",
       "      <td>0.382</td>\n",
       "      <td>0.930</td>\n",
       "      <td>D</td>\n",
       "      <td>3.823</td>\n",
       "      <td>0.516</td>\n",
       "      <td>23.4</td>\n",
       "      <td>0.997</td>\n",
       "      <td>0.827</td>\n",
       "      <td>0.914</td>\n",
       "      <td>0.533</td>\n",
       "      <td>D</td>\n",
       "      <td>c</td>\n",
       "      <td>0.417</td>\n",
       "      <td>0.387</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.747</td>\n",
       "      <td>0.628</td>\n",
       "      <td>0.401</td>\n",
       "      <td>0</td>\n",
       "      <td>5.59</td>\n",
       "      <td>0.846</td>\n",
       "      <td>2.453</td>\n",
       "      <td>0.444</td>\n",
       "      <td>0.843</td>\n",
       "      <td>0.344</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.715</td>\n",
       "      <td>0.386</td>\n",
       "      <td>0.252</td>\n",
       "      <td>18.175</td>\n",
       "      <td>0.896</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.381929214361</td>\n",
       "      <td>0.584</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988526</td>\n",
       "      <td>100988526</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.A316G:p.K106E,TWNK:N...</td>\n",
       "      <td>exonic</td>\n",
       "      <td>C10orf2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>C10orf2:uc001ksf.3:exon1:c.A316G:p.K106E,C10or...</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:ENST00000311916.6:exon1:c.A316G:p.K106E,T...</td>\n",
       "      <td>0.28</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>D</td>\n",
       "      <td>2.005</td>\n",
       "      <td>M</td>\n",
       "      <td>-3.86</td>\n",
       "      <td>D</td>\n",
       "      <td>0.962</td>\n",
       "      <td>D</td>\n",
       "      <td>0.916</td>\n",
       "      <td>D</td>\n",
       "      <td>0.923</td>\n",
       "      <td>4.280</td>\n",
       "      <td>22.4</td>\n",
       "      <td>5.51</td>\n",
       "      <td>2.096</td>\n",
       "      <td>9.119</td>\n",
       "      <td>14.462</td>\n",
       "      <td>.</td>\n",
       "      <td>0.008</td>\n",
       "      <td>0.586</td>\n",
       "      <td>D</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.899</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>0.850</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.843</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.516</td>\n",
       "      <td>D</td>\n",
       "      <td>2.395</td>\n",
       "      <td>0.694</td>\n",
       "      <td>M</td>\n",
       "      <td>-3.86</td>\n",
       "      <td>0.967</td>\n",
       "      <td>D</td>\n",
       "      <td>-2.23</td>\n",
       "      <td>0.506</td>\n",
       "      <td>N</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.915</td>\n",
       "      <td>0.962</td>\n",
       "      <td>0.966</td>\n",
       "      <td>D</td>\n",
       "      <td>0.916</td>\n",
       "      <td>0.972</td>\n",
       "      <td>D</td>\n",
       "      <td>0.208</td>\n",
       "      <td>0.872</td>\n",
       "      <td>D</td>\n",
       "      <td>3.906</td>\n",
       "      <td>0.526</td>\n",
       "      <td>23.5</td>\n",
       "      <td>0.999</td>\n",
       "      <td>0.945</td>\n",
       "      <td>0.989</td>\n",
       "      <td>0.884</td>\n",
       "      <td>D</td>\n",
       "      <td>c</td>\n",
       "      <td>0.718</td>\n",
       "      <td>0.714</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.747</td>\n",
       "      <td>0.628</td>\n",
       "      <td>0.401</td>\n",
       "      <td>0</td>\n",
       "      <td>5.51</td>\n",
       "      <td>0.817</td>\n",
       "      <td>9.107</td>\n",
       "      <td>0.936</td>\n",
       "      <td>1.088</td>\n",
       "      <td>0.866</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.715</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.888</td>\n",
       "      <td>14.462</td>\n",
       "      <td>0.669</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.207740715004</td>\n",
       "      <td>0.787</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3803</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964053</td>\n",
       "      <td>85964053</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>splicing</td>\n",
       "      <td>CHM</td>\n",
       "      <td>NM_000390:exon5:c.315-1G&gt;C;NM_001320959:exon5:...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>splicing</td>\n",
       "      <td>CHM</td>\n",
       "      <td>uc004eet.3:exon5:c.315-1G&gt;C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>splicing</td>\n",
       "      <td>CHM</td>\n",
       "      <td>ENST00000357749.6:exon5:c.315-1G&gt;C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.000</td>\n",
       "      <td>D</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.050</td>\n",
       "      <td>12.81</td>\n",
       "      <td>4.64</td>\n",
       "      <td>2.028</td>\n",
       "      <td>3.381</td>\n",
       "      <td>12.833</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>0.810</td>\n",
       "      <td>D</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.876</td>\n",
       "      <td>0.523</td>\n",
       "      <td>23.5</td>\n",
       "      <td>0.988</td>\n",
       "      <td>0.459</td>\n",
       "      <td>0.805</td>\n",
       "      <td>0.399</td>\n",
       "      <td>D</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.747</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.64</td>\n",
       "      <td>0.572</td>\n",
       "      <td>3.419</td>\n",
       "      <td>0.522</td>\n",
       "      <td>0.816</td>\n",
       "      <td>0.334</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.715</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.888</td>\n",
       "      <td>12.833</td>\n",
       "      <td>0.570</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.952</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.019730941704</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3804</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964054</td>\n",
       "      <td>85964054</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>splicing</td>\n",
       "      <td>CHM</td>\n",
       "      <td>NM_000390:exon5:c.315-2A&gt;G;NM_001320959:exon5:...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>splicing</td>\n",
       "      <td>CHM</td>\n",
       "      <td>uc004eet.3:exon5:c.315-2A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>splicing</td>\n",
       "      <td>CHM</td>\n",
       "      <td>ENST00000357749.6:exon5:c.315-2A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.000</td>\n",
       "      <td>D</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>2.494</td>\n",
       "      <td>14.30</td>\n",
       "      <td>4.64</td>\n",
       "      <td>1.627</td>\n",
       "      <td>3.454</td>\n",
       "      <td>13.494</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>0.810</td>\n",
       "      <td>D</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>3.595</td>\n",
       "      <td>0.489</td>\n",
       "      <td>23.2</td>\n",
       "      <td>0.989</td>\n",
       "      <td>0.470</td>\n",
       "      <td>0.828</td>\n",
       "      <td>0.417</td>\n",
       "      <td>D</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.747</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4.64</td>\n",
       "      <td>0.572</td>\n",
       "      <td>3.435</td>\n",
       "      <td>0.524</td>\n",
       "      <td>0.924</td>\n",
       "      <td>0.433</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.715</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.888</td>\n",
       "      <td>13.494</td>\n",
       "      <td>0.607</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>0.958</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0206278026906</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3805</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85965588</td>\n",
       "      <td>85965588</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3806</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85968639</td>\n",
       "      <td>85968639</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3807 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0     chr10  100988295  100988295   C   T       exonic         TWNK   \n",
       "1     chr10  100988415  100988415   A   T       exonic         TWNK   \n",
       "2     chr10  100988457  100988457   C   T       exonic         TWNK   \n",
       "3     chr10  100988526  100988526   A   G       exonic         TWNK   \n",
       "...     ...        ...        ...  ..  ..          ...          ...   \n",
       "3803   chrX   85964053   85964053   C   G     splicing          CHM   \n",
       "3804   chrX   85964054   85964054   T   C     splicing          CHM   \n",
       "3805   chrX   85965588   85965588   T   C     intronic          CHM   \n",
       "3806   chrX   85968639   85968639   A   T     intronic          CHM   \n",
       "\n",
       "                                     GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                                     .           stopgain   \n",
       "1                                                     .  nonsynonymous SNV   \n",
       "2                                                     .  nonsynonymous SNV   \n",
       "3                                                     .  nonsynonymous SNV   \n",
       "...                                                 ...                ...   \n",
       "3803  NM_000390:exon5:c.315-1G>C;NM_001320959:exon5:...                  .   \n",
       "3804  NM_000390:exon5:c.315-2A>G;NM_001320959:exon5:...                  .   \n",
       "3805                                                  .                  .   \n",
       "3806                                                  .                  .   \n",
       "\n",
       "                                       AAChange.refGene Func.knownGene  \\\n",
       "0     TWNK:NM_001163812:exon1:c.C85T:p.R29X,TWNK:NM_...         exonic   \n",
       "1     TWNK:NM_001163812:exon1:c.A205T:p.I69F,TWNK:NM...         exonic   \n",
       "2     TWNK:NM_001163812:exon1:c.C247T:p.P83S,TWNK:NM...         exonic   \n",
       "3     TWNK:NM_001163812:exon1:c.A316G:p.K106E,TWNK:N...         exonic   \n",
       "...                                                 ...            ...   \n",
       "3803                                                  .       splicing   \n",
       "3804                                                  .       splicing   \n",
       "3805                                                  .       intronic   \n",
       "3806                                                  .       intronic   \n",
       "\n",
       "     Gene.knownGene         GeneDetail.knownGene ExonicFunc.knownGene  \\\n",
       "0           C10orf2                            .             stopgain   \n",
       "1           C10orf2                            .    nonsynonymous SNV   \n",
       "2           C10orf2                            .    nonsynonymous SNV   \n",
       "3           C10orf2                            .    nonsynonymous SNV   \n",
       "...             ...                          ...                  ...   \n",
       "3803            CHM  uc004eet.3:exon5:c.315-1G>C                    .   \n",
       "3804            CHM  uc004eet.3:exon5:c.315-2A>G                    .   \n",
       "3805            CHM                            .                    .   \n",
       "3806            CHM                            .                    .   \n",
       "\n",
       "                                     AAChange.knownGene Func.ensGene  \\\n",
       "0     C10orf2:uc001ksf.3:exon1:c.C85T:p.R29X,C10orf2...       exonic   \n",
       "1     C10orf2:uc001ksf.3:exon1:c.A205T:p.I69F,C10orf...       exonic   \n",
       "2     C10orf2:uc001ksf.3:exon1:c.C247T:p.P83S,C10orf...       exonic   \n",
       "3     C10orf2:uc001ksf.3:exon1:c.A316G:p.K106E,C10or...       exonic   \n",
       "...                                                 ...          ...   \n",
       "3803                                                  .     splicing   \n",
       "3804                                                  .     splicing   \n",
       "3805                                                  .     intronic   \n",
       "3806                                                  .     intronic   \n",
       "\n",
       "     Gene.ensGene                  GeneDetail.ensGene ExonicFunc.ensGene  \\\n",
       "0            TWNK                                   .           stopgain   \n",
       "1            TWNK                                   .  nonsynonymous SNV   \n",
       "2            TWNK                                   .  nonsynonymous SNV   \n",
       "3            TWNK                                   .  nonsynonymous SNV   \n",
       "...           ...                                 ...                ...   \n",
       "3803          CHM  ENST00000357749.6:exon5:c.315-1G>C                  .   \n",
       "3804          CHM  ENST00000357749.6:exon5:c.315-2A>G                  .   \n",
       "3805          CHM                                   .                  .   \n",
       "3806          CHM                                   .                  .   \n",
       "\n",
       "                                       AAChange.ensGene SIFT_score SIFT_pred  \\\n",
       "0     TWNK:ENST00000311916.6:exon1:c.C85T:p.R29X,TWN...       0.76         T   \n",
       "1     TWNK:ENST00000311916.6:exon1:c.A205T:p.I69F,TW...       0.25         T   \n",
       "2     TWNK:ENST00000311916.6:exon1:c.C247T:p.P83S,TW...       0.01         D   \n",
       "3     TWNK:ENST00000311916.6:exon1:c.A316G:p.K106E,T...       0.28         T   \n",
       "...                                                 ...        ...       ...   \n",
       "3803                                                  .          .         .   \n",
       "3804                                                  .          .         .   \n",
       "3805                                                  .          .         .   \n",
       "3806                                                  .          .         .   \n",
       "\n",
       "     Polyphen2_HDIV_score Polyphen2_HDIV_pred Polyphen2_HVAR_score  \\\n",
       "0                       .                   .                    .   \n",
       "1                   0.996                   D                  0.9   \n",
       "2                   0.985                   D                0.827   \n",
       "3                     1.0                   D                0.997   \n",
       "...                   ...                 ...                  ...   \n",
       "3803                    .                   .                    .   \n",
       "3804                    .                   .                    .   \n",
       "3805                    .                   .                    .   \n",
       "3806                    .                   .                    .   \n",
       "\n",
       "     Polyphen2_HVAR_pred LRT_score LRT_pred MutationTaster_score  \\\n",
       "0                      .     0.018        N                1.000   \n",
       "1                      P     0.000        D                0.991   \n",
       "2                      P     0.000        D                0.883   \n",
       "3                      D     0.000        D                1.000   \n",
       "...                  ...       ...      ...                  ...   \n",
       "3803                   .         .        .                1.000   \n",
       "3804                   .         .        .                1.000   \n",
       "3805                   .         .        .                    .   \n",
       "3806                   .         .        .                    .   \n",
       "\n",
       "     MutationTaster_pred MutationAssessor_score MutationAssessor_pred  \\\n",
       "0                      A                      .                     .   \n",
       "1                      D                  1.845                     L   \n",
       "2                      D                  1.845                     L   \n",
       "3                      D                  2.005                     M   \n",
       "...                  ...                    ...                   ...   \n",
       "3803                   D                      .                     .   \n",
       "3804                   D                      .                     .   \n",
       "3805                   .                      .                     .   \n",
       "3806                   .                      .                     .   \n",
       "\n",
       "     FATHMM_score FATHMM_pred RadialSVM_score RadialSVM_pred LR_score LR_pred  \\\n",
       "0               .           .               .              .        .       .   \n",
       "1           -3.37           D           0.702              D    0.841       D   \n",
       "2           -3.33           D           1.027              D    0.868       D   \n",
       "3           -3.86           D           0.962              D    0.916       D   \n",
       "...           ...         ...             ...            ...      ...     ...   \n",
       "3803            .           .               .              .        .       .   \n",
       "3804            .           .               .              .        .       .   \n",
       "3805            .           .               .              .        .       .   \n",
       "3806            .           .               .              .        .       .   \n",
       "\n",
       "     VEST3_score CADD_raw CADD_phred GERP++_RS phyloP46way_placental  \\\n",
       "0              .    2.804      15.34    -0.054                -0.047   \n",
       "1          0.551    2.870      15.56       5.5                 2.089   \n",
       "2          0.695    1.917      12.37      5.59                 2.631   \n",
       "3          0.923    4.280       22.4      5.51                 2.096   \n",
       "...          ...      ...        ...       ...                   ...   \n",
       "3803           .    2.050      12.81      4.64                 2.028   \n",
       "3804           .    2.494      14.30      4.64                 1.627   \n",
       "3805           .        .          .         .                     .   \n",
       "3806           .        .          .         .                     .   \n",
       "\n",
       "     phyloP100way_vertebrate SiPhy_29way_logOdds Interpro_domain SIFT_score.1  \\\n",
       "0                     -0.078               6.492               .            .   \n",
       "1                      1.505               9.028               .         0.19   \n",
       "2                      2.411              18.175               .        0.208   \n",
       "3                      9.119              14.462               .        0.008   \n",
       "...                      ...                 ...             ...          ...   \n",
       "3803                   3.381              12.833               .            .   \n",
       "3804                   3.454              13.494               .            .   \n",
       "3805                       .                   .               .            .   \n",
       "3806                       .                   .               .            .   \n",
       "\n",
       "     SIFT_converted_rankscore SIFT_pred.1 Polyphen2_HDIV_score.1  \\\n",
       "0                           .           .                      .   \n",
       "1                       0.241           T                  0.996   \n",
       "2                       0.416           T                  0.985   \n",
       "3                       0.586           D                    1.0   \n",
       "...                       ...         ...                    ...   \n",
       "3803                        .           .                      .   \n",
       "3804                        .           .                      .   \n",
       "3805                        .           .                      .   \n",
       "3806                        .           .                      .   \n",
       "\n",
       "     Polyphen2_HDIV_rankscore Polyphen2_HDIV_pred.1 Polyphen2_HVAR_score.1  \\\n",
       "0                           .                     .                      .   \n",
       "1                       0.670                     D                    0.9   \n",
       "2                       0.592                     D                  0.827   \n",
       "3                       0.899                     D                  0.997   \n",
       "...                       ...                   ...                    ...   \n",
       "3803                        .                     .                      .   \n",
       "3804                        .                     .                      .   \n",
       "3805                        .                     .                      .   \n",
       "3806                        .                     .                      .   \n",
       "\n",
       "     Polyphen2_HVAR_rankscore Polyphen2_HVAR_pred.1 LRT_score.1  \\\n",
       "0                           .                     .       0.018   \n",
       "1                       0.621                     P       0.000   \n",
       "2                       0.576                     P       0.000   \n",
       "3                       0.850                     D       0.000   \n",
       "...                       ...                   ...         ...   \n",
       "3803                        .                     .           .   \n",
       "3804                        .                     .           .   \n",
       "3805                        .                     .           .   \n",
       "3806                        .                     .           .   \n",
       "\n",
       "     LRT_converted_rankscore LRT_pred.1 MutationTaster_score.1  \\\n",
       "0                      0.276          N                      1   \n",
       "1                      0.504          D                  0.971   \n",
       "2                      0.504          D                  0.685   \n",
       "3                      0.843          D                  1.000   \n",
       "...                      ...        ...                    ...   \n",
       "3803                       .          .                      1   \n",
       "3804                       .          .                      1   \n",
       "3805                       .          .                      .   \n",
       "3806                       .          .                      .   \n",
       "\n",
       "     MutationTaster_converted_rankscore MutationTaster_pred.1  \\\n",
       "0                                 0.810                     A   \n",
       "1                                 0.414                     D   \n",
       "2                                 0.358                     D   \n",
       "3                                 0.516                     D   \n",
       "...                                 ...                   ...   \n",
       "3803                              0.810                     D   \n",
       "3804                              0.810                     D   \n",
       "3805                                  .                     .   \n",
       "3806                                  .                     .   \n",
       "\n",
       "     MutationAssessor_score.1 MutationAssessor_score_rankscore  \\\n",
       "0                           .                                .   \n",
       "1                        2.28                            0.651   \n",
       "2                        2.28                            0.651   \n",
       "3                       2.395                            0.694   \n",
       "...                       ...                              ...   \n",
       "3803                        .                                .   \n",
       "3804                        .                                .   \n",
       "3805                        .                                .   \n",
       "3806                        .                                .   \n",
       "\n",
       "     MutationAssessor_pred.1 FATHMM_score.1 FATHMM_converted_rankscore  \\\n",
       "0                          .              .                          .   \n",
       "1                          M          -3.37                      0.953   \n",
       "2                          M          -3.33                      0.951   \n",
       "3                          M          -3.86                      0.967   \n",
       "...                      ...            ...                        ...   \n",
       "3803                       .              .                          .   \n",
       "3804                       .              .                          .   \n",
       "3805                       .              .                          .   \n",
       "3806                       .              .                          .   \n",
       "\n",
       "     FATHMM_pred.1 PROVEAN_score PROVEAN_converted_rankscore PROVEAN_pred  \\\n",
       "0                .             .                           .            .   \n",
       "1                D         -1.84                       0.431            N   \n",
       "2                D         -1.84                       0.431            N   \n",
       "3                D         -2.23                       0.506            N   \n",
       "...            ...           ...                         ...          ...   \n",
       "3803             .             .                           .            .   \n",
       "3804             .             .                           .            .   \n",
       "3805             .             .                           .            .   \n",
       "3806             .             .                           .            .   \n",
       "\n",
       "     VEST3_score.1 VEST3_rankscore MetaSVM_score MetaSVM_rankscore  \\\n",
       "0                .               .             .                 .   \n",
       "1            0.551           0.587         0.702             0.932   \n",
       "2             0.68           0.700         1.027             0.977   \n",
       "3             0.92           0.915         0.962             0.966   \n",
       "...            ...             ...           ...               ...   \n",
       "3803             .               .             .                 .   \n",
       "3804             .               .             .                 .   \n",
       "3805             .               .             .                 .   \n",
       "3806             .               .             .                 .   \n",
       "\n",
       "     MetaSVM_pred MetaLR_score MetaLR_rankscore MetaLR_pred M-CAP_score  \\\n",
       "0               .            .                .           .           .   \n",
       "1               D        0.841            0.947           D       0.093   \n",
       "2               D        0.868            0.956           D       0.382   \n",
       "3               D        0.916            0.972           D       0.208   \n",
       "...           ...          ...              ...         ...         ...   \n",
       "3803            .            .                .           .           .   \n",
       "3804            .            .                .           .           .   \n",
       "3805            .            .                .           .           .   \n",
       "3806            .            .                .           .           .   \n",
       "\n",
       "     M-CAP_rankscore M-CAP_pred CADD_raw.1 CADD_raw_rankscore CADD_phred.1  \\\n",
       "0                  .          .      6.409              0.885         29.6   \n",
       "1              0.760          D      1.435              0.235        12.97   \n",
       "2              0.930          D      3.823              0.516         23.4   \n",
       "3              0.872          D      3.906              0.526         23.5   \n",
       "...              ...        ...        ...                ...          ...   \n",
       "3803               .          .      3.876              0.523         23.5   \n",
       "3804               .          .      3.595              0.489         23.2   \n",
       "3805               .          .          .                  .            .   \n",
       "3806               .          .          .                  .            .   \n",
       "\n",
       "     DANN_score DANN_rankscore fathmm-MKL_coding_score  \\\n",
       "0         0.993          0.591                   0.191   \n",
       "1         0.978          0.358                   0.907   \n",
       "2         0.997          0.827                   0.914   \n",
       "3         0.999          0.945                   0.989   \n",
       "...         ...            ...                     ...   \n",
       "3803      0.988          0.459                   0.805   \n",
       "3804      0.989          0.470                   0.828   \n",
       "3805          .              .                       .   \n",
       "3806          .              .                       .   \n",
       "\n",
       "     fathmm-MKL_coding_rankscore fathmm-MKL_coding_pred  \\\n",
       "0                          0.203                      N   \n",
       "1                          0.518                      D   \n",
       "2                          0.533                      D   \n",
       "3                          0.884                      D   \n",
       "...                          ...                    ...   \n",
       "3803                       0.399                      D   \n",
       "3804                       0.417                      D   \n",
       "3805                           .                      .   \n",
       "3806                           .                      .   \n",
       "\n",
       "     Eigen_coding_or_noncoding Eigen-raw Eigen-PC-raw GenoCanyon_score  \\\n",
       "0                            c     0.086       -0.217            1.000   \n",
       "1                            c     0.407        0.402            1.000   \n",
       "2                            c     0.417        0.387            1.000   \n",
       "3                            c     0.718        0.714            1.000   \n",
       "...                        ...       ...          ...              ...   \n",
       "3803                         .         .            .            1.000   \n",
       "3804                         .         .            .            1.000   \n",
       "3805                         .         .            .                .   \n",
       "3806                         .         .            .                .   \n",
       "\n",
       "     GenoCanyon_score_rankscore integrated_fitCons_score  \\\n",
       "0                         0.454                    0.543   \n",
       "1                         0.747                    0.628   \n",
       "2                         0.747                    0.628   \n",
       "3                         0.747                    0.628   \n",
       "...                         ...                      ...   \n",
       "3803                      0.747                        .   \n",
       "3804                      0.747                        .   \n",
       "3805                          .                        .   \n",
       "3806                          .                        .   \n",
       "\n",
       "     integrated_fitCons_score_rankscore integrated_confidence_value  \\\n",
       "0                                 0.218                           0   \n",
       "1                                 0.401                           0   \n",
       "2                                 0.401                           0   \n",
       "3                                 0.401                           0   \n",
       "...                                 ...                         ...   \n",
       "3803                                  .                           .   \n",
       "3804                                  .                           .   \n",
       "3805                                  .                           .   \n",
       "3806                                  .                           .   \n",
       "\n",
       "     GERP++_RS.1 GERP++_RS_rankscore phyloP100way_vertebrate.1  \\\n",
       "0         -0.054               0.131                     0.224   \n",
       "1            5.5               0.813                     1.493   \n",
       "2           5.59               0.846                     2.453   \n",
       "3           5.51               0.817                     9.107   \n",
       "...          ...                 ...                       ...   \n",
       "3803        4.64               0.572                     3.419   \n",
       "3804        4.64               0.572                     3.435   \n",
       "3805           .                   .                         .   \n",
       "3806           .                   .                         .   \n",
       "\n",
       "     phyloP100way_vertebrate_rankscore phyloP20way_mammalian  \\\n",
       "0                                0.174                -0.142   \n",
       "1                                0.350                 1.076   \n",
       "2                                0.444                 0.843   \n",
       "3                                0.936                 1.088   \n",
       "...                                ...                   ...   \n",
       "3803                             0.522                 0.816   \n",
       "3804                             0.524                 0.924   \n",
       "3805                                 .                     .   \n",
       "3806                                 .                     .   \n",
       "\n",
       "     phyloP20way_mammalian_rankscore phastCons100way_vertebrate  \\\n",
       "0                              0.109                      0.001   \n",
       "1                              0.850                      0.953   \n",
       "2                              0.344                      1.000   \n",
       "3                              0.866                      1.000   \n",
       "...                              ...                        ...   \n",
       "3803                           0.334                      1.000   \n",
       "3804                           0.433                      1.000   \n",
       "3805                               .                          .   \n",
       "3806                               .                          .   \n",
       "\n",
       "     phastCons100way_vertebrate_rankscore phastCons20way_mammalian  \\\n",
       "0                                   0.137                    0.932   \n",
       "1                                   0.330                    0.990   \n",
       "2                                   0.715                    0.386   \n",
       "3                                   0.715                    1.000   \n",
       "...                                   ...                      ...   \n",
       "3803                                0.715                    1.000   \n",
       "3804                                0.715                    1.000   \n",
       "3805                                    .                        .   \n",
       "3806                                    .                        .   \n",
       "\n",
       "     phastCons20way_mammalian_rankscore SiPhy_29way_logOdds.1  \\\n",
       "0                                 0.405                 6.492   \n",
       "1                                 0.544                 9.028   \n",
       "2                                 0.252                18.175   \n",
       "3                                 0.888                14.462   \n",
       "...                                 ...                   ...   \n",
       "3803                              0.888                12.833   \n",
       "3804                              0.888                13.494   \n",
       "3805                                  .                     .   \n",
       "3806                                  .                     .   \n",
       "\n",
       "     SiPhy_29way_logOdds_rankscore Interpro_domain.1 GTEx_V6_gene  \\\n",
       "0                            0.212                 .            .   \n",
       "1                            0.352                 .            .   \n",
       "2                            0.896                 .            .   \n",
       "3                            0.669                 .            .   \n",
       "...                            ...               ...          ...   \n",
       "3803                         0.570                 .            .   \n",
       "3804                         0.607                 .            .   \n",
       "3805                             .                 .            .   \n",
       "3806                             .                 .            .   \n",
       "\n",
       "     GTEx_V6_tissue dbscSNV_ADA_SCORE dbscSNV_RF_SCORE  \\\n",
       "0                 .                 .                .   \n",
       "1                 .                 .                .   \n",
       "2                 .                 .                .   \n",
       "3                 .                 .                .   \n",
       "...             ...               ...              ...   \n",
       "3803              .            1.0000            0.952   \n",
       "3804              .            1.0000            0.958   \n",
       "3805              .                 .                .   \n",
       "3806              .                 .                .   \n",
       "\n",
       "          InterVar_automated PVS1 PS1 PS2 PS3 PS4 PM1 PM2 PM3 PM4 PM5 PM6 PP1  \\\n",
       "0     Uncertain significance    0   0   0   0   0   0   1   0   0   0   0   0   \n",
       "1     Uncertain significance    0   0   0   0   0   0   1   0   0   0   0   0   \n",
       "2     Uncertain significance    0   0   0   0   0   0   1   0   0   0   0   0   \n",
       "3     Uncertain significance    0   0   0   0   0   0   1   0   0   0   0   0   \n",
       "...                      ...  ...  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..   \n",
       "3803                       .    .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "3804                       .    .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "3805                       .    .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "3806                       .    .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "\n",
       "     PP2 PP3 PP4 PP5 BA1 BS1 BS2 BS3 BS4 BP1 BP2 BP3 BP4 BP5 BP6 BP7 cosmic70  \\\n",
       "0      0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0        .   \n",
       "1      0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0        .   \n",
       "2      0   1   0   1   0   0   0   0   0   0   0   0   0   0   0   0        .   \n",
       "3      0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0        .   \n",
       "...   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..      ...   \n",
       "3803   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .        .   \n",
       "3804   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .        .   \n",
       "3805   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .        .   \n",
       "3806   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .        .   \n",
       "\n",
       "                 MCAP  REVEL       regsnp_fpr regsnp_disease  \\\n",
       "0                   .      .                .              .   \n",
       "1     0.0927775052736  0.592                .              .   \n",
       "2      0.381929214361  0.584                .              .   \n",
       "3      0.207740715004  0.787                .              .   \n",
       "...               ...    ...              ...            ...   \n",
       "3803                .      .   0.019730941704              D   \n",
       "3804                .      .  0.0206278026906              D   \n",
       "3805                .      .                .              .   \n",
       "3806                .      .                .              .   \n",
       "\n",
       "     regsnp_splicing_site  \n",
       "0                       .  \n",
       "1                       .  \n",
       "2                       .  \n",
       "3                       .  \n",
       "...                   ...  \n",
       "3803                   on  \n",
       "3804                   on  \n",
       "3805                    .  \n",
       "3806                    .  \n",
       "\n",
       "[3807 rows x 149 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_annotated_with_annovar = pd.read_csv('hgmd_noncoding_annotated.hg38_multianno.csv')\n",
    "hgmd_noncoding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated results for hgmd noncoding data from CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (13,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEName</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>motifEScoreChng</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>FeatureID</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>Intron</th>\n",
       "      <th>Exon</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>relcDNApos</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>relCDSpos</th>\n",
       "      <th>protPos</th>\n",
       "      <th>relProtPos</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Dst2Splice</th>\n",
       "      <th>Dst2SplType</th>\n",
       "      <th>minDistTSS</th>\n",
       "      <th>minDistTSE</th>\n",
       "      <th>SIFTcat</th>\n",
       "      <th>SIFTval</th>\n",
       "      <th>PolyPhenCat</th>\n",
       "      <th>PolyPhenVal</th>\n",
       "      <th>priPhCons</th>\n",
       "      <th>mamPhCons</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>priPhyloP</th>\n",
       "      <th>mamPhyloP</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>bStatistic</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>mirSVR-Score</th>\n",
       "      <th>mirSVR-E</th>\n",
       "      <th>mirSVR-Aln</th>\n",
       "      <th>cHmm_E1</th>\n",
       "      <th>cHmm_E2</th>\n",
       "      <th>cHmm_E3</th>\n",
       "      <th>cHmm_E4</th>\n",
       "      <th>cHmm_E5</th>\n",
       "      <th>cHmm_E6</th>\n",
       "      <th>cHmm_E7</th>\n",
       "      <th>cHmm_E8</th>\n",
       "      <th>cHmm_E9</th>\n",
       "      <th>cHmm_E10</th>\n",
       "      <th>cHmm_E11</th>\n",
       "      <th>cHmm_E12</th>\n",
       "      <th>cHmm_E13</th>\n",
       "      <th>cHmm_E14</th>\n",
       "      <th>cHmm_E15</th>\n",
       "      <th>cHmm_E16</th>\n",
       "      <th>cHmm_E17</th>\n",
       "      <th>cHmm_E18</th>\n",
       "      <th>cHmm_E19</th>\n",
       "      <th>cHmm_E20</th>\n",
       "      <th>cHmm_E21</th>\n",
       "      <th>cHmm_E22</th>\n",
       "      <th>cHmm_E23</th>\n",
       "      <th>cHmm_E24</th>\n",
       "      <th>cHmm_E25</th>\n",
       "      <th>GerpRS</th>\n",
       "      <th>GerpRSpval</th>\n",
       "      <th>GerpN</th>\n",
       "      <th>GerpS</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>motifDist</th>\n",
       "      <th>EncodeH3K4me1-sum</th>\n",
       "      <th>EncodeH3K4me1-max</th>\n",
       "      <th>EncodeH3K4me2-sum</th>\n",
       "      <th>EncodeH3K4me2-max</th>\n",
       "      <th>EncodeH3K4me3-sum</th>\n",
       "      <th>EncodeH3K4me3-max</th>\n",
       "      <th>EncodeH3K9ac-sum</th>\n",
       "      <th>EncodeH3K9ac-max</th>\n",
       "      <th>EncodeH3K9me3-sum</th>\n",
       "      <th>EncodeH3K9me3-max</th>\n",
       "      <th>EncodeH3K27ac-sum</th>\n",
       "      <th>EncodeH3K27ac-max</th>\n",
       "      <th>EncodeH3K27me3-sum</th>\n",
       "      <th>EncodeH3K27me3-max</th>\n",
       "      <th>EncodeH3K36me3-sum</th>\n",
       "      <th>EncodeH3K36me3-max</th>\n",
       "      <th>EncodeH3K79me2-sum</th>\n",
       "      <th>EncodeH3K79me2-max</th>\n",
       "      <th>EncodeH4K20me1-sum</th>\n",
       "      <th>EncodeH4K20me1-max</th>\n",
       "      <th>EncodeH2AFZ-sum</th>\n",
       "      <th>EncodeH2AFZ-max</th>\n",
       "      <th>EncodeDNase-sum</th>\n",
       "      <th>EncodeDNase-max</th>\n",
       "      <th>EncodetotalRNA-sum</th>\n",
       "      <th>EncodetotalRNA-max</th>\n",
       "      <th>Grantham</th>\n",
       "      <th>Dist2Mutation</th>\n",
       "      <th>Freq100bp</th>\n",
       "      <th>Rare100bp</th>\n",
       "      <th>Sngl100bp</th>\n",
       "      <th>Freq1000bp</th>\n",
       "      <th>Rare1000bp</th>\n",
       "      <th>Sngl1000bp</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>0.781</td>\n",
       "      <td>0.253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSR00000000851</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36</td>\n",
       "      <td>9190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.690</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.399</td>\n",
       "      <td>-0.152</td>\n",
       "      <td>-0.334</td>\n",
       "      <td>824.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.58</td>\n",
       "      <td>-9.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.18</td>\n",
       "      <td>1.11</td>\n",
       "      <td>104.49</td>\n",
       "      <td>25.01</td>\n",
       "      <td>525.02</td>\n",
       "      <td>88.14</td>\n",
       "      <td>411.25</td>\n",
       "      <td>64.60</td>\n",
       "      <td>5.76</td>\n",
       "      <td>1.06</td>\n",
       "      <td>359.53</td>\n",
       "      <td>65.41</td>\n",
       "      <td>9.83</td>\n",
       "      <td>2.28</td>\n",
       "      <td>6.51</td>\n",
       "      <td>2.64</td>\n",
       "      <td>82.68</td>\n",
       "      <td>28.05</td>\n",
       "      <td>9.52</td>\n",
       "      <td>2.49</td>\n",
       "      <td>177.88</td>\n",
       "      <td>29.14</td>\n",
       "      <td>9.10</td>\n",
       "      <td>2.57</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>229</td>\n",
       "      <td>26</td>\n",
       "      <td>48</td>\n",
       "      <td>1451</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>208.0</td>\n",
       "      <td>475.0</td>\n",
       "      <td>1.043084</td>\n",
       "      <td>12.600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>0.781</td>\n",
       "      <td>0.253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000116288</td>\n",
       "      <td>ENST00000493678</td>\n",
       "      <td>PARK7</td>\n",
       "      <td>CCDS93.1</td>\n",
       "      <td>1/6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36</td>\n",
       "      <td>9190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.690</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.399</td>\n",
       "      <td>-0.152</td>\n",
       "      <td>-0.334</td>\n",
       "      <td>824.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.58</td>\n",
       "      <td>-9.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.18</td>\n",
       "      <td>1.11</td>\n",
       "      <td>104.49</td>\n",
       "      <td>25.01</td>\n",
       "      <td>525.02</td>\n",
       "      <td>88.14</td>\n",
       "      <td>411.25</td>\n",
       "      <td>64.60</td>\n",
       "      <td>5.76</td>\n",
       "      <td>1.06</td>\n",
       "      <td>359.53</td>\n",
       "      <td>65.41</td>\n",
       "      <td>9.83</td>\n",
       "      <td>2.28</td>\n",
       "      <td>6.51</td>\n",
       "      <td>2.64</td>\n",
       "      <td>82.68</td>\n",
       "      <td>28.05</td>\n",
       "      <td>9.52</td>\n",
       "      <td>2.49</td>\n",
       "      <td>177.88</td>\n",
       "      <td>29.14</td>\n",
       "      <td>9.10</td>\n",
       "      <td>2.57</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>229</td>\n",
       "      <td>26</td>\n",
       "      <td>48</td>\n",
       "      <td>1451</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>208.0</td>\n",
       "      <td>475.0</td>\n",
       "      <td>1.043084</td>\n",
       "      <td>12.600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>0.603</td>\n",
       "      <td>0.067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSR00000346835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9823</td>\n",
       "      <td>7207</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.227</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-1.578</td>\n",
       "      <td>-0.179</td>\n",
       "      <td>-1.643</td>\n",
       "      <td>835.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.30</td>\n",
       "      <td>-30.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.93</td>\n",
       "      <td>11.08</td>\n",
       "      <td>14.49</td>\n",
       "      <td>4.51</td>\n",
       "      <td>9.86</td>\n",
       "      <td>2.26</td>\n",
       "      <td>18.01</td>\n",
       "      <td>7.50</td>\n",
       "      <td>7.67</td>\n",
       "      <td>1.73</td>\n",
       "      <td>50.36</td>\n",
       "      <td>17.98</td>\n",
       "      <td>7.69</td>\n",
       "      <td>2.64</td>\n",
       "      <td>45.86</td>\n",
       "      <td>13.42</td>\n",
       "      <td>24.37</td>\n",
       "      <td>6.03</td>\n",
       "      <td>36.66</td>\n",
       "      <td>24.95</td>\n",
       "      <td>16.91</td>\n",
       "      <td>4.03</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>142</td>\n",
       "      <td>9</td>\n",
       "      <td>74</td>\n",
       "      <td>1406</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.014</td>\n",
       "      <td>12.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-0.385227</td>\n",
       "      <td>0.135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>0.603</td>\n",
       "      <td>0.067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000171608</td>\n",
       "      <td>ENST00000377346</td>\n",
       "      <td>PIK3CD</td>\n",
       "      <td>CCDS104.1</td>\n",
       "      <td>10/23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.0</td>\n",
       "      <td>DONOR</td>\n",
       "      <td>9823</td>\n",
       "      <td>7207</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.227</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-1.578</td>\n",
       "      <td>-0.179</td>\n",
       "      <td>-1.643</td>\n",
       "      <td>835.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15.30</td>\n",
       "      <td>-30.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.93</td>\n",
       "      <td>11.08</td>\n",
       "      <td>14.49</td>\n",
       "      <td>4.51</td>\n",
       "      <td>9.86</td>\n",
       "      <td>2.26</td>\n",
       "      <td>18.01</td>\n",
       "      <td>7.50</td>\n",
       "      <td>7.67</td>\n",
       "      <td>1.73</td>\n",
       "      <td>50.36</td>\n",
       "      <td>17.98</td>\n",
       "      <td>7.69</td>\n",
       "      <td>2.64</td>\n",
       "      <td>45.86</td>\n",
       "      <td>13.42</td>\n",
       "      <td>24.37</td>\n",
       "      <td>6.03</td>\n",
       "      <td>36.66</td>\n",
       "      <td>24.95</td>\n",
       "      <td>16.91</td>\n",
       "      <td>4.03</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>142</td>\n",
       "      <td>9</td>\n",
       "      <td>74</td>\n",
       "      <td>1406</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.014</td>\n",
       "      <td>12.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-0.385227</td>\n",
       "      <td>0.135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8609</th>\n",
       "      <td>X</td>\n",
       "      <td>154965965</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>0.371</td>\n",
       "      <td>0.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000185010</td>\n",
       "      <td>ENST00000360256</td>\n",
       "      <td>F8</td>\n",
       "      <td>CCDS35457.1</td>\n",
       "      <td>9/25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-5.0</td>\n",
       "      <td>DONOR</td>\n",
       "      <td>552</td>\n",
       "      <td>193</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.878</td>\n",
       "      <td>0.999</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.382</td>\n",
       "      <td>2.468</td>\n",
       "      <td>4.940</td>\n",
       "      <td>225.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1982.59</td>\n",
       "      <td>3.442070e-106</td>\n",
       "      <td>11.70</td>\n",
       "      <td>11.70</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.50</td>\n",
       "      <td>1.08</td>\n",
       "      <td>2.25</td>\n",
       "      <td>0.75</td>\n",
       "      <td>8.21</td>\n",
       "      <td>2.26</td>\n",
       "      <td>2.94</td>\n",
       "      <td>0.77</td>\n",
       "      <td>8.37</td>\n",
       "      <td>3.04</td>\n",
       "      <td>5.54</td>\n",
       "      <td>1.52</td>\n",
       "      <td>8.49</td>\n",
       "      <td>2.11</td>\n",
       "      <td>3.04</td>\n",
       "      <td>1.29</td>\n",
       "      <td>2.19</td>\n",
       "      <td>1.00</td>\n",
       "      <td>3.39</td>\n",
       "      <td>2.69</td>\n",
       "      <td>7.86</td>\n",
       "      <td>2.57</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>95</td>\n",
       "      <td>7</td>\n",
       "      <td>36</td>\n",
       "      <td>843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99521</td>\n",
       "      <td>0.924</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.271896</td>\n",
       "      <td>20.600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8610</th>\n",
       "      <td>X</td>\n",
       "      <td>154965965</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>0.371</td>\n",
       "      <td>0.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000185010</td>\n",
       "      <td>ENST00000360256</td>\n",
       "      <td>F8</td>\n",
       "      <td>CCDS35457.1</td>\n",
       "      <td>9/25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-5.0</td>\n",
       "      <td>DONOR</td>\n",
       "      <td>552</td>\n",
       "      <td>193</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.878</td>\n",
       "      <td>0.999</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.382</td>\n",
       "      <td>2.468</td>\n",
       "      <td>4.940</td>\n",
       "      <td>225.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1982.59</td>\n",
       "      <td>3.442070e-106</td>\n",
       "      <td>11.70</td>\n",
       "      <td>11.70</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.50</td>\n",
       "      <td>1.08</td>\n",
       "      <td>2.25</td>\n",
       "      <td>0.75</td>\n",
       "      <td>8.21</td>\n",
       "      <td>2.26</td>\n",
       "      <td>2.94</td>\n",
       "      <td>0.77</td>\n",
       "      <td>8.37</td>\n",
       "      <td>3.04</td>\n",
       "      <td>5.54</td>\n",
       "      <td>1.52</td>\n",
       "      <td>8.49</td>\n",
       "      <td>2.11</td>\n",
       "      <td>3.04</td>\n",
       "      <td>1.29</td>\n",
       "      <td>2.19</td>\n",
       "      <td>1.00</td>\n",
       "      <td>3.39</td>\n",
       "      <td>2.69</td>\n",
       "      <td>7.86</td>\n",
       "      <td>2.57</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>95</td>\n",
       "      <td>7</td>\n",
       "      <td>36</td>\n",
       "      <td>843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98832</td>\n",
       "      <td>0.926</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.302688</td>\n",
       "      <td>20.800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8611</th>\n",
       "      <td>X</td>\n",
       "      <td>155492384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.464</td>\n",
       "      <td>0.053</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>D</td>\n",
       "      <td>ENSG00000185973</td>\n",
       "      <td>ENST00000334398</td>\n",
       "      <td>TMLHE</td>\n",
       "      <td>CCDS14768.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7/8</td>\n",
       "      <td>1253.0</td>\n",
       "      <td>0.442</td>\n",
       "      <td>1107.0</td>\n",
       "      <td>0.874</td>\n",
       "      <td>369.0</td>\n",
       "      <td>0.876</td>\n",
       "      <td>ndomain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>106</td>\n",
       "      <td>152</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>0.01</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>0.968</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.979</td>\n",
       "      <td>0.990</td>\n",
       "      <td>-0.483</td>\n",
       "      <td>-0.176</td>\n",
       "      <td>0.142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2929.08</td>\n",
       "      <td>1.851780e-164</td>\n",
       "      <td>16.20</td>\n",
       "      <td>5.09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.69</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.31</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.18</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.0</td>\n",
       "      <td>753</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.463914</td>\n",
       "      <td>21.700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8612</th>\n",
       "      <td>X</td>\n",
       "      <td>155492384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>0.464</td>\n",
       "      <td>0.053</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000224533</td>\n",
       "      <td>ENST00000433624</td>\n",
       "      <td>TMLHE-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>106</td>\n",
       "      <td>152</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.163</td>\n",
       "      <td>0.979</td>\n",
       "      <td>0.990</td>\n",
       "      <td>-0.483</td>\n",
       "      <td>-0.176</td>\n",
       "      <td>0.142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2929.08</td>\n",
       "      <td>1.851780e-164</td>\n",
       "      <td>16.20</td>\n",
       "      <td>5.09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.69</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.31</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.18</td>\n",
       "      <td>0.27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>753</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.463914</td>\n",
       "      <td>21.700</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8613 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0         1    7961859   C   G  SNV       0  RegulatoryFeature   \n",
       "1         1    7961859   C   G  SNV       0         Transcript   \n",
       "2         1    9720021   G   A  SNV       0  RegulatoryFeature   \n",
       "3         1    9720021   G   A  SNV       0         Transcript   \n",
       "...     ...        ...  ..  ..  ...     ...                ...   \n",
       "8609      X  154965965   C   G  SNV       0         Transcript   \n",
       "8610      X  154965965   C   T  SNV       0         Transcript   \n",
       "8611      X  155492384   C   A  SNV       0   CodingTranscript   \n",
       "8612      X  155492384   C   A  SNV       0         Transcript   \n",
       "\n",
       "         Consequence  ConsScore         ConsDetail     GC    CpG  motifECount  \\\n",
       "0         REGULATORY          4         regulatory  0.781  0.253          NaN   \n",
       "1           INTRONIC          2             intron  0.781  0.253          NaN   \n",
       "2         REGULATORY          4         regulatory  0.603  0.067          NaN   \n",
       "3        SPLICE_SITE          5      splice,intron  0.603  0.067          NaN   \n",
       "...              ...        ...                ...    ...    ...          ...   \n",
       "8609     SPLICE_SITE          5      splice,intron  0.371  0.000          NaN   \n",
       "8610     SPLICE_SITE          5      splice,intron  0.371  0.000          NaN   \n",
       "8611  NON_SYNONYMOUS          7           missense  0.464  0.053          NaN   \n",
       "8612        INTRONIC          2  intron,non_coding  0.464  0.053          NaN   \n",
       "\n",
       "     motifEName  motifEHIPos  motifEScoreChng  oAA  nAA           GeneID  \\\n",
       "0           NaN          NaN              NaN  NaN  NaN              NaN   \n",
       "1           NaN          NaN              NaN  NaN  NaN  ENSG00000116288   \n",
       "2           NaN          NaN              NaN  NaN  NaN              NaN   \n",
       "3           NaN          NaN              NaN  NaN  NaN  ENSG00000171608   \n",
       "...         ...          ...              ...  ...  ...              ...   \n",
       "8609        NaN          NaN              NaN  NaN  NaN  ENSG00000185010   \n",
       "8610        NaN          NaN              NaN  NaN  NaN  ENSG00000185010   \n",
       "8611        NaN          NaN              NaN    E    D  ENSG00000185973   \n",
       "8612        NaN          NaN              NaN  NaN  NaN  ENSG00000224533   \n",
       "\n",
       "            FeatureID   GeneName         CCDS Intron Exon  cDNApos  \\\n",
       "0     ENSR00000000851        NaN          NaN    NaN  NaN      NaN   \n",
       "1     ENST00000493678      PARK7     CCDS93.1    1/6  NaN      NaN   \n",
       "2     ENSR00000346835        NaN          NaN    NaN  NaN      NaN   \n",
       "3     ENST00000377346     PIK3CD    CCDS104.1  10/23  NaN      NaN   \n",
       "...               ...        ...          ...    ...  ...      ...   \n",
       "8609  ENST00000360256         F8  CCDS35457.1   9/25  NaN      NaN   \n",
       "8610  ENST00000360256         F8  CCDS35457.1   9/25  NaN      NaN   \n",
       "8611  ENST00000334398      TMLHE  CCDS14768.1    NaN  7/8   1253.0   \n",
       "8612  ENST00000433624  TMLHE-AS1          NaN    3/3  NaN      NaN   \n",
       "\n",
       "      relcDNApos  CDSpos  relCDSpos  protPos  relProtPos   Domain  Dst2Splice  \\\n",
       "0            NaN     NaN        NaN      NaN         NaN      NaN         NaN   \n",
       "1            NaN     NaN        NaN      NaN         NaN      NaN         NaN   \n",
       "2            NaN     NaN        NaN      NaN         NaN      NaN         NaN   \n",
       "3            NaN     NaN        NaN      NaN         NaN      NaN        -4.0   \n",
       "...          ...     ...        ...      ...         ...      ...         ...   \n",
       "8609         NaN     NaN        NaN      NaN         NaN      NaN        -5.0   \n",
       "8610         NaN     NaN        NaN      NaN         NaN      NaN        -5.0   \n",
       "8611       0.442  1107.0      0.874    369.0       0.876  ndomain         NaN   \n",
       "8612         NaN     NaN        NaN      NaN         NaN      NaN         NaN   \n",
       "\n",
       "     Dst2SplType  minDistTSS  minDistTSE      SIFTcat  SIFTval  \\\n",
       "0            NaN          36        9190          NaN      NaN   \n",
       "1            NaN          36        9190          NaN      NaN   \n",
       "2            NaN        9823        7207          NaN      NaN   \n",
       "3          DONOR        9823        7207          NaN      NaN   \n",
       "...          ...         ...         ...          ...      ...   \n",
       "8609       DONOR         552         193          NaN      NaN   \n",
       "8610       DONOR         552         193          NaN      NaN   \n",
       "8611         NaN         106         152  deleterious     0.01   \n",
       "8612         NaN         106         152          NaN      NaN   \n",
       "\n",
       "            PolyPhenCat  PolyPhenVal  priPhCons  mamPhCons  verPhCons  \\\n",
       "0                   NaN          NaN      0.690      0.000      0.001   \n",
       "1                   NaN          NaN      0.690      0.000      0.001   \n",
       "2                   NaN          NaN      0.227      0.002      0.000   \n",
       "3                   NaN          NaN      0.227      0.002      0.000   \n",
       "...                 ...          ...        ...        ...        ...   \n",
       "8609                NaN          NaN      0.878      0.999      1.000   \n",
       "8610                NaN          NaN      0.878      0.999      1.000   \n",
       "8611  probably_damaging        0.968      0.163      0.979      0.990   \n",
       "8612                NaN          NaN      0.163      0.979      0.990   \n",
       "\n",
       "      priPhyloP  mamPhyloP  verPhyloP  bStatistic  targetScan  mirSVR-Score  \\\n",
       "0         0.399     -0.152     -0.334       824.0         NaN           NaN   \n",
       "1         0.399     -0.152     -0.334       824.0         NaN           NaN   \n",
       "2        -1.578     -0.179     -1.643       835.0         NaN           NaN   \n",
       "3        -1.578     -0.179     -1.643       835.0         NaN           NaN   \n",
       "...         ...        ...        ...         ...         ...           ...   \n",
       "8609      0.382      2.468      4.940       225.0         NaN           NaN   \n",
       "8610      0.382      2.468      4.940       225.0         NaN           NaN   \n",
       "8611     -0.483     -0.176      0.142         NaN         NaN           NaN   \n",
       "8612     -0.483     -0.176      0.142         NaN         NaN           NaN   \n",
       "\n",
       "      mirSVR-E  mirSVR-Aln  cHmm_E1  cHmm_E2  cHmm_E3  cHmm_E4  cHmm_E5  \\\n",
       "0          NaN         NaN      0.0      0.0      0.0      0.0      0.0   \n",
       "1          NaN         NaN      0.0      0.0      0.0      0.0      0.0   \n",
       "2          NaN         NaN      0.0      0.0      0.0      1.0      0.0   \n",
       "3          NaN         NaN      0.0      0.0      0.0      1.0      0.0   \n",
       "...        ...         ...      ...      ...      ...      ...      ...   \n",
       "8609       NaN         NaN      0.0      0.0      1.0     33.0      0.0   \n",
       "8610       NaN         NaN      0.0      0.0      1.0     33.0      0.0   \n",
       "8611       NaN         NaN      0.0      0.0      3.0     22.0      0.0   \n",
       "8612       NaN         NaN      0.0      0.0      3.0     22.0      0.0   \n",
       "\n",
       "      cHmm_E6  cHmm_E7  cHmm_E8  cHmm_E9  cHmm_E10  cHmm_E11  cHmm_E12  \\\n",
       "0         0.0      0.0      0.0      0.0       0.0       0.0       0.0   \n",
       "1         0.0      0.0      0.0      0.0       0.0       0.0       0.0   \n",
       "2         3.0     10.0      5.0      3.0       7.0       2.0       1.0   \n",
       "3         3.0     10.0      5.0      3.0       7.0       2.0       1.0   \n",
       "...       ...      ...      ...      ...       ...       ...       ...   \n",
       "8609      0.0      1.0      0.0      0.0       0.0       0.0       0.0   \n",
       "8610      0.0      1.0      0.0      0.0       0.0       0.0       0.0   \n",
       "8611      0.0      6.0      1.0      0.0       0.0       0.0       0.0   \n",
       "8612      0.0      6.0      1.0      0.0       0.0       0.0       0.0   \n",
       "\n",
       "      cHmm_E13  cHmm_E14  cHmm_E15  cHmm_E16  cHmm_E17  cHmm_E18  cHmm_E19  \\\n",
       "0          0.0       0.0       0.0      43.0       2.0       0.0       3.0   \n",
       "1          0.0       0.0       0.0      43.0       2.0       0.0       3.0   \n",
       "2          3.0       0.0      12.0       0.0       0.0       0.0       0.0   \n",
       "3          3.0       0.0      12.0       0.0       0.0       0.0       0.0   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "8609       0.0       0.0      11.0       0.0       0.0       0.0       0.0   \n",
       "8610       0.0       0.0      11.0       0.0       0.0       0.0       0.0   \n",
       "8611       0.0       0.0      16.0       0.0       0.0       0.0       0.0   \n",
       "8612       0.0       0.0      16.0       0.0       0.0       0.0       0.0   \n",
       "\n",
       "      cHmm_E20  cHmm_E21  cHmm_E22  cHmm_E23  cHmm_E24  cHmm_E25   GerpRS  \\\n",
       "0          0.0       0.0       0.0       0.0       0.0       0.0      NaN   \n",
       "1          0.0       0.0       0.0       0.0       0.0       0.0      NaN   \n",
       "2          0.0       0.0       0.0       1.0       0.0       0.0      NaN   \n",
       "3          0.0       0.0       0.0       1.0       0.0       0.0      NaN   \n",
       "...        ...       ...       ...       ...       ...       ...      ...   \n",
       "8609       0.0       2.0       0.0       0.0       0.0       0.0  1982.59   \n",
       "8610       0.0       2.0       0.0       0.0       0.0       0.0  1982.59   \n",
       "8611       0.0       0.0       0.0       0.0       0.0       0.0  2929.08   \n",
       "8612       0.0       0.0       0.0       0.0       0.0       0.0  2929.08   \n",
       "\n",
       "         GerpRSpval  GerpN  GerpS  tOverlapMotifs  motifDist  \\\n",
       "0               NaN   4.58  -9.17             NaN        NaN   \n",
       "1               NaN   4.58  -9.17             NaN        NaN   \n",
       "2               NaN  15.30 -30.60             NaN        NaN   \n",
       "3               NaN  15.30 -30.60             NaN        NaN   \n",
       "...             ...    ...    ...             ...        ...   \n",
       "8609  3.442070e-106  11.70  11.70             NaN        NaN   \n",
       "8610  3.442070e-106  11.70  11.70             NaN        NaN   \n",
       "8611  1.851780e-164  16.20   5.09             NaN        NaN   \n",
       "8612  1.851780e-164  16.20   5.09             NaN        NaN   \n",
       "\n",
       "      EncodeH3K4me1-sum  EncodeH3K4me1-max  EncodeH3K4me2-sum  \\\n",
       "0                  3.18               1.11             104.49   \n",
       "1                  3.18               1.11             104.49   \n",
       "2                 36.93              11.08              14.49   \n",
       "3                 36.93              11.08              14.49   \n",
       "...                 ...                ...                ...   \n",
       "8609               5.50               1.08               2.25   \n",
       "8610               5.50               1.08               2.25   \n",
       "8611                NaN                NaN                NaN   \n",
       "8612                NaN                NaN                NaN   \n",
       "\n",
       "      EncodeH3K4me2-max  EncodeH3K4me3-sum  EncodeH3K4me3-max  \\\n",
       "0                 25.01             525.02              88.14   \n",
       "1                 25.01             525.02              88.14   \n",
       "2                  4.51               9.86               2.26   \n",
       "3                  4.51               9.86               2.26   \n",
       "...                 ...                ...                ...   \n",
       "8609               0.75               8.21               2.26   \n",
       "8610               0.75               8.21               2.26   \n",
       "8611                NaN               0.23               0.23   \n",
       "8612                NaN               0.23               0.23   \n",
       "\n",
       "      EncodeH3K9ac-sum  EncodeH3K9ac-max  EncodeH3K9me3-sum  \\\n",
       "0               411.25             64.60               5.76   \n",
       "1               411.25             64.60               5.76   \n",
       "2                18.01              7.50               7.67   \n",
       "3                18.01              7.50               7.67   \n",
       "...                ...               ...                ...   \n",
       "8609              2.94              0.77               8.37   \n",
       "8610              2.94              0.77               8.37   \n",
       "8611               NaN               NaN                NaN   \n",
       "8612               NaN               NaN                NaN   \n",
       "\n",
       "      EncodeH3K9me3-max  EncodeH3K27ac-sum  EncodeH3K27ac-max  \\\n",
       "0                  1.06             359.53              65.41   \n",
       "1                  1.06             359.53              65.41   \n",
       "2                  1.73              50.36              17.98   \n",
       "3                  1.73              50.36              17.98   \n",
       "...                 ...                ...                ...   \n",
       "8609               3.04               5.54               1.52   \n",
       "8610               3.04               5.54               1.52   \n",
       "8611                NaN                NaN                NaN   \n",
       "8612                NaN                NaN                NaN   \n",
       "\n",
       "      EncodeH3K27me3-sum  EncodeH3K27me3-max  EncodeH3K36me3-sum  \\\n",
       "0                   9.83                2.28                6.51   \n",
       "1                   9.83                2.28                6.51   \n",
       "2                   7.69                2.64               45.86   \n",
       "3                   7.69                2.64               45.86   \n",
       "...                  ...                 ...                 ...   \n",
       "8609                8.49                2.11                3.04   \n",
       "8610                8.49                2.11                3.04   \n",
       "8611                 NaN                 NaN                0.69   \n",
       "8612                 NaN                 NaN                0.69   \n",
       "\n",
       "      EncodeH3K36me3-max  EncodeH3K79me2-sum  EncodeH3K79me2-max  \\\n",
       "0                   2.64               82.68               28.05   \n",
       "1                   2.64               82.68               28.05   \n",
       "2                  13.42               24.37                6.03   \n",
       "3                  13.42               24.37                6.03   \n",
       "...                  ...                 ...                 ...   \n",
       "8609                1.29                2.19                1.00   \n",
       "8610                1.29                2.19                1.00   \n",
       "8611                0.35                0.31                0.31   \n",
       "8612                0.35                0.31                0.31   \n",
       "\n",
       "      EncodeH4K20me1-sum  EncodeH4K20me1-max  EncodeH2AFZ-sum  \\\n",
       "0                   9.52                2.49           177.88   \n",
       "1                   9.52                2.49           177.88   \n",
       "2                  36.66               24.95            16.91   \n",
       "3                  36.66               24.95            16.91   \n",
       "...                  ...                 ...              ...   \n",
       "8609                3.39                2.69             7.86   \n",
       "8610                3.39                2.69             7.86   \n",
       "8611                 NaN                 NaN             0.68   \n",
       "8612                 NaN                 NaN             0.68   \n",
       "\n",
       "      EncodeH2AFZ-max  EncodeDNase-sum  EncodeDNase-max  EncodetotalRNA-sum  \\\n",
       "0               29.14             9.10             2.57                0.34   \n",
       "1               29.14             9.10             2.57                0.34   \n",
       "2                4.03             0.89             0.36                0.20   \n",
       "3                4.03             0.89             0.36                0.20   \n",
       "...               ...              ...              ...                 ...   \n",
       "8609             2.57             0.33             0.16                0.01   \n",
       "8610             2.57             0.33             0.16                0.01   \n",
       "8611             0.68             0.00             0.00                1.18   \n",
       "8612             0.68             0.00             0.00                1.18   \n",
       "\n",
       "      EncodetotalRNA-max  Grantham  Dist2Mutation  Freq100bp  Rare100bp  \\\n",
       "0                   0.24       NaN              6        0.0        0.0   \n",
       "1                   0.24       NaN              6        0.0        0.0   \n",
       "2                   0.05       NaN              9        0.0        1.0   \n",
       "3                   0.05       NaN              9        0.0        1.0   \n",
       "...                  ...       ...            ...        ...        ...   \n",
       "8609                0.01       NaN             13        0.0        0.0   \n",
       "8610                0.01       NaN             13        0.0        0.0   \n",
       "8611                0.27       0.0            753        NaN        NaN   \n",
       "8612                0.27       NaN            753        NaN        NaN   \n",
       "\n",
       "      Sngl100bp  Freq1000bp  Rare1000bp  Sngl1000bp  Freq10000bp  Rare10000bp  \\\n",
       "0          20.0           7           8         229           26           48   \n",
       "1          20.0           7           8         229           26           48   \n",
       "2          16.0           1           6         142            9           74   \n",
       "3          16.0           1           6         142            9           74   \n",
       "...         ...         ...         ...         ...          ...          ...   \n",
       "8609       11.0           0           3          95            7           36   \n",
       "8610       11.0           0           3          95            7           36   \n",
       "8611        NaN           0           1           3            3            3   \n",
       "8612        NaN           0           1           3            3            3   \n",
       "\n",
       "      Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0            1451                  Promoter                NaN   \n",
       "1            1451                  Promoter                NaN   \n",
       "2            1406         CTCF Binding Site            0.00003   \n",
       "3            1406         CTCF Binding Site            0.00003   \n",
       "...           ...                       ...                ...   \n",
       "8609          843                       NaN            0.99521   \n",
       "8610          843                       NaN            0.98832   \n",
       "8611           19                       NaN                NaN   \n",
       "8612           19                       NaN                NaN   \n",
       "\n",
       "     dbscSNV-rf_score  RemapOverlapTF  RemapOverlapCL  RawScore   PHRED  \n",
       "0                 NaN           208.0           475.0  1.043084  12.600  \n",
       "1                 NaN           208.0           475.0  1.043084  12.600  \n",
       "2               0.014            12.0            15.0 -0.385227   0.135  \n",
       "3               0.014            12.0            15.0 -0.385227   0.135  \n",
       "...               ...             ...             ...       ...     ...  \n",
       "8609            0.924             NaN             NaN  2.271896  20.600  \n",
       "8610            0.926             NaN             NaN  2.302688  20.800  \n",
       "8611              NaN             NaN             NaN  2.463914  21.700  \n",
       "8612              NaN             NaN             NaN  2.463914  21.700  \n",
       "\n",
       "[8613 rows x 125 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_annotated_with_cadd = pd.read_table('hgmd_noncoding_cadd_annotations_noheader.tsv')\n",
    "hgmd_noncoding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Human Derived"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated CADD file for human derived coding region:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>5</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>944699</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>117</td>\n",
       "      <td>2573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>2.980499</td>\n",
       "      <td>23.200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122229</th>\n",
       "      <td>X</td>\n",
       "      <td>155260197</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3PRIME_UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>3_prime_UTR</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>873</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.219204</td>\n",
       "      <td>0.473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122230</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122231</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122232</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>122233 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0           1     943329   C   T  SNV       0  RegulatoryFeature   \n",
       "1           1     943329   C   T  SNV       0   CodingTranscript   \n",
       "2           1     943329   C   T  SNV       0         Intergenic   \n",
       "3           1     944699   C   T  SNV       0   CodingTranscript   \n",
       "...       ...        ...  ..  ..  ...     ...                ...   \n",
       "122229      X  155260197   C   A  SNV       0         Transcript   \n",
       "122230      X  155612732   T   C  SNV       0  RegulatoryFeature   \n",
       "122231      X  155612732   T   C  SNV       0         Transcript   \n",
       "122232      X  155612732   T   C  SNV       0         Intergenic   \n",
       "\n",
       "           Consequence  ConsScore   ConsDetail  ...  Freq10000bp  Rare10000bp  \\\n",
       "0           REGULATORY          4   regulatory  ...           18          117   \n",
       "1           SYNONYMOUS          5   synonymous  ...           18          117   \n",
       "2           DOWNSTREAM          1   downstream  ...           18          117   \n",
       "3       NON_SYNONYMOUS          7     missense  ...           16          117   \n",
       "...                ...        ...          ...  ...          ...          ...   \n",
       "122229      3PRIME_UTR          2  3_prime_UTR  ...            9           36   \n",
       "122230      REGULATORY          4   regulatory  ...            7           27   \n",
       "122231        INTRONIC          2       intron  ...            7           27   \n",
       "122232      DOWNSTREAM          1   downstream  ...            7           27   \n",
       "\n",
       "        Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              2586         CTCF Binding Site                NaN   \n",
       "1              2586         CTCF Binding Site                NaN   \n",
       "2              2586         CTCF Binding Site                NaN   \n",
       "3              2573                       NaN                NaN   \n",
       "...             ...                       ...                ...   \n",
       "122229          873                       NaN                NaN   \n",
       "122230          866                  Promoter                NaN   \n",
       "122231          866                  Promoter                NaN   \n",
       "122232          866                  Promoter                NaN   \n",
       "\n",
       "        dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore   PHRED  \n",
       "0                    NaN           88.0          121.0  0.590702   8.117  \n",
       "1                    NaN           88.0          121.0  0.590702   8.117  \n",
       "2                    NaN           88.0          121.0  0.590702   8.117  \n",
       "3                    NaN           12.0           13.0  2.980499  23.200  \n",
       "...                  ...            ...            ...       ...     ...  \n",
       "122229               NaN            3.0            6.0 -0.219204   0.473  \n",
       "122230               NaN          112.0          202.0  0.661682   8.709  \n",
       "122231               NaN          112.0          202.0  0.661682   8.709  \n",
       "122232               NaN          112.0          202.0  0.661682   8.709  \n",
       "\n",
       "[122233 rows x 125 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_annotated_with_CADD = pd.read_table('human_derived_coding_for_cadd_noheader.tsv')\n",
    "human_derived_coding_annotated_with_CADD"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### ANNOVAR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we will do the same with the human derived coding data for ANNOVAR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10</th>\n",
       "      <th>100020652</th>\n",
       "      <th>.</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100190879</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100233196</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100267615</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100347207</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53793</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719539</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53794</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719939</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53795</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99720084</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53796</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99721008</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53797 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr10  100020652  .  G  A\n",
       "0      chr10  100190879  .  T  C\n",
       "1      chr10  100233196  .  G  A\n",
       "2      chr10  100267615  .  C  T\n",
       "3      chr10  100347207  .  T  C\n",
       "...      ...        ... .. .. ..\n",
       "53793   chrX   99719539  .  A  G\n",
       "53794   chrX   99719939  .  C  A\n",
       "53795   chrX   99720084  .  G  A\n",
       "53796   chrX   99721008  .  G  A\n",
       "\n",
       "[53797 rows x 5 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "cannot set a row with mismatched columns",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-82-39e3757c36ee>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mhuman_derived_coding_region\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'...'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'.'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mhuman_derived_coding_region\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'....'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'.'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0mhuman_derived_coding_region\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'chr10'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'100020652'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'.'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'G'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'A'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'.'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'.'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'.'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m \u001b[0mhuman_derived_coding_region\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhuman_derived_coding_region\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m1\u001b[0m  \u001b[0;31m# shifting index\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0mhuman_derived_coding_region\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhuman_derived_coding_region\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msort_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/lib/python3.7/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__setitem__\u001b[0;34m(self, key, value)\u001b[0m\n\u001b[1;32m    188\u001b[0m             \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_setitem_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 190\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_setitem_with_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    191\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    192\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_validate_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/lib/python3.7/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_setitem_with_indexer\u001b[0;34m(self, indexer, value)\u001b[0m\n\u001b[1;32m    443\u001b[0m                         \u001b[0;32mif\u001b[0m \u001b[0mis_list_like_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    444\u001b[0m                             \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalue\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 445\u001b[0;31m                                 raise ValueError(\"cannot set a row with \"\n\u001b[0m\u001b[1;32m    446\u001b[0m                                                  \"mismatched columns\")\n\u001b[1;32m    447\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: cannot set a row with mismatched columns"
     ]
    }
   ],
   "source": [
    "human_derived_coding_region['..'] = '.'\n",
    "human_derived_coding_region['...'] = '.'\n",
    "human_derived_coding_region['....'] = '.'\n",
    "human_derived_coding_region.loc[-1] = ['chr10', '100020652', '.', 'G', 'A', '.', '.', '.']\n",
    "human_derived_coding_region.index = human_derived_coding_region.index + 1  # shifting index\n",
    "human_derived_coding_region = human_derived_coding_region.sort_index()\n",
    "human_derived_coding_region.columns = ['', '', '', '', '', '', '', '']\n",
    "human_derived_coding_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "output_VCF = \"human_derived_coding_region_for_annovar.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "human_derived_coding_region.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the final annotated ANNOVAR data for the human derived coding region:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>100020652</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>DNMBP;CPN1</td>\n",
       "      <td>dist=10699;dist=21656</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100190879</td>\n",
       "      <td>100190879</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CHUK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>CHUK:NM_001278:exon20:c.A2198G:p.N733S</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00447999043875</td>\n",
       "      <td>0.039</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100233196</td>\n",
       "      <td>100233196</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>NM_018294:c.*31C&gt;T;NM_001303407:c.*31C&gt;T;NM_00...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100267615</td>\n",
       "      <td>100267615</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>NM_018294:c.-22G&gt;A;NM_001303407:c.-20707G&gt;A;NM...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53794</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719539</td>\n",
       "      <td>99719539</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>XRCC6P5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53795</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719939</td>\n",
       "      <td>99719939</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>XRCC6P5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53796</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99720084</td>\n",
       "      <td>99720084</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>XRCC6P5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53797</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99721008</td>\n",
       "      <td>99721008</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>XRCC6P5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53798 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr      Start        End Ref Alt  Func.refGene Gene.refGene  \\\n",
       "0      chr10  100020652  100020652   G   A    intergenic   DNMBP;CPN1   \n",
       "1      chr10  100190879  100190879   T   C        exonic         CHUK   \n",
       "2      chr10  100233196  100233196   G   A          UTR3      CWF19L1   \n",
       "3      chr10  100267615  100267615   C   T          UTR5      CWF19L1   \n",
       "...      ...        ...        ...  ..  ..           ...          ...   \n",
       "53794   chrX   99719539   99719539   A   G  ncRNA_exonic      XRCC6P5   \n",
       "53795   chrX   99719939   99719939   C   A  ncRNA_exonic      XRCC6P5   \n",
       "53796   chrX   99720084   99720084   G   A  ncRNA_exonic      XRCC6P5   \n",
       "53797   chrX   99721008   99721008   G   A  ncRNA_exonic      XRCC6P5   \n",
       "\n",
       "                                      GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                  dist=10699;dist=21656                  .   \n",
       "1                                                      .  nonsynonymous SNV   \n",
       "2      NM_018294:c.*31C>T;NM_001303407:c.*31C>T;NM_00...                  .   \n",
       "3      NM_018294:c.-22G>A;NM_001303407:c.-20707G>A;NM...                  .   \n",
       "...                                                  ...                ...   \n",
       "53794                                                  .                  .   \n",
       "53795                                                  .                  .   \n",
       "53796                                                  .                  .   \n",
       "53797                                                  .                  .   \n",
       "\n",
       "                             AAChange.refGene  ... BP4 BP5 BP6 BP7 cosmic70  \\\n",
       "0                                           .  ...   .   .   .   .        .   \n",
       "1      CHUK:NM_001278:exon20:c.A2198G:p.N733S  ...   1   0   0   0        .   \n",
       "2                                           .  ...   .   .   .   .        .   \n",
       "3                                           .  ...   .   .   .   .        .   \n",
       "...                                       ...  ...  ..  ..  ..  ..      ...   \n",
       "53794                                       .  ...   .   .   .   .        .   \n",
       "53795                                       .  ...   .   .   .   .        .   \n",
       "53796                                       .  ...   .   .   .   .        .   \n",
       "53797                                       .  ...   .   .   .   .        .   \n",
       "\n",
       "                   MCAP  REVEL regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0                     .      .          .              .                    .  \n",
       "1      0.00447999043875  0.039          .              .                    .  \n",
       "2                     .      .          .              .                    .  \n",
       "3                     .      .          .              .                    .  \n",
       "...                 ...    ...        ...            ...                  ...  \n",
       "53794                 .      .          .              .                    .  \n",
       "53795                 .      .          .              .                    .  \n",
       "53796                 .      .          .              .                    .  \n",
       "53797                 .      .          .              .                    .  \n",
       "\n",
       "[53798 rows x 149 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_annotated_with_annovar = pd.read_csv('human_derived_coding_annotated.hg38_multianno.csv')\n",
    "human_derived_coding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since we have a lot of files for the noncoding region for the human derived data in a lot of separate dataframes, we must combine them into one so that we can export the resulting final and only have to run ANNOVAR once for the noncoding region. The reason we have them in separate dataframes is that we initially split them for CADD annotations. Now we will use these files again for ANNOVAR. The process for this is shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_1 = read_vcf('human_derived_noncoding_for_cadd_1_for_annovar.vcf')\n",
    "human_derived_noncoding_region_2 = read_vcf('human_derived_noncoding_for_cadd_2_for_annovar.vcf')\n",
    "human_derived_noncoding_region_3 = read_vcf('human_derived_noncoding_for_cadd_3_for_annovar.vcf')\n",
    "human_derived_noncoding_region_4 = read_vcf('human_derived_noncoding_for_cadd_4_for_annovar.vcf')\n",
    "human_derived_noncoding_region_5 = read_vcf('human_derived_noncoding_for_cadd_5_for_annovar.vcf')\n",
    "human_derived_noncoding_region_6 = read_vcf('human_derived_noncoding_for_cadd_6_for_annovar.vcf')\n",
    "human_derived_noncoding_region_7 = read_vcf('human_derived_noncoding_for_cadd_7_for_annovar.vcf')\n",
    "human_derived_noncoding_region_8 = read_vcf('human_derived_noncoding_for_cadd_8_for_annovar.vcf')\n",
    "human_derived_noncoding_region_9 = read_vcf('human_derived_noncoding_for_cadd_9_for_annovar.vcf')\n",
    "human_derived_noncoding_region_10 = read_vcf('human_derived_noncoding_for_cadd_10_for_annovar.vcf')\n",
    "human_derived_noncoding_region_11 = read_vcf('human_derived_noncoding_for_cadd_11_for_annovar.vcf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000013</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000297</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000555</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89996</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312241</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89997</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312402</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89998</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312428</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89999</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312687</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90000 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         \n",
       "0      chr10    1000013  .  G  A  .  .  .\n",
       "1      chr10  100020652  .  G  A  .  .  .\n",
       "2      chr10    1000297  .  T  G  .  .  .\n",
       "3      chr10    1000555  .  A  T  .  .  .\n",
       "...      ...        ... .. .. .. .. .. ..\n",
       "89996  chr11   47312241  .  G  A  .  .  .\n",
       "89997  chr11   47312402  .  G  C  .  .  .\n",
       "89998  chr11   47312428  .  C  T  .  .  .\n",
       "89999  chr11   47312687  .  T  C  .  .  .\n",
       "\n",
       "[90000 rows x 8 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_region_1['..'] = '.'\n",
    "human_derived_noncoding_region_1['...'] = '.'\n",
    "human_derived_noncoding_region_1['....'] = '.'\n",
    "human_derived_noncoding_region_1.loc[-1] = ['chr10', '1000013', '.', 'G', 'A', '.', '.', '.']\n",
    "human_derived_noncoding_region_1.index = human_derived_noncoding_region_1.index + 1  # shifting index\n",
    "human_derived_noncoding_region_1 = human_derived_noncoding_region_1.sort_index()\n",
    "human_derived_noncoding_region_1.columns = ['', '', '', '', '', '', '', '']\n",
    "human_derived_noncoding_region_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312814</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312979</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47313760</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47313853</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89996</th>\n",
       "      <td>chr12</td>\n",
       "      <td>65348795</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89997</th>\n",
       "      <td>chr12</td>\n",
       "      <td>65348797</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89998</th>\n",
       "      <td>chr12</td>\n",
       "      <td>65348836</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89999</th>\n",
       "      <td>chr12</td>\n",
       "      <td>65348914</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90000 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        \n",
       "0      chr11  47312814  .  G  A  .  .  .\n",
       "1      chr11  47312979  .  T  G  .  .  .\n",
       "2      chr11  47313760  .  G  T  .  .  .\n",
       "3      chr11  47313853  .  A  G  .  .  .\n",
       "...      ...       ... .. .. .. .. .. ..\n",
       "89996  chr12  65348795  .  C  A  .  .  .\n",
       "89997  chr12  65348797  .  C  T  .  .  .\n",
       "89998  chr12  65348836  .  C  T  .  .  .\n",
       "89999  chr12  65348914  .  C  T  .  .  .\n",
       "\n",
       "[90000 rows x 8 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_region_2['..'] = '.'\n",
    "human_derived_noncoding_region_2['...'] = '.'\n",
    "human_derived_noncoding_region_2['....'] = '.'\n",
    "human_derived_noncoding_region_2.loc[-1] = ['chr11', '47312814', '.', 'G', 'A', '.', '.', '.']\n",
    "human_derived_noncoding_region_2.index = human_derived_noncoding_region_2.index + 1  # shifting index\n",
    "human_derived_noncoding_region_2 = human_derived_noncoding_region_2.sort_index()\n",
    "human_derived_noncoding_region_2.columns = ['', '', '', '', '', '', '', '']\n",
    "human_derived_noncoding_region_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_3['..'] = '.'\n",
    "human_derived_noncoding_region_3['...'] = '.'\n",
    "human_derived_noncoding_region_3['....'] = '.'\n",
    "human_derived_noncoding_region_3.loc[-1] = ['chr12', '65348986', '.', 'C', 'A', '.', '.', '.']\n",
    "human_derived_noncoding_region_3.index = human_derived_noncoding_region_3.index + 1  # shifting index\n",
    "human_derived_noncoding_region_3 = human_derived_noncoding_region_3.sort_index()\n",
    "human_derived_noncoding_region_3.columns = ['', '', '', '', '', '', '', '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_4['..'] = '.'\n",
    "human_derived_noncoding_region_4['...'] = '.'\n",
    "human_derived_noncoding_region_4['....'] = '.'\n",
    "human_derived_noncoding_region_4.loc[-1] = ['chr15', '64116948', '.', 'G', 'A', '.', '.', '.']\n",
    "human_derived_noncoding_region_4.index = human_derived_noncoding_region_4.index + 1  # shifting index\n",
    "human_derived_noncoding_region_4 = human_derived_noncoding_region_4.sort_index()\n",
    "human_derived_noncoding_region_4.columns = ['', '', '', '', '', '', '', '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_5['..'] = '.'\n",
    "human_derived_noncoding_region_5['...'] = '.'\n",
    "human_derived_noncoding_region_5['....'] = '.'\n",
    "human_derived_noncoding_region_5.loc[-1] = ['chr17', '48835049', '.', 'G', 'C', '.', '.', '.']\n",
    "human_derived_noncoding_region_5.index = human_derived_noncoding_region_5.index + 1  # shifting index\n",
    "human_derived_noncoding_region_5 = human_derived_noncoding_region_5.sort_index()\n",
    "human_derived_noncoding_region_5.columns = ['', '', '', '', '', '', '', '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_6['..'] = '.'\n",
    "human_derived_noncoding_region_6['...'] = '.'\n",
    "human_derived_noncoding_region_6['....'] = '.'\n",
    "human_derived_noncoding_region_6.loc[-1] = ['chr20', '35887279', '.', 'C', 'T', '.', '.', '.']\n",
    "human_derived_noncoding_region_6.index = human_derived_noncoding_region_6.index + 1  # shifting index\n",
    "human_derived_noncoding_region_6 = human_derived_noncoding_region_6.sort_index()\n",
    "human_derived_noncoding_region_6.columns = ['', '', '', '', '', '', '', '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_7['..'] = '.'\n",
    "human_derived_noncoding_region_7['...'] = '.'\n",
    "human_derived_noncoding_region_7['....'] = '.'\n",
    "human_derived_noncoding_region_7.loc[-1] = ['chr22', '35418721', '.', 'A', 'T', '.', '.', '.']\n",
    "human_derived_noncoding_region_7.index = human_derived_noncoding_region_7.index + 1  # shifting index\n",
    "human_derived_noncoding_region_7 = human_derived_noncoding_region_7.sort_index()\n",
    "human_derived_noncoding_region_7.columns = ['', '', '', '', '', '', '', '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_8['..'] = '.'\n",
    "human_derived_noncoding_region_8['...'] = '.'\n",
    "human_derived_noncoding_region_8['....'] = '.'\n",
    "human_derived_noncoding_region_8.loc[-1] = ['chr3', '28600649', '.', 'C', 'T', '.', '.', '.']\n",
    "human_derived_noncoding_region_8.index = human_derived_noncoding_region_8.index + 1  # shifting index\n",
    "human_derived_noncoding_region_8 = human_derived_noncoding_region_8.sort_index()\n",
    "human_derived_noncoding_region_8.columns = ['', '', '', '', '', '', '', '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_9['..'] = '.'\n",
    "human_derived_noncoding_region_9['...'] = '.'\n",
    "human_derived_noncoding_region_9['....'] = '.'\n",
    "human_derived_noncoding_region_9.loc[-1] = ['chr5', '1377732', '.', 'G', 'A', '.', '.', '.']\n",
    "human_derived_noncoding_region_9.index = human_derived_noncoding_region_9.index + 1  # shifting index\n",
    "human_derived_noncoding_region_9 = human_derived_noncoding_region_9.sort_index()\n",
    "human_derived_noncoding_region_9.columns = ['', '', '', '', '', '', '', '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_10['..'] = '.'\n",
    "human_derived_noncoding_region_10['...'] = '.'\n",
    "human_derived_noncoding_region_10['....'] = '.'\n",
    "human_derived_noncoding_region_10.loc[-1] = ['chr7', '107252617', '.', 'A', 'T', '.', '.', '.']\n",
    "human_derived_noncoding_region_10.index = human_derived_noncoding_region_10.index + 1  # shifting index\n",
    "human_derived_noncoding_region_10 = human_derived_noncoding_region_10.sort_index()\n",
    "human_derived_noncoding_region_10.columns = ['', '', '', '', '', '', '', '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_11['..'] = '.'\n",
    "human_derived_noncoding_region_11['...'] = '.'\n",
    "human_derived_noncoding_region_11['....'] = '.'\n",
    "human_derived_noncoding_region_11.loc[-1] = ['chr8', '43782972', '.', 'A', 'G', '.', '.', '.']\n",
    "human_derived_noncoding_region_11.index = human_derived_noncoding_region_11.index + 1  # shifting index\n",
    "human_derived_noncoding_region_11 = human_derived_noncoding_region_11.sort_index()\n",
    "human_derived_noncoding_region_11.columns = ['', '', '', '', '', '', '', '']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we want to combine these dataframes into one for ANNOVAR annotation and export them for ANNOVAR annotation:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000013</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000297</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000555</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66378</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931817</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66379</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931818</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66380</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931993</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66381</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9932000</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>966382 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         \n",
       "0      chr10    1000013  .  G  A  .  .  .\n",
       "1      chr10  100020652  .  G  A  .  .  .\n",
       "2      chr10    1000297  .  T  G  .  .  .\n",
       "3      chr10    1000555  .  A  T  .  .  .\n",
       "...      ...        ... .. .. .. .. .. ..\n",
       "66378   chrX    9931817  .  T  C  .  .  .\n",
       "66379   chrX    9931818  .  G  A  .  .  .\n",
       "66380   chrX    9931993  .  T  C  .  .  .\n",
       "66381   chrX    9932000  .  C  T  .  .  .\n",
       "\n",
       "[966382 rows x 8 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding = pd.concat([human_derived_noncoding_region_1, human_derived_noncoding_region_2, human_derived_noncoding_region_3,\n",
    "                                    human_derived_noncoding_region_4, human_derived_noncoding_region_5, human_derived_noncoding_region_6,\n",
    "                                    human_derived_noncoding_region_7, human_derived_noncoding_region_8, human_derived_noncoding_region_9,\n",
    "                                    human_derived_noncoding_region_10, human_derived_noncoding_region_11])\n",
    "human_derived_noncoding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "output_VCF = \"human_derived_noncoding_for_annovar.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "human_derived_noncoding.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have read in the ANNOVAR annotations using a chunking method, which takes a small sample of files at a time and then groups them together, using less memory than if read all at once using pandas:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000013</td>\n",
       "      <td>1000013</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>100020652</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>DNMBP;CPN1</td>\n",
       "      <td>dist=10699;dist=21656</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000297</td>\n",
       "      <td>1000297</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000555</td>\n",
       "      <td>1000555</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.482876712329</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966378</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931817</td>\n",
       "      <td>9931817</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SHROOM2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966379</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931818</td>\n",
       "      <td>9931818</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SHROOM2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966380</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931993</td>\n",
       "      <td>9931993</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SHROOM2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.438356164384</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966381</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9932000</td>\n",
       "      <td>9932000</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SHROOM2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.794520547945</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>966382 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       chr10    1000013    1000013   G   A     intronic       GTPBP4   \n",
       "1       chr10  100020652  100020652   G   A   intergenic   DNMBP;CPN1   \n",
       "2       chr10    1000297    1000297   T   G     intronic       GTPBP4   \n",
       "3       chr10    1000555    1000555   A   T     intronic       GTPBP4   \n",
       "...       ...        ...        ...  ..  ..          ...          ...   \n",
       "966378   chrX    9931817    9931817   T   C     intronic      SHROOM2   \n",
       "966379   chrX    9931818    9931818   G   A     intronic      SHROOM2   \n",
       "966380   chrX    9931993    9931993   T   C     intronic      SHROOM2   \n",
       "966381   chrX    9932000    9932000   C   T     intronic      SHROOM2   \n",
       "\n",
       "           GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  ... BP4  \\\n",
       "0                           .                  .                .  ...   .   \n",
       "1       dist=10699;dist=21656                  .                .  ...   .   \n",
       "2                           .                  .                .  ...   .   \n",
       "3                           .                  .                .  ...   .   \n",
       "...                       ...                ...              ...  ...  ..   \n",
       "966378                      .                  .                .  ...   .   \n",
       "966379                      .                  .                .  ...   .   \n",
       "966380                      .                  .                .  ...   .   \n",
       "966381                      .                  .                .  ...   .   \n",
       "\n",
       "       BP5 BP6 BP7 cosmic70 MCAP REVEL      regsnp_fpr regsnp_disease  \\\n",
       "0        .   .   .        .    .     .               .              .   \n",
       "1        .   .   .        .    .     .               .              .   \n",
       "2        .   .   .        .    .     .               .              .   \n",
       "3        .   .   .        .    .     .  0.482876712329              B   \n",
       "...     ..  ..  ..      ...  ...   ...             ...            ...   \n",
       "966378   .   .   .        .    .     .               .              .   \n",
       "966379   .   .   .        .    .     .               .              .   \n",
       "966380   .   .   .        .    .     .  0.438356164384              B   \n",
       "966381   .   .   .        .    .     .  0.794520547945              B   \n",
       "\n",
       "       regsnp_splicing_site  \n",
       "0                         .  \n",
       "1                         .  \n",
       "2                         .  \n",
       "3                       off  \n",
       "...                     ...  \n",
       "966378                    .  \n",
       "966379                    .  \n",
       "966380                  off  \n",
       "966381                  off  \n",
       "\n",
       "[966382 rows x 149 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_list = []\n",
    "for chunk in pd.read_csv('human_derived_noncoding_annotated.hg38_multianno.csv', chunksize = 5000, low_memory = False):\n",
    "    my_list.append(chunk)\n",
    "human_derived_noncoding_annotated_with_annovar = pd.concat(my_list, axis = 0)\n",
    "del my_list\n",
    "human_derived_noncoding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the final dataframe for CADD annotations for noncoding human derived data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>926666</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>42.0</td>\n",
       "      <td>108.0</td>\n",
       "      <td>2217.0</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>-0.115096</td>\n",
       "      <td>0.918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>926666</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>42.0</td>\n",
       "      <td>108.0</td>\n",
       "      <td>2217.0</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>-0.115096</td>\n",
       "      <td>0.918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>926854</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>41.0</td>\n",
       "      <td>109.0</td>\n",
       "      <td>2203.0</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.158437</td>\n",
       "      <td>3.522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>926854</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>41.0</td>\n",
       "      <td>109.0</td>\n",
       "      <td>2203.0</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.158437</td>\n",
       "      <td>3.522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101318</th>\n",
       "      <td>X</td>\n",
       "      <td>155939495</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>1553.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-0.315127</td>\n",
       "      <td>0.234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101319</th>\n",
       "      <td>X</td>\n",
       "      <td>155939592</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>1545.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.107893</td>\n",
       "      <td>2.886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101320</th>\n",
       "      <td>X</td>\n",
       "      <td>155939607</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>1545.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.515402</td>\n",
       "      <td>0.047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101321</th>\n",
       "      <td>X</td>\n",
       "      <td>155939608</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>9.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>1545.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.208316</td>\n",
       "      <td>0.509</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1535745 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom        Pos Ref Alt Type  Length           AnnoType Consequence  \\\n",
       "0           1     926666   G   A  SNV       0  RegulatoryFeature  REGULATORY   \n",
       "1           1     926666   G   A  SNV       0         Transcript    INTRONIC   \n",
       "2           1     926854   C   G  SNV       0  RegulatoryFeature  REGULATORY   \n",
       "3           1     926854   C   G  SNV       0         Transcript    INTRONIC   \n",
       "...       ...        ...  ..  ..  ...     ...                ...         ...   \n",
       "101318      X  155939495   G   C  SNV       0         Transcript    INTRONIC   \n",
       "101319      X  155939592   G   A  SNV       0         Transcript    INTRONIC   \n",
       "101320      X  155939607   T   C  SNV       0         Transcript    INTRONIC   \n",
       "101321      X  155939608   G   A  SNV       0         Transcript    INTRONIC   \n",
       "\n",
       "        ConsScore  ConsDetail  ...  Freq10000bp  Rare10000bp  Sngl10000bp  \\\n",
       "0               4  regulatory  ...         42.0        108.0       2217.0   \n",
       "1               2      intron  ...         42.0        108.0       2217.0   \n",
       "2               4  regulatory  ...         41.0        109.0       2203.0   \n",
       "3               2      intron  ...         41.0        109.0       2203.0   \n",
       "...           ...         ...  ...          ...          ...          ...   \n",
       "101318          2      intron  ...          9.0         73.0       1553.0   \n",
       "101319          2      intron  ...          9.0         73.0       1545.0   \n",
       "101320          2      intron  ...          9.0         73.0       1545.0   \n",
       "101321          2      intron  ...          9.0         73.0       1545.0   \n",
       "\n",
       "       EnsembleRegulatoryFeature  dbscSNV-ada_score  dbscSNV-rf_score  \\\n",
       "0                       Promoter                NaN               NaN   \n",
       "1                       Promoter                NaN               NaN   \n",
       "2                       Promoter                NaN               NaN   \n",
       "3                       Promoter                NaN               NaN   \n",
       "...                          ...                ...               ...   \n",
       "101318                       NaN                NaN               NaN   \n",
       "101319                       NaN                NaN               NaN   \n",
       "101320                       NaN                NaN               NaN   \n",
       "101321                       NaN                NaN               NaN   \n",
       "\n",
       "       RemapOverlapTF RemapOverlapCL  RawScore  PHRED  \n",
       "0                13.0           24.0 -0.115096  0.918  \n",
       "1                13.0           24.0 -0.115096  0.918  \n",
       "2                10.0           20.0  0.158437  3.522  \n",
       "3                10.0           20.0  0.158437  3.522  \n",
       "...               ...            ...       ...    ...  \n",
       "101318            1.0            1.0 -0.315127  0.234  \n",
       "101319            NaN            NaN  0.107893  2.886  \n",
       "101320            NaN            NaN -0.515402  0.047  \n",
       "101321            NaN            NaN -0.208316  0.509  \n",
       "\n",
       "[1535745 rows x 125 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_annotated_with_cadd = pd.concat([human_derived_noncoding_annotated_with_cadd_concat_1, human_derived_noncoding_annotated_with_cadd_concat_2, human_derived_noncoding_annotated_with_cadd_concat_3])\n",
    "human_derived_noncoding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
