{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import io\n",
    "import os\n",
    "import vcf\n",
    "import pysam\n",
    "import vcfpy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_rows', 8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_vcf(path):\n",
    "    with open(path, 'r') as f:\n",
    "        lines = [l for l in f if not l.startswith('##')]\n",
    "    return pd.read_csv(io.StringIO(''.join(lines)), dtype={'#CHROM': str, 'POS':int, 'ID':str, 'REF': str, 'ALT': str, 'QUAL': str, 'FILTER': str, 'INFO': str}, sep='\\t').rename(columns={'#CHROM': 'CHROM'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exporting Dataframes for CADD Annotation and Final Annotation Results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to annotate the files using CADD, they must be a very small size (2 MB) per file. Therefore, we will have to split up some of the larger dataframes into smaller sets of variants. In addition, we should cut off the last three columns of the dataframes, 'QUAL, FILTER, INFO' because they take up file space and are not needed for the annotations. Once the prior is finished, we need to export the dataframes using the format above with the long header, which we have already done a few times. Once we download them from the Jupyter home to the computer, we must open them in a text editor and cut out the header, but keep the column names. This way, it will work in the CADD tool. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## HGMD "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CADD:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the process used to export files for annotation for the non-coding region variants for HGMD for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988295</td>\n",
       "      <td>CM114899</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988415</td>\n",
       "      <td>CM164756</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988457</td>\n",
       "      <td>CM127719</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988526</td>\n",
       "      <td>CM1610318</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64559</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964053</td>\n",
       "      <td>CS1810957</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64560</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964054</td>\n",
       "      <td>CS1723659</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64561</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85965588</td>\n",
       "      <td>CS173873</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64562</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85968639</td>\n",
       "      <td>CS032064</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3807 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT\n",
       "9      chr10  100988295   CM114899   C   T\n",
       "10     chr10  100988415   CM164756   A   T\n",
       "11     chr10  100988457   CM127719   C   T\n",
       "12     chr10  100988526  CM1610318   A   G\n",
       "...      ...        ...        ...  ..  ..\n",
       "64559   chrX   85964053  CS1810957   C   G\n",
       "64560   chrX   85964054  CS1723659   T   C\n",
       "64561   chrX   85965588   CS173873   T   C\n",
       "64562   chrX   85968639   CS032064   A   T\n",
       "\n",
       "[3807 rows x 5 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_region_variants_only_for_cadd = hgmd_noncoding_region_variants_only.drop(['INFO', 'QUAL', 'FILTER'], axis = 1)\n",
    "hgmd_noncoding_region_variants_only_for_cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"hgmd_noncoding_for_cadd.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "hgmd_noncoding_region_variants_only_for_cadd.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated file for the HGMD non-coding region for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (13,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>26</td>\n",
       "      <td>48</td>\n",
       "      <td>1451</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>208.0</td>\n",
       "      <td>475.0</td>\n",
       "      <td>1.043084</td>\n",
       "      <td>12.600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>26</td>\n",
       "      <td>48</td>\n",
       "      <td>1451</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>208.0</td>\n",
       "      <td>475.0</td>\n",
       "      <td>1.043084</td>\n",
       "      <td>12.600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>74</td>\n",
       "      <td>1406</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.014</td>\n",
       "      <td>12.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-0.385227</td>\n",
       "      <td>0.135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>74</td>\n",
       "      <td>1406</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.014</td>\n",
       "      <td>12.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-0.385227</td>\n",
       "      <td>0.135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8609</th>\n",
       "      <td>X</td>\n",
       "      <td>154965965</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>36</td>\n",
       "      <td>843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99521</td>\n",
       "      <td>0.924</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.271896</td>\n",
       "      <td>20.600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8610</th>\n",
       "      <td>X</td>\n",
       "      <td>154965965</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>36</td>\n",
       "      <td>843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98832</td>\n",
       "      <td>0.926</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.302688</td>\n",
       "      <td>20.800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8611</th>\n",
       "      <td>X</td>\n",
       "      <td>155492384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.463914</td>\n",
       "      <td>21.700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8612</th>\n",
       "      <td>X</td>\n",
       "      <td>155492384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.463914</td>\n",
       "      <td>21.700</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8613 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0         1    7961859   C   G  SNV       0  RegulatoryFeature   \n",
       "1         1    7961859   C   G  SNV       0         Transcript   \n",
       "2         1    9720021   G   A  SNV       0  RegulatoryFeature   \n",
       "3         1    9720021   G   A  SNV       0         Transcript   \n",
       "...     ...        ...  ..  ..  ...     ...                ...   \n",
       "8609      X  154965965   C   G  SNV       0         Transcript   \n",
       "8610      X  154965965   C   T  SNV       0         Transcript   \n",
       "8611      X  155492384   C   A  SNV       0   CodingTranscript   \n",
       "8612      X  155492384   C   A  SNV       0         Transcript   \n",
       "\n",
       "         Consequence  ConsScore         ConsDetail  ...  Freq10000bp  \\\n",
       "0         REGULATORY          4         regulatory  ...           26   \n",
       "1           INTRONIC          2             intron  ...           26   \n",
       "2         REGULATORY          4         regulatory  ...            9   \n",
       "3        SPLICE_SITE          5      splice,intron  ...            9   \n",
       "...              ...        ...                ...  ...          ...   \n",
       "8609     SPLICE_SITE          5      splice,intron  ...            7   \n",
       "8610     SPLICE_SITE          5      splice,intron  ...            7   \n",
       "8611  NON_SYNONYMOUS          7           missense  ...            3   \n",
       "8612        INTRONIC          2  intron,non_coding  ...            3   \n",
       "\n",
       "      Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              48         1451                  Promoter                NaN   \n",
       "1              48         1451                  Promoter                NaN   \n",
       "2              74         1406         CTCF Binding Site            0.00003   \n",
       "3              74         1406         CTCF Binding Site            0.00003   \n",
       "...           ...          ...                       ...                ...   \n",
       "8609           36          843                       NaN            0.99521   \n",
       "8610           36          843                       NaN            0.98832   \n",
       "8611            3           19                       NaN                NaN   \n",
       "8612            3           19                       NaN                NaN   \n",
       "\n",
       "      dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore   PHRED  \n",
       "0                  NaN          208.0          475.0  1.043084  12.600  \n",
       "1                  NaN          208.0          475.0  1.043084  12.600  \n",
       "2                0.014           12.0           15.0 -0.385227   0.135  \n",
       "3                0.014           12.0           15.0 -0.385227   0.135  \n",
       "...                ...            ...            ...       ...     ...  \n",
       "8609             0.924            NaN            NaN  2.271896  20.600  \n",
       "8610             0.926            NaN            NaN  2.302688  20.800  \n",
       "8611               NaN            NaN            NaN  2.463914  21.700  \n",
       "8612               NaN            NaN            NaN  2.463914  21.700  \n",
       "\n",
       "[8613 rows x 125 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_annotated_cadd = pd.read_table('hgmd_noncoding_noheader.tsv', sep='\\t')\n",
    "hgmd_noncoding_annotated_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the process used to export files for annotation for the coding region variants for HGMD for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59426</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59427</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59428</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59429</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47645 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT\n",
       "0      chr10  100154922   CM140970   G   A\n",
       "1      chr10  100183802   CM140971   C   A\n",
       "2      chr10  100253438  CI1824020   A  AT\n",
       "3      chr10  100256298   CD162836  TG   T\n",
       "...      ...        ...        ...  ..  ..\n",
       "59426   chrX    9760731   CM981395   A   G\n",
       "59427   chrX    9760732   CI183806   G  GA\n",
       "59428   chrX    9760736   CD171619  GC   G\n",
       "59429   chrX    9760741   CI115195   A  AG\n",
       "\n",
       "[47645 rows x 5 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_region_variants_only_for_cadd = hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp_coding_region.drop(['INFO', 'QUAL', 'FILTER'], axis = 1)\n",
    "hgmd_coding_region_variants_only_for_cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"hgmd_coding_for_cadd.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "hgmd_coding_region_variants_only_for_cadd.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated file for the HGMD coding region for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89254</th>\n",
       "      <td>X</td>\n",
       "      <td>154966616</td>\n",
       "      <td>CTTCA</td>\n",
       "      <td>C</td>\n",
       "      <td>DEL</td>\n",
       "      <td>4</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>35</td>\n",
       "      <td>840</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.413846</td>\n",
       "      <td>21.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89255</th>\n",
       "      <td>X</td>\n",
       "      <td>154966617</td>\n",
       "      <td>TTC</td>\n",
       "      <td>T</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>35</td>\n",
       "      <td>840</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.392576</td>\n",
       "      <td>21.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89256</th>\n",
       "      <td>X</td>\n",
       "      <td>155524585</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>26</td>\n",
       "      <td>682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.231926</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89257</th>\n",
       "      <td>X</td>\n",
       "      <td>155524585</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>26</td>\n",
       "      <td>682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.231926</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89258 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos    Ref Alt Type  Length           AnnoType  \\\n",
       "0          1    1014143      C   T  SNV       0   CodingTranscript   \n",
       "1          1    1014143      C   T  SNV       0         Intergenic   \n",
       "2          1    1014143      C   T  SNV       0  RegulatoryFeature   \n",
       "3          1    1014143      C   T  SNV       0  RegulatoryFeature   \n",
       "...      ...        ...    ...  ..  ...     ...                ...   \n",
       "89254      X  154966616  CTTCA   C  DEL       4   CodingTranscript   \n",
       "89255      X  154966617    TTC   T  DEL       2   CodingTranscript   \n",
       "89256      X  155524585      G   A  SNV       0   CodingTranscript   \n",
       "89257      X  155524585      G   A  SNV       0         Transcript   \n",
       "\n",
       "       Consequence  ConsScore         ConsDetail  ...  Freq10000bp  \\\n",
       "0      STOP_GAINED          8        stop_gained  ...           22   \n",
       "1         UPSTREAM          1           upstream  ...           22   \n",
       "2       REGULATORY          4         regulatory  ...           22   \n",
       "3       REGULATORY          4         regulatory  ...           22   \n",
       "...            ...        ...                ...  ...          ...   \n",
       "89254  FRAME_SHIFT          7         frameshift  ...            7   \n",
       "89255  FRAME_SHIFT          7         frameshift  ...            7   \n",
       "89256  STOP_GAINED          8        stop_gained  ...            9   \n",
       "89257     INTRONIC          2  intron,non_coding  ...            9   \n",
       "\n",
       "       Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0               87         1622                  Promoter                NaN   \n",
       "1               87         1622                  Promoter                NaN   \n",
       "2               87         1622                  Promoter                NaN   \n",
       "3               87         1622                  Promoter                NaN   \n",
       "...            ...          ...                       ...                ...   \n",
       "89254           35          840                       NaN                NaN   \n",
       "89255           35          840                       NaN                NaN   \n",
       "89256           26          682                       NaN                NaN   \n",
       "89257           26          682                       NaN                NaN   \n",
       "\n",
       "       dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore PHRED  \n",
       "0                   NaN           75.0          107.0  5.410558  35.0  \n",
       "1                   NaN           75.0          107.0  5.410558  35.0  \n",
       "2                   NaN           75.0          107.0  5.410558  35.0  \n",
       "3                   NaN           75.0          107.0  5.410558  35.0  \n",
       "...                 ...            ...            ...       ...   ...  \n",
       "89254               NaN            NaN            NaN  2.413846  21.5  \n",
       "89255               NaN            NaN            NaN  2.392576  21.3  \n",
       "89256               NaN            4.0            4.0  6.231926  35.0  \n",
       "89257               NaN            4.0            4.0  6.231926  35.0  \n",
       "\n",
       "[89258 rows x 125 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_annotated_cadd = pd.read_table('hgmd_coding_noheader.tsv', sep='\\t')\n",
    "hgmd_coding_annotated_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ClinVar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### FOR ALL CLINVAR:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**I made the CHROM column values begin with chrchr# on accident. It should be set as chr# only. Therefore, I needed to change this. The documentation to do so is found under the ClinVar section of the \"3)annotation using CADD script\" section. I removed the string 'chrchr', cut out the column names row, and added the string 'chr.' I did this for every ClinVar dataframe here since they all share the same format.**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CADD:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the process used to export files for annotation for the non-coding regions for pathogenic variants from ClinVar for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100988540</td>\n",
       "      <td>225837</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100989084</td>\n",
       "      <td>488187</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100989118</td>\n",
       "      <td>4628</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100989154</td>\n",
       "      <td>4620</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121537</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>74529428</td>\n",
       "      <td>212188</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121688</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>77902641</td>\n",
       "      <td>39768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121696</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>78003237</td>\n",
       "      <td>11786</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121713</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>78122954</td>\n",
       "      <td>9955</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1114 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID REF ALT\n",
       "38      chrchr10  100988540  225837  CT   C\n",
       "51      chrchr10  100989084  488187   C   A\n",
       "54      chrchr10  100989118    4628   G   A\n",
       "56      chrchr10  100989154    4620   G   T\n",
       "...          ...        ...     ...  ..  ..\n",
       "121537   chrchrX   74529428  212188   G  GC\n",
       "121688   chrchrX   77902641   39768   A   G\n",
       "121696   chrchrX   78003237   11786   G   A\n",
       "121713   chrchrX   78122954    9955   G   A\n",
       "\n",
       "[1114 rows x 5 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_noncoding_region_pathogenic_variants_only_for_cadd = clinvar_noncoding_region_pathogenic_variants_only.drop(['INFO', 'QUAL', 'FILTER'], axis = 1)\n",
    "clinvar_noncoding_region_pathogenic_variants_only_for_cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"clinvar_noncoding_pathogenic_for_cadd.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "clinvar_noncoding_region_pathogenic_variants_only_for_cadd.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated file for the ClinVar non-coding region pathogenic variants for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>11960768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>90</td>\n",
       "      <td>1434</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>0.99998</td>\n",
       "      <td>0.942</td>\n",
       "      <td>27.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>5.268674</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11960768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_donor</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>90</td>\n",
       "      <td>1434</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>0.99998</td>\n",
       "      <td>0.942</td>\n",
       "      <td>27.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>5.268674</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>11964787</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_donor</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>99</td>\n",
       "      <td>1332</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99987</td>\n",
       "      <td>0.886</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>4.686023</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>11972977</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>95</td>\n",
       "      <td>1304</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.497634</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2641</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>splice,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>7.476349</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2642</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>7.476349</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2643</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "      <td>828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>0.944</td>\n",
       "      <td>38.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>4.640280</td>\n",
       "      <td>29.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2644</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_acceptor</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "      <td>828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>0.944</td>\n",
       "      <td>38.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>4.640280</td>\n",
       "      <td>29.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2645 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0         1   11960768   G   A  SNV       0  RegulatoryFeature   \n",
       "1         1   11960768   G   A  SNV       0         Transcript   \n",
       "2         1   11964787   T   C  SNV       0         Transcript   \n",
       "3         1   11972977   C   T  SNV       0   CodingTranscript   \n",
       "...     ...        ...  ..  ..  ...     ...                ...   \n",
       "2641      X  154419746   G   T  SNV       0   CodingTranscript   \n",
       "2642      X  154419746   G   T  SNV       0         Intergenic   \n",
       "2643      X  154420211   G   C  SNV       0         Intergenic   \n",
       "2644      X  154420211   G   C  SNV       0         Transcript   \n",
       "\n",
       "           Consequence  ConsScore          ConsDetail  ...  Freq10000bp  \\\n",
       "0           REGULATORY          4          regulatory  ...           21   \n",
       "1     CANONICAL_SPLICE          6        splice_donor  ...           21   \n",
       "2     CANONICAL_SPLICE          6        splice_donor  ...           19   \n",
       "3          STOP_GAINED          8         stop_gained  ...           20   \n",
       "...                ...        ...                 ...  ...          ...   \n",
       "2641       STOP_GAINED          8  splice,stop_gained  ...           11   \n",
       "2642        DOWNSTREAM          1          downstream  ...           11   \n",
       "2643        DOWNSTREAM          1          downstream  ...           12   \n",
       "2644  CANONICAL_SPLICE          6     splice_acceptor  ...           12   \n",
       "\n",
       "      Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              90         1434  Promoter Flanking Region            0.99998   \n",
       "1              90         1434  Promoter Flanking Region            0.99998   \n",
       "2              99         1332                       NaN            0.99987   \n",
       "3              95         1304  Promoter Flanking Region                NaN   \n",
       "...           ...          ...                       ...                ...   \n",
       "2641           29          822                       NaN            0.99999   \n",
       "2642           29          822                       NaN            0.99999   \n",
       "2643           32          828                       NaN            0.99999   \n",
       "2644           32          828                       NaN            0.99999   \n",
       "\n",
       "      dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore PHRED  \n",
       "0                0.942           27.0           36.0  5.268674  35.0  \n",
       "1                0.942           27.0           36.0  5.268674  35.0  \n",
       "2                0.886           10.0           10.0  4.686023  31.0  \n",
       "3                  NaN           19.0           23.0  7.497634  38.0  \n",
       "...                ...            ...            ...       ...   ...  \n",
       "2641               1.0            9.0           10.0  7.476349  38.0  \n",
       "2642               1.0            9.0           10.0  7.476349  38.0  \n",
       "2643             0.944           38.0           46.0  4.640280  29.8  \n",
       "2644             0.944           38.0           46.0  4.640280  29.8  \n",
       "\n",
       "[2645 rows x 125 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_noncoding_pathogenic_annotated = pd.read_table('clinvar_noncoding_pathogenic_noheader.tsv', sep='\\t')\n",
    "clinvar_noncoding_pathogenic_annotated"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the process used to export files for annotation for the non-coding regions for benign variants from ClinVar for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100989312</td>\n",
       "      <td>136588</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100990864</td>\n",
       "      <td>136589</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100990866</td>\n",
       "      <td>136590</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100991026</td>\n",
       "      <td>136591</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121408</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>213614</td>\n",
       "      <td>CCTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121410</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>95249</td>\n",
       "      <td>CCTCTT</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121412</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>95251</td>\n",
       "      <td>CCTCTTCTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121709</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>78118027</td>\n",
       "      <td>558817</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1560 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID                              REF ALT\n",
       "73      chrchr10  100989312  136588                                G   A\n",
       "102     chrchr10  100990864  136589                                C   T\n",
       "103     chrchr10  100990866  136590                                T   C\n",
       "108     chrchr10  100991026  136591                                C   A\n",
       "...          ...        ...     ...                              ...  ..\n",
       "121408   chrchrX   71132767  213614       CCTCTTCTCTTCTCTTCTCTTCTCTT   C\n",
       "121410   chrchrX   71132767   95249                           CCTCTT   C\n",
       "121412   chrchrX   71132767   95251  CCTCTTCTCTTCTCTTCTCTTCTCTTCTCTT   C\n",
       "121709   chrchrX   78118027  558817                                C   T\n",
       "\n",
       "[1560 rows x 5 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_noncoding_region_benign_variants_only_for_cadd = clinvar_noncoding_region_benign_variants_only.drop(['INFO', 'QUAL', 'FILTER'], axis = 1)\n",
    "clinvar_noncoding_region_benign_variants_only_for_cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"clinvar_noncoding_benign_for_cadd.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "clinvar_noncoding_region_benign_variants_only_for_cadd.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated file for the ClinVar non-coding region benign variants for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1041950</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>100</td>\n",
       "      <td>1931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>0.002</td>\n",
       "      <td>10.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.092155</td>\n",
       "      <td>2.700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1042190</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>109</td>\n",
       "      <td>1955</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>-0.401276</td>\n",
       "      <td>0.118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>117</td>\n",
       "      <td>2004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-0.012669</td>\n",
       "      <td>1.640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1045707</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>131</td>\n",
       "      <td>2126</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-0.057159</td>\n",
       "      <td>1.287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3200</th>\n",
       "      <td>X</td>\n",
       "      <td>154419685</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.033909</td>\n",
       "      <td>1.464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>X</td>\n",
       "      <td>154420108</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>31</td>\n",
       "      <td>826</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>-0.549155</td>\n",
       "      <td>0.036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3202</th>\n",
       "      <td>X</td>\n",
       "      <td>154420108</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>31</td>\n",
       "      <td>826</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>-0.549155</td>\n",
       "      <td>0.036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3203</th>\n",
       "      <td>X</td>\n",
       "      <td>154961190</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>45</td>\n",
       "      <td>823</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.351934</td>\n",
       "      <td>5.879</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3204 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos  Ref Alt Type  Length    AnnoType  Consequence  \\\n",
       "0         1    1041950    T   C  SNV       0  Transcript  SPLICE_SITE   \n",
       "1         1    1042190    G   A  SNV       0  Transcript     INTRONIC   \n",
       "2         1    1043223  CCT   C  DEL       2  Transcript     INTRONIC   \n",
       "3         1    1045707    A   G  SNV       0  Transcript     INTRONIC   \n",
       "...     ...        ...  ...  ..  ...     ...         ...          ...   \n",
       "3200      X  154419685    C   T  SNV       0  Transcript     INTRONIC   \n",
       "3201      X  154420108    C   T  SNV       0  Intergenic   DOWNSTREAM   \n",
       "3202      X  154420108    C   T  SNV       0  Transcript     INTRONIC   \n",
       "3203      X  154961190    A   G  SNV       0  Transcript     INTRONIC   \n",
       "\n",
       "      ConsScore     ConsDetail  ...  Freq10000bp  Rare10000bp  Sngl10000bp  \\\n",
       "0             5  splice,intron  ...           23          100         1931   \n",
       "1             2         intron  ...           25          109         1955   \n",
       "2             2         intron  ...           24          117         2004   \n",
       "3             2         intron  ...           22          131         2126   \n",
       "...         ...            ...  ...          ...          ...          ...   \n",
       "3200          2         intron  ...           11           29          820   \n",
       "3201          1     downstream  ...           12           31          826   \n",
       "3202          2         intron  ...           12           31          826   \n",
       "3203          2         intron  ...            7           45          823   \n",
       "\n",
       "     EnsembleRegulatoryFeature  dbscSNV-ada_score  dbscSNV-rf_score  \\\n",
       "0                          NaN            0.00001             0.002   \n",
       "1                          NaN                NaN               NaN   \n",
       "2                          NaN                NaN               NaN   \n",
       "3                          NaN                NaN               NaN   \n",
       "...                        ...                ...               ...   \n",
       "3200                       NaN                NaN               NaN   \n",
       "3201                       NaN                NaN               NaN   \n",
       "3202                       NaN                NaN               NaN   \n",
       "3203                       NaN                NaN               NaN   \n",
       "\n",
       "     RemapOverlapTF RemapOverlapCL  RawScore  PHRED  \n",
       "0              10.0           13.0  0.092155  2.700  \n",
       "1              12.0           14.0 -0.401276  0.118  \n",
       "2               8.0            9.0 -0.012669  1.640  \n",
       "3               9.0           10.0 -0.057159  1.287  \n",
       "...             ...            ...       ...    ...  \n",
       "3200            4.0            4.0 -0.033909  1.464  \n",
       "3201           36.0           47.0 -0.549155  0.036  \n",
       "3202           36.0           47.0 -0.549155  0.036  \n",
       "3203            1.0            1.0  0.351934  5.879  \n",
       "\n",
       "[3204 rows x 125 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_noncoding_pathogenic_annotated = pd.read_table('clinvar_noncoding_benign_noheader.tsv', sep='\\t')\n",
    "clinvar_noncoding_pathogenic_annotated"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the process used to export files for annotation for the coding regions for pathogenic coding variants from ClinVar for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>226426</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>226427</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100246864</td>\n",
       "      <td>504028</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100253422</td>\n",
       "      <td>419610</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101084</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85981796</td>\n",
       "      <td>279771</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101094</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9759332</td>\n",
       "      <td>10517</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101098</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>10516</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101099</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>10519</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16775 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID REF ALT\n",
       "1       chrchr10  100154922  226426   G   A\n",
       "3       chrchr10  100183802  226427   C   A\n",
       "7       chrchr10  100246864  504028  AT   A\n",
       "11      chrchr10  100253422  419610   G   A\n",
       "...          ...        ...     ...  ..  ..\n",
       "101084   chrchrX   85981796  279771   C   A\n",
       "101094   chrchrX    9759332   10517   C   T\n",
       "101098   chrchrX    9759390   10516   A   G\n",
       "101099   chrchrX    9759390   10519   A   T\n",
       "\n",
       "[16775 rows x 5 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_coding_region_pathogenic_variants_only_for_cadd = clinvar_coding_region_pathogenic_variants_only.drop(['INFO', 'QUAL', 'FILTER'], axis = 1)\n",
    "clinvar_coding_region_pathogenic_variants_only_for_cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"clinvar_coding_pathogenic_for_cadd.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "clinvar_coding_region_pathogenic_variants_only_for_cadd.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated file for the ClinVar coding region pathogenic variants for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8.0</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1.0</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31074</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7.0</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>33</td>\n",
       "      <td>827</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.198603</td>\n",
       "      <td>13.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31075</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>33</td>\n",
       "      <td>827</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.198603</td>\n",
       "      <td>13.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31076</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7.0</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>39</td>\n",
       "      <td>652</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.018377</td>\n",
       "      <td>25.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31077</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>39</td>\n",
       "      <td>652</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.018377</td>\n",
       "      <td>25.80</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31078 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos  Ref Alt Type  Length           AnnoType  \\\n",
       "0          1    1014143    C   T  SNV       0   CodingTranscript   \n",
       "1          1    1014143    C   T  SNV       0         Intergenic   \n",
       "2          1    1014143    C   T  SNV       0  RegulatoryFeature   \n",
       "3          1    1014143    C   T  SNV       0  RegulatoryFeature   \n",
       "...      ...        ...  ...  ..  ...     ...                ...   \n",
       "31074      X  155280059    G   C  SNV       0   CodingTranscript   \n",
       "31075      X  155280059    G   C  SNV       0  RegulatoryFeature   \n",
       "31076      X  155506930  GAT   G  DEL       2   CodingTranscript   \n",
       "31077      X  155506930  GAT   G  DEL       2         Transcript   \n",
       "\n",
       "          Consequence  ConsScore         ConsDetail  ...  Freq10000bp  \\\n",
       "0         STOP_GAINED        8.0        stop_gained  ...           22   \n",
       "1            UPSTREAM        1.0           upstream  ...           22   \n",
       "2          REGULATORY        4.0         regulatory  ...           22   \n",
       "3          REGULATORY        4.0         regulatory  ...           22   \n",
       "...               ...        ...                ...  ...          ...   \n",
       "31074  NON_SYNONYMOUS        7.0           missense  ...           11   \n",
       "31075      REGULATORY        4.0         regulatory  ...           11   \n",
       "31076     FRAME_SHIFT        7.0         frameshift  ...            6   \n",
       "31077        INTRONIC        2.0  intron,non_coding  ...            6   \n",
       "\n",
       "       Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0               87         1622                  Promoter                NaN   \n",
       "1               87         1622                  Promoter                NaN   \n",
       "2               87         1622                  Promoter                NaN   \n",
       "3               87         1622                  Promoter                NaN   \n",
       "...            ...          ...                       ...                ...   \n",
       "31074           33          827  Promoter Flanking Region                NaN   \n",
       "31075           33          827  Promoter Flanking Region                NaN   \n",
       "31076           39          652                       NaN                NaN   \n",
       "31077           39          652                       NaN                NaN   \n",
       "\n",
       "       dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore  PHRED  \n",
       "0                   NaN           75.0          107.0  5.410558  35.00  \n",
       "1                   NaN           75.0          107.0  5.410558  35.00  \n",
       "2                   NaN           75.0          107.0  5.410558  35.00  \n",
       "3                   NaN           75.0          107.0  5.410558  35.00  \n",
       "...                 ...            ...            ...       ...    ...  \n",
       "31074               NaN            NaN            NaN  1.198603  13.85  \n",
       "31075               NaN            NaN            NaN  1.198603  13.85  \n",
       "31076               NaN            1.0            1.0  4.018377  25.80  \n",
       "31077               NaN            1.0            1.0  4.018377  25.80  \n",
       "\n",
       "[31078 rows x 125 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_coding_pathogenic_annotated = pd.read_table('clinvar_coding_pathogenic_noheader.tsv', sep='\\t')\n",
    "clinvar_coding_pathogenic_annotated"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the process used to export files for annotation for the coding regions for benign coding variants from ClinVar for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100987606</td>\n",
       "      <td>136593</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>102065918</td>\n",
       "      <td>284200</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>102396271</td>\n",
       "      <td>474786</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>102399466</td>\n",
       "      <td>541631</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101076</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85964016</td>\n",
       "      <td>255991</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101079</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85978770</td>\n",
       "      <td>497462</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101081</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85978816</td>\n",
       "      <td>377662</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101087</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>93671995</td>\n",
       "      <td>208906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4659 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID REF ALT\n",
       "18      chrchr10  100987606  136593   G   T\n",
       "57      chrchr10  102065918  284200   C   G\n",
       "82      chrchr10  102396271  474786   G   A\n",
       "87      chrchr10  102399466  541631   C   T\n",
       "...          ...        ...     ...  ..  ..\n",
       "101076   chrchrX   85964016  255991   T   C\n",
       "101079   chrchrX   85978770  497462   G   C\n",
       "101081   chrchrX   85978816  377662   T   A\n",
       "101087   chrchrX   93671995  208906   G   C\n",
       "\n",
       "[4659 rows x 5 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_coding_region_benign_variants_only_for_cadd = clinvar_coding_region_benign_variants_only.drop(['INFO', 'QUAL', 'FILTER'], axis = 1)\n",
    "clinvar_coding_region_benign_variants_only_for_cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"clinvar_coding_benign_for_cadd.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "clinvar_coding_region_benign_variants_only_for_cadd.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated file for the ClinVar coding region benign variants for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1616</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1616</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1616</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1616</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9134</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>3</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>INFRAME</td>\n",
       "      <td>6</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.893695</td>\n",
       "      <td>10.880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9135</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>3</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.893695</td>\n",
       "      <td>10.880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9136</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>3</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.893695</td>\n",
       "      <td>10.880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9137</th>\n",
       "      <td>X</td>\n",
       "      <td>154929926</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>5</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>878</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.089486</td>\n",
       "      <td>2.669</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9138 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref   Alt Type  Length           AnnoType  \\\n",
       "0         1    1014042   G     A  SNV       0   CodingTranscript   \n",
       "1         1    1014042   G     A  SNV       0         Intergenic   \n",
       "2         1    1014042   G     A  SNV       0  RegulatoryFeature   \n",
       "3         1    1014042   G     A  SNV       0  RegulatoryFeature   \n",
       "...     ...        ...  ..   ...  ...     ...                ...   \n",
       "9134      X  154776813   C  CAAG  INS       3   CodingTranscript   \n",
       "9135      X  154776813   C  CAAG  INS       3         Intergenic   \n",
       "9136      X  154776813   C  CAAG  INS       3         Intergenic   \n",
       "9137      X  154929926   T     G  SNV       0   CodingTranscript   \n",
       "\n",
       "         Consequence  ConsScore         ConsDetail  ...  Freq10000bp  \\\n",
       "0     NON_SYNONYMOUS          7           missense  ...           22   \n",
       "1           UPSTREAM          1           upstream  ...           22   \n",
       "2         REGULATORY          4         regulatory  ...           22   \n",
       "3         REGULATORY          4         regulatory  ...           22   \n",
       "...              ...        ...                ...  ...          ...   \n",
       "9134         INFRAME          6  inframe_insertion  ...            4   \n",
       "9135      DOWNSTREAM          1         downstream  ...            4   \n",
       "9136      DOWNSTREAM          1         downstream  ...            4   \n",
       "9137      SYNONYMOUS          5         synonymous  ...            3   \n",
       "\n",
       "      Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              87         1616                  Promoter                NaN   \n",
       "1              87         1616                  Promoter                NaN   \n",
       "2              87         1616                  Promoter                NaN   \n",
       "3              87         1616                  Promoter                NaN   \n",
       "...           ...          ...                       ...                ...   \n",
       "9134           22          980                       NaN                NaN   \n",
       "9135           22          980                       NaN                NaN   \n",
       "9136           22          980                       NaN                NaN   \n",
       "9137           31          878                       NaN                NaN   \n",
       "\n",
       "      dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore   PHRED  \n",
       "0                  NaN           69.0           97.0  0.076262   2.519  \n",
       "1                  NaN           69.0           97.0  0.076262   2.519  \n",
       "2                  NaN           69.0           97.0  0.076262   2.519  \n",
       "3                  NaN           69.0           97.0  0.076262   2.519  \n",
       "...                ...            ...            ...       ...     ...  \n",
       "9134               NaN            7.0            7.0  0.893695  10.880  \n",
       "9135               NaN            7.0            7.0  0.893695  10.880  \n",
       "9136               NaN            7.0            7.0  0.893695  10.880  \n",
       "9137               NaN            NaN            NaN  0.089486   2.669  \n",
       "\n",
       "[9138 rows x 125 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_coding_benign_annotated = pd.read_table('clinvar_coding_benign_noheader.tsv', sep='\\t')\n",
    "clinvar_coding_benign_annotated"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Human Derived "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CADD:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the process used to export files for annotation for the noncoding regions from the Human Derived data set:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since there are already only 5 columns for this dataset, we do not have to code anything to drop any columns as we have had to do with the other dataframes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "human_derived_noncoding_region_variants_only - File we will use for our annotations for CADD."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000013</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000297</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000555</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020168</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931817</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020169</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931818</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020170</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931993</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020171</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9932000</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>966382 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM        POS ID REF ALT\n",
       "0        chr10    1000013  .   G   A\n",
       "1        chr10  100020652  .   G   A\n",
       "2        chr10    1000297  .   T   G\n",
       "3        chr10    1000555  .   A   T\n",
       "...        ...        ... ..  ..  ..\n",
       "1020168   chrX    9931817  .   T   C\n",
       "1020169   chrX    9931818  .   G   A\n",
       "1020170   chrX    9931993  .   T   C\n",
       "1020171   chrX    9932000  .   C   T\n",
       "\n",
       "[966382 rows x 5 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_region_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since this file is so large, I will break it into 10 separate pieces, each containing about 90,000 variants. The  code for this is shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000013</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000297</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000555</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94277</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312241</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94278</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312402</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94279</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312428</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94280</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312687</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90000 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS ID REF ALT\n",
       "0      chr10    1000013  .   G   A\n",
       "1      chr10  100020652  .   G   A\n",
       "2      chr10    1000297  .   T   G\n",
       "3      chr10    1000555  .   A   T\n",
       "...      ...        ... ..  ..  ..\n",
       "94277  chr11   47312241  .   G   A\n",
       "94278  chr11   47312402  .   G   C\n",
       "94279  chr11   47312428  .   C   T\n",
       "94280  chr11   47312687  .   T   C\n",
       "\n",
       "[90000 rows x 5 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_region_variants_only_1 = human_derived_noncoding_region_variants_only[:90000]\n",
    "human_derived_noncoding_region_variants_only_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_1.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_1.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>94281</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312814</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94282</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47312979</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94283</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47313760</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94284</th>\n",
       "      <td>chr11</td>\n",
       "      <td>47313853</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190691</th>\n",
       "      <td>chr12</td>\n",
       "      <td>65348795</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190692</th>\n",
       "      <td>chr12</td>\n",
       "      <td>65348797</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190693</th>\n",
       "      <td>chr12</td>\n",
       "      <td>65348836</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190694</th>\n",
       "      <td>chr12</td>\n",
       "      <td>65348914</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90000 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM       POS ID REF ALT\n",
       "94281   chr11  47312814  .   G   A\n",
       "94282   chr11  47312979  .   T   G\n",
       "94283   chr11  47313760  .   G   T\n",
       "94284   chr11  47313853  .   A   G\n",
       "...       ...       ... ..  ..  ..\n",
       "190691  chr12  65348795  .   C   A\n",
       "190692  chr12  65348797  .   C   T\n",
       "190693  chr12  65348836  .   C   T\n",
       "190694  chr12  65348914  .   C   T\n",
       "\n",
       "[90000 rows x 5 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_region_variants_only_2 = human_derived_noncoding_region_variants_only[90000:180000]\n",
    "human_derived_noncoding_region_variants_only_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_2.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_2.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_variants_only_3 = human_derived_noncoding_region_variants_only[180000:270000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_3.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_3.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_variants_only_4 = human_derived_noncoding_region_variants_only[270000:360000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_4.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_4.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_variants_only_5 = human_derived_noncoding_region_variants_only[360000:450000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_5.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_5.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_variants_only_6 = human_derived_noncoding_region_variants_only[450000:540000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_6.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_6.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_variants_only_7 = human_derived_noncoding_region_variants_only[540000:630000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_7.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_7.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_variants_only_8 = human_derived_noncoding_region_variants_only[630000:720000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_8.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_8.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_region_variants_only_9 = human_derived_noncoding_region_variants_only[720000:810000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_9.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_9.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>855720</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107252617</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>855721</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107252854</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>855722</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107252864</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>855723</th>\n",
       "      <td>chr7</td>\n",
       "      <td>107283879</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949468</th>\n",
       "      <td>chr8</td>\n",
       "      <td>47381618</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949469</th>\n",
       "      <td>chr8</td>\n",
       "      <td>47381868</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949470</th>\n",
       "      <td>chr8</td>\n",
       "      <td>47382570</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949471</th>\n",
       "      <td>chr8</td>\n",
       "      <td>47382847</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90000 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS ID REF ALT\n",
       "855720  chr7  107252617  .   A   T\n",
       "855721  chr7  107252854  .   T   C\n",
       "855722  chr7  107252864  .   A   T\n",
       "855723  chr7  107283879  .   C   T\n",
       "...      ...        ... ..  ..  ..\n",
       "949468  chr8   47381618  .   T   C\n",
       "949469  chr8   47381868  .   A   C\n",
       "949470  chr8   47382570  .   A   G\n",
       "949471  chr8   47382847  .   A   C\n",
       "\n",
       "[90000 rows x 5 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_region_variants_only_10 = human_derived_noncoding_region_variants_only[810000:900000]\n",
    "human_derived_noncoding_region_variants_only_10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_10.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_10.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>949472</th>\n",
       "      <td>chr8</td>\n",
       "      <td>47382972</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949473</th>\n",
       "      <td>chr8</td>\n",
       "      <td>47382981</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949474</th>\n",
       "      <td>chr8</td>\n",
       "      <td>47383038</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>949475</th>\n",
       "      <td>chr8</td>\n",
       "      <td>47383524</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020168</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931817</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020169</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931818</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020170</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931993</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020171</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9932000</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66382 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM       POS ID REF ALT\n",
       "949472   chr8  47382972  .   A   G\n",
       "949473   chr8  47382981  .   T   C\n",
       "949474   chr8  47383038  .   A   G\n",
       "949475   chr8  47383524  .   G   A\n",
       "...       ...       ... ..  ..  ..\n",
       "1020168  chrX   9931817  .   T   C\n",
       "1020169  chrX   9931818  .   G   A\n",
       "1020170  chrX   9931993  .   T   C\n",
       "1020171  chrX   9932000  .   C   T\n",
       "\n",
       "[66382 rows x 5 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_region_variants_only_11 = human_derived_noncoding_region_variants_only[900000:]\n",
    "human_derived_noncoding_region_variants_only_11"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_noncoding_for_cadd_11.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_noncoding_region_variants_only_11.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below are the annotated files for the human derived non-coding region variants for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (16,17,30,35,37,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1001313</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>25.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>1931.0</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>0.053718</td>\n",
       "      <td>2.275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1001313</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>25.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>1931.0</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>0.053718</td>\n",
       "      <td>2.275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1001313</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>25.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>1931.0</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>0.053718</td>\n",
       "      <td>2.275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1001313</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>25.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>1931.0</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>0.053718</td>\n",
       "      <td>2.275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135092</th>\n",
       "      <td>11</td>\n",
       "      <td>134383329</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>23.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>1401.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.154648</td>\n",
       "      <td>0.721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135093</th>\n",
       "      <td>11</td>\n",
       "      <td>134383329</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>23.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>1401.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.154648</td>\n",
       "      <td>0.721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135094</th>\n",
       "      <td>11</td>\n",
       "      <td>134383599</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>24.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>1411.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>-0.097920</td>\n",
       "      <td>1.017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135095</th>\n",
       "      <td>11</td>\n",
       "      <td>134383614</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>24.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>1412.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.049698</td>\n",
       "      <td>2.233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>135096 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #Chrom        Pos Ref Alt Type  Length           AnnoType Consequence  \\\n",
       "0            1    1001313   G   T  SNV       0  RegulatoryFeature  REGULATORY   \n",
       "1            1    1001313   G   T  SNV       0  RegulatoryFeature  REGULATORY   \n",
       "2            1    1001313   G   T  SNV       0         Intergenic    UPSTREAM   \n",
       "3            1    1001313   G   T  SNV       0         Transcript    INTRONIC   \n",
       "...        ...        ...  ..  ..  ...     ...                ...         ...   \n",
       "135092      11  134383329   G   C  SNV       0         Intergenic  DOWNSTREAM   \n",
       "135093      11  134383329   G   C  SNV       0         Transcript    INTRONIC   \n",
       "135094      11  134383599   G   A  SNV       0         Transcript    INTRONIC   \n",
       "135095      11  134383614   T   C  SNV       0         Transcript    INTRONIC   \n",
       "\n",
       "        ConsScore  ConsDetail  ...  Freq10000bp  Rare10000bp  Sngl10000bp  \\\n",
       "0               4  regulatory  ...         25.0         88.0       1931.0   \n",
       "1               4  regulatory  ...         25.0         88.0       1931.0   \n",
       "2               1    upstream  ...         25.0         88.0       1931.0   \n",
       "3               2      intron  ...         25.0         88.0       1931.0   \n",
       "...           ...         ...  ...          ...          ...          ...   \n",
       "135092          1  downstream  ...         23.0         62.0       1401.0   \n",
       "135093          2      intron  ...         23.0         62.0       1401.0   \n",
       "135094          2      intron  ...         24.0         59.0       1411.0   \n",
       "135095          2      intron  ...         24.0         60.0       1412.0   \n",
       "\n",
       "       EnsembleRegulatoryFeature  dbscSNV-ada_score  dbscSNV-rf_score  \\\n",
       "0                       Promoter                NaN               NaN   \n",
       "1                       Promoter                NaN               NaN   \n",
       "2                       Promoter                NaN               NaN   \n",
       "3                       Promoter                NaN               NaN   \n",
       "...                          ...                ...               ...   \n",
       "135092                       NaN                NaN               NaN   \n",
       "135093                       NaN                NaN               NaN   \n",
       "135094                       NaN                NaN               NaN   \n",
       "135095                       NaN                NaN               NaN   \n",
       "\n",
       "       RemapOverlapTF RemapOverlapCL  RawScore  PHRED  \n",
       "0                60.0           90.0  0.053718  2.275  \n",
       "1                60.0           90.0  0.053718  2.275  \n",
       "2                60.0           90.0  0.053718  2.275  \n",
       "3                60.0           90.0  0.053718  2.275  \n",
       "...               ...            ...       ...    ...  \n",
       "135092            NaN            NaN -0.154648  0.721  \n",
       "135093            NaN            NaN -0.154648  0.721  \n",
       "135094           10.0           12.0 -0.097920  1.017  \n",
       "135095           10.0           12.0  0.049698  2.233  \n",
       "\n",
       "[135096 rows x 125 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_annotated_1 = pd.read_table('human_derived_noncoding_for_cadd_1_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (13,16,17,30,35,37,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1668612</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>80</td>\n",
       "      <td>1740</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.161503</td>\n",
       "      <td>0.690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1668612</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>80</td>\n",
       "      <td>1740</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.161503</td>\n",
       "      <td>0.690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1668740</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>84</td>\n",
       "      <td>1743</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.209266</td>\n",
       "      <td>0.506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1668740</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>84</td>\n",
       "      <td>1743</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.209266</td>\n",
       "      <td>0.506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148269</th>\n",
       "      <td>12</td>\n",
       "      <td>133056532</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>60</td>\n",
       "      <td>1336</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.331295</td>\n",
       "      <td>0.206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148270</th>\n",
       "      <td>12</td>\n",
       "      <td>133056612</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>59</td>\n",
       "      <td>1330</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.448011</td>\n",
       "      <td>0.081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148271</th>\n",
       "      <td>12</td>\n",
       "      <td>133056705</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>59</td>\n",
       "      <td>1337</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.019107</td>\n",
       "      <td>1.585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148272</th>\n",
       "      <td>12</td>\n",
       "      <td>133056935</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>59</td>\n",
       "      <td>1336</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.231915</td>\n",
       "      <td>0.433</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>148273 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #Chrom        Pos Ref Alt Type  Length    AnnoType Consequence  \\\n",
       "0            1    1668612   C   G  SNV       0  Transcript    INTRONIC   \n",
       "1            1    1668612   C   G  SNV       0  Intergenic    UPSTREAM   \n",
       "2            1    1668740   G   A  SNV       0  Transcript    INTRONIC   \n",
       "3            1    1668740   G   A  SNV       0  Intergenic    UPSTREAM   \n",
       "...        ...        ...  ..  ..  ...     ...         ...         ...   \n",
       "148269      12  133056532   T   C  SNV       0  Transcript    INTRONIC   \n",
       "148270      12  133056612   G   A  SNV       0  Transcript    INTRONIC   \n",
       "148271      12  133056705   T   C  SNV       0  Transcript    INTRONIC   \n",
       "148272      12  133056935   G   T  SNV       0  Transcript    INTRONIC   \n",
       "\n",
       "        ConsScore ConsDetail  ...  Freq10000bp  Rare10000bp  Sngl10000bp  \\\n",
       "0               2     intron  ...           20           80         1740   \n",
       "1               1   upstream  ...           20           80         1740   \n",
       "2               2     intron  ...           20           84         1743   \n",
       "3               1   upstream  ...           20           84         1743   \n",
       "...           ...        ...  ...          ...          ...          ...   \n",
       "148269          2     intron  ...           23           60         1336   \n",
       "148270          2     intron  ...           23           59         1330   \n",
       "148271          2     intron  ...           21           59         1337   \n",
       "148272          2     intron  ...           21           59         1336   \n",
       "\n",
       "       EnsembleRegulatoryFeature  dbscSNV-ada_score  dbscSNV-rf_score  \\\n",
       "0                            NaN                NaN               NaN   \n",
       "1                            NaN                NaN               NaN   \n",
       "2                            NaN                NaN               NaN   \n",
       "3                            NaN                NaN               NaN   \n",
       "...                          ...                ...               ...   \n",
       "148269                       NaN                NaN               NaN   \n",
       "148270                       NaN                NaN               NaN   \n",
       "148271                       NaN                NaN               NaN   \n",
       "148272                       NaN                NaN               NaN   \n",
       "\n",
       "       RemapOverlapTF RemapOverlapCL  RawScore  PHRED  \n",
       "0                 NaN            NaN -0.161503  0.690  \n",
       "1                 NaN            NaN -0.161503  0.690  \n",
       "2                 NaN            NaN -0.209266  0.506  \n",
       "3                 NaN            NaN -0.209266  0.506  \n",
       "...               ...            ...       ...    ...  \n",
       "148269            2.0            2.0 -0.331295  0.206  \n",
       "148270            3.0            3.0 -0.448011  0.081  \n",
       "148271            3.0            4.0 -0.019107  1.585  \n",
       "148272            5.0            6.0 -0.231915  0.433  \n",
       "\n",
       "[148273 rows x 125 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_annotated_2 = pd.read_table('human_derived_noncoding_for_cadd_2_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (35,37,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>3385306</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>30</td>\n",
       "      <td>95</td>\n",
       "      <td>1620</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.105947</td>\n",
       "      <td>2.863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>3385442</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>95</td>\n",
       "      <td>1626</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.097632</td>\n",
       "      <td>2.764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>3385843</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>91</td>\n",
       "      <td>1637</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-0.120348</td>\n",
       "      <td>0.890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>3385881</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>91</td>\n",
       "      <td>1636</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-0.299183</td>\n",
       "      <td>0.264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146449</th>\n",
       "      <td>15</td>\n",
       "      <td>101781876</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>96</td>\n",
       "      <td>1738</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.300977</td>\n",
       "      <td>0.261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146450</th>\n",
       "      <td>15</td>\n",
       "      <td>101781883</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>96</td>\n",
       "      <td>1739</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.225834</td>\n",
       "      <td>4.395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146451</th>\n",
       "      <td>15</td>\n",
       "      <td>101781883</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>96</td>\n",
       "      <td>1739</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.225834</td>\n",
       "      <td>4.395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146452</th>\n",
       "      <td>15</td>\n",
       "      <td>101781883</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>96</td>\n",
       "      <td>1739</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.225834</td>\n",
       "      <td>4.395</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>146453 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #Chrom        Pos Ref Alt Type  Length           AnnoType Consequence  \\\n",
       "0            1    3385306   A   G  SNV       0         Transcript    INTRONIC   \n",
       "1            1    3385442   T   C  SNV       0         Transcript    INTRONIC   \n",
       "2            1    3385843   T   C  SNV       0         Transcript    INTRONIC   \n",
       "3            1    3385881   G   C  SNV       0         Transcript    INTRONIC   \n",
       "...        ...        ...  ..  ..  ...     ...                ...         ...   \n",
       "146449      15  101781876   C   T  SNV       0         Transcript    INTRONIC   \n",
       "146450      15  101781883   C   G  SNV       0  RegulatoryFeature  REGULATORY   \n",
       "146451      15  101781883   C   G  SNV       0         Transcript    INTRONIC   \n",
       "146452      15  101781883   C   G  SNV       0         Transcript    INTRONIC   \n",
       "\n",
       "        ConsScore         ConsDetail  ...  Freq10000bp  Rare10000bp  \\\n",
       "0               2             intron  ...           30           95   \n",
       "1               2             intron  ...           29           95   \n",
       "2               2             intron  ...           29           91   \n",
       "3               2             intron  ...           29           91   \n",
       "...           ...                ...  ...          ...          ...   \n",
       "146449          2  intron,non_coding  ...           14           96   \n",
       "146450          4         regulatory  ...           14           96   \n",
       "146451          2  intron,non_coding  ...           14           96   \n",
       "146452          2  intron,non_coding  ...           14           96   \n",
       "\n",
       "        Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              1620                       NaN                NaN   \n",
       "1              1626                       NaN                NaN   \n",
       "2              1637                       NaN                NaN   \n",
       "3              1636                       NaN                NaN   \n",
       "...             ...                       ...                ...   \n",
       "146449         1738  Promoter Flanking Region                NaN   \n",
       "146450         1739  Promoter Flanking Region                NaN   \n",
       "146451         1739  Promoter Flanking Region                NaN   \n",
       "146452         1739  Promoter Flanking Region                NaN   \n",
       "\n",
       "        dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore  PHRED  \n",
       "0                    NaN            3.0            4.0  0.105947  2.863  \n",
       "1                    NaN            6.0            7.0  0.097632  2.764  \n",
       "2                    NaN            8.0           10.0 -0.120348  0.890  \n",
       "3                    NaN            8.0           10.0 -0.299183  0.264  \n",
       "...                  ...            ...            ...       ...    ...  \n",
       "146449               NaN            2.0            2.0 -0.300977  0.261  \n",
       "146450               NaN            2.0            2.0  0.225834  4.395  \n",
       "146451               NaN            2.0            2.0  0.225834  4.395  \n",
       "146452               NaN            2.0            2.0  0.225834  4.395  \n",
       "\n",
       "[146453 rows x 125 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_annotated_3 = pd.read_table('human_derived_noncoding_for_cadd_3_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (16,17,30,35,37,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>5875122</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>31</td>\n",
       "      <td>84</td>\n",
       "      <td>1744</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.188326</td>\n",
       "      <td>0.581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>5875194</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>30</td>\n",
       "      <td>85</td>\n",
       "      <td>1749</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.517745</td>\n",
       "      <td>0.046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>5875324</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>30</td>\n",
       "      <td>86</td>\n",
       "      <td>1755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.343136</td>\n",
       "      <td>0.188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>5875346</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>30</td>\n",
       "      <td>86</td>\n",
       "      <td>1758</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.074180</td>\n",
       "      <td>1.168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148917</th>\n",
       "      <td>17</td>\n",
       "      <td>48834602</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>45</td>\n",
       "      <td>1296</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.578898</td>\n",
       "      <td>8.017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148918</th>\n",
       "      <td>17</td>\n",
       "      <td>48834679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>45</td>\n",
       "      <td>1294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.811848</td>\n",
       "      <td>10.050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148919</th>\n",
       "      <td>17</td>\n",
       "      <td>48834896</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>44</td>\n",
       "      <td>1308</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-0.007444</td>\n",
       "      <td>1.685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148920</th>\n",
       "      <td>17</td>\n",
       "      <td>48834902</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>44</td>\n",
       "      <td>1308</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>-0.027706</td>\n",
       "      <td>1.514</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>148921 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #Chrom       Pos Ref Alt Type  Length    AnnoType Consequence  \\\n",
       "0            1   5875122   T   C  SNV       0  Transcript    INTRONIC   \n",
       "1            1   5875194   G   C  SNV       0  Transcript    INTRONIC   \n",
       "2            1   5875324   T   C  SNV       0  Transcript    INTRONIC   \n",
       "3            1   5875346   T   C  SNV       0  Transcript    INTRONIC   \n",
       "...        ...       ...  ..  ..  ...     ...         ...         ...   \n",
       "148917      17  48834602   C   T  SNV       0  Transcript    INTRONIC   \n",
       "148918      17  48834679   G   A  SNV       0  Transcript    INTRONIC   \n",
       "148919      17  48834896   C   A  SNV       0  Transcript    INTRONIC   \n",
       "148920      17  48834902   G   C  SNV       0  Transcript    INTRONIC   \n",
       "\n",
       "        ConsScore ConsDetail  ...  Freq10000bp  Rare10000bp  Sngl10000bp  \\\n",
       "0               2     intron  ...           31           84         1744   \n",
       "1               2     intron  ...           30           85         1749   \n",
       "2               2     intron  ...           30           86         1755   \n",
       "3               2     intron  ...           30           86         1758   \n",
       "...           ...        ...  ...          ...          ...          ...   \n",
       "148917          2     intron  ...           22           45         1296   \n",
       "148918          2     intron  ...           23           45         1294   \n",
       "148919          2     intron  ...           24           44         1308   \n",
       "148920          2     intron  ...           24           44         1308   \n",
       "\n",
       "       EnsembleRegulatoryFeature  dbscSNV-ada_score  dbscSNV-rf_score  \\\n",
       "0                            NaN                NaN               NaN   \n",
       "1                            NaN                NaN               NaN   \n",
       "2                            NaN                NaN               NaN   \n",
       "3                            NaN                NaN               NaN   \n",
       "...                          ...                ...               ...   \n",
       "148917                       NaN                NaN               NaN   \n",
       "148918                       NaN                NaN               NaN   \n",
       "148919                       NaN                NaN               NaN   \n",
       "148920                       NaN                NaN               NaN   \n",
       "\n",
       "       RemapOverlapTF RemapOverlapCL  RawScore   PHRED  \n",
       "0                 5.0            5.0 -0.188326   0.581  \n",
       "1                 3.0            3.0 -0.517745   0.046  \n",
       "2                 4.0            6.0 -0.343136   0.188  \n",
       "3                 4.0            6.0 -0.074180   1.168  \n",
       "...               ...            ...       ...     ...  \n",
       "148917            2.0            3.0  0.578898   8.017  \n",
       "148918            2.0            2.0  0.811848  10.050  \n",
       "148919            2.0            2.0 -0.007444   1.685  \n",
       "148920            3.0            3.0 -0.027706   1.514  \n",
       "\n",
       "[148921 rows x 125 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_annotated_4 = pd.read_table('human_derived_noncoding_for_cadd_4_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated_5 = pd.read_table('human_derived_noncoding_for_cadd_5_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated_6 = pd.read_table('human_derived_noncoding_for_cadd_6_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated_7 = pd.read_table('human_derived_noncoding_for_cadd_7_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated_8 = pd.read_table('human_derived_noncoding_for_cadd_8_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated_9 = pd.read_table('human_derived_noncoding_for_cadd_9_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated_10 = pd.read_table('human_derived_noncoding_for_cadd_10_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated_11 = pd.read_table('human_derived_noncoding_for_cadd_11_noheader.tsv', sep='\\t')\n",
    "human_derived_noncoding_annotated_11"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we have to combine all of these files back into one complete dataframe to give the total annotated non-coding variants from the human derived data for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated = pd.concat('human_derived_noncoding_annotated_1', 'human_derived_noncoding_annotated_2', \n",
    "                                             'human_derived_noncoding_annotated_3', 'human_derived_noncoding_annotated_4', \n",
    "                                             'human_derived_noncoding_annotated_5', 'human_derived_noncoding_annotated_6',\n",
    "                                             'human_derived_noncoding_annotated_7', 'human_derived_noncoding_annotated_8',\n",
    "                                             'human_derived_noncoding_annotated_9', 'human_derived_noncoding_annotated_10', \n",
    "                                             'human_derived_noncoding_annotated_11')\n",
    "human_derived_noncoding_annotated"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the process used to export files for annotation for the coding regions from the Human Derived data set for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100190879</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100233196</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100267615</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53794</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719539</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53795</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719939</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53796</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99720084</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53797</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99721008</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53798 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS ID REF ALT\n",
       "0      chr10  100020652  .   G   A\n",
       "1      chr10  100190879  .   T   C\n",
       "2      chr10  100233196  .   G   A\n",
       "3      chr10  100267615  .   C   T\n",
       "...      ...        ... ..  ..  ..\n",
       "53794   chrX   99719539  .   A   G\n",
       "53795   chrX   99719939  .   C   A\n",
       "53796   chrX   99720084  .   G   A\n",
       "53797   chrX   99721008  .   G   A\n",
       "\n",
       "[53798 rows x 5 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_region_variants_only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'human_derived_coding_region_variants_only' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-fc05542da766>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     10\u001b[0m     \u001b[0mvcf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mheader\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 12\u001b[0;31m \u001b[0mhuman_derived_coding_region_variants_only\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moutput_VCF\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"\\t\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'a'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'human_derived_coding_region_variants_only' is not defined"
     ]
    }
   ],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"human_derived_coding_for_cadd.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_coding_region_variants_only.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annotated file for the human derived coding region variants for CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>5</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>944699</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>117</td>\n",
       "      <td>2573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>2.980499</td>\n",
       "      <td>23.200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122229</th>\n",
       "      <td>X</td>\n",
       "      <td>155260197</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3PRIME_UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>3_prime_UTR</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>873</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.219204</td>\n",
       "      <td>0.473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122230</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122231</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122232</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>122233 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0           1     943329   C   T  SNV       0  RegulatoryFeature   \n",
       "1           1     943329   C   T  SNV       0   CodingTranscript   \n",
       "2           1     943329   C   T  SNV       0         Intergenic   \n",
       "3           1     944699   C   T  SNV       0   CodingTranscript   \n",
       "...       ...        ...  ..  ..  ...     ...                ...   \n",
       "122229      X  155260197   C   A  SNV       0         Transcript   \n",
       "122230      X  155612732   T   C  SNV       0  RegulatoryFeature   \n",
       "122231      X  155612732   T   C  SNV       0         Transcript   \n",
       "122232      X  155612732   T   C  SNV       0         Intergenic   \n",
       "\n",
       "           Consequence  ConsScore   ConsDetail  ...  Freq10000bp  Rare10000bp  \\\n",
       "0           REGULATORY          4   regulatory  ...           18          117   \n",
       "1           SYNONYMOUS          5   synonymous  ...           18          117   \n",
       "2           DOWNSTREAM          1   downstream  ...           18          117   \n",
       "3       NON_SYNONYMOUS          7     missense  ...           16          117   \n",
       "...                ...        ...          ...  ...          ...          ...   \n",
       "122229      3PRIME_UTR          2  3_prime_UTR  ...            9           36   \n",
       "122230      REGULATORY          4   regulatory  ...            7           27   \n",
       "122231        INTRONIC          2       intron  ...            7           27   \n",
       "122232      DOWNSTREAM          1   downstream  ...            7           27   \n",
       "\n",
       "        Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              2586         CTCF Binding Site                NaN   \n",
       "1              2586         CTCF Binding Site                NaN   \n",
       "2              2586         CTCF Binding Site                NaN   \n",
       "3              2573                       NaN                NaN   \n",
       "...             ...                       ...                ...   \n",
       "122229          873                       NaN                NaN   \n",
       "122230          866                  Promoter                NaN   \n",
       "122231          866                  Promoter                NaN   \n",
       "122232          866                  Promoter                NaN   \n",
       "\n",
       "        dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore   PHRED  \n",
       "0                    NaN           88.0          121.0  0.590702   8.117  \n",
       "1                    NaN           88.0          121.0  0.590702   8.117  \n",
       "2                    NaN           88.0          121.0  0.590702   8.117  \n",
       "3                    NaN           12.0           13.0  2.980499  23.200  \n",
       "...                  ...            ...            ...       ...     ...  \n",
       "122229               NaN            3.0            6.0 -0.219204   0.473  \n",
       "122230               NaN          112.0          202.0  0.661682   8.709  \n",
       "122231               NaN          112.0          202.0  0.661682   8.709  \n",
       "122232               NaN          112.0          202.0  0.661682   8.709  \n",
       "\n",
       "[122233 rows x 125 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_annotated = pd.read_table('human_derived_coding_noheader.tsv', sep='\\t')\n",
    "human_derived_coding_annotated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
