{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Manipulation Before CADD Annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import io"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since dataframes are very big, we want to show only a few rows using the following code:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_rows', 8)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to use a vcf in a dataframe, the following function has been created. This function will work on any vcf file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_vcf(path):\n",
    "    with open(path, 'r') as f:\n",
    "        lines = [l for l in f if not l.startswith('##')]\n",
    "    return pd.read_csv(io.StringIO(''.join(lines)), dtype={'#CHROM': str, 'POS':int, 'ID':str, 'REF': str, 'ALT': str, 'QUAL': str, 'FILTER': str, 'INFO': str}, sep='\\t').rename(columns={'#CHROM': 'CHROM'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ClinVar Analysis Below:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First, we import all variants from the ClinVar dataset and read them into a dataframe:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>475283</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00546;AF_EXAC=0.00165;AF_TGP=0.00619;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014122</td>\n",
       "      <td>542074</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00015;AF_EXAC=0.00010;ALLELEID=514926...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>183381</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=181485;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014179</td>\n",
       "      <td>542075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=514896;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453956</th>\n",
       "      <td>MT</td>\n",
       "      <td>15965</td>\n",
       "      <td>9571</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=24610;CLNDISDB=MedGen:C1838867,OMIM:5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453957</th>\n",
       "      <td>MT</td>\n",
       "      <td>15967</td>\n",
       "      <td>9572</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=24611;CLNDISDB=MedGen:C0162672,OMIM:5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453958</th>\n",
       "      <td>MT</td>\n",
       "      <td>15990</td>\n",
       "      <td>9570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=24609;CLNDISDB=Human_Phenotype_Ontolo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453959</th>\n",
       "      <td>NW_009646201.1</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=32774;CLNDISDB=MedGen:C0000778,OMIM:6...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>453960 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 CHROM      POS      ID REF ALT QUAL FILTER  \\\n",
       "0                    1  1014042  475283   G   A    .      .   \n",
       "1                    1  1014122  542074   C   T    .      .   \n",
       "2                    1  1014143  183381   C   T    .      .   \n",
       "3                    1  1014179  542075   C   T    .      .   \n",
       "...                ...      ...     ...  ..  ..  ...    ...   \n",
       "453956              MT    15965    9571   A   G    .      .   \n",
       "453957              MT    15967    9572   G   A    .      .   \n",
       "453958              MT    15990    9570   C   T    .      .   \n",
       "453959  NW_009646201.1    83614   17735  TC   T    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "0       AF_ESP=0.00546;AF_EXAC=0.00165;AF_TGP=0.00619;...  \n",
       "1       AF_ESP=0.00015;AF_EXAC=0.00010;ALLELEID=514926...  \n",
       "2       ALLELEID=181485;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "3       ALLELEID=514896;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "...                                                   ...  \n",
       "453956  ALLELEID=24610;CLNDISDB=MedGen:C1838867,OMIM:5...  \n",
       "453957  ALLELEID=24611;CLNDISDB=MedGen:C0162672,OMIM:5...  \n",
       "453958  ALLELEID=24609;CLNDISDB=Human_Phenotype_Ontolo...  \n",
       "453959  ALLELEID=32774;CLNDISDB=MedGen:C0000778,OMIM:6...  \n",
       "\n",
       "[453960 rows x 8 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_data = read_vcf('clinvar_20190701.vcf')\n",
    "clinvar_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using the INFO section of the ClinVar database, we can split the variants into two separate groups, benign and pathogenic. All we need to do is search through the INFO column for the Clinical Significance (CLNSIG) and take those whose values are equal to benign or pathogenic."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below are all the benign variants from the clinvar database:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>475283</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00546;AF_EXAC=0.00165;AF_TGP=0.00619;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1014217</td>\n",
       "      <td>475278</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00515;AF_EXAC=0.00831;AF_TGP=0.00339;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>1014228</td>\n",
       "      <td>402986</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.40158;AF_EXAC=0.37025;AF_TGP=0.33886;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1</td>\n",
       "      <td>1014451</td>\n",
       "      <td>475281</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00987;AF_EXAC=0.00772;AF_TGP=0.01558;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453895</th>\n",
       "      <td>MT</td>\n",
       "      <td>15235</td>\n",
       "      <td>377041</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=363919;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453922</th>\n",
       "      <td>MT</td>\n",
       "      <td>15470</td>\n",
       "      <td>143892</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=153617;CLNDISDB=MedGen:C0006142,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453939</th>\n",
       "      <td>MT</td>\n",
       "      <td>15746</td>\n",
       "      <td>235623</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=237304;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453947</th>\n",
       "      <td>MT</td>\n",
       "      <td>15884</td>\n",
       "      <td>252455</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=247210;CLNDISDB=MedGen:CN169374;CLNDN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29963 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM      POS      ID REF ALT QUAL FILTER  \\\n",
       "0          1  1014042  475283   G   A    .      .   \n",
       "4          1  1014217  475278   C   T    .      .   \n",
       "5          1  1014228  402986   G   A    .      .   \n",
       "11         1  1014451  475281   C   T    .      .   \n",
       "...      ...      ...     ...  ..  ..  ...    ...   \n",
       "453895    MT    15235  377041   A   G    .      .   \n",
       "453922    MT    15470  143892   T   C    .      .   \n",
       "453939    MT    15746  235623   A   G    .      .   \n",
       "453947    MT    15884  252455   G   C    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "0       AF_ESP=0.00546;AF_EXAC=0.00165;AF_TGP=0.00619;...  \n",
       "4       AF_ESP=0.00515;AF_EXAC=0.00831;AF_TGP=0.00339;...  \n",
       "5       AF_ESP=0.40158;AF_EXAC=0.37025;AF_TGP=0.33886;...  \n",
       "11      AF_ESP=0.00987;AF_EXAC=0.00772;AF_TGP=0.01558;...  \n",
       "...                                                   ...  \n",
       "453895  ALLELEID=363919;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "453922  ALLELEID=153617;CLNDISDB=MedGen:C0006142,OMIM:...  \n",
       "453939  ALLELEID=237304;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "453947  ALLELEID=247210;CLNDISDB=MedGen:CN169374;CLNDN...  \n",
       "\n",
       "[29963 rows x 8 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "benign_variants = clinvar_data[clinvar_data['INFO'].str.contains('CLNSIG=Benign') & ~clinvar_data['INFO'].str.contains('CLNSIG=Benign/Likely_benign') & ~clinvar_data['INFO'].str.contains('CLNSIG=Benign/Likely benign')& ~clinvar_data['INFO'].str.contains('CLNSIG=Benign/Likely_benign') & ~clinvar_data['INFO'].str.contains('CLNSIG=Benign/Likely')]\n",
    "benign_variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below are all the pathogenic variants from the clinvar database:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>183381</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=181485;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>1014316</td>\n",
       "      <td>161455</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=171289;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>1014359</td>\n",
       "      <td>161454</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00001;ALLELEID=171288;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>1</td>\n",
       "      <td>1022225</td>\n",
       "      <td>243036</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00001;ALLELEID=244110;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453933</th>\n",
       "      <td>MT</td>\n",
       "      <td>15615</td>\n",
       "      <td>9678</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=24717;CLNDISDB=Human_Phenotype_Ontolo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453943</th>\n",
       "      <td>MT</td>\n",
       "      <td>15812</td>\n",
       "      <td>9675</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=24714;CLNDISDB=Human_Phenotype_Ontolo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453957</th>\n",
       "      <td>MT</td>\n",
       "      <td>15967</td>\n",
       "      <td>9572</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=24611;CLNDISDB=MedGen:C0162672,OMIM:5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453958</th>\n",
       "      <td>MT</td>\n",
       "      <td>15990</td>\n",
       "      <td>9570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=24609;CLNDISDB=Human_Phenotype_Ontolo...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61388 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM      POS      ID REF ALT QUAL FILTER  \\\n",
       "2          1  1014143  183381   C   T    .      .   \n",
       "8          1  1014316  161455   C  CG    .      .   \n",
       "9          1  1014359  161454   G   T    .      .   \n",
       "24         1  1022225  243036   G   A    .      .   \n",
       "...      ...      ...     ...  ..  ..  ...    ...   \n",
       "453933    MT    15615    9678   G   A    .      .   \n",
       "453943    MT    15812    9675   G   A    .      .   \n",
       "453957    MT    15967    9572   G   A    .      .   \n",
       "453958    MT    15990    9570   C   T    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "2       ALLELEID=181485;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "8       ALLELEID=171289;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "9       AF_EXAC=0.00001;ALLELEID=171288;CLNDISDB=MedGe...  \n",
       "24      AF_EXAC=0.00001;ALLELEID=244110;CLNDISDB=MedGe...  \n",
       "...                                                   ...  \n",
       "453933  ALLELEID=24717;CLNDISDB=Human_Phenotype_Ontolo...  \n",
       "453943  ALLELEID=24714;CLNDISDB=Human_Phenotype_Ontolo...  \n",
       "453957  ALLELEID=24611;CLNDISDB=MedGen:C0162672,OMIM:5...  \n",
       "453958  ALLELEID=24609;CLNDISDB=Human_Phenotype_Ontolo...  \n",
       "\n",
       "[61388 rows x 8 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_variants = clinvar_data[clinvar_data['INFO'].str.contains('CLNSIG=Pathogenic') & ~clinvar_data['INFO'].str.contains('CLNSIG=non-Pathogenic')& ~clinvar_data['INFO'].str.contains('CLNSIG=Pathogenic/Likely_pathogenic') & ~clinvar_data['INFO'].str.contains('CLNSIG=Pathogenic/LikelyPathogenic')]\n",
    "pathogenic_variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ClinVar data mapped to nORF regions using BedTools Below:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using the BedTools intersect tool in a Ubuntu command line, we were able to locate which variants in nORFs. These are the only variants we are interested in for this study. The code for bedtools intersect can be found in the google doc entitled sORFc_Felix's_work_replication Finalized 27/6."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following code gives us all variants which mapped to nORF regions from the ClinVar database:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100154912</td>\n",
       "      <td>476196</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00001;ALLELEID=459810;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>226426</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=228227;CLNDISDB=MedGen:C4284588,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100183753</td>\n",
       "      <td>596294</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=587355;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>226427</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=228228;CLNDISDB=MedGen:C4284588,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121810</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9760717</td>\n",
       "      <td>98631</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00002;ALLELEID=104521;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121811</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9760724</td>\n",
       "      <td>98628</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=104518;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121812</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9760730</td>\n",
       "      <td>98627</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=104517;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121813</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>98626</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=104516;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>121814 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID REF ALT QUAL FILTER  \\\n",
       "0       chrchr10  100154912  476196   G   A    .      .   \n",
       "1       chrchr10  100154922  226426   G   A    .      .   \n",
       "2       chrchr10  100183753  596294   C   T    .      .   \n",
       "3       chrchr10  100183802  226427   C   A    .      .   \n",
       "...          ...        ...     ...  ..  ..  ...    ...   \n",
       "121810   chrchrX    9760717   98631   C   T    .      .   \n",
       "121811   chrchrX    9760724   98628   C   T    .      .   \n",
       "121812   chrchrX    9760730   98627   C   G    .      .   \n",
       "121813   chrchrX    9760731   98626   A   G    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "0       AF_EXAC=0.00001;ALLELEID=459810;CLNDISDB=MedGe...  \n",
       "1       ALLELEID=228227;CLNDISDB=MedGen:C4284588,OMIM:...  \n",
       "2       ALLELEID=587355;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "3       ALLELEID=228228;CLNDISDB=MedGen:C4284588,OMIM:...  \n",
       "...                                                   ...  \n",
       "121810  AF_EXAC=0.00002;ALLELEID=104521;CLNDISDB=MedGe...  \n",
       "121811  ALLELEID=104518;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "121812  ALLELEID=104517;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "121813  ALLELEID=104516;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "\n",
       "[121814 rows x 8 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_variants_mapped_to_nORFs = read_vcf('clinvar_mapped_to_norfs_with_header.vcf')\n",
    "clinvar_variants_mapped_to_nORFs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the dataframe for the coding region only. This dataset was found using bedtools intersect with the command word -slice. This dataframe is very important to our study since the two classifiers we will make after all dataframes are created will be based on if variants are located in the coding or non-coding regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100154912</td>\n",
       "      <td>476196</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00001;ALLELEID=459810;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>226426</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=228227;CLNDISDB=MedGen:C4284588,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100183753</td>\n",
       "      <td>596294</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=587355;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>226427</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=228228;CLNDISDB=MedGen:C4284588,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101101</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9760717</td>\n",
       "      <td>98631</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00002;ALLELEID=104521;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101102</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9760724</td>\n",
       "      <td>98628</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=104518;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101103</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9760730</td>\n",
       "      <td>98627</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=104517;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101104</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>98626</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=104516;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>101105 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID REF ALT QUAL FILTER  \\\n",
       "0       chrchr10  100154912  476196   G   A    .      .   \n",
       "1       chrchr10  100154922  226426   G   A    .      .   \n",
       "2       chrchr10  100183753  596294   C   T    .      .   \n",
       "3       chrchr10  100183802  226427   C   A    .      .   \n",
       "...          ...        ...     ...  ..  ..  ...    ...   \n",
       "101101   chrchrX    9760717   98631   C   T    .      .   \n",
       "101102   chrchrX    9760724   98628   C   T    .      .   \n",
       "101103   chrchrX    9760730   98627   C   G    .      .   \n",
       "101104   chrchrX    9760731   98626   A   G    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "0       AF_EXAC=0.00001;ALLELEID=459810;CLNDISDB=MedGe...  \n",
       "1       ALLELEID=228227;CLNDISDB=MedGen:C4284588,OMIM:...  \n",
       "2       ALLELEID=587355;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "3       ALLELEID=228228;CLNDISDB=MedGen:C4284588,OMIM:...  \n",
       "...                                                   ...  \n",
       "101101  AF_EXAC=0.00002;ALLELEID=104521;CLNDISDB=MedGe...  \n",
       "101102  ALLELEID=104518;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "101103  ALLELEID=104517;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "101104  ALLELEID=104516;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "\n",
       "[101105 rows x 8 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_coding_region_variants_only = read_vcf('clinvar_mapped_to_norfs_split_with_header.vcf')\n",
    "clinvar_coding_region_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the dataframe for clinvar benign variants mapped to coding nORF regions. This has been found by taking the Clinical Significance (CLINSIG) for the values of benign only (not likely_benign)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100987606</td>\n",
       "      <td>136593</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_TGP=0.41214;ALLELEID=140296;CLNDISDB=MedGen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>102065918</td>\n",
       "      <td>284200</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_TGP=0.00260;ALLELEID=268437;CLNDISDB=MedGen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>102396271</td>\n",
       "      <td>474786</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00431;AF_EXAC=0.00140;AF_TGP=0.00579;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>102399466</td>\n",
       "      <td>541631</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00158;AF_TGP=0.00339;ALLELEID=525432...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101076</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85964016</td>\n",
       "      <td>255991</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.16937;AF_EXAC=0.19154;AF_TGP=0.12000;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101079</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85978770</td>\n",
       "      <td>497462</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00246;AF_EXAC=0.00079;AF_TGP=0.00318;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101081</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85978816</td>\n",
       "      <td>377662</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.01335;AF_EXAC=0.01456;AF_TGP=0.00742;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101087</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>93671995</td>\n",
       "      <td>208906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=205531;CLNDISDB=Human_Phenotype_Ontol...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4659 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID REF ALT QUAL FILTER  \\\n",
       "18      chrchr10  100987606  136593   G   T    .      .   \n",
       "57      chrchr10  102065918  284200   C   G    .      .   \n",
       "82      chrchr10  102396271  474786   G   A    .      .   \n",
       "87      chrchr10  102399466  541631   C   T    .      .   \n",
       "...          ...        ...     ...  ..  ..  ...    ...   \n",
       "101076   chrchrX   85964016  255991   T   C    .      .   \n",
       "101079   chrchrX   85978770  497462   G   C    .      .   \n",
       "101081   chrchrX   85978816  377662   T   A    .      .   \n",
       "101087   chrchrX   93671995  208906   G   C    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "18      AF_TGP=0.41214;ALLELEID=140296;CLNDISDB=MedGen...  \n",
       "57      AF_TGP=0.00260;ALLELEID=268437;CLNDISDB=MedGen...  \n",
       "82      AF_ESP=0.00431;AF_EXAC=0.00140;AF_TGP=0.00579;...  \n",
       "87      AF_EXAC=0.00158;AF_TGP=0.00339;ALLELEID=525432...  \n",
       "...                                                   ...  \n",
       "101076  AF_ESP=0.16937;AF_EXAC=0.19154;AF_TGP=0.12000;...  \n",
       "101079  AF_ESP=0.00246;AF_EXAC=0.00079;AF_TGP=0.00318;...  \n",
       "101081  AF_ESP=0.01335;AF_EXAC=0.01456;AF_TGP=0.00742;...  \n",
       "101087  ALLELEID=205531;CLNDISDB=Human_Phenotype_Ontol...  \n",
       "\n",
       "[4659 rows x 8 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_coding_region_benign_variants_only = clinvar_coding_region_variants_only[clinvar_coding_region_variants_only['INFO'].str.contains('CLNSIG=Benign')\n",
    "                                                                                 & ~clinvar_coding_region_variants_only['INFO'].str.contains('CLNSIG=Benign/Likely_benign') \n",
    "                                                                                 & ~clinvar_coding_region_variants_only['INFO'].str.contains('CLNSIG=Benign/Likely benign') \n",
    "                                                                                 & ~clinvar_coding_region_variants_only['INFO'].str.contains('CLNSIG=Benign/Likely_benign') \n",
    "                                                                                 & ~clinvar_coding_region_variants_only['INFO'].str.contains('CLNSIG=Benign/Likely')]\n",
    "clinvar_coding_region_benign_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the dataframe for clinvar pathogenic variants mapped to nORF regions. This has been found by taking the Clinical Significance (CLINSIG) for the values of pathogenic only (not likely_pathogenic)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>226426</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=228227;CLNDISDB=MedGen:C4284588,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>226427</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=228228;CLNDISDB=MedGen:C4284588,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100246864</td>\n",
       "      <td>504028</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=495446;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100253422</td>\n",
       "      <td>419610</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00023;AF_EXAC=0.00004;ALLELEID=407798...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101084</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85981796</td>\n",
       "      <td>279771</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=265177;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101094</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9759332</td>\n",
       "      <td>10517</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=25556;CLNDISDB=MedGen:C0342684,OMIM:3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101098</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>10516</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=25555;CLNDISDB=MedGen:C0342684,OMIM:3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101099</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>10519</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=25558;CLNDISDB=MedGen:C0342684,OMIM:3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16775 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID REF ALT QUAL FILTER  \\\n",
       "1       chrchr10  100154922  226426   G   A    .      .   \n",
       "3       chrchr10  100183802  226427   C   A    .      .   \n",
       "7       chrchr10  100246864  504028  AT   A    .      .   \n",
       "11      chrchr10  100253422  419610   G   A    .      .   \n",
       "...          ...        ...     ...  ..  ..  ...    ...   \n",
       "101084   chrchrX   85981796  279771   C   A    .      .   \n",
       "101094   chrchrX    9759332   10517   C   T    .      .   \n",
       "101098   chrchrX    9759390   10516   A   G    .      .   \n",
       "101099   chrchrX    9759390   10519   A   T    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "1       ALLELEID=228227;CLNDISDB=MedGen:C4284588,OMIM:...  \n",
       "3       ALLELEID=228228;CLNDISDB=MedGen:C4284588,OMIM:...  \n",
       "7       ALLELEID=495446;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "11      AF_ESP=0.00023;AF_EXAC=0.00004;ALLELEID=407798...  \n",
       "...                                                   ...  \n",
       "101084  ALLELEID=265177;CLNDISDB=MedGen:CN517202;CLNDN...  \n",
       "101094  ALLELEID=25556;CLNDISDB=MedGen:C0342684,OMIM:3...  \n",
       "101098  ALLELEID=25555;CLNDISDB=MedGen:C0342684,OMIM:3...  \n",
       "101099  ALLELEID=25558;CLNDISDB=MedGen:C0342684,OMIM:3...  \n",
       "\n",
       "[16775 rows x 8 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_coding_region_pathogenic_variants_only = clinvar_coding_region_variants_only[clinvar_coding_region_variants_only['INFO'].str.contains('CLNSIG=Pathogenic') & ~clinvar_coding_region_variants_only['INFO'].str.contains('CLNSIG=non-Pathogenic')& ~clinvar_coding_region_variants_only['INFO'].str.contains('CLNSIG=Pathogenic/Likely_pathogenic') & ~clinvar_coding_region_variants_only['INFO'].str.contains('CLNSIG=Pathogenic/LikelyPathogenic')]\n",
    "clinvar_coding_region_pathogenic_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we want to take only the non-coding region variants by using the concat function below. This command takes all the variants mapped to nORFs and all the variants mapped to the coding region only in nORFs and drops those in common. Thus we get the variants in the non-coding region only:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100987662</td>\n",
       "      <td>298487</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=319783;CLNDISDB=MedGen:C0342782,Orpha...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100987970</td>\n",
       "      <td>298492</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_TGP=0.00539;ALLELEID=309197;CLNDISDB=MedGen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100988106</td>\n",
       "      <td>298493</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=309201;CLNDISDB=MedGen:C0342782,Orpha...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100988240</td>\n",
       "      <td>380984</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=371478;CLNDISDB=MedGen:CN169374;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121783</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85965588</td>\n",
       "      <td>438064</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=431849;CLNDISDB=Human_Phenotype_Ontol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121784</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85968493</td>\n",
       "      <td>389953</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_TGP=0.00053;ALLELEID=378679;CLNDISDB=MedGen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121785</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>85968616</td>\n",
       "      <td>425508</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_TGP=0.00026;ALLELEID=413842;CLNDISDB=MedGen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121796</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>9653674</td>\n",
       "      <td>283429</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00013;ALLELEID=267666;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20709 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID REF ALT QUAL FILTER  \\\n",
       "20      chrchr10  100987662  298487   G   A    .      .   \n",
       "25      chrchr10  100987970  298492   C   T    .      .   \n",
       "26      chrchr10  100988106  298493   T   C    .      .   \n",
       "27      chrchr10  100988240  380984   C   T    .      .   \n",
       "...          ...        ...     ...  ..  ..  ...    ...   \n",
       "121783   chrchrX   85965588  438064   T   C    .      .   \n",
       "121784   chrchrX   85968493  389953   C   G    .      .   \n",
       "121785   chrchrX   85968616  425508   G   C    .      .   \n",
       "121796   chrchrX    9653674  283429   C   T    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "20      ALLELEID=319783;CLNDISDB=MedGen:C0342782,Orpha...  \n",
       "25      AF_TGP=0.00539;ALLELEID=309197;CLNDISDB=MedGen...  \n",
       "26      ALLELEID=309201;CLNDISDB=MedGen:C0342782,Orpha...  \n",
       "27      ALLELEID=371478;CLNDISDB=MedGen:CN169374;CLNDN...  \n",
       "...                                                   ...  \n",
       "121783  ALLELEID=431849;CLNDISDB=Human_Phenotype_Ontol...  \n",
       "121784  AF_TGP=0.00053;ALLELEID=378679;CLNDISDB=MedGen...  \n",
       "121785  AF_TGP=0.00026;ALLELEID=413842;CLNDISDB=MedGen...  \n",
       "121796  AF_EXAC=0.00013;ALLELEID=267666;CLNDISDB=MedGe...  \n",
       "\n",
       "[20709 rows x 8 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_noncoding_region_variants_only = pd.concat([clinvar_variants_mapped_to_nORFs, clinvar_coding_region_variants_only, clinvar_coding_region_variants_only]).drop_duplicates(keep=False)\n",
    "clinvar_noncoding_region_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now let us take the benign variants from the non-coding set:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100989312</td>\n",
       "      <td>136588</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.05359;AF_TGP=0.04553;ALLELEID=140291;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100990864</td>\n",
       "      <td>136589</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.25819;AF_EXAC=0.27515;AF_TGP=0.37600;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100990866</td>\n",
       "      <td>136590</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.25573;AF_EXAC=0.27520;AF_TGP=0.37600;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100991026</td>\n",
       "      <td>136591</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.28510;AF_TGP=0.37580;ALLELEID=140294;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121408</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>213614</td>\n",
       "      <td>CCTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=210570;CLNDISDB=MedGen:CN169374;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121410</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>95249</td>\n",
       "      <td>CCTCTT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=101148;CLNDISDB=MedGen:CN169374;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121412</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>71132767</td>\n",
       "      <td>95251</td>\n",
       "      <td>CCTCTTCTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=101150;CLNDISDB=MedGen:CN169374;CLNDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121709</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>78118027</td>\n",
       "      <td>558817</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.26615;AF_EXAC=0.27274;AF_TGP=0.34782;...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1560 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID                              REF ALT QUAL  \\\n",
       "73      chrchr10  100989312  136588                                G   A    .   \n",
       "102     chrchr10  100990864  136589                                C   T    .   \n",
       "103     chrchr10  100990866  136590                                T   C    .   \n",
       "108     chrchr10  100991026  136591                                C   A    .   \n",
       "...          ...        ...     ...                              ...  ..  ...   \n",
       "121408   chrchrX   71132767  213614       CCTCTTCTCTTCTCTTCTCTTCTCTT   C    .   \n",
       "121410   chrchrX   71132767   95249                           CCTCTT   C    .   \n",
       "121412   chrchrX   71132767   95251  CCTCTTCTCTTCTCTTCTCTTCTCTTCTCTT   C    .   \n",
       "121709   chrchrX   78118027  558817                                C   T    .   \n",
       "\n",
       "       FILTER                                               INFO  \n",
       "73          .  AF_ESP=0.05359;AF_TGP=0.04553;ALLELEID=140291;...  \n",
       "102         .  AF_ESP=0.25819;AF_EXAC=0.27515;AF_TGP=0.37600;...  \n",
       "103         .  AF_ESP=0.25573;AF_EXAC=0.27520;AF_TGP=0.37600;...  \n",
       "108         .  AF_ESP=0.28510;AF_TGP=0.37580;ALLELEID=140294;...  \n",
       "...       ...                                                ...  \n",
       "121408      .  ALLELEID=210570;CLNDISDB=MedGen:CN169374;CLNDN...  \n",
       "121410      .  ALLELEID=101148;CLNDISDB=MedGen:CN169374;CLNDN...  \n",
       "121412      .  ALLELEID=101150;CLNDISDB=MedGen:CN169374;CLNDN...  \n",
       "121709      .  AF_ESP=0.26615;AF_EXAC=0.27274;AF_TGP=0.34782;...  \n",
       "\n",
       "[1560 rows x 8 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_noncoding_region_benign_variants_only = clinvar_noncoding_region_variants_only[clinvar_noncoding_region_variants_only['INFO'].str.contains('CLNSIG=Benign') \n",
    "                                                                                       & ~clinvar_noncoding_region_variants_only['INFO'].str.contains('CLNSIG=Benign/Likely_benign') \n",
    "                                                                                       & ~clinvar_noncoding_region_variants_only['INFO'].str.contains('CLNSIG=Benign/Likely benign') \n",
    "                                                                                       & ~clinvar_noncoding_region_variants_only['INFO'].str.contains('CLNSIG=Benign/Likely_benign') \n",
    "                                                                                       & ~clinvar_noncoding_region_variants_only['INFO'].str.contains('CLNSIG=Benign/Likely')]\n",
    "clinvar_noncoding_region_benign_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will do the same for the pathogenic variants below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100988540</td>\n",
       "      <td>225837</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=227652;CLNDISDB=MedGen:C1849096,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100989084</td>\n",
       "      <td>488187</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=481146;CLNDISDB=MedGen:C4015307,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100989118</td>\n",
       "      <td>4628</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00002;ALLELEID=19667;CLNDISDB=MedGen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>chrchr10</td>\n",
       "      <td>100989154</td>\n",
       "      <td>4620</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=19659;CLNDISDB=MedGen:C1836439,OMIM:6...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121537</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>74529428</td>\n",
       "      <td>212188</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=209181;CLNDISDB=MedGen:C0795889,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121688</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>77902641</td>\n",
       "      <td>39768</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=48367;CLNDISDB=MedGen:C3550921,OMIM:3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121696</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>78003237</td>\n",
       "      <td>11786</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=26825;CLNDISDB=MedGen:C0022716,OMIM:3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121713</th>\n",
       "      <td>chrchrX</td>\n",
       "      <td>78122954</td>\n",
       "      <td>9955</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=24994;CLNDISDB=MedGen:C1970848,OMIM:3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1114 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           CHROM        POS      ID REF ALT QUAL FILTER  \\\n",
       "38      chrchr10  100988540  225837  CT   C    .      .   \n",
       "51      chrchr10  100989084  488187   C   A    .      .   \n",
       "54      chrchr10  100989118    4628   G   A    .      .   \n",
       "56      chrchr10  100989154    4620   G   T    .      .   \n",
       "...          ...        ...     ...  ..  ..  ...    ...   \n",
       "121537   chrchrX   74529428  212188   G  GC    .      .   \n",
       "121688   chrchrX   77902641   39768   A   G    .      .   \n",
       "121696   chrchrX   78003237   11786   G   A    .      .   \n",
       "121713   chrchrX   78122954    9955   G   A    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "38      ALLELEID=227652;CLNDISDB=MedGen:C1849096,OMIM:...  \n",
       "51      ALLELEID=481146;CLNDISDB=MedGen:C4015307,OMIM:...  \n",
       "54      AF_EXAC=0.00002;ALLELEID=19667;CLNDISDB=MedGen...  \n",
       "56      ALLELEID=19659;CLNDISDB=MedGen:C1836439,OMIM:6...  \n",
       "...                                                   ...  \n",
       "121537  ALLELEID=209181;CLNDISDB=MedGen:C0795889,OMIM:...  \n",
       "121688  ALLELEID=48367;CLNDISDB=MedGen:C3550921,OMIM:3...  \n",
       "121696  ALLELEID=26825;CLNDISDB=MedGen:C0022716,OMIM:3...  \n",
       "121713  ALLELEID=24994;CLNDISDB=MedGen:C1970848,OMIM:3...  \n",
       "\n",
       "[1114 rows x 8 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_noncoding_region_pathogenic_variants_only = clinvar_noncoding_region_variants_only[clinvar_noncoding_region_variants_only['INFO'].str.contains('CLNSIG=Pathogenic') \n",
    "                                                                                           & ~clinvar_noncoding_region_variants_only['INFO'].str.contains('CLNSIG=non-Pathogenic') \n",
    "                                                                                           & ~clinvar_noncoding_region_variants_only['INFO'].str.contains('CLNSIG=Pathogenic/Likely_pathogenic') \n",
    "                                                                                           & ~clinvar_noncoding_region_variants_only['INFO'].str.contains('CLNSIG=Pathogenic/LikelyPathogenic')]\n",
    "clinvar_noncoding_region_pathogenic_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Human Derived SNVs Below:\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### hg19:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the entire benign variant dataset for the human derived EPO 6-way primates file mapped to hg19 which we received from Dr. Martin Kircher:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2901: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  if self.run_code(code, result):\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>379177</th>\n",
       "      <th>.</th>\n",
       "      <th>G</th>\n",
       "      <th>T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>379274</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>379476</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>379631</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>379724</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15684027</th>\n",
       "      <td>X</td>\n",
       "      <td>155172105</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15684028</th>\n",
       "      <td>X</td>\n",
       "      <td>155172597</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15684029</th>\n",
       "      <td>X</td>\n",
       "      <td>155173020</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15684030</th>\n",
       "      <td>X</td>\n",
       "      <td>155173117</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15684031 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          1     379177  .  G  T\n",
       "0         1     379274  .  G  C\n",
       "1         1     379476  .  T  A\n",
       "2         1     379631  .  G  C\n",
       "3         1     379724  .  G  A\n",
       "...      ..        ... .. .. ..\n",
       "15684027  X  155172105  .  T  C\n",
       "15684028  X  155172597  .  T  C\n",
       "15684029  X  155173020  .  T  C\n",
       "15684030  X  155173117  .  C  T\n",
       "\n",
       "[15684031 rows x 5 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_SNV_data = read_vcf('humanDerived_SNVs.vcf')\n",
    "human_derived_SNV_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We need to use liftover on this dataframe to change it to hg38 from hg19. To do this, we must manipulate the data in order to package it as a vcf file that can be used in command line:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_SNV_data['..'] = '.'\n",
    "human_derived_SNV_data['...'] = '.'\n",
    "human_derived_SNV_data['....'] = '.'\n",
    "human_derived_SNV_data.loc[-1] = ['1', '379177', '.', 'G', 'T', '.', '.', '.']\n",
    "human_derived_SNV_data.index = human_derived_SNV_data.index + 1  # shifting index\n",
    "human_derived_SNV_data = human_derived_SNV_data.sort_index()\n",
    "human_derived_SNV_data.columns = ['', '', '', '', '', '', '', '']\n",
    "human_derived_SNV_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The below creates a vcf header for the data. Once we execute this code, we will use command line to change it into a bed file so we can use liftover on the bed file. This is documented in the sORFc_Felix's_work_replication Finalized 27/6."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"humanderivedSNVsreal.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_SNV_data.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### hg38:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the entire benign variant dataset for the human derived EPO 6-way primates file mapped to hg38. It is necessary to only use the file below since we are building our classifier based on all data being from hg38. Therefore, for the rest of the analysis, the file below will be the only one we use and the hg19 file will be disregarded."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr1</th>\n",
       "      <th>440160</th>\n",
       "      <th>.</th>\n",
       "      <th>G</th>\n",
       "      <th>T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>440063</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>439861</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>439706</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>439613</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15679200</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155942441</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15679201</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155942933</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15679202</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155943356</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15679203</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155943453</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15679204 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          chr1     440160  .  G  T\n",
       "0         chr1     440063  .  G  C\n",
       "1         chr1     439861  .  T  A\n",
       "2         chr1     439706  .  G  C\n",
       "3         chr1     439613  .  G  A\n",
       "...        ...        ... .. .. ..\n",
       "15679200  chrX  155942441  .  T  C\n",
       "15679201  chrX  155942933  .  T  C\n",
       "15679202  chrX  155943356  .  T  C\n",
       "15679203  chrX  155943453  .  C  T\n",
       "\n",
       "[15679204 rows x 5 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_SNV_data_hg38 = read_vcf('humanderivedSNVsreal_hg38.vcf')\n",
    "human_derived_SNV_data_hg38"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting the dataframe as a vcf file to use bedtools on:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to use this dataframe for bedtools intersect, we must have it in vcf format. The following code adds three arbitrary columns so that we get to the standard 8 columns of a vcf file, creates a new row for the row currently held in the header, and gives the header all blank values. The reason for this is because bedtools intersect cannot run when there are column names. I also had to manipulate the file using command line, which is documented in the google doc with all the commands on it mentioned previously."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>440160</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>440063</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>439861</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>439706</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15679201</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155942441</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15679202</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155942933</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15679203</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155943356</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15679204</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155943453</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15679205 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           \n",
       "0         chr1     440160  .  G  T  .  .  .\n",
       "1         chr1     440063  .  G  C  .  .  .\n",
       "2         chr1     439861  .  T  A  .  .  .\n",
       "3         chr1     439706  .  G  C  .  .  .\n",
       "...        ...        ... .. .. .. .. .. ..\n",
       "15679201  chrX  155942441  .  T  C  .  .  .\n",
       "15679202  chrX  155942933  .  T  C  .  .  .\n",
       "15679203  chrX  155943356  .  T  C  .  .  .\n",
       "15679204  chrX  155943453  .  C  T  .  .  .\n",
       "\n",
       "[15679205 rows x 8 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_SNV_data_hg38['..'] = '.'\n",
    "human_derived_SNV_data_hg38['...'] = '.'\n",
    "human_derived_SNV_data_hg38['....'] = '.'\n",
    "human_derived_SNV_data_hg38.loc[-1] = ['chr1', '440160', '.', 'G', 'T', '.', '.', '.']\n",
    "human_derived_SNV_data_hg38.index = human_derived_SNV_data_hg38.index + 1  # shifting index\n",
    "human_derived_SNV_data_hg38 = human_derived_SNV_data_hg38.sort_index()\n",
    "human_derived_SNV_data_hg38.columns = ['', '', '', '', '', '', '', '']\n",
    "human_derived_SNV_data_hg38"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will need the following code to create the header of the vcf file:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"humanderivedSNVsreal_hg38forbedtoolsintersect.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "human_derived_SNV_data_hg38.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## After bedtools intersect, we got the following dataframe which shows the human derived variants in the nORF regions:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10</th>\n",
       "      <th>1000013</th>\n",
       "      <th>.</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000297</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000555</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000567</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020174</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719539</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020175</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719939</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020176</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99720084</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020177</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99721008</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1020178 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr10    1000013  .  G  A\n",
       "0        chr10  100020652  .  G  A\n",
       "1        chr10    1000297  .  T  G\n",
       "2        chr10    1000555  .  A  T\n",
       "3        chr10    1000567  .  G  C\n",
       "...        ...        ... .. .. ..\n",
       "1020174   chrX   99719539  .  A  G\n",
       "1020175   chrX   99719939  .  C  A\n",
       "1020176   chrX   99720084  .  G  A\n",
       "1020177   chrX   99721008  .  G  A\n",
       "\n",
       "[1020178 rows x 5 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_variants_mapped_to_nORFs = read_vcf('human_derived_mapped_to_norfs.vcf')\n",
    "human_derived_variants_mapped_to_nORFs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let us clean this up to make it look more like a VCF file:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000013</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000297</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000555</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020175</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719539</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020176</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719939</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020177</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99720084</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020178</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99721008</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1020179 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM        POS ID REF ALT\n",
       "0        chr10    1000013  .   G   A\n",
       "1        chr10  100020652  .   G   A\n",
       "2        chr10    1000297  .   T   G\n",
       "3        chr10    1000555  .   A   T\n",
       "...        ...        ... ..  ..  ..\n",
       "1020175   chrX   99719539  .   A   G\n",
       "1020176   chrX   99719939  .   C   A\n",
       "1020177   chrX   99720084  .   G   A\n",
       "1020178   chrX   99721008  .   G   A\n",
       "\n",
       "[1020179 rows x 5 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_variants_mapped_to_nORFs.loc[-1] = ['chr10', '1000013', '.', 'G', 'A']\n",
    "human_derived_variants_mapped_to_nORFs.index = human_derived_variants_mapped_to_nORFs.index + 1  # shifting index\n",
    "human_derived_variants_mapped_to_nORFs = human_derived_variants_mapped_to_nORFs.sort_index()\n",
    "human_derived_variants_mapped_to_nORFs.columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT']\n",
    "human_derived_variants_mapped_to_nORFs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below are the variants mapped to the coding region of nORFs only:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10</th>\n",
       "      <th>100020652</th>\n",
       "      <th>.</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100190879</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100233196</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100267615</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100347207</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53793</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719539</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53794</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719939</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53795</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99720084</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53796</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99721008</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53797 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr10  100020652  .  G  A\n",
       "0      chr10  100190879  .  T  C\n",
       "1      chr10  100233196  .  G  A\n",
       "2      chr10  100267615  .  C  T\n",
       "3      chr10  100347207  .  T  C\n",
       "...      ...        ... .. .. ..\n",
       "53793   chrX   99719539  .  A  G\n",
       "53794   chrX   99719939  .  C  A\n",
       "53795   chrX   99720084  .  G  A\n",
       "53796   chrX   99721008  .  G  A\n",
       "\n",
       "[53797 rows x 5 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_region_variants_only = read_vcf('human_derived_mapped_to_norfs_with_split.vcf')\n",
    "human_derived_coding_region_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Making this look more like our favorite vcf format, we get the following:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100190879</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100233196</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100267615</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53794</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719539</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53795</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99719939</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53796</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99720084</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53797</th>\n",
       "      <td>chrX</td>\n",
       "      <td>99721008</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53798 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS ID REF ALT\n",
       "0      chr10  100020652  .   G   A\n",
       "1      chr10  100190879  .   T   C\n",
       "2      chr10  100233196  .   G   A\n",
       "3      chr10  100267615  .   C   T\n",
       "...      ...        ... ..  ..  ..\n",
       "53794   chrX   99719539  .   A   G\n",
       "53795   chrX   99719939  .   C   A\n",
       "53796   chrX   99720084  .   G   A\n",
       "53797   chrX   99721008  .   G   A\n",
       "\n",
       "[53798 rows x 5 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_region_variants_only.loc[-1] = ['chr10', '100020652', '.', 'G', 'A']\n",
    "human_derived_coding_region_variants_only.index = human_derived_coding_region_variants_only.index + 1  # shifting index\n",
    "human_derived_coding_region_variants_only = human_derived_coding_region_variants_only.sort_index()\n",
    "human_derived_coding_region_variants_only.columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT']\n",
    "human_derived_coding_region_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below are the variants mapped to the non-coding region of nORFs only:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000013</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000297</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000555</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020168</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931817</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020169</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931818</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020170</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931993</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020171</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9932000</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>966382 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM        POS ID REF ALT\n",
       "0        chr10    1000013  .   G   A\n",
       "1        chr10  100020652  .   G   A\n",
       "2        chr10    1000297  .   T   G\n",
       "3        chr10    1000555  .   A   T\n",
       "...        ...        ... ..  ..  ..\n",
       "1020168   chrX    9931817  .   T   C\n",
       "1020169   chrX    9931818  .   G   A\n",
       "1020170   chrX    9931993  .   T   C\n",
       "1020171   chrX    9932000  .   C   T\n",
       "\n",
       "[966382 rows x 5 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_region_variants_only = pd.concat([human_derived_variants_mapped_to_nORFs, human_derived_coding_region_variants_only, human_derived_coding_region_variants_only]).drop_duplicates(keep=False)\n",
    "human_derived_noncoding_region_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HGMD Database Shown Below"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## The dataframe below is for GRCh37 - We will not be using it in our analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>865595</td>\n",
       "      <td>CM1613956</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>874491</td>\n",
       "      <td>CM1613954</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>877523</td>\n",
       "      <td>CM1511864</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>879286</td>\n",
       "      <td>CS1613955</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174266</th>\n",
       "      <td>Y</td>\n",
       "      <td>6931938</td>\n",
       "      <td>CM121018</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=TBL1Y;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174267</th>\n",
       "      <td>Y</td>\n",
       "      <td>6938305</td>\n",
       "      <td>CM121019</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=TBL1Y;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174268</th>\n",
       "      <td>Y</td>\n",
       "      <td>14847658</td>\n",
       "      <td>CD993525</td>\n",
       "      <td>TTAAG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=USP9Y;STRAND=+;DNA=NM_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174269</th>\n",
       "      <td>Y</td>\n",
       "      <td>16952726</td>\n",
       "      <td>CM086695</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=NLGN4Y;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>174270 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS         ID    REF ALT QUAL FILTER  \\\n",
       "0          1    865595  CM1613956      A   G    .      .   \n",
       "1          1    874491  CM1613954      C   T    .      .   \n",
       "2          1    877523  CM1511864      C   G    .      .   \n",
       "3          1    879286  CS1613955      A   C    .      .   \n",
       "...      ...       ...        ...    ...  ..  ...    ...   \n",
       "174266     Y   6931938   CM121018      G   C    .      .   \n",
       "174267     Y   6938305   CM121019      C   T    .      .   \n",
       "174268     Y  14847658   CD993525  TTAAG   T    .      .   \n",
       "174269     Y  16952726   CM086695      A   G    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "0       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "1       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "2       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "3       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "...                                                   ...  \n",
       "174266  CLASS=DM?;MUT=ALT;GENE=TBL1Y;STRAND=+;DNA=NM_0...  \n",
       "174267  CLASS=DM?;MUT=ALT;GENE=TBL1Y;STRAND=+;DNA=NM_0...  \n",
       "174268  CLASS=DM;MUT=ALT;GENE=USP9Y;STRAND=+;DNA=NM_00...  \n",
       "174269  CLASS=DM;MUT=ALT;GENE=NLGN4Y;STRAND=+;DNA=NM_0...  \n",
       "\n",
       "[174270 rows x 8 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "read_vcf('HGMD_PRO_2016.4_hg19_best.vcf')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## This dataframe is for GRCh38 and we will be using this for bedtools  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In a text editor, Sublime Text 3, I have removed the header of the file. This seemed to be causing some trouble when creating the python dataframe you see below. I was able to keep the column headers on the file to get a good look at the vcf format shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>930215</td>\n",
       "      <td>CM1613956</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>939111</td>\n",
       "      <td>CM1613954</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>942143</td>\n",
       "      <td>CM1511864</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>943906</td>\n",
       "      <td>CS1613955</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232614</th>\n",
       "      <td>Y</td>\n",
       "      <td>9467296</td>\n",
       "      <td>CD1112014</td>\n",
       "      <td>GCC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=TSPY1;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232615</th>\n",
       "      <td>Y</td>\n",
       "      <td>9467303</td>\n",
       "      <td>CD1112015</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=TSPY1;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232616</th>\n",
       "      <td>Y</td>\n",
       "      <td>12735724</td>\n",
       "      <td>CD993525</td>\n",
       "      <td>TTAAG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=USP9Y;STRAND=+;DNA=NM_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232617</th>\n",
       "      <td>Y</td>\n",
       "      <td>14840846</td>\n",
       "      <td>CM086695</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=NLGN4Y;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>232618 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS         ID    REF ALT QUAL FILTER  \\\n",
       "0          1    930215  CM1613956      A   G    .      .   \n",
       "1          1    939111  CM1613954      C   T    .      .   \n",
       "2          1    942143  CM1511864      C   G    .      .   \n",
       "3          1    943906  CS1613955      A   C    .      .   \n",
       "...      ...       ...        ...    ...  ..  ...    ...   \n",
       "232614     Y   9467296  CD1112014    GCC   G    .      .   \n",
       "232615     Y   9467303  CD1112015     GC   G    .      .   \n",
       "232616     Y  12735724   CD993525  TTAAG   T    .      .   \n",
       "232617     Y  14840846   CM086695      A   G    .      .   \n",
       "\n",
       "                                                     INFO  \n",
       "0       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "1       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "2       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "3       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "...                                                   ...  \n",
       "232614  CLASS=DM?;MUT=ALT;GENE=TSPY1;STRAND=+;DNA=NM_0...  \n",
       "232615  CLASS=DM?;MUT=ALT;GENE=TSPY1;STRAND=+;DNA=NM_0...  \n",
       "232616  CLASS=DM;MUT=ALT;GENE=USP9Y;STRAND=+;DNA=NM_00...  \n",
       "232617  CLASS=DM;MUT=ALT;GENE=NLGN4Y;STRAND=+;DNA=NM_0...  \n",
       "\n",
       "[232618 rows x 8 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_for_hg38 = read_vcf('hgmddatabasenew_noheader.vcf')\n",
    "hgmd_for_hg38"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to export to a vcf file, I removed the column headers:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>930215</td>\n",
       "      <td>CM1613956</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>939111</td>\n",
       "      <td>CM1613954</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>942143</td>\n",
       "      <td>CM1511864</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>943906</td>\n",
       "      <td>CS1613955</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232614</th>\n",
       "      <td>Y</td>\n",
       "      <td>9467296</td>\n",
       "      <td>CD1112014</td>\n",
       "      <td>GCC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=TSPY1;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232615</th>\n",
       "      <td>Y</td>\n",
       "      <td>9467303</td>\n",
       "      <td>CD1112015</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=TSPY1;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232616</th>\n",
       "      <td>Y</td>\n",
       "      <td>12735724</td>\n",
       "      <td>CD993525</td>\n",
       "      <td>TTAAG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=USP9Y;STRAND=+;DNA=NM_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232617</th>\n",
       "      <td>Y</td>\n",
       "      <td>14840846</td>\n",
       "      <td>CM086695</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=NLGN4Y;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>232618 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                \\\n",
       "0       1    930215  CM1613956      A  G  .  .   \n",
       "1       1    939111  CM1613954      C  T  .  .   \n",
       "2       1    942143  CM1511864      C  G  .  .   \n",
       "3       1    943906  CS1613955      A  C  .  .   \n",
       "...    ..       ...        ...    ... .. .. ..   \n",
       "232614  Y   9467296  CD1112014    GCC  G  .  .   \n",
       "232615  Y   9467303  CD1112015     GC  G  .  .   \n",
       "232616  Y  12735724   CD993525  TTAAG  T  .  .   \n",
       "232617  Y  14840846   CM086695      A  G  .  .   \n",
       "\n",
       "                                                           \n",
       "0       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "1       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "2       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "3       CLASS=DM?;MUT=ALT;GENE=SAMD11;STRAND=+;DNA=NM_...  \n",
       "...                                                   ...  \n",
       "232614  CLASS=DM?;MUT=ALT;GENE=TSPY1;STRAND=+;DNA=NM_0...  \n",
       "232615  CLASS=DM?;MUT=ALT;GENE=TSPY1;STRAND=+;DNA=NM_0...  \n",
       "232616  CLASS=DM;MUT=ALT;GENE=USP9Y;STRAND=+;DNA=NM_00...  \n",
       "232617  CLASS=DM;MUT=ALT;GENE=NLGN4Y;STRAND=+;DNA=NM_0...  \n",
       "\n",
       "[232618 rows x 8 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_for_hg38.columns = ['', '', '', '', '', '', '', '']\n",
    "hgmd_for_hg38"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Just as with the Human Derived files, I need to create a header in order to use this on bedtools, shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "header = \"\"\"##fileformat=VCFv4.1\n",
    "##fileDate=20090805\n",
    "##source=myImputationProgramV3.1\n",
    "##reference=file:///seq/references/\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "\n",
    "output_VCF = \"hgmd_for_bedtools_intersect.vcf\"\n",
    "with open(output_VCF, 'w') as vcf:\n",
    "    vcf.write(header)\n",
    "\n",
    "hgmd_for_hg38.to_csv(output_VCF, sep=\"\\t\", mode='a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# After Running HGMD through Bedtools Intersect"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below are all the variants that were mapped to nORF regions for the HGMD dataset:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10</th>\n",
       "      <th>100154922</th>\n",
       "      <th>CM140970</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <th>.</th>\n",
       "      <th>..1</th>\n",
       "      <th>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_006459.3:c.763C&gt;T;PROT=NP_006450.2:p.R255*;DB=rs876657413;PHEN=\"Spastic_paraplegia_62\";RANKSCORE=0.99</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100262050</td>\n",
       "      <td>CM162834</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64630</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64631</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64632</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64633</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64634 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr10  100154922   CM140970   G   A  . ..1  \\\n",
       "0      chr10  100183802   CM140971   C   A  .   .   \n",
       "1      chr10  100253438  CI1824020   A  AT  .   .   \n",
       "2      chr10  100256298   CD162836  TG   T  .   .   \n",
       "3      chr10  100262050   CM162834   C   G  .   .   \n",
       "...      ...        ...        ...  ..  .. ..  ..   \n",
       "64630   chrX    9760731   CM981395   A   G  .   .   \n",
       "64631   chrX    9760732   CI183806   G  GA  .   .   \n",
       "64632   chrX    9760736   CD171619  GC   G  .   .   \n",
       "64633   chrX    9760741   CI115195   A  AG  .   .   \n",
       "\n",
       "      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_006459.3:c.763C>T;PROT=NP_006450.2:p.R255*;DB=rs876657413;PHEN=\"Spastic_paraplegia_62\";RANKSCORE=0.99  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...                                                                                                  \n",
       "1      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...                                                                                                  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...                                                                                                  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...                                                                                                  \n",
       "...                                                  ...                                                                                                  \n",
       "64630  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...                                                                                                  \n",
       "64631  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...                                                                                                  \n",
       "64632  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...                                                                                                  \n",
       "64633  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...                                                                                                  \n",
       "\n",
       "[64634 rows x 8 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_variants_mapped_to_nORFs = read_vcf('hgmd_mapped_to_norfs_real.vcf')\n",
    "hgmd_variants_mapped_to_nORFs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As with others, we must manipulate this to get to the correct vcf format:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64631</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64632</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64633</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64634</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64635 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "64631   chrX    9760731   CM981395   A   G    .      .   \n",
       "64632   chrX    9760732   CI183806   G  GA    .      .   \n",
       "64633   chrX    9760736   CD171619  GC   G    .      .   \n",
       "64634   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "64631  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64632  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64633  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64634  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[64635 rows x 8 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_variants_mapped_to_nORFs.loc[-1] = ['chr10', '100154922', 'CM140970', 'G', 'A', '.', '.', 'CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_006459.3:c.763C>T;PROT=NP_006450.2:p.R255*;DB=rs876657413;PHEN=\"Spastic_paraplegia_62\";RANKSCORE=0.99']\n",
    "hgmd_variants_mapped_to_nORFs.index = hgmd_variants_mapped_to_nORFs.index + 1  # shifting index\n",
    "hgmd_variants_mapped_to_nORFs = hgmd_variants_mapped_to_nORFs.sort_index()\n",
    "hgmd_variants_mapped_to_nORFs.columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO']\n",
    "hgmd_variants_mapped_to_nORFs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since we want the best classifier possible, we want to filter out all variants with uncertainty in whether or not the variant is pathogenic. Therefore, we have to find the variants which contain under the INFO column 'CLASS=DM?', since these are the variants which may not be pathogenic and cut them from the dataframe. This process is shown below:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First, we need to figure out where these uncertain variants are located, which is shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989064</td>\n",
       "      <td>CM193135</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_02...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989148</td>\n",
       "      <td>CM1720497</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_02...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989312</td>\n",
       "      <td>CM012076</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_02...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989400</td>\n",
       "      <td>CM182916</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_02...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64581</th>\n",
       "      <td>chrX</td>\n",
       "      <td>86027503</td>\n",
       "      <td>CM137329</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64593</th>\n",
       "      <td>chrX</td>\n",
       "      <td>8623669</td>\n",
       "      <td>CM133761</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64595</th>\n",
       "      <td>chrX</td>\n",
       "      <td>8623670</td>\n",
       "      <td>CM183496</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64598</th>\n",
       "      <td>chrX</td>\n",
       "      <td>87518266</td>\n",
       "      <td>CM1417120</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=KLHL4;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11123 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "18     chr10  100989064   CM193135   G   A    .      .   \n",
       "23     chr10  100989148  CM1720497   G   A    .      .   \n",
       "41     chr10  100989312   CM012076   G   A    .      .   \n",
       "53     chr10  100989400   CM182916   A   G    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "64581   chrX   86027503   CM137329   A   G    .      .   \n",
       "64593   chrX    8623669   CM133761   C   T    .      .   \n",
       "64595   chrX    8623670   CM183496   A   T    .      .   \n",
       "64598   chrX   87518266  CM1417120   C   G    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "18     CLASS=DM?;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_02...  \n",
       "23     CLASS=DM?;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_02...  \n",
       "41     CLASS=DM?;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_02...  \n",
       "53     CLASS=DM?;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_02...  \n",
       "...                                                  ...  \n",
       "64581  CLASS=DM?;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_000...  \n",
       "64593  CLASS=DM?;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_0...  \n",
       "64595  CLASS=DM?;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_0...  \n",
       "64598  CLASS=DM?;MUT=ALT;GENE=KLHL4;STRAND=+;DNA=NM_0...  \n",
       "\n",
       "[11123 rows x 8 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "finding_the_uncertainty = hgmd_variants_mapped_to_nORFs[~hgmd_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DM?;') & \n",
    "                                                        ~hgmd_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DP') \n",
    "                                                        & ~hgmd_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DFP') \n",
    "                                                        & ~hgmd_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=FP') \n",
    "                                                        & ~hgmd_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=R')]\n",
    "finding_the_uncertainty"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, we want to cut these from the dataframe since we do not need them. We do this like we did for ClinVar and the human derived variants when searching for the non-coding region. This gives the complete dataframe for all pathogenic variants mapped to the nORF regions:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64631</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64632</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64633</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64634</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53512 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "64631   chrX    9760731   CM981395   A   G    .      .   \n",
       "64632   chrX    9760732   CI183806   G  GA    .      .   \n",
       "64633   chrX    9760736   CD171619  GC   G    .      .   \n",
       "64634   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "64631  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64632  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64633  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64634  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[53512 rows x 8 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_disease_related_variants_mapped_to_nORFs = pd.concat([hgmd_variants_mapped_to_nORFs, finding_the_uncertainty, finding_the_uncertainty]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we will filter out every variant with 'CLASS=R' since these variants also do not pertain to our study. I repeat the same process above to do this:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100990866</td>\n",
       "      <td>CS1715008</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=REF;GENE=TWNK;STRAND=+;DNA=NM_0218...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>chr10</td>\n",
       "      <td>119677227</td>\n",
       "      <td>CM117921</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=BAG3;STRAND=+;DNA=NM_0042...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>chr10</td>\n",
       "      <td>13298236</td>\n",
       "      <td>CM001290</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=PHYH;STRAND=-;DNA=NM_0062...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3266</th>\n",
       "      <td>chr1</td>\n",
       "      <td>119955217</td>\n",
       "      <td>CM1110715</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=NOTCH2;STRAND=-;DNA=NM_02...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62204</th>\n",
       "      <td>chrX</td>\n",
       "      <td>18575402</td>\n",
       "      <td>CM081205</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=CDKL5;STRAND=+;DNA=NM_003...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64303</th>\n",
       "      <td>chrX</td>\n",
       "      <td>74524444</td>\n",
       "      <td>CM1613116</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=SLC16A2;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64587</th>\n",
       "      <td>chrX</td>\n",
       "      <td>8623612</td>\n",
       "      <td>CD146454</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64588</th>\n",
       "      <td>chrX</td>\n",
       "      <td>8623617</td>\n",
       "      <td>CM146452</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_000...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>94 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "75     chr10  100990866  CS1715008   T   C    .      .   \n",
       "291    chr10  119677227   CM117921   C   T    .      .   \n",
       "476    chr10   13298236   CM001290   G   A    .      .   \n",
       "3266    chr1  119955217  CM1110715   A   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "62204   chrX   18575402   CM081205   G   A    .      .   \n",
       "64303   chrX   74524444  CM1613116   G   A    .      .   \n",
       "64587   chrX    8623612   CD146454  CA   C    .      .   \n",
       "64588   chrX    8623617   CM146452   G   T    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "75     CLASS=R;MUT=REF;GENE=TWNK;STRAND=+;DNA=NM_0218...  \n",
       "291    CLASS=R;MUT=ALT;GENE=BAG3;STRAND=+;DNA=NM_0042...  \n",
       "476    CLASS=R;MUT=ALT;GENE=PHYH;STRAND=-;DNA=NM_0062...  \n",
       "3266   CLASS=R;MUT=ALT;GENE=NOTCH2;STRAND=-;DNA=NM_02...  \n",
       "...                                                  ...  \n",
       "62204  CLASS=R;MUT=ALT;GENE=CDKL5;STRAND=+;DNA=NM_003...  \n",
       "64303  CLASS=R;MUT=ALT;GENE=SLC16A2;STRAND=+;DNA=NM_0...  \n",
       "64587  CLASS=R;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_000...  \n",
       "64588  CLASS=R;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_000...  \n",
       "\n",
       "[94 rows x 8 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_r = hgmd_disease_related_variants_mapped_to_nORFs[hgmd_disease_related_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=R')]\n",
    "only_r"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cutting out the variants of the above dataframe, we get:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64631</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64632</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64633</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64634</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53418 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "64631   chrX    9760731   CM981395   A   G    .      .   \n",
       "64632   chrX    9760732   CI183806   G  GA    .      .   \n",
       "64633   chrX    9760736   CD171619  GC   G    .      .   \n",
       "64634   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "64631  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64632  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64633  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64634  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[53418 rows x 8 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs, only_r, only_r]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will follow the same process as above, but this time for 'CLASS=FP', since these are not known to be pathogenic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64631</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64632</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64633</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64634</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52655 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "64631   chrX    9760731   CM981395   A   G    .      .   \n",
       "64632   chrX    9760732   CI183806   G  GA    .      .   \n",
       "64633   chrX    9760736   CD171619  GC   G    .      .   \n",
       "64634   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "64631  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64632  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64633  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64634  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[52655 rows x 8 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_fp = hgmd_disease_related_variants_mapped_to_nORFs[hgmd_disease_related_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=FP') \n",
    "                                                        & ~hgmd_disease_related_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DFP')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_no_r, only_fp, only_fp]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will also take out 'CLASS = DP' and 'CLASS = DFP' below using the same steps as above:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64631</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64632</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64633</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64634</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52238 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "64631   chrX    9760731   CM981395   A   G    .      .   \n",
       "64632   chrX    9760732   CI183806   G  GA    .      .   \n",
       "64633   chrX    9760736   CD171619  GC   G    .      .   \n",
       "64634   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "64631  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64632  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64633  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64634  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[52238 rows x 8 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_dfp = hgmd_disease_related_variants_mapped_to_nORFs[hgmd_disease_related_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DFP')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp, only_dfp, only_dfp]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64631</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64632</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64633</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64634</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>51452 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "64631   chrX    9760731   CM981395   A   G    .      .   \n",
       "64632   chrX    9760732   CI183806   G  GA    .      .   \n",
       "64633   chrX    9760736   CD171619  GC   G    .      .   \n",
       "64634   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "64631  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64632  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64633  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64634  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[51452 rows x 8 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_dp = hgmd_disease_related_variants_mapped_to_nORFs[hgmd_disease_related_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DP')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp, only_dp, only_dp]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As a check to show there are no longer any variants without 'CLASS=DM', we run the following code which gives an empty output, signifying our success in filtering the dataframe:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [CHROM, POS, ID, REF, ALT, QUAL, FILTER, INFO]\n",
       "Index: []"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_dm = hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp[~hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp['INFO'].str.contains('CLASS=DM')]\n",
    "not_dm"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### A compact version of the code that gives the same result is below. I performed the code step by step so that it is easier to follow."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64631</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64632</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64633</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64634</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>51452 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "64631   chrX    9760731   CM981395   A   G    .      .   \n",
       "64632   chrX    9760732   CI183806   G  GA    .      .   \n",
       "64633   chrX    9760736   CD171619  GC   G    .      .   \n",
       "64634   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "64631  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64632  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64633  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "64634  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[51452 rows x 8 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "finding_the_uncertainty = hgmd_variants_mapped_to_nORFs[~hgmd_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DM?;') & \n",
    "                                                        ~hgmd_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DP') \n",
    "                                                        & ~hgmd_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DFP') \n",
    "                                                        & ~hgmd_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=FP') \n",
    "                                                        & ~hgmd_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=R')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs = pd.concat([hgmd_variants_mapped_to_nORFs, finding_the_uncertainty, finding_the_uncertainty]).drop_duplicates(keep=False)\n",
    "only_r = hgmd_disease_related_variants_mapped_to_nORFs[hgmd_disease_related_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=R')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs, only_r, only_r]).drop_duplicates(keep=False)\n",
    "only_fp = hgmd_disease_related_variants_mapped_to_nORFs[hgmd_disease_related_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=FP') \n",
    "                                                        & ~hgmd_disease_related_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DFP')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_no_r, only_fp, only_fp]).drop_duplicates(keep=False)\n",
    "only_dfp = hgmd_disease_related_variants_mapped_to_nORFs[hgmd_disease_related_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DFP')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp, only_dfp, only_dfp]).drop_duplicates(keep=False)\n",
    "only_dp = hgmd_disease_related_variants_mapped_to_nORFs[hgmd_disease_related_variants_mapped_to_nORFs['INFO'].str.contains('CLASS=DP')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp, only_dp, only_dp]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we will create a dataframe for the coding region nORFs only from the -split command on bedtools intersect:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10</th>\n",
       "      <th>100154922</th>\n",
       "      <th>CM140970</th>\n",
       "      <th>G</th>\n",
       "      <th>A</th>\n",
       "      <th>.</th>\n",
       "      <th>..1</th>\n",
       "      <th>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_006459.3:c.763C&gt;T;PROT=NP_006450.2:p.R255*;DB=rs876657413;PHEN=\"Spastic_paraplegia_62\";RANKSCORE=0.99</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100262050</td>\n",
       "      <td>CM162834</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59425</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59426</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59427</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59428</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59429 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr10  100154922   CM140970   G   A  . ..1  \\\n",
       "0      chr10  100183802   CM140971   C   A  .   .   \n",
       "1      chr10  100253438  CI1824020   A  AT  .   .   \n",
       "2      chr10  100256298   CD162836  TG   T  .   .   \n",
       "3      chr10  100262050   CM162834   C   G  .   .   \n",
       "...      ...        ...        ...  ..  .. ..  ..   \n",
       "59425   chrX    9760731   CM981395   A   G  .   .   \n",
       "59426   chrX    9760732   CI183806   G  GA  .   .   \n",
       "59427   chrX    9760736   CD171619  GC   G  .   .   \n",
       "59428   chrX    9760741   CI115195   A  AG  .   .   \n",
       "\n",
       "      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_006459.3:c.763C>T;PROT=NP_006450.2:p.R255*;DB=rs876657413;PHEN=\"Spastic_paraplegia_62\";RANKSCORE=0.99  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...                                                                                                  \n",
       "1      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...                                                                                                  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...                                                                                                  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...                                                                                                  \n",
       "...                                                  ...                                                                                                  \n",
       "59425  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...                                                                                                  \n",
       "59426  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...                                                                                                  \n",
       "59427  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...                                                                                                  \n",
       "59428  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...                                                                                                  \n",
       "\n",
       "[59429 rows x 8 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_region_variants_only = read_vcf('hgmd_mapped_to_norfs_real_with_split.vcf')\n",
    "hgmd_coding_region_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We apply the same technique as above to get this into a usable form. Luckily the first entry has the same information as our previous code so we can copy and paste line for line, only changing the file name we are calling on:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59426</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59427</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59428</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59429</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>59430 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "59426   chrX    9760731   CM981395   A   G    .      .   \n",
       "59427   chrX    9760732   CI183806   G  GA    .      .   \n",
       "59428   chrX    9760736   CD171619  GC   G    .      .   \n",
       "59429   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "59426  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59427  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59428  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59429  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[59430 rows x 8 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_region_variants_only.loc[-1] = ['chr10', '100154922', 'CM140970', 'G', 'A', '.', '.', 'CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_006459.3:c.763C>T;PROT=NP_006450.2:p.R255*;DB=rs876657413;PHEN=\"Spastic_paraplegia_62\";RANKSCORE=0.99']\n",
    "hgmd_coding_region_variants_only.index = hgmd_coding_region_variants_only.index + 1  # shifting index\n",
    "hgmd_coding_region_variants_only = hgmd_coding_region_variants_only.sort_index()\n",
    "hgmd_coding_region_variants_only.columns = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO']\n",
    "hgmd_coding_region_variants_only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, we will cut out the uncertainties like we did above, leaving only the truly pathogenic variants:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101580292</td>\n",
       "      <td>CD175017</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=POLL;STRAND=-;DNA=NM_01...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102109470</td>\n",
       "      <td>CM188932</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=LDB1;STRAND=-;DNA=NM_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102157525</td>\n",
       "      <td>CM1416537</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=NOLC1;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102375375</td>\n",
       "      <td>CM160744</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=GBF1;STRAND=+;DNA=NM_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59376</th>\n",
       "      <td>chrX</td>\n",
       "      <td>86027503</td>\n",
       "      <td>CM137329</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59388</th>\n",
       "      <td>chrX</td>\n",
       "      <td>8623669</td>\n",
       "      <td>CM133761</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59390</th>\n",
       "      <td>chrX</td>\n",
       "      <td>8623670</td>\n",
       "      <td>CM183496</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59393</th>\n",
       "      <td>chrX</td>\n",
       "      <td>87518266</td>\n",
       "      <td>CM1417120</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM?;MUT=ALT;GENE=KLHL4;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10104 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "30     chr10  101580292   CD175017  TG   T    .      .   \n",
       "44     chr10  102109470   CM188932   G   A    .      .   \n",
       "45     chr10  102157525  CM1416537   T   G    .      .   \n",
       "54     chr10  102375375   CM160744   G   A    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "59376   chrX   86027503   CM137329   A   G    .      .   \n",
       "59388   chrX    8623669   CM133761   C   T    .      .   \n",
       "59390   chrX    8623670   CM183496   A   T    .      .   \n",
       "59393   chrX   87518266  CM1417120   C   G    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "30     CLASS=DM?;MUT=ALT;GENE=POLL;STRAND=-;DNA=NM_01...  \n",
       "44     CLASS=DM?;MUT=ALT;GENE=LDB1;STRAND=-;DNA=NM_00...  \n",
       "45     CLASS=DM?;MUT=ALT;GENE=NOLC1;STRAND=+;DNA=NM_0...  \n",
       "54     CLASS=DM?;MUT=ALT;GENE=GBF1;STRAND=+;DNA=NM_00...  \n",
       "...                                                  ...  \n",
       "59376  CLASS=DM?;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_000...  \n",
       "59388  CLASS=DM?;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_0...  \n",
       "59390  CLASS=DM?;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_0...  \n",
       "59393  CLASS=DM?;MUT=ALT;GENE=KLHL4;STRAND=+;DNA=NM_0...  \n",
       "\n",
       "[10104 rows x 8 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "finding_the_uncertainty_coding_region = hgmd_coding_region_variants_only[~hgmd_coding_region_variants_only['INFO'].str.contains('CLASS=DM?;') & \n",
    "                                                        ~hgmd_coding_region_variants_only['INFO'].str.contains('CLASS=DP') \n",
    "                                                        & ~hgmd_coding_region_variants_only['INFO'].str.contains('CLASS=DFP') \n",
    "                                                        & ~hgmd_coding_region_variants_only['INFO'].str.contains('CLASS=FP') \n",
    "                                                        & ~hgmd_coding_region_variants_only['INFO'].str.contains('CLASS=R')]\n",
    "finding_the_uncertainty_coding_region"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Again, we will cut these out of the dataframe using the concatenation function:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59426</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59427</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59428</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59429</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49326 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "59426   chrX    9760731   CM981395   A   G    .      .   \n",
       "59427   chrX    9760732   CI183806   G  GA    .      .   \n",
       "59428   chrX    9760736   CD171619  GC   G    .      .   \n",
       "59429   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "59426  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59427  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59428  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59429  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[49326 rows x 8 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_disease_related_variants_mapped_to_nORFs_coding_region = pd.concat([hgmd_coding_region_variants_only, finding_the_uncertainty_coding_region, finding_the_uncertainty_coding_region]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_coding_region"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will also take out the 'CLASS=R' from this dataframe since it is unnecessary to our classifier:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>chr10</td>\n",
       "      <td>119677227</td>\n",
       "      <td>CM117921</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=BAG3;STRAND=+;DNA=NM_0042...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402</th>\n",
       "      <td>chr10</td>\n",
       "      <td>13298236</td>\n",
       "      <td>CM001290</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=PHYH;STRAND=-;DNA=NM_0062...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2966</th>\n",
       "      <td>chr1</td>\n",
       "      <td>119955217</td>\n",
       "      <td>CM1110715</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=NOTCH2;STRAND=-;DNA=NM_02...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5900</th>\n",
       "      <td>chr11</td>\n",
       "      <td>6390705</td>\n",
       "      <td>CM154141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=SMPD1;STRAND=+;DNA=NM_000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57205</th>\n",
       "      <td>chrX</td>\n",
       "      <td>18575402</td>\n",
       "      <td>CM081205</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=CDKL5;STRAND=+;DNA=NM_003...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59121</th>\n",
       "      <td>chrX</td>\n",
       "      <td>74524444</td>\n",
       "      <td>CM1613116</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=SLC16A2;STRAND=+;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59382</th>\n",
       "      <td>chrX</td>\n",
       "      <td>8623612</td>\n",
       "      <td>CD146454</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_000...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59383</th>\n",
       "      <td>chrX</td>\n",
       "      <td>8623617</td>\n",
       "      <td>CM146452</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=R;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_000...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>85 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "223    chr10  119677227   CM117921   C   T    .      .   \n",
       "402    chr10   13298236   CM001290   G   A    .      .   \n",
       "2966    chr1  119955217  CM1110715   A   T    .      .   \n",
       "5900   chr11    6390705   CM154141   T   C    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "57205   chrX   18575402   CM081205   G   A    .      .   \n",
       "59121   chrX   74524444  CM1613116   G   A    .      .   \n",
       "59382   chrX    8623612   CD146454  CA   C    .      .   \n",
       "59383   chrX    8623617   CM146452   G   T    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "223    CLASS=R;MUT=ALT;GENE=BAG3;STRAND=+;DNA=NM_0042...  \n",
       "402    CLASS=R;MUT=ALT;GENE=PHYH;STRAND=-;DNA=NM_0062...  \n",
       "2966   CLASS=R;MUT=ALT;GENE=NOTCH2;STRAND=-;DNA=NM_02...  \n",
       "5900   CLASS=R;MUT=ALT;GENE=SMPD1;STRAND=+;DNA=NM_000...  \n",
       "...                                                  ...  \n",
       "57205  CLASS=R;MUT=ALT;GENE=CDKL5;STRAND=+;DNA=NM_003...  \n",
       "59121  CLASS=R;MUT=ALT;GENE=SLC16A2;STRAND=+;DNA=NM_0...  \n",
       "59382  CLASS=R;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_000...  \n",
       "59383  CLASS=R;MUT=ALT;GENE=ANOS1;STRAND=-;DNA=NM_000...  \n",
       "\n",
       "[85 rows x 8 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_r_coding_region = hgmd_disease_related_variants_mapped_to_nORFs_coding_region[hgmd_disease_related_variants_mapped_to_nORFs_coding_region['INFO'].str.contains('CLASS=R')]\n",
    "only_r_coding_region"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we will cut these rows out of the dataframe. This is shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59426</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59427</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59428</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59429</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49241 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "59426   chrX    9760731   CM981395   A   G    .      .   \n",
       "59427   chrX    9760732   CI183806   G  GA    .      .   \n",
       "59428   chrX    9760736   CD171619  GC   G    .      .   \n",
       "59429   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "59426  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59427  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59428  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59429  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[49241 rows x 8 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_coding_region = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_coding_region, only_r_coding_region, only_r_coding_region]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_coding_region"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Just like for the whole region, we do this again with 'CLASS=FP' for the coding region:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59426</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59427</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59428</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59429</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48574 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "59426   chrX    9760731   CM981395   A   G    .      .   \n",
       "59427   chrX    9760732   CI183806   G  GA    .      .   \n",
       "59428   chrX    9760736   CD171619  GC   G    .      .   \n",
       "59429   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "59426  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59427  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59428  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59429  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[48574 rows x 8 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_fp_coding_region = hgmd_coding_region_variants_only[hgmd_coding_region_variants_only['INFO'].str.contains('CLASS=FP') \n",
    "                                                        & ~hgmd_coding_region_variants_only['INFO'].str.contains('CLASS=DFP')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_coding_region = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_no_r_coding_region, only_fp_coding_region, only_fp_coding_region]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_coding_region"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We again do this for 'CLASS=DFP':"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59426</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59427</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59428</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59429</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48252 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "59426   chrX    9760731   CM981395   A   G    .      .   \n",
       "59427   chrX    9760732   CI183806   G  GA    .      .   \n",
       "59428   chrX    9760736   CD171619  GC   G    .      .   \n",
       "59429   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "59426  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59427  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59428  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59429  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[48252 rows x 8 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_dfp_coding_region = hgmd_coding_region_variants_only[hgmd_coding_region_variants_only['INFO'].str.contains('CLASS=DFP')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_coding_region = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_coding_region, only_dfp_coding_region, only_dfp_coding_region]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_coding_region"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We again do this for 'CLASS=DP'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>CM140970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>CM140971</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>CI1824020</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256298</td>\n",
       "      <td>CD162836</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59426</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>CM981395</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59427</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>CI183806</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59428</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760736</td>\n",
       "      <td>CD171619</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59429</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>CI115195</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47645 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "0      chr10  100154922   CM140970   G   A    .      .   \n",
       "1      chr10  100183802   CM140971   C   A    .      .   \n",
       "2      chr10  100253438  CI1824020   A  AT    .      .   \n",
       "3      chr10  100256298   CD162836  TG   T    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "59426   chrX    9760731   CM981395   A   G    .      .   \n",
       "59427   chrX    9760732   CI183806   G  GA    .      .   \n",
       "59428   chrX    9760736   CD171619  GC   G    .      .   \n",
       "59429   chrX    9760741   CI115195   A  AG    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "0      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "1      CLASS=DM;MUT=ALT;GENE=ERLIN1;STRAND=-;DNA=NM_0...  \n",
       "2      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "3      CLASS=DM;MUT=ALT;GENE=CWF19L1;STRAND=-;DNA=NM_...  \n",
       "...                                                  ...  \n",
       "59426  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59427  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59428  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "59429  CLASS=DM;MUT=ALT;GENE=GPR143;STRAND=-;DNA=NM_0...  \n",
       "\n",
       "[47645 rows x 8 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_dp_coding_region = hgmd_coding_region_variants_only[hgmd_coding_region_variants_only['INFO'].str.contains('CLASS=DP')]\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp_coding_region = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_coding_region, only_dp_coding_region, only_dp_coding_region]).drop_duplicates(keep=False)\n",
    "hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp_coding_region"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Again, it is possible to condese the code above for the coding region only into one cell. I will not do this, however, for the coding region only case."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we can use the above dataframes to create a separate dataframe showing only the non-coding region variants using, again, the concat function. This will give us the true number of HGMD variants found in the non-coding regions of nORFs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988295</td>\n",
       "      <td>CM114899</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_021...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988415</td>\n",
       "      <td>CM164756</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_021...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988457</td>\n",
       "      <td>CM127719</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_021...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988526</td>\n",
       "      <td>CM1610318</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_021...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64559</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964053</td>\n",
       "      <td>CS1810957</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_0003...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64560</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964054</td>\n",
       "      <td>CS1723659</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_0003...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64561</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85965588</td>\n",
       "      <td>CS173873</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_0003...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64562</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85968639</td>\n",
       "      <td>CS032064</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>CLASS=DM;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_0003...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3807 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS         ID REF ALT QUAL FILTER  \\\n",
       "9      chr10  100988295   CM114899   C   T    .      .   \n",
       "10     chr10  100988415   CM164756   A   T    .      .   \n",
       "11     chr10  100988457   CM127719   C   T    .      .   \n",
       "12     chr10  100988526  CM1610318   A   G    .      .   \n",
       "...      ...        ...        ...  ..  ..  ...    ...   \n",
       "64559   chrX   85964053  CS1810957   C   G    .      .   \n",
       "64560   chrX   85964054  CS1723659   T   C    .      .   \n",
       "64561   chrX   85965588   CS173873   T   C    .      .   \n",
       "64562   chrX   85968639   CS032064   A   T    .      .   \n",
       "\n",
       "                                                    INFO  \n",
       "9      CLASS=DM;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_021...  \n",
       "10     CLASS=DM;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_021...  \n",
       "11     CLASS=DM;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_021...  \n",
       "12     CLASS=DM;MUT=ALT;GENE=TWNK;STRAND=+;DNA=NM_021...  \n",
       "...                                                  ...  \n",
       "64559  CLASS=DM;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_0003...  \n",
       "64560  CLASS=DM;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_0003...  \n",
       "64561  CLASS=DM;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_0003...  \n",
       "64562  CLASS=DM;MUT=ALT;GENE=CHM;STRAND=-;DNA=NM_0003...  \n",
       "\n",
       "[3807 rows x 8 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_region_variants_only = pd.concat([hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp, hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp_coding_region, hgmd_disease_related_variants_mapped_to_nORFs_no_r_no_fp_no_dfp_no_dp_coding_region]).drop_duplicates(keep=False)\n",
    "hgmd_noncoding_region_variants_only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
