{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merging Annotations For Different Regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "pd.options.display.max_rows = 8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below we import our vcf reading function:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_vcf(path):\n",
    "    with open(path, 'r') as f:\n",
    "        lines = [l for l in f if not l.startswith('##')]\n",
    "    return pd.read_csv(io.StringIO(''.join(lines)), dtype={'#CHROM': str, 'POS':int, 'ID':str, 'REF': str, 'ALT': str, 'QUAL': str, 'FILTER': str, 'INFO': str}, sep='\\t').rename(columns={'#CHROM': 'CHROM'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Benign Coding Region:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Need to rename a few columns in ANNOVAR dataframes in order to merge them with CADD annotations and also remove 'chr' in all the values under the #Chrom column. Notice how we merge the dataframes using how = 'left' notation. The reason for this is that there are more annotations using CADD and this allows us to keep all of them. Where there is a corresponding annotation for ANNNOVAR, the columns following the CADD annotations get filled in with the ANNOVAR annotations. If not, the remaining columns are left blank."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Human Derived"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the annovar annotation file for the human derived coding region. We have cut the string 'chr' out of the first column and changed the first and second column names to #Chrom and Pos respectively:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>100020652</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>DNMBP;CPN1</td>\n",
       "      <td>dist=10699;dist=21656</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>100190879</td>\n",
       "      <td>100190879</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CHUK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>CHUK:NM_001278:exon20:c.A2198G:p.N733S</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00447999043875</td>\n",
       "      <td>0.039</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>100233196</td>\n",
       "      <td>100233196</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>NM_018294:c.*31C&gt;T;NM_001303407:c.*31C&gt;T;NM_00...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>100267615</td>\n",
       "      <td>100267615</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>NM_018294:c.-22G&gt;A;NM_001303407:c.-20707G&gt;A;NM...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53794</th>\n",
       "      <td>X</td>\n",
       "      <td>99719539</td>\n",
       "      <td>99719539</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>XRCC6P5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53795</th>\n",
       "      <td>X</td>\n",
       "      <td>99719939</td>\n",
       "      <td>99719939</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>XRCC6P5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53796</th>\n",
       "      <td>X</td>\n",
       "      <td>99720084</td>\n",
       "      <td>99720084</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>XRCC6P5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53797</th>\n",
       "      <td>X</td>\n",
       "      <td>99721008</td>\n",
       "      <td>99721008</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>XRCC6P5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53798 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos        End Ref Alt  Func.refGene Gene.refGene  \\\n",
       "0         10  100020652  100020652   G   A    intergenic   DNMBP;CPN1   \n",
       "1         10  100190879  100190879   T   C        exonic         CHUK   \n",
       "2         10  100233196  100233196   G   A          UTR3      CWF19L1   \n",
       "3         10  100267615  100267615   C   T          UTR5      CWF19L1   \n",
       "...      ...        ...        ...  ..  ..           ...          ...   \n",
       "53794      X   99719539   99719539   A   G  ncRNA_exonic      XRCC6P5   \n",
       "53795      X   99719939   99719939   C   A  ncRNA_exonic      XRCC6P5   \n",
       "53796      X   99720084   99720084   G   A  ncRNA_exonic      XRCC6P5   \n",
       "53797      X   99721008   99721008   G   A  ncRNA_exonic      XRCC6P5   \n",
       "\n",
       "                                      GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                  dist=10699;dist=21656                  .   \n",
       "1                                                      .  nonsynonymous SNV   \n",
       "2      NM_018294:c.*31C>T;NM_001303407:c.*31C>T;NM_00...                  .   \n",
       "3      NM_018294:c.-22G>A;NM_001303407:c.-20707G>A;NM...                  .   \n",
       "...                                                  ...                ...   \n",
       "53794                                                  .                  .   \n",
       "53795                                                  .                  .   \n",
       "53796                                                  .                  .   \n",
       "53797                                                  .                  .   \n",
       "\n",
       "                             AAChange.refGene  ... BP4 BP5 BP6 BP7 cosmic70  \\\n",
       "0                                           .  ...   .   .   .   .        .   \n",
       "1      CHUK:NM_001278:exon20:c.A2198G:p.N733S  ...   1   0   0   0        .   \n",
       "2                                           .  ...   .   .   .   .        .   \n",
       "3                                           .  ...   .   .   .   .        .   \n",
       "...                                       ...  ...  ..  ..  ..  ..      ...   \n",
       "53794                                       .  ...   .   .   .   .        .   \n",
       "53795                                       .  ...   .   .   .   .        .   \n",
       "53796                                       .  ...   .   .   .   .        .   \n",
       "53797                                       .  ...   .   .   .   .        .   \n",
       "\n",
       "                   MCAP  REVEL regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0                     .      .          .              .                    .  \n",
       "1      0.00447999043875  0.039          .              .                    .  \n",
       "2                     .      .          .              .                    .  \n",
       "3                     .      .          .              .                    .  \n",
       "...                 ...    ...        ...            ...                  ...  \n",
       "53794                 .      .          .              .                    .  \n",
       "53795                 .      .          .              .                    .  \n",
       "53796                 .      .          .              .                    .  \n",
       "53797                 .      .          .              .                    .  \n",
       "\n",
       "[53798 rows x 149 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_annotated_with_annovar = pd.read_csv('human_derived_coding_annotated.hg38_multianno.csv')\n",
    "human_derived_coding_annotated_with_annovar['Chr'] = human_derived_coding_annotated_with_annovar['Chr'].map(lambda x: x.lstrip('chr').rstrip('aAbBcC'))\n",
    "human_derived_coding_annotated_with_annovar = human_derived_coding_annotated_with_annovar.rename(columns = {'Chr': '#Chrom', 'Start': 'Pos'})\n",
    "human_derived_coding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the file for the CADD annotations for the human derived coding region:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>5</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>944699</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>117</td>\n",
       "      <td>2573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>2.980499</td>\n",
       "      <td>23.200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122229</th>\n",
       "      <td>X</td>\n",
       "      <td>155260197</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3PRIME_UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>3_prime_UTR</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>873</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.219204</td>\n",
       "      <td>0.473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122230</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122231</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122232</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>122233 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0           1     943329   C   T  SNV       0  RegulatoryFeature   \n",
       "1           1     943329   C   T  SNV       0   CodingTranscript   \n",
       "2           1     943329   C   T  SNV       0         Intergenic   \n",
       "3           1     944699   C   T  SNV       0   CodingTranscript   \n",
       "...       ...        ...  ..  ..  ...     ...                ...   \n",
       "122229      X  155260197   C   A  SNV       0         Transcript   \n",
       "122230      X  155612732   T   C  SNV       0  RegulatoryFeature   \n",
       "122231      X  155612732   T   C  SNV       0         Transcript   \n",
       "122232      X  155612732   T   C  SNV       0         Intergenic   \n",
       "\n",
       "           Consequence  ConsScore   ConsDetail  ...  Freq10000bp  Rare10000bp  \\\n",
       "0           REGULATORY          4   regulatory  ...           18          117   \n",
       "1           SYNONYMOUS          5   synonymous  ...           18          117   \n",
       "2           DOWNSTREAM          1   downstream  ...           18          117   \n",
       "3       NON_SYNONYMOUS          7     missense  ...           16          117   \n",
       "...                ...        ...          ...  ...          ...          ...   \n",
       "122229      3PRIME_UTR          2  3_prime_UTR  ...            9           36   \n",
       "122230      REGULATORY          4   regulatory  ...            7           27   \n",
       "122231        INTRONIC          2       intron  ...            7           27   \n",
       "122232      DOWNSTREAM          1   downstream  ...            7           27   \n",
       "\n",
       "        Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              2586         CTCF Binding Site                NaN   \n",
       "1              2586         CTCF Binding Site                NaN   \n",
       "2              2586         CTCF Binding Site                NaN   \n",
       "3              2573                       NaN                NaN   \n",
       "...             ...                       ...                ...   \n",
       "122229          873                       NaN                NaN   \n",
       "122230          866                  Promoter                NaN   \n",
       "122231          866                  Promoter                NaN   \n",
       "122232          866                  Promoter                NaN   \n",
       "\n",
       "        dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore   PHRED  \n",
       "0                    NaN           88.0          121.0  0.590702   8.117  \n",
       "1                    NaN           88.0          121.0  0.590702   8.117  \n",
       "2                    NaN           88.0          121.0  0.590702   8.117  \n",
       "3                    NaN           12.0           13.0  2.980499  23.200  \n",
       "...                  ...            ...            ...       ...     ...  \n",
       "122229               NaN            3.0            6.0 -0.219204   0.473  \n",
       "122230               NaN          112.0          202.0  0.661682   8.709  \n",
       "122231               NaN          112.0          202.0  0.661682   8.709  \n",
       "122232               NaN          112.0          202.0  0.661682   8.709  \n",
       "\n",
       "[122233 rows x 125 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_annotated_with_cadd = pd.read_table('human_derived_coding_for_cadd_noheader.tsv')\n",
    "human_derived_coding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below we have merged the two dataframes above. We have merged using the CADD annotation dataframe (how = 'left') since there were more variants and more annotations given by CADD than ANNOVAR. This fact is shown below:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Merged annotations based on CADD:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>5</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>944699</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122229</th>\n",
       "      <td>X</td>\n",
       "      <td>155260197</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3PRIME_UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>3_prime_UTR</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122230</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122231</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122232</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>122233 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0           1     943329   C   T  SNV       0  RegulatoryFeature   \n",
       "1           1     943329   C   T  SNV       0   CodingTranscript   \n",
       "2           1     943329   C   T  SNV       0         Intergenic   \n",
       "3           1     944699   C   T  SNV       0   CodingTranscript   \n",
       "...       ...        ...  ..  ..  ...     ...                ...   \n",
       "122229      X  155260197   C   A  SNV       0         Transcript   \n",
       "122230      X  155612732   T   C  SNV       0  RegulatoryFeature   \n",
       "122231      X  155612732   T   C  SNV       0         Transcript   \n",
       "122232      X  155612732   T   C  SNV       0         Intergenic   \n",
       "\n",
       "           Consequence  ConsScore   ConsDetail  ...  BP4  BP5  BP6  BP7  \\\n",
       "0           REGULATORY          4   regulatory  ...  NaN  NaN  NaN  NaN   \n",
       "1           SYNONYMOUS          5   synonymous  ...  NaN  NaN  NaN  NaN   \n",
       "2           DOWNSTREAM          1   downstream  ...  NaN  NaN  NaN  NaN   \n",
       "3       NON_SYNONYMOUS          7     missense  ...  NaN  NaN  NaN  NaN   \n",
       "...                ...        ...          ...  ...  ...  ...  ...  ...   \n",
       "122229      3PRIME_UTR          2  3_prime_UTR  ...    .    .    .    .   \n",
       "122230      REGULATORY          4   regulatory  ...    .    .    .    .   \n",
       "122231        INTRONIC          2       intron  ...    .    .    .    .   \n",
       "122232      DOWNSTREAM          1   downstream  ...    .    .    .    .   \n",
       "\n",
       "        cosmic70  MCAP REVEL regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0            NaN   NaN   NaN        NaN            NaN                  NaN  \n",
       "1            NaN   NaN   NaN        NaN            NaN                  NaN  \n",
       "2            NaN   NaN   NaN        NaN            NaN                  NaN  \n",
       "3            NaN   NaN   NaN        NaN            NaN                  NaN  \n",
       "...          ...   ...   ...        ...            ...                  ...  \n",
       "122229         .     .     .          .              .                    .  \n",
       "122230         .     .     .          .              .                    .  \n",
       "122231         .     .     .          .              .                    .  \n",
       "122232         .     .     .          .              .                    .  \n",
       "\n",
       "[122233 rows x 270 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_merged = pd.merge(human_derived_coding_annotated_with_cadd, human_derived_coding_annotated_with_annovar,\n",
    "                                         how = 'left', on = ['#Chrom', 'Pos', 'Ref', 'Alt'])\n",
    "human_derived_coding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Merged annotations based on ANNOVAR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEName</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>motifEScoreChng</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>FeatureID</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>Intron</th>\n",
       "      <th>Exon</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>relcDNApos</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>relCDSpos</th>\n",
       "      <th>protPos</th>\n",
       "      <th>relProtPos</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Dst2Splice</th>\n",
       "      <th>Dst2SplType</th>\n",
       "      <th>minDistTSS</th>\n",
       "      <th>minDistTSE</th>\n",
       "      <th>SIFTcat</th>\n",
       "      <th>SIFTval</th>\n",
       "      <th>PolyPhenCat</th>\n",
       "      <th>PolyPhenVal</th>\n",
       "      <th>priPhCons</th>\n",
       "      <th>mamPhCons</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>priPhyloP</th>\n",
       "      <th>mamPhyloP</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>bStatistic</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>mirSVR-Score</th>\n",
       "      <th>mirSVR-E</th>\n",
       "      <th>mirSVR-Aln</th>\n",
       "      <th>cHmm_E1</th>\n",
       "      <th>cHmm_E2</th>\n",
       "      <th>cHmm_E3</th>\n",
       "      <th>cHmm_E4</th>\n",
       "      <th>cHmm_E5</th>\n",
       "      <th>cHmm_E6</th>\n",
       "      <th>cHmm_E7</th>\n",
       "      <th>cHmm_E8</th>\n",
       "      <th>cHmm_E9</th>\n",
       "      <th>cHmm_E10</th>\n",
       "      <th>cHmm_E11</th>\n",
       "      <th>cHmm_E12</th>\n",
       "      <th>cHmm_E13</th>\n",
       "      <th>cHmm_E14</th>\n",
       "      <th>cHmm_E15</th>\n",
       "      <th>cHmm_E16</th>\n",
       "      <th>cHmm_E17</th>\n",
       "      <th>cHmm_E18</th>\n",
       "      <th>cHmm_E19</th>\n",
       "      <th>cHmm_E20</th>\n",
       "      <th>cHmm_E21</th>\n",
       "      <th>cHmm_E22</th>\n",
       "      <th>cHmm_E23</th>\n",
       "      <th>cHmm_E24</th>\n",
       "      <th>cHmm_E25</th>\n",
       "      <th>GerpRS</th>\n",
       "      <th>GerpRSpval</th>\n",
       "      <th>GerpN</th>\n",
       "      <th>GerpS</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>motifDist</th>\n",
       "      <th>EncodeH3K4me1-sum</th>\n",
       "      <th>EncodeH3K4me1-max</th>\n",
       "      <th>EncodeH3K4me2-sum</th>\n",
       "      <th>EncodeH3K4me2-max</th>\n",
       "      <th>EncodeH3K4me3-sum</th>\n",
       "      <th>EncodeH3K4me3-max</th>\n",
       "      <th>EncodeH3K9ac-sum</th>\n",
       "      <th>EncodeH3K9ac-max</th>\n",
       "      <th>EncodeH3K9me3-sum</th>\n",
       "      <th>EncodeH3K9me3-max</th>\n",
       "      <th>EncodeH3K27ac-sum</th>\n",
       "      <th>EncodeH3K27ac-max</th>\n",
       "      <th>EncodeH3K27me3-sum</th>\n",
       "      <th>EncodeH3K27me3-max</th>\n",
       "      <th>EncodeH3K36me3-sum</th>\n",
       "      <th>EncodeH3K36me3-max</th>\n",
       "      <th>EncodeH3K79me2-sum</th>\n",
       "      <th>EncodeH3K79me2-max</th>\n",
       "      <th>EncodeH4K20me1-sum</th>\n",
       "      <th>EncodeH4K20me1-max</th>\n",
       "      <th>EncodeH2AFZ-sum</th>\n",
       "      <th>EncodeH2AFZ-max</th>\n",
       "      <th>EncodeDNase-sum</th>\n",
       "      <th>EncodeDNase-max</th>\n",
       "      <th>EncodetotalRNA-sum</th>\n",
       "      <th>EncodetotalRNA-max</th>\n",
       "      <th>Grantham</th>\n",
       "      <th>Dist2Mutation</th>\n",
       "      <th>Freq100bp</th>\n",
       "      <th>Rare100bp</th>\n",
       "      <th>Sngl100bp</th>\n",
       "      <th>Freq1000bp</th>\n",
       "      <th>Rare1000bp</th>\n",
       "      <th>Sngl1000bp</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>End</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>Func.knownGene</th>\n",
       "      <th>Gene.knownGene</th>\n",
       "      <th>GeneDetail.knownGene</th>\n",
       "      <th>ExonicFunc.knownGene</th>\n",
       "      <th>AAChange.knownGene</th>\n",
       "      <th>Func.ensGene</th>\n",
       "      <th>Gene.ensGene</th>\n",
       "      <th>GeneDetail.ensGene</th>\n",
       "      <th>ExonicFunc.ensGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>SIFT_pred</th>\n",
       "      <th>Polyphen2_HDIV_score</th>\n",
       "      <th>Polyphen2_HDIV_pred</th>\n",
       "      <th>Polyphen2_HVAR_score</th>\n",
       "      <th>Polyphen2_HVAR_pred</th>\n",
       "      <th>LRT_score</th>\n",
       "      <th>LRT_pred</th>\n",
       "      <th>MutationTaster_score</th>\n",
       "      <th>MutationTaster_pred</th>\n",
       "      <th>MutationAssessor_score</th>\n",
       "      <th>MutationAssessor_pred</th>\n",
       "      <th>FATHMM_score</th>\n",
       "      <th>FATHMM_pred</th>\n",
       "      <th>RadialSVM_score</th>\n",
       "      <th>RadialSVM_pred</th>\n",
       "      <th>LR_score</th>\n",
       "      <th>LR_pred</th>\n",
       "      <th>VEST3_score</th>\n",
       "      <th>CADD_raw</th>\n",
       "      <th>CADD_phred</th>\n",
       "      <th>GERP++_RS</th>\n",
       "      <th>phyloP46way_placental</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>Interpro_domain</th>\n",
       "      <th>SIFT_score.1</th>\n",
       "      <th>SIFT_converted_rankscore</th>\n",
       "      <th>SIFT_pred.1</th>\n",
       "      <th>Polyphen2_HDIV_score.1</th>\n",
       "      <th>Polyphen2_HDIV_rankscore</th>\n",
       "      <th>Polyphen2_HDIV_pred.1</th>\n",
       "      <th>Polyphen2_HVAR_score.1</th>\n",
       "      <th>Polyphen2_HVAR_rankscore</th>\n",
       "      <th>Polyphen2_HVAR_pred.1</th>\n",
       "      <th>LRT_score.1</th>\n",
       "      <th>LRT_converted_rankscore</th>\n",
       "      <th>LRT_pred.1</th>\n",
       "      <th>MutationTaster_score.1</th>\n",
       "      <th>MutationTaster_converted_rankscore</th>\n",
       "      <th>MutationTaster_pred.1</th>\n",
       "      <th>MutationAssessor_score.1</th>\n",
       "      <th>MutationAssessor_score_rankscore</th>\n",
       "      <th>MutationAssessor_pred.1</th>\n",
       "      <th>FATHMM_score.1</th>\n",
       "      <th>FATHMM_converted_rankscore</th>\n",
       "      <th>FATHMM_pred.1</th>\n",
       "      <th>PROVEAN_score</th>\n",
       "      <th>PROVEAN_converted_rankscore</th>\n",
       "      <th>PROVEAN_pred</th>\n",
       "      <th>VEST3_score.1</th>\n",
       "      <th>VEST3_rankscore</th>\n",
       "      <th>MetaSVM_score</th>\n",
       "      <th>MetaSVM_rankscore</th>\n",
       "      <th>MetaSVM_pred</th>\n",
       "      <th>MetaLR_score</th>\n",
       "      <th>MetaLR_rankscore</th>\n",
       "      <th>MetaLR_pred</th>\n",
       "      <th>M-CAP_score</th>\n",
       "      <th>M-CAP_rankscore</th>\n",
       "      <th>M-CAP_pred</th>\n",
       "      <th>CADD_raw.1</th>\n",
       "      <th>CADD_raw_rankscore</th>\n",
       "      <th>CADD_phred.1</th>\n",
       "      <th>DANN_score</th>\n",
       "      <th>DANN_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_score</th>\n",
       "      <th>fathmm-MKL_coding_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_pred</th>\n",
       "      <th>Eigen_coding_or_noncoding</th>\n",
       "      <th>Eigen-raw</th>\n",
       "      <th>Eigen-PC-raw</th>\n",
       "      <th>GenoCanyon_score</th>\n",
       "      <th>GenoCanyon_score_rankscore</th>\n",
       "      <th>integrated_fitCons_score</th>\n",
       "      <th>integrated_fitCons_score_rankscore</th>\n",
       "      <th>integrated_confidence_value</th>\n",
       "      <th>GERP++_RS.1</th>\n",
       "      <th>GERP++_RS_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate.1</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP20way_mammalian</th>\n",
       "      <th>phyloP20way_mammalian_rankscore</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons20way_mammalian</th>\n",
       "      <th>phastCons20way_mammalian_rankscore</th>\n",
       "      <th>SiPhy_29way_logOdds.1</th>\n",
       "      <th>SiPhy_29way_logOdds_rankscore</th>\n",
       "      <th>Interpro_domain.1</th>\n",
       "      <th>GTEx_V6_gene</th>\n",
       "      <th>GTEx_V6_tissue</th>\n",
       "      <th>dbscSNV_ADA_SCORE</th>\n",
       "      <th>dbscSNV_RF_SCORE</th>\n",
       "      <th>InterVar_automated</th>\n",
       "      <th>PVS1</th>\n",
       "      <th>PS1</th>\n",
       "      <th>PS2</th>\n",
       "      <th>PS3</th>\n",
       "      <th>PS4</th>\n",
       "      <th>PM1</th>\n",
       "      <th>PM2</th>\n",
       "      <th>PM3</th>\n",
       "      <th>PM4</th>\n",
       "      <th>PM5</th>\n",
       "      <th>PM6</th>\n",
       "      <th>PP1</th>\n",
       "      <th>PP2</th>\n",
       "      <th>PP3</th>\n",
       "      <th>PP4</th>\n",
       "      <th>PP5</th>\n",
       "      <th>BA1</th>\n",
       "      <th>BS1</th>\n",
       "      <th>BS2</th>\n",
       "      <th>BS3</th>\n",
       "      <th>BS4</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>BP3</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>1046375</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1.0</td>\n",
       "      <td>upstream</td>\n",
       "      <td>0.649</td>\n",
       "      <td>0.147</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000173253</td>\n",
       "      <td>ENST00000382251</td>\n",
       "      <td>DMRT2</td>\n",
       "      <td>CCDS6444.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>750.0</td>\n",
       "      <td>2267.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.382</td>\n",
       "      <td>-0.373</td>\n",
       "      <td>-0.324</td>\n",
       "      <td>952.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.61</td>\n",
       "      <td>-3.21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.28</td>\n",
       "      <td>2.72</td>\n",
       "      <td>8.93</td>\n",
       "      <td>2.22</td>\n",
       "      <td>4.07</td>\n",
       "      <td>1.13</td>\n",
       "      <td>2.37</td>\n",
       "      <td>1.3</td>\n",
       "      <td>14.40</td>\n",
       "      <td>3.64</td>\n",
       "      <td>3.24</td>\n",
       "      <td>1.20</td>\n",
       "      <td>73.12</td>\n",
       "      <td>39.14</td>\n",
       "      <td>4.04</td>\n",
       "      <td>1.18</td>\n",
       "      <td>3.34</td>\n",
       "      <td>0.93</td>\n",
       "      <td>4.55</td>\n",
       "      <td>1.07</td>\n",
       "      <td>9.45</td>\n",
       "      <td>3.27</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>213.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>1972.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>-0.106065</td>\n",
       "      <td>0.969</td>\n",
       "      <td>1046375</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>DMRT3;LINC01230</td>\n",
       "      <td>dist=54643;dist=1768</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>LINC01230</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>LINC01230</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>1046375</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NonCodingTranscript</td>\n",
       "      <td>NONCODING_CHANGE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>non_coding_exon</td>\n",
       "      <td>0.649</td>\n",
       "      <td>0.147</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000281769</td>\n",
       "      <td>ENST00000625222</td>\n",
       "      <td>LINC01230</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>751.0</td>\n",
       "      <td>0.249</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>750.0</td>\n",
       "      <td>2267.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.382</td>\n",
       "      <td>-0.373</td>\n",
       "      <td>-0.324</td>\n",
       "      <td>952.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.61</td>\n",
       "      <td>-3.21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.28</td>\n",
       "      <td>2.72</td>\n",
       "      <td>8.93</td>\n",
       "      <td>2.22</td>\n",
       "      <td>4.07</td>\n",
       "      <td>1.13</td>\n",
       "      <td>2.37</td>\n",
       "      <td>1.3</td>\n",
       "      <td>14.40</td>\n",
       "      <td>3.64</td>\n",
       "      <td>3.24</td>\n",
       "      <td>1.20</td>\n",
       "      <td>73.12</td>\n",
       "      <td>39.14</td>\n",
       "      <td>4.04</td>\n",
       "      <td>1.18</td>\n",
       "      <td>3.34</td>\n",
       "      <td>0.93</td>\n",
       "      <td>4.55</td>\n",
       "      <td>1.07</td>\n",
       "      <td>9.45</td>\n",
       "      <td>3.27</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>213.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>1972.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>-0.106065</td>\n",
       "      <td>0.969</td>\n",
       "      <td>1046375</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>DMRT3;LINC01230</td>\n",
       "      <td>dist=54643;dist=1768</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>LINC01230</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>LINC01230</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>1046421</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NonCodingTranscript</td>\n",
       "      <td>NONCODING_CHANGE</td>\n",
       "      <td>5.0</td>\n",
       "      <td>non_coding_exon</td>\n",
       "      <td>0.517</td>\n",
       "      <td>0.080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000281769</td>\n",
       "      <td>ENST00000625222</td>\n",
       "      <td>LINC01230</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/1</td>\n",
       "      <td>797.0</td>\n",
       "      <td>0.264</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>796.0</td>\n",
       "      <td>2221.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.595</td>\n",
       "      <td>-0.197</td>\n",
       "      <td>-0.221</td>\n",
       "      <td>952.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.73</td>\n",
       "      <td>-3.59</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.95</td>\n",
       "      <td>2.37</td>\n",
       "      <td>8.60</td>\n",
       "      <td>2.21</td>\n",
       "      <td>5.45</td>\n",
       "      <td>1.94</td>\n",
       "      <td>3.29</td>\n",
       "      <td>1.3</td>\n",
       "      <td>14.74</td>\n",
       "      <td>4.85</td>\n",
       "      <td>2.39</td>\n",
       "      <td>0.94</td>\n",
       "      <td>63.16</td>\n",
       "      <td>37.78</td>\n",
       "      <td>1.95</td>\n",
       "      <td>0.81</td>\n",
       "      <td>2.95</td>\n",
       "      <td>0.93</td>\n",
       "      <td>3.75</td>\n",
       "      <td>1.07</td>\n",
       "      <td>8.08</td>\n",
       "      <td>1.95</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>1971.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.197295</td>\n",
       "      <td>4.027</td>\n",
       "      <td>1046421</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>DMRT3;LINC01230</td>\n",
       "      <td>dist=54689;dist=1722</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>LINC01230</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>LINC01230</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>1046421</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1.0</td>\n",
       "      <td>upstream</td>\n",
       "      <td>0.517</td>\n",
       "      <td>0.080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000173253</td>\n",
       "      <td>ENST00000382251</td>\n",
       "      <td>DMRT2</td>\n",
       "      <td>CCDS6444.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>796.0</td>\n",
       "      <td>2221.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.595</td>\n",
       "      <td>-0.197</td>\n",
       "      <td>-0.221</td>\n",
       "      <td>952.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.73</td>\n",
       "      <td>-3.59</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.95</td>\n",
       "      <td>2.37</td>\n",
       "      <td>8.60</td>\n",
       "      <td>2.21</td>\n",
       "      <td>5.45</td>\n",
       "      <td>1.94</td>\n",
       "      <td>3.29</td>\n",
       "      <td>1.3</td>\n",
       "      <td>14.74</td>\n",
       "      <td>4.85</td>\n",
       "      <td>2.39</td>\n",
       "      <td>0.94</td>\n",
       "      <td>63.16</td>\n",
       "      <td>37.78</td>\n",
       "      <td>1.95</td>\n",
       "      <td>0.81</td>\n",
       "      <td>2.95</td>\n",
       "      <td>0.93</td>\n",
       "      <td>3.75</td>\n",
       "      <td>1.07</td>\n",
       "      <td>8.08</td>\n",
       "      <td>1.95</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>1971.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.197295</td>\n",
       "      <td>4.027</td>\n",
       "      <td>1046421</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>DMRT3;LINC01230</td>\n",
       "      <td>dist=54689;dist=1722</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>LINC01230</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ncRNA_exonic</td>\n",
       "      <td>LINC01230</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57690</th>\n",
       "      <td>X</td>\n",
       "      <td>153182814</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>153182814</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>MAGEA1:NM_004988:exon3:c.A425A:p.E142E</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>MAGEA1:uc004fhf.3:exon3:c.A425A:p.E142E</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>MAGEA1:ENST00000356661.6:exon3:c.A425A:p.E142E</td>\n",
       "      <td>0.35</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0</td>\n",
       "      <td>B</td>\n",
       "      <td>0.003</td>\n",
       "      <td>B</td>\n",
       "      <td>0.147</td>\n",
       "      <td>N</td>\n",
       "      <td>1.000</td>\n",
       "      <td>N</td>\n",
       "      <td>0</td>\n",
       "      <td>N</td>\n",
       "      <td>3.44</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.962</td>\n",
       "      <td>T</td>\n",
       "      <td>0.007</td>\n",
       "      <td>T</td>\n",
       "      <td>0.112</td>\n",
       "      <td>-0.967</td>\n",
       "      <td>0.313</td>\n",
       "      <td>-2.09</td>\n",
       "      <td>-1.453</td>\n",
       "      <td>-0.012</td>\n",
       "      <td>0.067</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00480613091631</td>\n",
       "      <td>0.048</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57691</th>\n",
       "      <td>X</td>\n",
       "      <td>153182926</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>153182926</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>MAGEA1:NM_004988:exon3:c.T537G:p.Y179X</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>MAGEA1:uc004fhf.3:exon3:c.T537G:p.Y179X</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>MAGEA1:ENST00000356661.6:exon3:c.T537G:p.Y179X</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.500</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>-2.55</td>\n",
       "      <td>0.059</td>\n",
       "      <td>-4.507</td>\n",
       "      <td>0.003</td>\n",
       "      <td>-4.751</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.063</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.016</td>\n",
       "      <td>9.852</td>\n",
       "      <td>0.400</td>\n",
       "      <td>MAGE homology domain</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57692</th>\n",
       "      <td>X</td>\n",
       "      <td>153183077</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>153183077</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>MAGEA1:NM_004988:exon3:c.A688A:p.S230S</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>MAGEA1:uc004fhf.3:exon3:c.A688A:p.S230S</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>MAGEA1:ENST00000356661.6:exon3:c.A688A:p.S230S</td>\n",
       "      <td>0.07</td>\n",
       "      <td>T</td>\n",
       "      <td>0.996</td>\n",
       "      <td>D</td>\n",
       "      <td>0.905</td>\n",
       "      <td>P</td>\n",
       "      <td>0.006</td>\n",
       "      <td>N</td>\n",
       "      <td>1.000</td>\n",
       "      <td>N</td>\n",
       "      <td>1.63</td>\n",
       "      <td>L</td>\n",
       "      <td>3.43</td>\n",
       "      <td>T</td>\n",
       "      <td>-1.021</td>\n",
       "      <td>T</td>\n",
       "      <td>0.037</td>\n",
       "      <td>T</td>\n",
       "      <td>0.061</td>\n",
       "      <td>1.324</td>\n",
       "      <td>10.34</td>\n",
       "      <td>-2.55</td>\n",
       "      <td>-0.677</td>\n",
       "      <td>-0.478</td>\n",
       "      <td>4.633</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00127721042215</td>\n",
       "      <td>0.038</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57693</th>\n",
       "      <td>X</td>\n",
       "      <td>153183202</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>153183202</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>MAGEA1:NM_004988:exon3:c.C813T:p.L271L</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>MAGEA1:uc004fhf.3:exon3:c.C813T:p.L271L</td>\n",
       "      <td>exonic</td>\n",
       "      <td>MAGEA1</td>\n",
       "      <td>.</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>MAGEA1:ENST00000356661.6:exon3:c.C813T:p.L271L</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>57694 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos Ref Alt Type  Length             AnnoType  \\\n",
       "0          9    1046375   G   T  SNV     0.0           Intergenic   \n",
       "1          9    1046375   G   T  SNV     0.0  NonCodingTranscript   \n",
       "2          9    1046421   A   G  SNV     0.0  NonCodingTranscript   \n",
       "3          9    1046421   A   G  SNV     0.0           Intergenic   \n",
       "...      ...        ...  ..  ..  ...     ...                  ...   \n",
       "57690      X  153182814   T   A  NaN     NaN                  NaN   \n",
       "57691      X  153182926   A   G  NaN     NaN                  NaN   \n",
       "57692      X  153183077   T   A  NaN     NaN                  NaN   \n",
       "57693      X  153183202   G   T  NaN     NaN                  NaN   \n",
       "\n",
       "            Consequence  ConsScore       ConsDetail     GC    CpG  \\\n",
       "0              UPSTREAM        1.0         upstream  0.649  0.147   \n",
       "1      NONCODING_CHANGE        5.0  non_coding_exon  0.649  0.147   \n",
       "2      NONCODING_CHANGE        5.0  non_coding_exon  0.517  0.080   \n",
       "3              UPSTREAM        1.0         upstream  0.517  0.080   \n",
       "...                 ...        ...              ...    ...    ...   \n",
       "57690               NaN        NaN              NaN    NaN    NaN   \n",
       "57691               NaN        NaN              NaN    NaN    NaN   \n",
       "57692               NaN        NaN              NaN    NaN    NaN   \n",
       "57693               NaN        NaN              NaN    NaN    NaN   \n",
       "\n",
       "       motifECount motifEName  motifEHIPos  motifEScoreChng  oAA  nAA  \\\n",
       "0              NaN        NaN          NaN              NaN  NaN  NaN   \n",
       "1              NaN        NaN          NaN              NaN  NaN  NaN   \n",
       "2              NaN        NaN          NaN              NaN  NaN  NaN   \n",
       "3              NaN        NaN          NaN              NaN  NaN  NaN   \n",
       "...            ...        ...          ...              ...  ...  ...   \n",
       "57690          NaN        NaN          NaN              NaN  NaN  NaN   \n",
       "57691          NaN        NaN          NaN              NaN  NaN  NaN   \n",
       "57692          NaN        NaN          NaN              NaN  NaN  NaN   \n",
       "57693          NaN        NaN          NaN              NaN  NaN  NaN   \n",
       "\n",
       "                GeneID        FeatureID   GeneName        CCDS Intron Exon  \\\n",
       "0      ENSG00000173253  ENST00000382251      DMRT2  CCDS6444.1    NaN  NaN   \n",
       "1      ENSG00000281769  ENST00000625222  LINC01230         NaN    NaN  1/1   \n",
       "2      ENSG00000281769  ENST00000625222  LINC01230         NaN    NaN  1/1   \n",
       "3      ENSG00000173253  ENST00000382251      DMRT2  CCDS6444.1    NaN  NaN   \n",
       "...                ...              ...        ...         ...    ...  ...   \n",
       "57690              NaN              NaN        NaN         NaN    NaN  NaN   \n",
       "57691              NaN              NaN        NaN         NaN    NaN  NaN   \n",
       "57692              NaN              NaN        NaN         NaN    NaN  NaN   \n",
       "57693              NaN              NaN        NaN         NaN    NaN  NaN   \n",
       "\n",
       "       cDNApos  relcDNApos  CDSpos  relCDSpos  protPos  relProtPos Domain  \\\n",
       "0          NaN         NaN     NaN        NaN      NaN         NaN    NaN   \n",
       "1        751.0       0.249     NaN        NaN      NaN         NaN    NaN   \n",
       "2        797.0       0.264     NaN        NaN      NaN         NaN    NaN   \n",
       "3          NaN         NaN     NaN        NaN      NaN         NaN    NaN   \n",
       "...        ...         ...     ...        ...      ...         ...    ...   \n",
       "57690      NaN         NaN     NaN        NaN      NaN         NaN    NaN   \n",
       "57691      NaN         NaN     NaN        NaN      NaN         NaN    NaN   \n",
       "57692      NaN         NaN     NaN        NaN      NaN         NaN    NaN   \n",
       "57693      NaN         NaN     NaN        NaN      NaN         NaN    NaN   \n",
       "\n",
       "       Dst2Splice Dst2SplType  minDistTSS  minDistTSE SIFTcat  SIFTval  \\\n",
       "0             NaN         NaN       750.0      2267.0     NaN      NaN   \n",
       "1             NaN         NaN       750.0      2267.0     NaN      NaN   \n",
       "2             NaN         NaN       796.0      2221.0     NaN      NaN   \n",
       "3             NaN         NaN       796.0      2221.0     NaN      NaN   \n",
       "...           ...         ...         ...         ...     ...      ...   \n",
       "57690         NaN         NaN         NaN         NaN     NaN      NaN   \n",
       "57691         NaN         NaN         NaN         NaN     NaN      NaN   \n",
       "57692         NaN         NaN         NaN         NaN     NaN      NaN   \n",
       "57693         NaN         NaN         NaN         NaN     NaN      NaN   \n",
       "\n",
       "      PolyPhenCat  PolyPhenVal  priPhCons  mamPhCons  verPhCons  priPhyloP  \\\n",
       "0             NaN          NaN      0.002      0.001        0.0     -0.382   \n",
       "1             NaN          NaN      0.002      0.001        0.0     -0.382   \n",
       "2             NaN          NaN      0.006      0.000        0.0      0.595   \n",
       "3             NaN          NaN      0.006      0.000        0.0      0.595   \n",
       "...           ...          ...        ...        ...        ...        ...   \n",
       "57690         NaN          NaN        NaN        NaN        NaN        NaN   \n",
       "57691         NaN          NaN        NaN        NaN        NaN        NaN   \n",
       "57692         NaN          NaN        NaN        NaN        NaN        NaN   \n",
       "57693         NaN          NaN        NaN        NaN        NaN        NaN   \n",
       "\n",
       "       mamPhyloP  verPhyloP  bStatistic  targetScan  mirSVR-Score  mirSVR-E  \\\n",
       "0         -0.373     -0.324       952.0         NaN           NaN       NaN   \n",
       "1         -0.373     -0.324       952.0         NaN           NaN       NaN   \n",
       "2         -0.197     -0.221       952.0         NaN           NaN       NaN   \n",
       "3         -0.197     -0.221       952.0         NaN           NaN       NaN   \n",
       "...          ...        ...         ...         ...           ...       ...   \n",
       "57690        NaN        NaN         NaN         NaN           NaN       NaN   \n",
       "57691        NaN        NaN         NaN         NaN           NaN       NaN   \n",
       "57692        NaN        NaN         NaN         NaN           NaN       NaN   \n",
       "57693        NaN        NaN         NaN         NaN           NaN       NaN   \n",
       "\n",
       "       mirSVR-Aln  cHmm_E1  cHmm_E2  cHmm_E3  cHmm_E4  cHmm_E5  cHmm_E6  \\\n",
       "0             NaN      1.0      0.0      1.0      0.0      2.0      0.0   \n",
       "1             NaN      1.0      0.0      1.0      0.0      2.0      0.0   \n",
       "2             NaN      1.0      0.0      1.0      0.0      2.0      0.0   \n",
       "3             NaN      1.0      0.0      1.0      0.0      2.0      0.0   \n",
       "...           ...      ...      ...      ...      ...      ...      ...   \n",
       "57690         NaN      NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "57691         NaN      NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "57692         NaN      NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "57693         NaN      NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "\n",
       "       cHmm_E7  cHmm_E8  cHmm_E9  cHmm_E10  cHmm_E11  cHmm_E12  cHmm_E13  \\\n",
       "0          0.0      0.0      0.0       0.0       0.0       0.0       0.0   \n",
       "1          0.0      0.0      0.0       0.0       0.0       0.0       0.0   \n",
       "2          0.0      0.0      0.0       0.0       0.0       0.0       0.0   \n",
       "3          0.0      0.0      0.0       0.0       0.0       0.0       0.0   \n",
       "...        ...      ...      ...       ...       ...       ...       ...   \n",
       "57690      NaN      NaN      NaN       NaN       NaN       NaN       NaN   \n",
       "57691      NaN      NaN      NaN       NaN       NaN       NaN       NaN   \n",
       "57692      NaN      NaN      NaN       NaN       NaN       NaN       NaN   \n",
       "57693      NaN      NaN      NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "       cHmm_E14  cHmm_E15  cHmm_E16  cHmm_E17  cHmm_E18  cHmm_E19  cHmm_E20  \\\n",
       "0           0.0       6.0       0.0       0.0       0.0       0.0       9.0   \n",
       "1           0.0       6.0       0.0       0.0       0.0       0.0       9.0   \n",
       "2           0.0       6.0       0.0       0.0       0.0       0.0       7.0   \n",
       "3           0.0       6.0       0.0       0.0       0.0       0.0       7.0   \n",
       "...         ...       ...       ...       ...       ...       ...       ...   \n",
       "57690       NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "57691       NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "57692       NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "57693       NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "       cHmm_E21  cHmm_E22  cHmm_E23  cHmm_E24  cHmm_E25  GerpRS  GerpRSpval  \\\n",
       "0           8.0      17.0       4.0       0.0       0.0     NaN         NaN   \n",
       "1           8.0      17.0       4.0       0.0       0.0     NaN         NaN   \n",
       "2           8.0      20.0       3.0       0.0       0.0     NaN         NaN   \n",
       "3           8.0      20.0       3.0       0.0       0.0     NaN         NaN   \n",
       "...         ...       ...       ...       ...       ...     ...         ...   \n",
       "57690       NaN       NaN       NaN       NaN       NaN     NaN         NaN   \n",
       "57691       NaN       NaN       NaN       NaN       NaN     NaN         NaN   \n",
       "57692       NaN       NaN       NaN       NaN       NaN     NaN         NaN   \n",
       "57693       NaN       NaN       NaN       NaN       NaN     NaN         NaN   \n",
       "\n",
       "       GerpN  GerpS  tOverlapMotifs  motifDist  EncodeH3K4me1-sum  \\\n",
       "0       1.61  -3.21             NaN        NaN               9.28   \n",
       "1       1.61  -3.21             NaN        NaN               9.28   \n",
       "2       4.73  -3.59             NaN        NaN               6.95   \n",
       "3       4.73  -3.59             NaN        NaN               6.95   \n",
       "...      ...    ...             ...        ...                ...   \n",
       "57690    NaN    NaN             NaN        NaN                NaN   \n",
       "57691    NaN    NaN             NaN        NaN                NaN   \n",
       "57692    NaN    NaN             NaN        NaN                NaN   \n",
       "57693    NaN    NaN             NaN        NaN                NaN   \n",
       "\n",
       "       EncodeH3K4me1-max  EncodeH3K4me2-sum  EncodeH3K4me2-max  \\\n",
       "0                   2.72               8.93               2.22   \n",
       "1                   2.72               8.93               2.22   \n",
       "2                   2.37               8.60               2.21   \n",
       "3                   2.37               8.60               2.21   \n",
       "...                  ...                ...                ...   \n",
       "57690                NaN                NaN                NaN   \n",
       "57691                NaN                NaN                NaN   \n",
       "57692                NaN                NaN                NaN   \n",
       "57693                NaN                NaN                NaN   \n",
       "\n",
       "       EncodeH3K4me3-sum  EncodeH3K4me3-max  EncodeH3K9ac-sum  \\\n",
       "0                   4.07               1.13              2.37   \n",
       "1                   4.07               1.13              2.37   \n",
       "2                   5.45               1.94              3.29   \n",
       "3                   5.45               1.94              3.29   \n",
       "...                  ...                ...               ...   \n",
       "57690                NaN                NaN               NaN   \n",
       "57691                NaN                NaN               NaN   \n",
       "57692                NaN                NaN               NaN   \n",
       "57693                NaN                NaN               NaN   \n",
       "\n",
       "       EncodeH3K9ac-max  EncodeH3K9me3-sum  EncodeH3K9me3-max  \\\n",
       "0                   1.3              14.40               3.64   \n",
       "1                   1.3              14.40               3.64   \n",
       "2                   1.3              14.74               4.85   \n",
       "3                   1.3              14.74               4.85   \n",
       "...                 ...                ...                ...   \n",
       "57690               NaN                NaN                NaN   \n",
       "57691               NaN                NaN                NaN   \n",
       "57692               NaN                NaN                NaN   \n",
       "57693               NaN                NaN                NaN   \n",
       "\n",
       "       EncodeH3K27ac-sum  EncodeH3K27ac-max  EncodeH3K27me3-sum  \\\n",
       "0                   3.24               1.20               73.12   \n",
       "1                   3.24               1.20               73.12   \n",
       "2                   2.39               0.94               63.16   \n",
       "3                   2.39               0.94               63.16   \n",
       "...                  ...                ...                 ...   \n",
       "57690                NaN                NaN                 NaN   \n",
       "57691                NaN                NaN                 NaN   \n",
       "57692                NaN                NaN                 NaN   \n",
       "57693                NaN                NaN                 NaN   \n",
       "\n",
       "       EncodeH3K27me3-max  EncodeH3K36me3-sum  EncodeH3K36me3-max  \\\n",
       "0                   39.14                4.04                1.18   \n",
       "1                   39.14                4.04                1.18   \n",
       "2                   37.78                1.95                0.81   \n",
       "3                   37.78                1.95                0.81   \n",
       "...                   ...                 ...                 ...   \n",
       "57690                 NaN                 NaN                 NaN   \n",
       "57691                 NaN                 NaN                 NaN   \n",
       "57692                 NaN                 NaN                 NaN   \n",
       "57693                 NaN                 NaN                 NaN   \n",
       "\n",
       "       EncodeH3K79me2-sum  EncodeH3K79me2-max  EncodeH4K20me1-sum  \\\n",
       "0                    3.34                0.93                4.55   \n",
       "1                    3.34                0.93                4.55   \n",
       "2                    2.95                0.93                3.75   \n",
       "3                    2.95                0.93                3.75   \n",
       "...                   ...                 ...                 ...   \n",
       "57690                 NaN                 NaN                 NaN   \n",
       "57691                 NaN                 NaN                 NaN   \n",
       "57692                 NaN                 NaN                 NaN   \n",
       "57693                 NaN                 NaN                 NaN   \n",
       "\n",
       "       EncodeH4K20me1-max  EncodeH2AFZ-sum  EncodeH2AFZ-max  EncodeDNase-sum  \\\n",
       "0                    1.07             9.45             3.27             0.35   \n",
       "1                    1.07             9.45             3.27             0.35   \n",
       "2                    1.07             8.08             1.95             0.35   \n",
       "3                    1.07             8.08             1.95             0.35   \n",
       "...                   ...              ...              ...              ...   \n",
       "57690                 NaN              NaN              NaN              NaN   \n",
       "57691                 NaN              NaN              NaN              NaN   \n",
       "57692                 NaN              NaN              NaN              NaN   \n",
       "57693                 NaN              NaN              NaN              NaN   \n",
       "\n",
       "       EncodeDNase-max  EncodetotalRNA-sum  EncodetotalRNA-max  Grantham  \\\n",
       "0                 0.17                 NaN                 NaN       NaN   \n",
       "1                 0.17                 NaN                 NaN       NaN   \n",
       "2                 0.19                 NaN                 NaN       NaN   \n",
       "3                 0.19                 NaN                 NaN       NaN   \n",
       "...                ...                 ...                 ...       ...   \n",
       "57690              NaN                 NaN                 NaN       NaN   \n",
       "57691              NaN                 NaN                 NaN       NaN   \n",
       "57692              NaN                 NaN                 NaN       NaN   \n",
       "57693              NaN                 NaN                 NaN       NaN   \n",
       "\n",
       "       Dist2Mutation  Freq100bp  Rare100bp  Sngl100bp  Freq1000bp  Rare1000bp  \\\n",
       "0                3.0        0.0        2.0       21.0         2.0        13.0   \n",
       "1                3.0        0.0        2.0       21.0         2.0        13.0   \n",
       "2                5.0        0.0        0.0       26.0         2.0        13.0   \n",
       "3                5.0        0.0        0.0       26.0         2.0        13.0   \n",
       "...              ...        ...        ...        ...         ...         ...   \n",
       "57690            NaN        NaN        NaN        NaN         NaN         NaN   \n",
       "57691            NaN        NaN        NaN        NaN         NaN         NaN   \n",
       "57692            NaN        NaN        NaN        NaN         NaN         NaN   \n",
       "57693            NaN        NaN        NaN        NaN         NaN         NaN   \n",
       "\n",
       "       Sngl1000bp  Freq10000bp  Rare10000bp  Sngl10000bp  \\\n",
       "0           213.0         40.0         99.0       1972.0   \n",
       "1           213.0         40.0         99.0       1972.0   \n",
       "2           214.0         40.0         99.0       1971.0   \n",
       "3           214.0         40.0         99.0       1971.0   \n",
       "...           ...          ...          ...          ...   \n",
       "57690         NaN          NaN          NaN          NaN   \n",
       "57691         NaN          NaN          NaN          NaN   \n",
       "57692         NaN          NaN          NaN          NaN   \n",
       "57693         NaN          NaN          NaN          NaN   \n",
       "\n",
       "      EnsembleRegulatoryFeature  dbscSNV-ada_score dbscSNV-rf_score  \\\n",
       "0                           NaN                NaN              NaN   \n",
       "1                           NaN                NaN              NaN   \n",
       "2                           NaN                NaN              NaN   \n",
       "3                           NaN                NaN              NaN   \n",
       "...                         ...                ...              ...   \n",
       "57690                       NaN                NaN              NaN   \n",
       "57691                       NaN                NaN              NaN   \n",
       "57692                       NaN                NaN              NaN   \n",
       "57693                       NaN                NaN              NaN   \n",
       "\n",
       "       RemapOverlapTF  RemapOverlapCL  RawScore  PHRED        End  \\\n",
       "0                 7.0            16.0 -0.106065  0.969    1046375   \n",
       "1                 7.0            16.0 -0.106065  0.969    1046375   \n",
       "2                 3.0            12.0  0.197295  4.027    1046421   \n",
       "3                 3.0            12.0  0.197295  4.027    1046421   \n",
       "...               ...             ...       ...    ...        ...   \n",
       "57690             NaN             NaN       NaN    NaN  153182814   \n",
       "57691             NaN             NaN       NaN    NaN  153182926   \n",
       "57692             NaN             NaN       NaN    NaN  153183077   \n",
       "57693             NaN             NaN       NaN    NaN  153183202   \n",
       "\n",
       "      Func.refGene     Gene.refGene    GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0       intergenic  DMRT3;LINC01230  dist=54643;dist=1768                  .   \n",
       "1       intergenic  DMRT3;LINC01230  dist=54643;dist=1768                  .   \n",
       "2       intergenic  DMRT3;LINC01230  dist=54689;dist=1722                  .   \n",
       "3       intergenic  DMRT3;LINC01230  dist=54689;dist=1722                  .   \n",
       "...            ...              ...                   ...                ...   \n",
       "57690       exonic           MAGEA1                     .     synonymous SNV   \n",
       "57691       exonic           MAGEA1                     .           stopgain   \n",
       "57692       exonic           MAGEA1                     .     synonymous SNV   \n",
       "57693       exonic           MAGEA1                     .     synonymous SNV   \n",
       "\n",
       "                             AAChange.refGene Func.knownGene Gene.knownGene  \\\n",
       "0                                           .   ncRNA_exonic      LINC01230   \n",
       "1                                           .   ncRNA_exonic      LINC01230   \n",
       "2                                           .   ncRNA_exonic      LINC01230   \n",
       "3                                           .   ncRNA_exonic      LINC01230   \n",
       "...                                       ...            ...            ...   \n",
       "57690  MAGEA1:NM_004988:exon3:c.A425A:p.E142E         exonic         MAGEA1   \n",
       "57691  MAGEA1:NM_004988:exon3:c.T537G:p.Y179X         exonic         MAGEA1   \n",
       "57692  MAGEA1:NM_004988:exon3:c.A688A:p.S230S         exonic         MAGEA1   \n",
       "57693  MAGEA1:NM_004988:exon3:c.C813T:p.L271L         exonic         MAGEA1   \n",
       "\n",
       "      GeneDetail.knownGene ExonicFunc.knownGene  \\\n",
       "0                        .                    .   \n",
       "1                        .                    .   \n",
       "2                        .                    .   \n",
       "3                        .                    .   \n",
       "...                    ...                  ...   \n",
       "57690                    .       synonymous SNV   \n",
       "57691                    .             stopgain   \n",
       "57692                    .       synonymous SNV   \n",
       "57693                    .       synonymous SNV   \n",
       "\n",
       "                            AAChange.knownGene  Func.ensGene Gene.ensGene  \\\n",
       "0                                            .  ncRNA_exonic    LINC01230   \n",
       "1                                            .  ncRNA_exonic    LINC01230   \n",
       "2                                            .  ncRNA_exonic    LINC01230   \n",
       "3                                            .  ncRNA_exonic    LINC01230   \n",
       "...                                        ...           ...          ...   \n",
       "57690  MAGEA1:uc004fhf.3:exon3:c.A425A:p.E142E        exonic       MAGEA1   \n",
       "57691  MAGEA1:uc004fhf.3:exon3:c.T537G:p.Y179X        exonic       MAGEA1   \n",
       "57692  MAGEA1:uc004fhf.3:exon3:c.A688A:p.S230S        exonic       MAGEA1   \n",
       "57693  MAGEA1:uc004fhf.3:exon3:c.C813T:p.L271L        exonic       MAGEA1   \n",
       "\n",
       "      GeneDetail.ensGene ExonicFunc.ensGene  \\\n",
       "0                      .                  .   \n",
       "1                      .                  .   \n",
       "2                      .                  .   \n",
       "3                      .                  .   \n",
       "...                  ...                ...   \n",
       "57690                  .     synonymous SNV   \n",
       "57691                  .           stopgain   \n",
       "57692                  .     synonymous SNV   \n",
       "57693                  .     synonymous SNV   \n",
       "\n",
       "                                     AAChange.ensGene SIFT_score SIFT_pred  \\\n",
       "0                                                   .          .         .   \n",
       "1                                                   .          .         .   \n",
       "2                                                   .          .         .   \n",
       "3                                                   .          .         .   \n",
       "...                                               ...        ...       ...   \n",
       "57690  MAGEA1:ENST00000356661.6:exon3:c.A425A:p.E142E       0.35         T   \n",
       "57691  MAGEA1:ENST00000356661.6:exon3:c.T537G:p.Y179X          .         .   \n",
       "57692  MAGEA1:ENST00000356661.6:exon3:c.A688A:p.S230S       0.07         T   \n",
       "57693  MAGEA1:ENST00000356661.6:exon3:c.C813T:p.L271L          .         .   \n",
       "\n",
       "      Polyphen2_HDIV_score Polyphen2_HDIV_pred Polyphen2_HVAR_score  \\\n",
       "0                        .                   .                    .   \n",
       "1                        .                   .                    .   \n",
       "2                        .                   .                    .   \n",
       "3                        .                   .                    .   \n",
       "...                    ...                 ...                  ...   \n",
       "57690                  0.0                   B                0.003   \n",
       "57691                    .                   .                    .   \n",
       "57692                0.996                   D                0.905   \n",
       "57693                    .                   .                    .   \n",
       "\n",
       "      Polyphen2_HVAR_pred LRT_score LRT_pred MutationTaster_score  \\\n",
       "0                       .         .        .                    .   \n",
       "1                       .         .        .                    .   \n",
       "2                       .         .        .                    .   \n",
       "3                       .         .        .                    .   \n",
       "...                   ...       ...      ...                  ...   \n",
       "57690                   B     0.147        N                1.000   \n",
       "57691                   .         .        .                    .   \n",
       "57692                   P     0.006        N                1.000   \n",
       "57693                   .         .        .                    .   \n",
       "\n",
       "      MutationTaster_pred MutationAssessor_score MutationAssessor_pred  \\\n",
       "0                       .                      .                     .   \n",
       "1                       .                      .                     .   \n",
       "2                       .                      .                     .   \n",
       "3                       .                      .                     .   \n",
       "...                   ...                    ...                   ...   \n",
       "57690                   N                      0                     N   \n",
       "57691                   .                      .                     .   \n",
       "57692                   N                   1.63                     L   \n",
       "57693                   .                      .                     .   \n",
       "\n",
       "      FATHMM_score FATHMM_pred RadialSVM_score RadialSVM_pred LR_score  \\\n",
       "0                .           .               .              .        .   \n",
       "1                .           .               .              .        .   \n",
       "2                .           .               .              .        .   \n",
       "3                .           .               .              .        .   \n",
       "...            ...         ...             ...            ...      ...   \n",
       "57690         3.44           T          -0.962              T    0.007   \n",
       "57691            .           .               .              .        .   \n",
       "57692         3.43           T          -1.021              T    0.037   \n",
       "57693            .           .               .              .        .   \n",
       "\n",
       "      LR_pred VEST3_score CADD_raw CADD_phred GERP++_RS phyloP46way_placental  \\\n",
       "0           .           .        .          .         .                     .   \n",
       "1           .           .        .          .         .                     .   \n",
       "2           .           .        .          .         .                     .   \n",
       "3           .           .        .          .         .                     .   \n",
       "...       ...         ...      ...        ...       ...                   ...   \n",
       "57690       T       0.112   -0.967      0.313     -2.09                -1.453   \n",
       "57691       .           .        .          .         .                     .   \n",
       "57692       T       0.061    1.324      10.34     -2.55                -0.677   \n",
       "57693       .           .        .          .         .                     .   \n",
       "\n",
       "      phyloP100way_vertebrate SiPhy_29way_logOdds Interpro_domain  \\\n",
       "0                           .                   .               .   \n",
       "1                           .                   .               .   \n",
       "2                           .                   .               .   \n",
       "3                           .                   .               .   \n",
       "...                       ...                 ...             ...   \n",
       "57690                  -0.012               0.067               .   \n",
       "57691                       .                   .               .   \n",
       "57692                  -0.478               4.633               .   \n",
       "57693                       .                   .               .   \n",
       "\n",
       "      SIFT_score.1 SIFT_converted_rankscore SIFT_pred.1  \\\n",
       "0                .                        .           .   \n",
       "1                .                        .           .   \n",
       "2                .                        .           .   \n",
       "3                .                        .           .   \n",
       "...            ...                      ...         ...   \n",
       "57690            .                        .           .   \n",
       "57691            .                        .           .   \n",
       "57692            .                        .           .   \n",
       "57693            .                        .           .   \n",
       "\n",
       "      Polyphen2_HDIV_score.1 Polyphen2_HDIV_rankscore Polyphen2_HDIV_pred.1  \\\n",
       "0                          .                        .                     .   \n",
       "1                          .                        .                     .   \n",
       "2                          .                        .                     .   \n",
       "3                          .                        .                     .   \n",
       "...                      ...                      ...                   ...   \n",
       "57690                      .                        .                     .   \n",
       "57691                      .                        .                     .   \n",
       "57692                      .                        .                     .   \n",
       "57693                      .                        .                     .   \n",
       "\n",
       "      Polyphen2_HVAR_score.1 Polyphen2_HVAR_rankscore Polyphen2_HVAR_pred.1  \\\n",
       "0                          .                        .                     .   \n",
       "1                          .                        .                     .   \n",
       "2                          .                        .                     .   \n",
       "3                          .                        .                     .   \n",
       "...                      ...                      ...                   ...   \n",
       "57690                      .                        .                     .   \n",
       "57691                      .                        .                     .   \n",
       "57692                      .                        .                     .   \n",
       "57693                      .                        .                     .   \n",
       "\n",
       "      LRT_score.1 LRT_converted_rankscore LRT_pred.1 MutationTaster_score.1  \\\n",
       "0               .                       .          .                      .   \n",
       "1               .                       .          .                      .   \n",
       "2               .                       .          .                      .   \n",
       "3               .                       .          .                      .   \n",
       "...           ...                     ...        ...                    ...   \n",
       "57690           .                       .          .                      .   \n",
       "57691           .                       .          .                      .   \n",
       "57692           .                       .          .                      .   \n",
       "57693           .                       .          .                      .   \n",
       "\n",
       "      MutationTaster_converted_rankscore MutationTaster_pred.1  \\\n",
       "0                                      .                     .   \n",
       "1                                      .                     .   \n",
       "2                                      .                     .   \n",
       "3                                      .                     .   \n",
       "...                                  ...                   ...   \n",
       "57690                                  .                     .   \n",
       "57691                                  .                     .   \n",
       "57692                                  .                     .   \n",
       "57693                                  .                     .   \n",
       "\n",
       "      MutationAssessor_score.1 MutationAssessor_score_rankscore  \\\n",
       "0                            .                                .   \n",
       "1                            .                                .   \n",
       "2                            .                                .   \n",
       "3                            .                                .   \n",
       "...                        ...                              ...   \n",
       "57690                        .                                .   \n",
       "57691                        .                                .   \n",
       "57692                        .                                .   \n",
       "57693                        .                                .   \n",
       "\n",
       "      MutationAssessor_pred.1 FATHMM_score.1 FATHMM_converted_rankscore  \\\n",
       "0                           .              .                          .   \n",
       "1                           .              .                          .   \n",
       "2                           .              .                          .   \n",
       "3                           .              .                          .   \n",
       "...                       ...            ...                        ...   \n",
       "57690                       .              .                          .   \n",
       "57691                       .              .                          .   \n",
       "57692                       .              .                          .   \n",
       "57693                       .              .                          .   \n",
       "\n",
       "      FATHMM_pred.1 PROVEAN_score PROVEAN_converted_rankscore PROVEAN_pred  \\\n",
       "0                 .             .                           .            .   \n",
       "1                 .             .                           .            .   \n",
       "2                 .             .                           .            .   \n",
       "3                 .             .                           .            .   \n",
       "...             ...           ...                         ...          ...   \n",
       "57690             .             .                           .            .   \n",
       "57691             .             .                           .            .   \n",
       "57692             .             .                           .            .   \n",
       "57693             .             .                           .            .   \n",
       "\n",
       "      VEST3_score.1 VEST3_rankscore MetaSVM_score MetaSVM_rankscore  \\\n",
       "0                 .               .             .                 .   \n",
       "1                 .               .             .                 .   \n",
       "2                 .               .             .                 .   \n",
       "3                 .               .             .                 .   \n",
       "...             ...             ...           ...               ...   \n",
       "57690             .               .             .                 .   \n",
       "57691             .               .             .                 .   \n",
       "57692             .               .             .                 .   \n",
       "57693             .               .             .                 .   \n",
       "\n",
       "      MetaSVM_pred MetaLR_score MetaLR_rankscore MetaLR_pred M-CAP_score  \\\n",
       "0                .            .                .           .           .   \n",
       "1                .            .                .           .           .   \n",
       "2                .            .                .           .           .   \n",
       "3                .            .                .           .           .   \n",
       "...            ...          ...              ...         ...         ...   \n",
       "57690            .            .                .           .           .   \n",
       "57691            .            .                .           .           .   \n",
       "57692            .            .                .           .           .   \n",
       "57693            .            .                .           .           .   \n",
       "\n",
       "      M-CAP_rankscore M-CAP_pred CADD_raw.1 CADD_raw_rankscore CADD_phred.1  \\\n",
       "0                   .          .          .                  .            .   \n",
       "1                   .          .          .                  .            .   \n",
       "2                   .          .          .                  .            .   \n",
       "3                   .          .          .                  .            .   \n",
       "...               ...        ...        ...                ...          ...   \n",
       "57690               .          .          .                  .            .   \n",
       "57691               .          .          .                  .            .   \n",
       "57692               .          .          .                  .            .   \n",
       "57693               .          .          .                  .            .   \n",
       "\n",
       "      DANN_score DANN_rankscore fathmm-MKL_coding_score  \\\n",
       "0              .              .                       .   \n",
       "1              .              .                       .   \n",
       "2              .              .                       .   \n",
       "3              .              .                       .   \n",
       "...          ...            ...                     ...   \n",
       "57690          .              .                       .   \n",
       "57691          .              .                       .   \n",
       "57692          .              .                       .   \n",
       "57693          .              .                       .   \n",
       "\n",
       "      fathmm-MKL_coding_rankscore fathmm-MKL_coding_pred  \\\n",
       "0                               .                      .   \n",
       "1                               .                      .   \n",
       "2                               .                      .   \n",
       "3                               .                      .   \n",
       "...                           ...                    ...   \n",
       "57690                           .                      .   \n",
       "57691                           .                      .   \n",
       "57692                           .                      .   \n",
       "57693                           .                      .   \n",
       "\n",
       "      Eigen_coding_or_noncoding Eigen-raw Eigen-PC-raw GenoCanyon_score  \\\n",
       "0                             .         .            .                .   \n",
       "1                             .         .            .                .   \n",
       "2                             .         .            .                .   \n",
       "3                             .         .            .                .   \n",
       "...                         ...       ...          ...              ...   \n",
       "57690                         .         .            .                .   \n",
       "57691                         .         .            .            1.000   \n",
       "57692                         .         .            .                .   \n",
       "57693                         .         .            .                .   \n",
       "\n",
       "      GenoCanyon_score_rankscore integrated_fitCons_score  \\\n",
       "0                              .                        .   \n",
       "1                              .                        .   \n",
       "2                              .                        .   \n",
       "3                              .                        .   \n",
       "...                          ...                      ...   \n",
       "57690                          .                        .   \n",
       "57691                      0.500                        .   \n",
       "57692                          .                        .   \n",
       "57693                          .                        .   \n",
       "\n",
       "      integrated_fitCons_score_rankscore integrated_confidence_value  \\\n",
       "0                                      .                           .   \n",
       "1                                      .                           .   \n",
       "2                                      .                           .   \n",
       "3                                      .                           .   \n",
       "...                                  ...                         ...   \n",
       "57690                                  .                           .   \n",
       "57691                                  .                           .   \n",
       "57692                                  .                           .   \n",
       "57693                                  .                           .   \n",
       "\n",
       "      GERP++_RS.1 GERP++_RS_rankscore phyloP100way_vertebrate.1  \\\n",
       "0               .                   .                         .   \n",
       "1               .                   .                         .   \n",
       "2               .                   .                         .   \n",
       "3               .                   .                         .   \n",
       "...           ...                 ...                       ...   \n",
       "57690           .                   .                         .   \n",
       "57691       -2.55               0.059                    -4.507   \n",
       "57692           .                   .                         .   \n",
       "57693           .                   .                         .   \n",
       "\n",
       "      phyloP100way_vertebrate_rankscore phyloP20way_mammalian  \\\n",
       "0                                     .                     .   \n",
       "1                                     .                     .   \n",
       "2                                     .                     .   \n",
       "3                                     .                     .   \n",
       "...                                 ...                   ...   \n",
       "57690                                 .                     .   \n",
       "57691                             0.003                -4.751   \n",
       "57692                                 .                     .   \n",
       "57693                                 .                     .   \n",
       "\n",
       "      phyloP20way_mammalian_rankscore phastCons100way_vertebrate  \\\n",
       "0                                   .                          .   \n",
       "1                                   .                          .   \n",
       "2                                   .                          .   \n",
       "3                                   .                          .   \n",
       "...                               ...                        ...   \n",
       "57690                               .                          .   \n",
       "57691                           0.000                      0.000   \n",
       "57692                               .                          .   \n",
       "57693                               .                          .   \n",
       "\n",
       "      phastCons100way_vertebrate_rankscore phastCons20way_mammalian  \\\n",
       "0                                        .                        .   \n",
       "1                                        .                        .   \n",
       "2                                        .                        .   \n",
       "3                                        .                        .   \n",
       "...                                    ...                      ...   \n",
       "57690                                    .                        .   \n",
       "57691                                0.063                    0.000   \n",
       "57692                                    .                        .   \n",
       "57693                                    .                        .   \n",
       "\n",
       "      phastCons20way_mammalian_rankscore SiPhy_29way_logOdds.1  \\\n",
       "0                                      .                     .   \n",
       "1                                      .                     .   \n",
       "2                                      .                     .   \n",
       "3                                      .                     .   \n",
       "...                                  ...                   ...   \n",
       "57690                                  .                     .   \n",
       "57691                              0.016                 9.852   \n",
       "57692                                  .                     .   \n",
       "57693                                  .                     .   \n",
       "\n",
       "      SiPhy_29way_logOdds_rankscore     Interpro_domain.1 GTEx_V6_gene  \\\n",
       "0                                 .                     .            .   \n",
       "1                                 .                     .            .   \n",
       "2                                 .                     .            .   \n",
       "3                                 .                     .            .   \n",
       "...                             ...                   ...          ...   \n",
       "57690                             .                     .            .   \n",
       "57691                         0.400  MAGE homology domain            .   \n",
       "57692                             .                     .            .   \n",
       "57693                             .                     .            .   \n",
       "\n",
       "      GTEx_V6_tissue dbscSNV_ADA_SCORE dbscSNV_RF_SCORE  \\\n",
       "0                  .                 .                .   \n",
       "1                  .                 .                .   \n",
       "2                  .                 .                .   \n",
       "3                  .                 .                .   \n",
       "...              ...               ...              ...   \n",
       "57690              .                 .                .   \n",
       "57691              .                 .                .   \n",
       "57692              .                 .                .   \n",
       "57693              .                 .                .   \n",
       "\n",
       "           InterVar_automated PVS1 PS1 PS2 PS3 PS4 PM1 PM2 PM3 PM4 PM5 PM6  \\\n",
       "0                           .    .   .   .   .   .   .   .   .   .   .   .   \n",
       "1                           .    .   .   .   .   .   .   .   .   .   .   .   \n",
       "2                           .    .   .   .   .   .   .   .   .   .   .   .   \n",
       "3                           .    .   .   .   .   .   .   .   .   .   .   .   \n",
       "...                       ...  ...  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..   \n",
       "57690                       .    .   .   .   .   .   .   .   .   .   .   .   \n",
       "57691  Uncertain significance    0   0   0   0   0   0   1   0   0   0   0   \n",
       "57692                       .    .   .   .   .   .   .   .   .   .   .   .   \n",
       "57693           Likely benign    0   0   0   0   0   0   1   0   0   0   0   \n",
       "\n",
       "      PP1 PP2 PP3 PP4 PP5 BA1 BS1 BS2 BS3 BS4 BP1 BP2 BP3 BP4 BP5 BP6 BP7  \\\n",
       "0       .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "1       .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "2       .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "3       .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "...    ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..   \n",
       "57690   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "57691   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   \n",
       "57692   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   .   \n",
       "57693   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   1   \n",
       "\n",
       "      cosmic70              MCAP  REVEL regsnp_fpr regsnp_disease  \\\n",
       "0            .                 .      .          .              .   \n",
       "1            .                 .      .          .              .   \n",
       "2            .                 .      .          .              .   \n",
       "3            .                 .      .          .              .   \n",
       "...        ...               ...    ...        ...            ...   \n",
       "57690        .  0.00480613091631  0.048          .              .   \n",
       "57691        .                 .      .          .              .   \n",
       "57692        .  0.00127721042215  0.038          .              .   \n",
       "57693        .                 .      .          .              .   \n",
       "\n",
       "      regsnp_splicing_site  \n",
       "0                        .  \n",
       "1                        .  \n",
       "2                        .  \n",
       "3                        .  \n",
       "...                    ...  \n",
       "57690                    .  \n",
       "57691                    .  \n",
       "57692                    .  \n",
       "57693                    .  \n",
       "\n",
       "[57694 rows x 270 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_merged_annovar = pd.merge(human_derived_coding_annotated_with_cadd, human_derived_coding_annotated_with_annovar,\n",
    "                                         how = 'right', on = ['#Chrom', 'Pos', 'Ref', 'Alt'])\n",
    "pd.set_option('display.max_columns', 270)\n",
    "pd.options.display.max_rows = 8\n",
    "human_derived_coding_merged_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "When looking at all the columns, we see that for the first several variants, almost all the ANNOVAR annotations are NaN values. Therefore, we know that CADD annotations give us more values and therefore we should use the CADD dataframe to merge by since we have a greater output of variants."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below we have dropped any duplicate rows as to not have our model train on multiple variants that are exactly the same:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEName</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>motifEScoreChng</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>FeatureID</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>Intron</th>\n",
       "      <th>Exon</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>relcDNApos</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>relCDSpos</th>\n",
       "      <th>protPos</th>\n",
       "      <th>relProtPos</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Dst2Splice</th>\n",
       "      <th>Dst2SplType</th>\n",
       "      <th>minDistTSS</th>\n",
       "      <th>minDistTSE</th>\n",
       "      <th>SIFTcat</th>\n",
       "      <th>SIFTval</th>\n",
       "      <th>PolyPhenCat</th>\n",
       "      <th>PolyPhenVal</th>\n",
       "      <th>priPhCons</th>\n",
       "      <th>mamPhCons</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>priPhyloP</th>\n",
       "      <th>mamPhyloP</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>bStatistic</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>mirSVR-Score</th>\n",
       "      <th>mirSVR-E</th>\n",
       "      <th>mirSVR-Aln</th>\n",
       "      <th>cHmm_E1</th>\n",
       "      <th>cHmm_E2</th>\n",
       "      <th>cHmm_E3</th>\n",
       "      <th>cHmm_E4</th>\n",
       "      <th>cHmm_E5</th>\n",
       "      <th>cHmm_E6</th>\n",
       "      <th>cHmm_E7</th>\n",
       "      <th>cHmm_E8</th>\n",
       "      <th>cHmm_E9</th>\n",
       "      <th>cHmm_E10</th>\n",
       "      <th>cHmm_E11</th>\n",
       "      <th>cHmm_E12</th>\n",
       "      <th>cHmm_E13</th>\n",
       "      <th>cHmm_E14</th>\n",
       "      <th>cHmm_E15</th>\n",
       "      <th>cHmm_E16</th>\n",
       "      <th>cHmm_E17</th>\n",
       "      <th>cHmm_E18</th>\n",
       "      <th>cHmm_E19</th>\n",
       "      <th>cHmm_E20</th>\n",
       "      <th>cHmm_E21</th>\n",
       "      <th>cHmm_E22</th>\n",
       "      <th>cHmm_E23</th>\n",
       "      <th>cHmm_E24</th>\n",
       "      <th>cHmm_E25</th>\n",
       "      <th>GerpRS</th>\n",
       "      <th>GerpRSpval</th>\n",
       "      <th>GerpN</th>\n",
       "      <th>GerpS</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>motifDist</th>\n",
       "      <th>EncodeH3K4me1-sum</th>\n",
       "      <th>EncodeH3K4me1-max</th>\n",
       "      <th>EncodeH3K4me2-sum</th>\n",
       "      <th>EncodeH3K4me2-max</th>\n",
       "      <th>EncodeH3K4me3-sum</th>\n",
       "      <th>EncodeH3K4me3-max</th>\n",
       "      <th>EncodeH3K9ac-sum</th>\n",
       "      <th>EncodeH3K9ac-max</th>\n",
       "      <th>EncodeH3K9me3-sum</th>\n",
       "      <th>EncodeH3K9me3-max</th>\n",
       "      <th>EncodeH3K27ac-sum</th>\n",
       "      <th>EncodeH3K27ac-max</th>\n",
       "      <th>EncodeH3K27me3-sum</th>\n",
       "      <th>EncodeH3K27me3-max</th>\n",
       "      <th>EncodeH3K36me3-sum</th>\n",
       "      <th>EncodeH3K36me3-max</th>\n",
       "      <th>EncodeH3K79me2-sum</th>\n",
       "      <th>EncodeH3K79me2-max</th>\n",
       "      <th>EncodeH4K20me1-sum</th>\n",
       "      <th>EncodeH4K20me1-max</th>\n",
       "      <th>EncodeH2AFZ-sum</th>\n",
       "      <th>EncodeH2AFZ-max</th>\n",
       "      <th>EncodeDNase-sum</th>\n",
       "      <th>EncodeDNase-max</th>\n",
       "      <th>EncodetotalRNA-sum</th>\n",
       "      <th>EncodetotalRNA-max</th>\n",
       "      <th>Grantham</th>\n",
       "      <th>Dist2Mutation</th>\n",
       "      <th>Freq100bp</th>\n",
       "      <th>Rare100bp</th>\n",
       "      <th>Sngl100bp</th>\n",
       "      <th>Freq1000bp</th>\n",
       "      <th>Rare1000bp</th>\n",
       "      <th>Sngl1000bp</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>End</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>Func.knownGene</th>\n",
       "      <th>Gene.knownGene</th>\n",
       "      <th>GeneDetail.knownGene</th>\n",
       "      <th>ExonicFunc.knownGene</th>\n",
       "      <th>AAChange.knownGene</th>\n",
       "      <th>Func.ensGene</th>\n",
       "      <th>Gene.ensGene</th>\n",
       "      <th>GeneDetail.ensGene</th>\n",
       "      <th>ExonicFunc.ensGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>SIFT_pred</th>\n",
       "      <th>Polyphen2_HDIV_score</th>\n",
       "      <th>Polyphen2_HDIV_pred</th>\n",
       "      <th>Polyphen2_HVAR_score</th>\n",
       "      <th>Polyphen2_HVAR_pred</th>\n",
       "      <th>LRT_score</th>\n",
       "      <th>LRT_pred</th>\n",
       "      <th>MutationTaster_score</th>\n",
       "      <th>MutationTaster_pred</th>\n",
       "      <th>MutationAssessor_score</th>\n",
       "      <th>MutationAssessor_pred</th>\n",
       "      <th>FATHMM_score</th>\n",
       "      <th>FATHMM_pred</th>\n",
       "      <th>RadialSVM_score</th>\n",
       "      <th>RadialSVM_pred</th>\n",
       "      <th>LR_score</th>\n",
       "      <th>LR_pred</th>\n",
       "      <th>VEST3_score</th>\n",
       "      <th>CADD_raw</th>\n",
       "      <th>CADD_phred</th>\n",
       "      <th>GERP++_RS</th>\n",
       "      <th>phyloP46way_placental</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>Interpro_domain</th>\n",
       "      <th>SIFT_score.1</th>\n",
       "      <th>SIFT_converted_rankscore</th>\n",
       "      <th>SIFT_pred.1</th>\n",
       "      <th>Polyphen2_HDIV_score.1</th>\n",
       "      <th>Polyphen2_HDIV_rankscore</th>\n",
       "      <th>Polyphen2_HDIV_pred.1</th>\n",
       "      <th>Polyphen2_HVAR_score.1</th>\n",
       "      <th>Polyphen2_HVAR_rankscore</th>\n",
       "      <th>Polyphen2_HVAR_pred.1</th>\n",
       "      <th>LRT_score.1</th>\n",
       "      <th>LRT_converted_rankscore</th>\n",
       "      <th>LRT_pred.1</th>\n",
       "      <th>MutationTaster_score.1</th>\n",
       "      <th>MutationTaster_converted_rankscore</th>\n",
       "      <th>MutationTaster_pred.1</th>\n",
       "      <th>MutationAssessor_score.1</th>\n",
       "      <th>MutationAssessor_score_rankscore</th>\n",
       "      <th>MutationAssessor_pred.1</th>\n",
       "      <th>FATHMM_score.1</th>\n",
       "      <th>FATHMM_converted_rankscore</th>\n",
       "      <th>FATHMM_pred.1</th>\n",
       "      <th>PROVEAN_score</th>\n",
       "      <th>PROVEAN_converted_rankscore</th>\n",
       "      <th>PROVEAN_pred</th>\n",
       "      <th>VEST3_score.1</th>\n",
       "      <th>VEST3_rankscore</th>\n",
       "      <th>MetaSVM_score</th>\n",
       "      <th>MetaSVM_rankscore</th>\n",
       "      <th>MetaSVM_pred</th>\n",
       "      <th>MetaLR_score</th>\n",
       "      <th>MetaLR_rankscore</th>\n",
       "      <th>MetaLR_pred</th>\n",
       "      <th>M-CAP_score</th>\n",
       "      <th>M-CAP_rankscore</th>\n",
       "      <th>M-CAP_pred</th>\n",
       "      <th>CADD_raw.1</th>\n",
       "      <th>CADD_raw_rankscore</th>\n",
       "      <th>CADD_phred.1</th>\n",
       "      <th>DANN_score</th>\n",
       "      <th>DANN_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_score</th>\n",
       "      <th>fathmm-MKL_coding_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_pred</th>\n",
       "      <th>Eigen_coding_or_noncoding</th>\n",
       "      <th>Eigen-raw</th>\n",
       "      <th>Eigen-PC-raw</th>\n",
       "      <th>GenoCanyon_score</th>\n",
       "      <th>GenoCanyon_score_rankscore</th>\n",
       "      <th>integrated_fitCons_score</th>\n",
       "      <th>integrated_fitCons_score_rankscore</th>\n",
       "      <th>integrated_confidence_value</th>\n",
       "      <th>GERP++_RS.1</th>\n",
       "      <th>GERP++_RS_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate.1</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP20way_mammalian</th>\n",
       "      <th>phyloP20way_mammalian_rankscore</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons20way_mammalian</th>\n",
       "      <th>phastCons20way_mammalian_rankscore</th>\n",
       "      <th>SiPhy_29way_logOdds.1</th>\n",
       "      <th>SiPhy_29way_logOdds_rankscore</th>\n",
       "      <th>Interpro_domain.1</th>\n",
       "      <th>GTEx_V6_gene</th>\n",
       "      <th>GTEx_V6_tissue</th>\n",
       "      <th>dbscSNV_ADA_SCORE</th>\n",
       "      <th>dbscSNV_RF_SCORE</th>\n",
       "      <th>InterVar_automated</th>\n",
       "      <th>PVS1</th>\n",
       "      <th>PS1</th>\n",
       "      <th>PS2</th>\n",
       "      <th>PS3</th>\n",
       "      <th>PS4</th>\n",
       "      <th>PM1</th>\n",
       "      <th>PM2</th>\n",
       "      <th>PM3</th>\n",
       "      <th>PM4</th>\n",
       "      <th>PM5</th>\n",
       "      <th>PM6</th>\n",
       "      <th>PP1</th>\n",
       "      <th>PP2</th>\n",
       "      <th>PP3</th>\n",
       "      <th>PP4</th>\n",
       "      <th>PP5</th>\n",
       "      <th>BA1</th>\n",
       "      <th>BS1</th>\n",
       "      <th>BS2</th>\n",
       "      <th>BS3</th>\n",
       "      <th>BS4</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>BP3</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>0.675</td>\n",
       "      <td>0.093</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSR00000344437</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1163</td>\n",
       "      <td>335</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.817</td>\n",
       "      <td>0.197</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-1.217</td>\n",
       "      <td>-1.307</td>\n",
       "      <td>-1.541</td>\n",
       "      <td>940.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.30</td>\n",
       "      <td>-15.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.67</td>\n",
       "      <td>9.85</td>\n",
       "      <td>26.74</td>\n",
       "      <td>9.35</td>\n",
       "      <td>26.18</td>\n",
       "      <td>10.77</td>\n",
       "      <td>21.90</td>\n",
       "      <td>8.94</td>\n",
       "      <td>5.92</td>\n",
       "      <td>2.84</td>\n",
       "      <td>22.88</td>\n",
       "      <td>10.90</td>\n",
       "      <td>38.80</td>\n",
       "      <td>24.49</td>\n",
       "      <td>10.97</td>\n",
       "      <td>2.87</td>\n",
       "      <td>8.26</td>\n",
       "      <td>2.52</td>\n",
       "      <td>24.86</td>\n",
       "      <td>4.94</td>\n",
       "      <td>7.70</td>\n",
       "      <td>2.47</td>\n",
       "      <td>8.49</td>\n",
       "      <td>5.15</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>314.0</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>SYNONYMOUS</td>\n",
       "      <td>5</td>\n",
       "      <td>synonymous</td>\n",
       "      <td>0.675</td>\n",
       "      <td>0.093</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>D</td>\n",
       "      <td>D</td>\n",
       "      <td>ENSG00000187634</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>CCDS2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/14</td>\n",
       "      <td>1724.0</td>\n",
       "      <td>0.676</td>\n",
       "      <td>1641.0</td>\n",
       "      <td>0.802</td>\n",
       "      <td>547.0</td>\n",
       "      <td>0.803</td>\n",
       "      <td>ndomain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1163</td>\n",
       "      <td>335</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.817</td>\n",
       "      <td>0.197</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-1.217</td>\n",
       "      <td>-1.307</td>\n",
       "      <td>-1.541</td>\n",
       "      <td>940.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.30</td>\n",
       "      <td>-15.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.67</td>\n",
       "      <td>9.85</td>\n",
       "      <td>26.74</td>\n",
       "      <td>9.35</td>\n",
       "      <td>26.18</td>\n",
       "      <td>10.77</td>\n",
       "      <td>21.90</td>\n",
       "      <td>8.94</td>\n",
       "      <td>5.92</td>\n",
       "      <td>2.84</td>\n",
       "      <td>22.88</td>\n",
       "      <td>10.90</td>\n",
       "      <td>38.80</td>\n",
       "      <td>24.49</td>\n",
       "      <td>10.97</td>\n",
       "      <td>2.87</td>\n",
       "      <td>8.26</td>\n",
       "      <td>2.52</td>\n",
       "      <td>24.86</td>\n",
       "      <td>4.94</td>\n",
       "      <td>7.70</td>\n",
       "      <td>2.47</td>\n",
       "      <td>8.49</td>\n",
       "      <td>5.15</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>314.0</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>943329</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>0.675</td>\n",
       "      <td>0.093</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>CCDS3.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1163</td>\n",
       "      <td>335</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.817</td>\n",
       "      <td>0.197</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-1.217</td>\n",
       "      <td>-1.307</td>\n",
       "      <td>-1.541</td>\n",
       "      <td>940.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.30</td>\n",
       "      <td>-15.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.67</td>\n",
       "      <td>9.85</td>\n",
       "      <td>26.74</td>\n",
       "      <td>9.35</td>\n",
       "      <td>26.18</td>\n",
       "      <td>10.77</td>\n",
       "      <td>21.90</td>\n",
       "      <td>8.94</td>\n",
       "      <td>5.92</td>\n",
       "      <td>2.84</td>\n",
       "      <td>22.88</td>\n",
       "      <td>10.90</td>\n",
       "      <td>38.80</td>\n",
       "      <td>24.49</td>\n",
       "      <td>10.97</td>\n",
       "      <td>2.87</td>\n",
       "      <td>8.26</td>\n",
       "      <td>2.52</td>\n",
       "      <td>24.86</td>\n",
       "      <td>4.94</td>\n",
       "      <td>7.70</td>\n",
       "      <td>2.47</td>\n",
       "      <td>8.49</td>\n",
       "      <td>5.15</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>314.0</td>\n",
       "      <td>18</td>\n",
       "      <td>117</td>\n",
       "      <td>2586</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.590702</td>\n",
       "      <td>8.117</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>944699</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.695</td>\n",
       "      <td>0.053</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>D</td>\n",
       "      <td>N</td>\n",
       "      <td>ENSG00000188976</td>\n",
       "      <td>ENST00000327044</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>CCDS3.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19/19</td>\n",
       "      <td>2295.0</td>\n",
       "      <td>0.823</td>\n",
       "      <td>2245.0</td>\n",
       "      <td>0.998</td>\n",
       "      <td>749.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>lcompl</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>864</td>\n",
       "      <td>118</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>0.0</td>\n",
       "      <td>possibly_damaging</td>\n",
       "      <td>0.791</td>\n",
       "      <td>0.007</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.988</td>\n",
       "      <td>-0.494</td>\n",
       "      <td>1.459</td>\n",
       "      <td>2.526</td>\n",
       "      <td>934.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.30</td>\n",
       "      <td>-4.82</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18.34</td>\n",
       "      <td>4.96</td>\n",
       "      <td>15.56</td>\n",
       "      <td>5.30</td>\n",
       "      <td>12.11</td>\n",
       "      <td>3.01</td>\n",
       "      <td>14.44</td>\n",
       "      <td>9.34</td>\n",
       "      <td>1.38</td>\n",
       "      <td>0.52</td>\n",
       "      <td>37.51</td>\n",
       "      <td>20.55</td>\n",
       "      <td>3.00</td>\n",
       "      <td>1.73</td>\n",
       "      <td>8.22</td>\n",
       "      <td>2.08</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.61</td>\n",
       "      <td>22.94</td>\n",
       "      <td>5.98</td>\n",
       "      <td>6.13</td>\n",
       "      <td>2.90</td>\n",
       "      <td>2.33</td>\n",
       "      <td>0.94</td>\n",
       "      <td>35.12</td>\n",
       "      <td>23.19</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>290.0</td>\n",
       "      <td>16</td>\n",
       "      <td>117</td>\n",
       "      <td>2573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>2.980499</td>\n",
       "      <td>23.200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122229</th>\n",
       "      <td>X</td>\n",
       "      <td>155260197</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3PRIME_UTR</td>\n",
       "      <td>2</td>\n",
       "      <td>3_prime_UTR</td>\n",
       "      <td>0.377</td>\n",
       "      <td>0.013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000155961</td>\n",
       "      <td>ENST00000369454</td>\n",
       "      <td>RAB39B</td>\n",
       "      <td>CCDS14766.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>1549.0</td>\n",
       "      <td>0.442</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4393</td>\n",
       "      <td>1956</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.160</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.386</td>\n",
       "      <td>-1.109</td>\n",
       "      <td>-1.160</td>\n",
       "      <td>541.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2312</td>\n",
       "      <td>-16.8</td>\n",
       "      <td>150.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6692.35</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.81</td>\n",
       "      <td>-9.09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.13</td>\n",
       "      <td>1.65</td>\n",
       "      <td>3.88</td>\n",
       "      <td>1.26</td>\n",
       "      <td>10.09</td>\n",
       "      <td>2.15</td>\n",
       "      <td>4.57</td>\n",
       "      <td>1.38</td>\n",
       "      <td>9.09</td>\n",
       "      <td>3.38</td>\n",
       "      <td>3.72</td>\n",
       "      <td>1.38</td>\n",
       "      <td>6.08</td>\n",
       "      <td>1.24</td>\n",
       "      <td>7.35</td>\n",
       "      <td>2.16</td>\n",
       "      <td>5.53</td>\n",
       "      <td>1.78</td>\n",
       "      <td>9.23</td>\n",
       "      <td>2.24</td>\n",
       "      <td>10.70</td>\n",
       "      <td>3.78</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.70</td>\n",
       "      <td>0.52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>85.0</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>873</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>-0.219204</td>\n",
       "      <td>0.473</td>\n",
       "      <td>155260197.0</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>RAB39B</td>\n",
       "      <td>NM_171998:c.*606G&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>RAB39B</td>\n",
       "      <td>uc004fne.5:c.*606G&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>RAB39B</td>\n",
       "      <td>ENST00000369454.3:c.*606G&gt;T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122230</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>0.695</td>\n",
       "      <td>0.147</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSR00000249743</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>146</td>\n",
       "      <td>604</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.026</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.237</td>\n",
       "      <td>0.421</td>\n",
       "      <td>0.441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.17</td>\n",
       "      <td>1.47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.11</td>\n",
       "      <td>0.77</td>\n",
       "      <td>158.05</td>\n",
       "      <td>35.65</td>\n",
       "      <td>527.53</td>\n",
       "      <td>60.44</td>\n",
       "      <td>549.81</td>\n",
       "      <td>109.62</td>\n",
       "      <td>11.13</td>\n",
       "      <td>2.37</td>\n",
       "      <td>406.50</td>\n",
       "      <td>99.96</td>\n",
       "      <td>11.39</td>\n",
       "      <td>4.35</td>\n",
       "      <td>6.98</td>\n",
       "      <td>1.62</td>\n",
       "      <td>54.94</td>\n",
       "      <td>11.00</td>\n",
       "      <td>7.66</td>\n",
       "      <td>1.99</td>\n",
       "      <td>233.84</td>\n",
       "      <td>36.28</td>\n",
       "      <td>3.93</td>\n",
       "      <td>1.60</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "      <td>155612732.0</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SPRY3;TMLHE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TMLHE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TMLHE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122231</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>0.695</td>\n",
       "      <td>0.147</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000185973</td>\n",
       "      <td>ENST00000334398</td>\n",
       "      <td>TMLHE</td>\n",
       "      <td>CCDS14768.1</td>\n",
       "      <td>1/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>146</td>\n",
       "      <td>604</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.026</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.237</td>\n",
       "      <td>0.421</td>\n",
       "      <td>0.441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.17</td>\n",
       "      <td>1.47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.11</td>\n",
       "      <td>0.77</td>\n",
       "      <td>158.05</td>\n",
       "      <td>35.65</td>\n",
       "      <td>527.53</td>\n",
       "      <td>60.44</td>\n",
       "      <td>549.81</td>\n",
       "      <td>109.62</td>\n",
       "      <td>11.13</td>\n",
       "      <td>2.37</td>\n",
       "      <td>406.50</td>\n",
       "      <td>99.96</td>\n",
       "      <td>11.39</td>\n",
       "      <td>4.35</td>\n",
       "      <td>6.98</td>\n",
       "      <td>1.62</td>\n",
       "      <td>54.94</td>\n",
       "      <td>11.00</td>\n",
       "      <td>7.66</td>\n",
       "      <td>1.99</td>\n",
       "      <td>233.84</td>\n",
       "      <td>36.28</td>\n",
       "      <td>3.93</td>\n",
       "      <td>1.60</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "      <td>155612732.0</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SPRY3;TMLHE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TMLHE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TMLHE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122232</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>0.695</td>\n",
       "      <td>0.147</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000224533</td>\n",
       "      <td>ENST00000452506</td>\n",
       "      <td>TMLHE-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>146</td>\n",
       "      <td>604</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.026</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.237</td>\n",
       "      <td>0.421</td>\n",
       "      <td>0.441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.17</td>\n",
       "      <td>1.47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.11</td>\n",
       "      <td>0.77</td>\n",
       "      <td>158.05</td>\n",
       "      <td>35.65</td>\n",
       "      <td>527.53</td>\n",
       "      <td>60.44</td>\n",
       "      <td>549.81</td>\n",
       "      <td>109.62</td>\n",
       "      <td>11.13</td>\n",
       "      <td>2.37</td>\n",
       "      <td>406.50</td>\n",
       "      <td>99.96</td>\n",
       "      <td>11.39</td>\n",
       "      <td>4.35</td>\n",
       "      <td>6.98</td>\n",
       "      <td>1.62</td>\n",
       "      <td>54.94</td>\n",
       "      <td>11.00</td>\n",
       "      <td>7.66</td>\n",
       "      <td>1.99</td>\n",
       "      <td>233.84</td>\n",
       "      <td>36.28</td>\n",
       "      <td>3.93</td>\n",
       "      <td>1.60</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>866</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>112.0</td>\n",
       "      <td>202.0</td>\n",
       "      <td>0.661682</td>\n",
       "      <td>8.709</td>\n",
       "      <td>155612732.0</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SPRY3;TMLHE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TMLHE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TMLHE</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>122233 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0           1     943329   C   T  SNV       0  RegulatoryFeature   \n",
       "1           1     943329   C   T  SNV       0   CodingTranscript   \n",
       "2           1     943329   C   T  SNV       0         Intergenic   \n",
       "3           1     944699   C   T  SNV       0   CodingTranscript   \n",
       "...       ...        ...  ..  ..  ...     ...                ...   \n",
       "122229      X  155260197   C   A  SNV       0         Transcript   \n",
       "122230      X  155612732   T   C  SNV       0  RegulatoryFeature   \n",
       "122231      X  155612732   T   C  SNV       0         Transcript   \n",
       "122232      X  155612732   T   C  SNV       0         Intergenic   \n",
       "\n",
       "           Consequence  ConsScore   ConsDetail     GC    CpG  motifECount  \\\n",
       "0           REGULATORY          4   regulatory  0.675  0.093          NaN   \n",
       "1           SYNONYMOUS          5   synonymous  0.675  0.093          NaN   \n",
       "2           DOWNSTREAM          1   downstream  0.675  0.093          NaN   \n",
       "3       NON_SYNONYMOUS          7     missense  0.695  0.053          NaN   \n",
       "...                ...        ...          ...    ...    ...          ...   \n",
       "122229      3PRIME_UTR          2  3_prime_UTR  0.377  0.013          NaN   \n",
       "122230      REGULATORY          4   regulatory  0.695  0.147          NaN   \n",
       "122231        INTRONIC          2       intron  0.695  0.147          NaN   \n",
       "122232      DOWNSTREAM          1   downstream  0.695  0.147          NaN   \n",
       "\n",
       "       motifEName  motifEHIPos  motifEScoreChng  oAA  nAA           GeneID  \\\n",
       "0             NaN          NaN              NaN  NaN  NaN              NaN   \n",
       "1             NaN          NaN              NaN    D    D  ENSG00000187634   \n",
       "2             NaN          NaN              NaN  NaN  NaN  ENSG00000188976   \n",
       "3             NaN          NaN              NaN    D    N  ENSG00000188976   \n",
       "...           ...          ...              ...  ...  ...              ...   \n",
       "122229        NaN          NaN              NaN  NaN  NaN  ENSG00000155961   \n",
       "122230        NaN          NaN              NaN  NaN  NaN              NaN   \n",
       "122231        NaN          NaN              NaN  NaN  NaN  ENSG00000185973   \n",
       "122232        NaN          NaN              NaN  NaN  NaN  ENSG00000224533   \n",
       "\n",
       "              FeatureID   GeneName         CCDS Intron   Exon  cDNApos  \\\n",
       "0       ENSR00000344437        NaN          NaN    NaN    NaN      NaN   \n",
       "1       ENST00000342066     SAMD11      CCDS2.2    NaN  12/14   1724.0   \n",
       "2       ENST00000327044      NOC2L      CCDS3.1    NaN    NaN      NaN   \n",
       "3       ENST00000327044      NOC2L      CCDS3.1    NaN  19/19   2295.0   \n",
       "...                 ...        ...          ...    ...    ...      ...   \n",
       "122229  ENST00000369454     RAB39B  CCDS14766.1    NaN    2/2   1549.0   \n",
       "122230  ENSR00000249743        NaN          NaN    NaN    NaN      NaN   \n",
       "122231  ENST00000334398      TMLHE  CCDS14768.1    1/7    NaN      NaN   \n",
       "122232  ENST00000452506  TMLHE-AS1          NaN    NaN    NaN      NaN   \n",
       "\n",
       "        relcDNApos  CDSpos  relCDSpos  protPos  relProtPos   Domain  \\\n",
       "0              NaN     NaN        NaN      NaN         NaN      NaN   \n",
       "1            0.676  1641.0      0.802    547.0       0.803  ndomain   \n",
       "2              NaN     NaN        NaN      NaN         NaN      NaN   \n",
       "3            0.823  2245.0      0.998    749.0       1.000   lcompl   \n",
       "...            ...     ...        ...      ...         ...      ...   \n",
       "122229       0.442     NaN        NaN      NaN         NaN      NaN   \n",
       "122230         NaN     NaN        NaN      NaN         NaN      NaN   \n",
       "122231         NaN     NaN        NaN      NaN         NaN      NaN   \n",
       "122232         NaN     NaN        NaN      NaN         NaN      NaN   \n",
       "\n",
       "        Dst2Splice Dst2SplType  minDistTSS  minDistTSE      SIFTcat  SIFTval  \\\n",
       "0              NaN         NaN        1163         335          NaN      NaN   \n",
       "1              NaN         NaN        1163         335          NaN      NaN   \n",
       "2              NaN         NaN        1163         335          NaN      NaN   \n",
       "3              NaN         NaN         864         118  deleterious      0.0   \n",
       "...            ...         ...         ...         ...          ...      ...   \n",
       "122229         NaN         NaN        4393        1956          NaN      NaN   \n",
       "122230         NaN         NaN         146         604          NaN      NaN   \n",
       "122231         NaN         NaN         146         604          NaN      NaN   \n",
       "122232         NaN         NaN         146         604          NaN      NaN   \n",
       "\n",
       "              PolyPhenCat  PolyPhenVal  priPhCons  mamPhCons  verPhCons  \\\n",
       "0                     NaN          NaN      0.817      0.197      0.000   \n",
       "1                     NaN          NaN      0.817      0.197      0.000   \n",
       "2                     NaN          NaN      0.817      0.197      0.000   \n",
       "3       possibly_damaging        0.791      0.007      1.000      0.988   \n",
       "...                   ...          ...        ...        ...        ...   \n",
       "122229                NaN          NaN      0.160      0.000      0.000   \n",
       "122230                NaN          NaN      0.026      0.000      0.000   \n",
       "122231                NaN          NaN      0.026      0.000      0.000   \n",
       "122232                NaN          NaN      0.026      0.000      0.000   \n",
       "\n",
       "        priPhyloP  mamPhyloP  verPhyloP  bStatistic  targetScan  mirSVR-Score  \\\n",
       "0          -1.217     -1.307     -1.541       940.0         NaN           NaN   \n",
       "1          -1.217     -1.307     -1.541       940.0         NaN           NaN   \n",
       "2          -1.217     -1.307     -1.541       940.0         NaN           NaN   \n",
       "3          -0.494      1.459      2.526       934.0         NaN           NaN   \n",
       "...           ...        ...        ...         ...         ...           ...   \n",
       "122229      0.386     -1.109     -1.160       541.0         NaN       -0.2312   \n",
       "122230      0.237      0.421      0.441         NaN         NaN           NaN   \n",
       "122231      0.237      0.421      0.441         NaN         NaN           NaN   \n",
       "122232      0.237      0.421      0.441         NaN         NaN           NaN   \n",
       "\n",
       "        mirSVR-E  mirSVR-Aln  cHmm_E1  cHmm_E2  cHmm_E3  cHmm_E4  cHmm_E5  \\\n",
       "0            NaN         NaN        0        1        0        0        0   \n",
       "1            NaN         NaN        0        1        0        0        0   \n",
       "2            NaN         NaN        0        1        0        0        0   \n",
       "3            NaN         NaN        1        1        0        0        0   \n",
       "...          ...         ...      ...      ...      ...      ...      ...   \n",
       "122229     -16.8       150.0        0        0        1       16        0   \n",
       "122230       NaN         NaN        1        0        0        0        0   \n",
       "122231       NaN         NaN        1        0        0        0        0   \n",
       "122232       NaN         NaN        1        0        0        0        0   \n",
       "\n",
       "        cHmm_E6  cHmm_E7  cHmm_E8  cHmm_E9  cHmm_E10  cHmm_E11  cHmm_E12  \\\n",
       "0             0        1        0        0         0         1         5   \n",
       "1             0        1        0        0         0         1         5   \n",
       "2             0        1        0        0         0         1         5   \n",
       "3             5       11        3        1         5         5         0   \n",
       "...         ...      ...      ...      ...       ...       ...       ...   \n",
       "122229        0        3        1        0         0         0         0   \n",
       "122230        0        0        0        0         0         0         0   \n",
       "122231        0        0        0        0         0         0         0   \n",
       "122232        0        0        0        0         0         0         0   \n",
       "\n",
       "        cHmm_E13  cHmm_E14  cHmm_E15  cHmm_E16  cHmm_E17  cHmm_E18  cHmm_E19  \\\n",
       "0              1         2         3         0         1         1         0   \n",
       "1              1         2         3         0         1         1         0   \n",
       "2              1         2         3         0         1         1         0   \n",
       "3              1         3         3         0         1         0         0   \n",
       "...          ...       ...       ...       ...       ...       ...       ...   \n",
       "122229         0         0        23         0         0         0         0   \n",
       "122230         0         0         0        44         1         0         2   \n",
       "122231         0         0         0        44         1         0         2   \n",
       "122232         0         0         0        44         1         0         2   \n",
       "\n",
       "        cHmm_E20  cHmm_E21  cHmm_E22  cHmm_E23  cHmm_E24  cHmm_E25   GerpRS  \\\n",
       "0             15         0         5         8         2         2      NaN   \n",
       "1             15         0         5         8         2         2      NaN   \n",
       "2             15         0         5         8         2         2      NaN   \n",
       "3              0         0         1         2         3         2      NaN   \n",
       "...          ...       ...       ...       ...       ...       ...      ...   \n",
       "122229         0         4         0         0         0         0  6692.35   \n",
       "122230         0         0         0         0         0         0      NaN   \n",
       "122231         0         0         0         0         0         0      NaN   \n",
       "122232         0         0         0         0         0         0      NaN   \n",
       "\n",
       "        GerpRSpval  GerpN  GerpS  tOverlapMotifs  motifDist  \\\n",
       "0              NaN  18.30 -15.60             NaN        NaN   \n",
       "1              NaN  18.30 -15.60             NaN        NaN   \n",
       "2              NaN  18.30 -15.60             NaN        NaN   \n",
       "3              NaN  18.30  -4.82             NaN        NaN   \n",
       "...            ...    ...    ...             ...        ...   \n",
       "122229         0.0   5.81  -9.09             NaN        NaN   \n",
       "122230         NaN   4.17   1.47             NaN        NaN   \n",
       "122231         NaN   4.17   1.47             NaN        NaN   \n",
       "122232         NaN   4.17   1.47             NaN        NaN   \n",
       "\n",
       "        EncodeH3K4me1-sum  EncodeH3K4me1-max  EncodeH3K4me2-sum  \\\n",
       "0                   35.67               9.85              26.74   \n",
       "1                   35.67               9.85              26.74   \n",
       "2                   35.67               9.85              26.74   \n",
       "3                   18.34               4.96              15.56   \n",
       "...                   ...                ...                ...   \n",
       "122229               7.13               1.65               3.88   \n",
       "122230               5.11               0.77             158.05   \n",
       "122231               5.11               0.77             158.05   \n",
       "122232               5.11               0.77             158.05   \n",
       "\n",
       "        EncodeH3K4me2-max  EncodeH3K4me3-sum  EncodeH3K4me3-max  \\\n",
       "0                    9.35              26.18              10.77   \n",
       "1                    9.35              26.18              10.77   \n",
       "2                    9.35              26.18              10.77   \n",
       "3                    5.30              12.11               3.01   \n",
       "...                   ...                ...                ...   \n",
       "122229               1.26              10.09               2.15   \n",
       "122230              35.65             527.53              60.44   \n",
       "122231              35.65             527.53              60.44   \n",
       "122232              35.65             527.53              60.44   \n",
       "\n",
       "        EncodeH3K9ac-sum  EncodeH3K9ac-max  EncodeH3K9me3-sum  \\\n",
       "0                  21.90              8.94               5.92   \n",
       "1                  21.90              8.94               5.92   \n",
       "2                  21.90              8.94               5.92   \n",
       "3                  14.44              9.34               1.38   \n",
       "...                  ...               ...                ...   \n",
       "122229              4.57              1.38               9.09   \n",
       "122230            549.81            109.62              11.13   \n",
       "122231            549.81            109.62              11.13   \n",
       "122232            549.81            109.62              11.13   \n",
       "\n",
       "        EncodeH3K9me3-max  EncodeH3K27ac-sum  EncodeH3K27ac-max  \\\n",
       "0                    2.84              22.88              10.90   \n",
       "1                    2.84              22.88              10.90   \n",
       "2                    2.84              22.88              10.90   \n",
       "3                    0.52              37.51              20.55   \n",
       "...                   ...                ...                ...   \n",
       "122229               3.38               3.72               1.38   \n",
       "122230               2.37             406.50              99.96   \n",
       "122231               2.37             406.50              99.96   \n",
       "122232               2.37             406.50              99.96   \n",
       "\n",
       "        EncodeH3K27me3-sum  EncodeH3K27me3-max  EncodeH3K36me3-sum  \\\n",
       "0                    38.80               24.49               10.97   \n",
       "1                    38.80               24.49               10.97   \n",
       "2                    38.80               24.49               10.97   \n",
       "3                     3.00                1.73                8.22   \n",
       "...                    ...                 ...                 ...   \n",
       "122229                6.08                1.24                7.35   \n",
       "122230               11.39                4.35                6.98   \n",
       "122231               11.39                4.35                6.98   \n",
       "122232               11.39                4.35                6.98   \n",
       "\n",
       "        EncodeH3K36me3-max  EncodeH3K79me2-sum  EncodeH3K79me2-max  \\\n",
       "0                     2.87                8.26                2.52   \n",
       "1                     2.87                8.26                2.52   \n",
       "2                     2.87                8.26                2.52   \n",
       "3                     2.08                0.61                0.61   \n",
       "...                    ...                 ...                 ...   \n",
       "122229                2.16                5.53                1.78   \n",
       "122230                1.62               54.94               11.00   \n",
       "122231                1.62               54.94               11.00   \n",
       "122232                1.62               54.94               11.00   \n",
       "\n",
       "        EncodeH4K20me1-sum  EncodeH4K20me1-max  EncodeH2AFZ-sum  \\\n",
       "0                    24.86                4.94             7.70   \n",
       "1                    24.86                4.94             7.70   \n",
       "2                    24.86                4.94             7.70   \n",
       "3                    22.94                5.98             6.13   \n",
       "...                    ...                 ...              ...   \n",
       "122229                9.23                2.24            10.70   \n",
       "122230                7.66                1.99           233.84   \n",
       "122231                7.66                1.99           233.84   \n",
       "122232                7.66                1.99           233.84   \n",
       "\n",
       "        EncodeH2AFZ-max  EncodeDNase-sum  EncodeDNase-max  EncodetotalRNA-sum  \\\n",
       "0                  2.47             8.49             5.15                0.60   \n",
       "1                  2.47             8.49             5.15                0.60   \n",
       "2                  2.47             8.49             5.15                0.60   \n",
       "3                  2.90             2.33             0.94               35.12   \n",
       "...                 ...              ...              ...                 ...   \n",
       "122229             3.78             0.17             0.07                0.70   \n",
       "122230            36.28             3.93             1.60                0.38   \n",
       "122231            36.28             3.93             1.60                0.38   \n",
       "122232            36.28             3.93             1.60                0.38   \n",
       "\n",
       "        EncodetotalRNA-max  Grantham  Dist2Mutation  Freq100bp  Rare100bp  \\\n",
       "0                     0.32       NaN              6        0.0        2.0   \n",
       "1                     0.32       NaN              6        0.0        2.0   \n",
       "2                     0.32       NaN              6        0.0        2.0   \n",
       "3                    23.19      23.0              7        0.0        0.0   \n",
       "...                    ...       ...            ...        ...        ...   \n",
       "122229                0.52       NaN              9        0.0        0.0   \n",
       "122230                0.08       NaN             26        0.0        2.0   \n",
       "122231                0.08       NaN             26        0.0        2.0   \n",
       "122232                0.08       NaN             26        0.0        2.0   \n",
       "\n",
       "        Sngl100bp  Freq1000bp  Rare1000bp  Sngl1000bp  Freq10000bp  \\\n",
       "0            38.0         1.0        17.0       314.0           18   \n",
       "1            38.0         1.0        17.0       314.0           18   \n",
       "2            38.0         1.0        17.0       314.0           18   \n",
       "3            34.0         4.0        14.0       290.0           16   \n",
       "...           ...         ...         ...         ...          ...   \n",
       "122229        6.0         0.0         5.0        85.0            9   \n",
       "122230       12.0         1.0         3.0       122.0            7   \n",
       "122231       12.0         1.0         3.0       122.0            7   \n",
       "122232       12.0         1.0         3.0       122.0            7   \n",
       "\n",
       "        Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0               117         2586         CTCF Binding Site                NaN   \n",
       "1               117         2586         CTCF Binding Site                NaN   \n",
       "2               117         2586         CTCF Binding Site                NaN   \n",
       "3               117         2573                       NaN                NaN   \n",
       "...             ...          ...                       ...                ...   \n",
       "122229           36          873                       NaN                NaN   \n",
       "122230           27          866                  Promoter                NaN   \n",
       "122231           27          866                  Promoter                NaN   \n",
       "122232           27          866                  Promoter                NaN   \n",
       "\n",
       "       dbscSNV-rf_score  RemapOverlapTF  RemapOverlapCL  RawScore   PHRED  \\\n",
       "0                   NaN            88.0           121.0  0.590702   8.117   \n",
       "1                   NaN            88.0           121.0  0.590702   8.117   \n",
       "2                   NaN            88.0           121.0  0.590702   8.117   \n",
       "3                   NaN            12.0            13.0  2.980499  23.200   \n",
       "...                 ...             ...             ...       ...     ...   \n",
       "122229              NaN             3.0             6.0 -0.219204   0.473   \n",
       "122230              NaN           112.0           202.0  0.661682   8.709   \n",
       "122231              NaN           112.0           202.0  0.661682   8.709   \n",
       "122232              NaN           112.0           202.0  0.661682   8.709   \n",
       "\n",
       "                End Func.refGene Gene.refGene   GeneDetail.refGene  \\\n",
       "0               NaN          NaN          NaN                  NaN   \n",
       "1               NaN          NaN          NaN                  NaN   \n",
       "2               NaN          NaN          NaN                  NaN   \n",
       "3               NaN          NaN          NaN                  NaN   \n",
       "...             ...          ...          ...                  ...   \n",
       "122229  155260197.0         UTR3       RAB39B  NM_171998:c.*606G>T   \n",
       "122230  155612732.0     intronic  SPRY3;TMLHE                    .   \n",
       "122231  155612732.0     intronic  SPRY3;TMLHE                    .   \n",
       "122232  155612732.0     intronic  SPRY3;TMLHE                    .   \n",
       "\n",
       "       ExonicFunc.refGene AAChange.refGene Func.knownGene Gene.knownGene  \\\n",
       "0                     NaN              NaN            NaN            NaN   \n",
       "1                     NaN              NaN            NaN            NaN   \n",
       "2                     NaN              NaN            NaN            NaN   \n",
       "3                     NaN              NaN            NaN            NaN   \n",
       "...                   ...              ...            ...            ...   \n",
       "122229                  .                .           UTR3         RAB39B   \n",
       "122230                  .                .       intronic          TMLHE   \n",
       "122231                  .                .       intronic          TMLHE   \n",
       "122232                  .                .       intronic          TMLHE   \n",
       "\n",
       "        GeneDetail.knownGene ExonicFunc.knownGene AAChange.knownGene  \\\n",
       "0                        NaN                  NaN                NaN   \n",
       "1                        NaN                  NaN                NaN   \n",
       "2                        NaN                  NaN                NaN   \n",
       "3                        NaN                  NaN                NaN   \n",
       "...                      ...                  ...                ...   \n",
       "122229  uc004fne.5:c.*606G>T                    .                  .   \n",
       "122230                     .                    .                  .   \n",
       "122231                     .                    .                  .   \n",
       "122232                     .                    .                  .   \n",
       "\n",
       "       Func.ensGene Gene.ensGene           GeneDetail.ensGene  \\\n",
       "0               NaN          NaN                          NaN   \n",
       "1               NaN          NaN                          NaN   \n",
       "2               NaN          NaN                          NaN   \n",
       "3               NaN          NaN                          NaN   \n",
       "...             ...          ...                          ...   \n",
       "122229         UTR3       RAB39B  ENST00000369454.3:c.*606G>T   \n",
       "122230     intronic        TMLHE                            .   \n",
       "122231     intronic        TMLHE                            .   \n",
       "122232     intronic        TMLHE                            .   \n",
       "\n",
       "       ExonicFunc.ensGene AAChange.ensGene SIFT_score SIFT_pred  \\\n",
       "0                     NaN              NaN        NaN       NaN   \n",
       "1                     NaN              NaN        NaN       NaN   \n",
       "2                     NaN              NaN        NaN       NaN   \n",
       "3                     NaN              NaN        NaN       NaN   \n",
       "...                   ...              ...        ...       ...   \n",
       "122229                  .                .          .         .   \n",
       "122230                  .                .          .         .   \n",
       "122231                  .                .          .         .   \n",
       "122232                  .                .          .         .   \n",
       "\n",
       "       Polyphen2_HDIV_score Polyphen2_HDIV_pred Polyphen2_HVAR_score  \\\n",
       "0                       NaN                 NaN                  NaN   \n",
       "1                       NaN                 NaN                  NaN   \n",
       "2                       NaN                 NaN                  NaN   \n",
       "3                       NaN                 NaN                  NaN   \n",
       "...                     ...                 ...                  ...   \n",
       "122229                    .                   .                    .   \n",
       "122230                    .                   .                    .   \n",
       "122231                    .                   .                    .   \n",
       "122232                    .                   .                    .   \n",
       "\n",
       "       Polyphen2_HVAR_pred LRT_score LRT_pred MutationTaster_score  \\\n",
       "0                      NaN       NaN      NaN                  NaN   \n",
       "1                      NaN       NaN      NaN                  NaN   \n",
       "2                      NaN       NaN      NaN                  NaN   \n",
       "3                      NaN       NaN      NaN                  NaN   \n",
       "...                    ...       ...      ...                  ...   \n",
       "122229                   .         .        .                    .   \n",
       "122230                   .         .        .                    .   \n",
       "122231                   .         .        .                    .   \n",
       "122232                   .         .        .                    .   \n",
       "\n",
       "       MutationTaster_pred MutationAssessor_score MutationAssessor_pred  \\\n",
       "0                      NaN                    NaN                   NaN   \n",
       "1                      NaN                    NaN                   NaN   \n",
       "2                      NaN                    NaN                   NaN   \n",
       "3                      NaN                    NaN                   NaN   \n",
       "...                    ...                    ...                   ...   \n",
       "122229                   .                      .                     .   \n",
       "122230                   .                      .                     .   \n",
       "122231                   .                      .                     .   \n",
       "122232                   .                      .                     .   \n",
       "\n",
       "       FATHMM_score FATHMM_pred RadialSVM_score RadialSVM_pred LR_score  \\\n",
       "0               NaN         NaN             NaN            NaN      NaN   \n",
       "1               NaN         NaN             NaN            NaN      NaN   \n",
       "2               NaN         NaN             NaN            NaN      NaN   \n",
       "3               NaN         NaN             NaN            NaN      NaN   \n",
       "...             ...         ...             ...            ...      ...   \n",
       "122229            .           .               .              .        .   \n",
       "122230            .           .               .              .        .   \n",
       "122231            .           .               .              .        .   \n",
       "122232            .           .               .              .        .   \n",
       "\n",
       "       LR_pred VEST3_score CADD_raw CADD_phred GERP++_RS  \\\n",
       "0          NaN         NaN      NaN        NaN       NaN   \n",
       "1          NaN         NaN      NaN        NaN       NaN   \n",
       "2          NaN         NaN      NaN        NaN       NaN   \n",
       "3          NaN         NaN      NaN        NaN       NaN   \n",
       "...        ...         ...      ...        ...       ...   \n",
       "122229       .           .        .          .         .   \n",
       "122230       .           .        .          .         .   \n",
       "122231       .           .        .          .         .   \n",
       "122232       .           .        .          .         .   \n",
       "\n",
       "       phyloP46way_placental phyloP100way_vertebrate SiPhy_29way_logOdds  \\\n",
       "0                        NaN                     NaN                 NaN   \n",
       "1                        NaN                     NaN                 NaN   \n",
       "2                        NaN                     NaN                 NaN   \n",
       "3                        NaN                     NaN                 NaN   \n",
       "...                      ...                     ...                 ...   \n",
       "122229                     .                       .                   .   \n",
       "122230                     .                       .                   .   \n",
       "122231                     .                       .                   .   \n",
       "122232                     .                       .                   .   \n",
       "\n",
       "       Interpro_domain SIFT_score.1 SIFT_converted_rankscore SIFT_pred.1  \\\n",
       "0                  NaN          NaN                      NaN         NaN   \n",
       "1                  NaN          NaN                      NaN         NaN   \n",
       "2                  NaN          NaN                      NaN         NaN   \n",
       "3                  NaN          NaN                      NaN         NaN   \n",
       "...                ...          ...                      ...         ...   \n",
       "122229               .            .                        .           .   \n",
       "122230               .            .                        .           .   \n",
       "122231               .            .                        .           .   \n",
       "122232               .            .                        .           .   \n",
       "\n",
       "       Polyphen2_HDIV_score.1 Polyphen2_HDIV_rankscore Polyphen2_HDIV_pred.1  \\\n",
       "0                         NaN                      NaN                   NaN   \n",
       "1                         NaN                      NaN                   NaN   \n",
       "2                         NaN                      NaN                   NaN   \n",
       "3                         NaN                      NaN                   NaN   \n",
       "...                       ...                      ...                   ...   \n",
       "122229                      .                        .                     .   \n",
       "122230                      .                        .                     .   \n",
       "122231                      .                        .                     .   \n",
       "122232                      .                        .                     .   \n",
       "\n",
       "       Polyphen2_HVAR_score.1 Polyphen2_HVAR_rankscore Polyphen2_HVAR_pred.1  \\\n",
       "0                         NaN                      NaN                   NaN   \n",
       "1                         NaN                      NaN                   NaN   \n",
       "2                         NaN                      NaN                   NaN   \n",
       "3                         NaN                      NaN                   NaN   \n",
       "...                       ...                      ...                   ...   \n",
       "122229                      .                        .                     .   \n",
       "122230                      .                        .                     .   \n",
       "122231                      .                        .                     .   \n",
       "122232                      .                        .                     .   \n",
       "\n",
       "       LRT_score.1 LRT_converted_rankscore LRT_pred.1 MutationTaster_score.1  \\\n",
       "0              NaN                     NaN        NaN                    NaN   \n",
       "1              NaN                     NaN        NaN                    NaN   \n",
       "2              NaN                     NaN        NaN                    NaN   \n",
       "3              NaN                     NaN        NaN                    NaN   \n",
       "...            ...                     ...        ...                    ...   \n",
       "122229           .                       .          .                      .   \n",
       "122230           .                       .          .                      .   \n",
       "122231           .                       .          .                      .   \n",
       "122232           .                       .          .                      .   \n",
       "\n",
       "       MutationTaster_converted_rankscore MutationTaster_pred.1  \\\n",
       "0                                     NaN                   NaN   \n",
       "1                                     NaN                   NaN   \n",
       "2                                     NaN                   NaN   \n",
       "3                                     NaN                   NaN   \n",
       "...                                   ...                   ...   \n",
       "122229                                  .                     .   \n",
       "122230                                  .                     .   \n",
       "122231                                  .                     .   \n",
       "122232                                  .                     .   \n",
       "\n",
       "       MutationAssessor_score.1 MutationAssessor_score_rankscore  \\\n",
       "0                           NaN                              NaN   \n",
       "1                           NaN                              NaN   \n",
       "2                           NaN                              NaN   \n",
       "3                           NaN                              NaN   \n",
       "...                         ...                              ...   \n",
       "122229                        .                                .   \n",
       "122230                        .                                .   \n",
       "122231                        .                                .   \n",
       "122232                        .                                .   \n",
       "\n",
       "       MutationAssessor_pred.1 FATHMM_score.1 FATHMM_converted_rankscore  \\\n",
       "0                          NaN            NaN                        NaN   \n",
       "1                          NaN            NaN                        NaN   \n",
       "2                          NaN            NaN                        NaN   \n",
       "3                          NaN            NaN                        NaN   \n",
       "...                        ...            ...                        ...   \n",
       "122229                       .              .                          .   \n",
       "122230                       .              .                          .   \n",
       "122231                       .              .                          .   \n",
       "122232                       .              .                          .   \n",
       "\n",
       "       FATHMM_pred.1 PROVEAN_score PROVEAN_converted_rankscore PROVEAN_pred  \\\n",
       "0                NaN           NaN                         NaN          NaN   \n",
       "1                NaN           NaN                         NaN          NaN   \n",
       "2                NaN           NaN                         NaN          NaN   \n",
       "3                NaN           NaN                         NaN          NaN   \n",
       "...              ...           ...                         ...          ...   \n",
       "122229             .             .                           .            .   \n",
       "122230             .             .                           .            .   \n",
       "122231             .             .                           .            .   \n",
       "122232             .             .                           .            .   \n",
       "\n",
       "       VEST3_score.1 VEST3_rankscore MetaSVM_score MetaSVM_rankscore  \\\n",
       "0                NaN             NaN           NaN               NaN   \n",
       "1                NaN             NaN           NaN               NaN   \n",
       "2                NaN             NaN           NaN               NaN   \n",
       "3                NaN             NaN           NaN               NaN   \n",
       "...              ...             ...           ...               ...   \n",
       "122229             .               .             .                 .   \n",
       "122230             .               .             .                 .   \n",
       "122231             .               .             .                 .   \n",
       "122232             .               .             .                 .   \n",
       "\n",
       "       MetaSVM_pred MetaLR_score MetaLR_rankscore MetaLR_pred M-CAP_score  \\\n",
       "0               NaN          NaN              NaN         NaN         NaN   \n",
       "1               NaN          NaN              NaN         NaN         NaN   \n",
       "2               NaN          NaN              NaN         NaN         NaN   \n",
       "3               NaN          NaN              NaN         NaN         NaN   \n",
       "...             ...          ...              ...         ...         ...   \n",
       "122229            .            .                .           .           .   \n",
       "122230            .            .                .           .           .   \n",
       "122231            .            .                .           .           .   \n",
       "122232            .            .                .           .           .   \n",
       "\n",
       "       M-CAP_rankscore M-CAP_pred CADD_raw.1 CADD_raw_rankscore CADD_phred.1  \\\n",
       "0                  NaN        NaN        NaN                NaN          NaN   \n",
       "1                  NaN        NaN        NaN                NaN          NaN   \n",
       "2                  NaN        NaN        NaN                NaN          NaN   \n",
       "3                  NaN        NaN        NaN                NaN          NaN   \n",
       "...                ...        ...        ...                ...          ...   \n",
       "122229               .          .          .                  .            .   \n",
       "122230               .          .          .                  .            .   \n",
       "122231               .          .          .                  .            .   \n",
       "122232               .          .          .                  .            .   \n",
       "\n",
       "       DANN_score DANN_rankscore fathmm-MKL_coding_score  \\\n",
       "0             NaN            NaN                     NaN   \n",
       "1             NaN            NaN                     NaN   \n",
       "2             NaN            NaN                     NaN   \n",
       "3             NaN            NaN                     NaN   \n",
       "...           ...            ...                     ...   \n",
       "122229          .              .                       .   \n",
       "122230          .              .                       .   \n",
       "122231          .              .                       .   \n",
       "122232          .              .                       .   \n",
       "\n",
       "       fathmm-MKL_coding_rankscore fathmm-MKL_coding_pred  \\\n",
       "0                              NaN                    NaN   \n",
       "1                              NaN                    NaN   \n",
       "2                              NaN                    NaN   \n",
       "3                              NaN                    NaN   \n",
       "...                            ...                    ...   \n",
       "122229                           .                      .   \n",
       "122230                           .                      .   \n",
       "122231                           .                      .   \n",
       "122232                           .                      .   \n",
       "\n",
       "       Eigen_coding_or_noncoding Eigen-raw Eigen-PC-raw GenoCanyon_score  \\\n",
       "0                            NaN       NaN          NaN              NaN   \n",
       "1                            NaN       NaN          NaN              NaN   \n",
       "2                            NaN       NaN          NaN              NaN   \n",
       "3                            NaN       NaN          NaN              NaN   \n",
       "...                          ...       ...          ...              ...   \n",
       "122229                         .         .            .                .   \n",
       "122230                         .         .            .                .   \n",
       "122231                         .         .            .                .   \n",
       "122232                         .         .            .                .   \n",
       "\n",
       "       GenoCanyon_score_rankscore integrated_fitCons_score  \\\n",
       "0                             NaN                      NaN   \n",
       "1                             NaN                      NaN   \n",
       "2                             NaN                      NaN   \n",
       "3                             NaN                      NaN   \n",
       "...                           ...                      ...   \n",
       "122229                          .                        .   \n",
       "122230                          .                        .   \n",
       "122231                          .                        .   \n",
       "122232                          .                        .   \n",
       "\n",
       "       integrated_fitCons_score_rankscore integrated_confidence_value  \\\n",
       "0                                     NaN                         NaN   \n",
       "1                                     NaN                         NaN   \n",
       "2                                     NaN                         NaN   \n",
       "3                                     NaN                         NaN   \n",
       "...                                   ...                         ...   \n",
       "122229                                  .                           .   \n",
       "122230                                  .                           .   \n",
       "122231                                  .                           .   \n",
       "122232                                  .                           .   \n",
       "\n",
       "       GERP++_RS.1 GERP++_RS_rankscore phyloP100way_vertebrate.1  \\\n",
       "0              NaN                 NaN                       NaN   \n",
       "1              NaN                 NaN                       NaN   \n",
       "2              NaN                 NaN                       NaN   \n",
       "3              NaN                 NaN                       NaN   \n",
       "...            ...                 ...                       ...   \n",
       "122229           .                   .                         .   \n",
       "122230           .                   .                         .   \n",
       "122231           .                   .                         .   \n",
       "122232           .                   .                         .   \n",
       "\n",
       "       phyloP100way_vertebrate_rankscore phyloP20way_mammalian  \\\n",
       "0                                    NaN                   NaN   \n",
       "1                                    NaN                   NaN   \n",
       "2                                    NaN                   NaN   \n",
       "3                                    NaN                   NaN   \n",
       "...                                  ...                   ...   \n",
       "122229                                 .                     .   \n",
       "122230                                 .                     .   \n",
       "122231                                 .                     .   \n",
       "122232                                 .                     .   \n",
       "\n",
       "       phyloP20way_mammalian_rankscore phastCons100way_vertebrate  \\\n",
       "0                                  NaN                        NaN   \n",
       "1                                  NaN                        NaN   \n",
       "2                                  NaN                        NaN   \n",
       "3                                  NaN                        NaN   \n",
       "...                                ...                        ...   \n",
       "122229                               .                          .   \n",
       "122230                               .                          .   \n",
       "122231                               .                          .   \n",
       "122232                               .                          .   \n",
       "\n",
       "       phastCons100way_vertebrate_rankscore phastCons20way_mammalian  \\\n",
       "0                                       NaN                      NaN   \n",
       "1                                       NaN                      NaN   \n",
       "2                                       NaN                      NaN   \n",
       "3                                       NaN                      NaN   \n",
       "...                                     ...                      ...   \n",
       "122229                                    .                        .   \n",
       "122230                                    .                        .   \n",
       "122231                                    .                        .   \n",
       "122232                                    .                        .   \n",
       "\n",
       "       phastCons20way_mammalian_rankscore SiPhy_29way_logOdds.1  \\\n",
       "0                                     NaN                   NaN   \n",
       "1                                     NaN                   NaN   \n",
       "2                                     NaN                   NaN   \n",
       "3                                     NaN                   NaN   \n",
       "...                                   ...                   ...   \n",
       "122229                                  .                     .   \n",
       "122230                                  .                     .   \n",
       "122231                                  .                     .   \n",
       "122232                                  .                     .   \n",
       "\n",
       "       SiPhy_29way_logOdds_rankscore Interpro_domain.1 GTEx_V6_gene  \\\n",
       "0                                NaN               NaN          NaN   \n",
       "1                                NaN               NaN          NaN   \n",
       "2                                NaN               NaN          NaN   \n",
       "3                                NaN               NaN          NaN   \n",
       "...                              ...               ...          ...   \n",
       "122229                             .                 .            .   \n",
       "122230                             .                 .            .   \n",
       "122231                             .                 .            .   \n",
       "122232                             .                 .            .   \n",
       "\n",
       "       GTEx_V6_tissue dbscSNV_ADA_SCORE dbscSNV_RF_SCORE InterVar_automated  \\\n",
       "0                 NaN               NaN              NaN                NaN   \n",
       "1                 NaN               NaN              NaN                NaN   \n",
       "2                 NaN               NaN              NaN                NaN   \n",
       "3                 NaN               NaN              NaN                NaN   \n",
       "...               ...               ...              ...                ...   \n",
       "122229              .                 .                .                  .   \n",
       "122230              .                 .                .                  .   \n",
       "122231              .                 .                .                  .   \n",
       "122232              .                 .                .                  .   \n",
       "\n",
       "       PVS1  PS1  PS2  PS3  PS4  PM1  PM2  PM3  PM4  PM5  PM6  PP1  PP2  PP3  \\\n",
       "0       NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "1       NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "2       NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "3       NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "...     ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   \n",
       "122229    .    .    .    .    .    .    .    .    .    .    .    .    .    .   \n",
       "122230    .    .    .    .    .    .    .    .    .    .    .    .    .    .   \n",
       "122231    .    .    .    .    .    .    .    .    .    .    .    .    .    .   \n",
       "122232    .    .    .    .    .    .    .    .    .    .    .    .    .    .   \n",
       "\n",
       "        PP4  PP5  BA1  BS1  BS2  BS3  BS4  BP1  BP2  BP3  BP4  BP5  BP6  BP7  \\\n",
       "0       NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "1       NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "2       NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "3       NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN   \n",
       "...     ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   \n",
       "122229    .    .    .    .    .    .    .    .    .    .    .    .    .    .   \n",
       "122230    .    .    .    .    .    .    .    .    .    .    .    .    .    .   \n",
       "122231    .    .    .    .    .    .    .    .    .    .    .    .    .    .   \n",
       "122232    .    .    .    .    .    .    .    .    .    .    .    .    .    .   \n",
       "\n",
       "       cosmic70 MCAP REVEL regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0           NaN  NaN   NaN        NaN            NaN                  NaN  \n",
       "1           NaN  NaN   NaN        NaN            NaN                  NaN  \n",
       "2           NaN  NaN   NaN        NaN            NaN                  NaN  \n",
       "3           NaN  NaN   NaN        NaN            NaN                  NaN  \n",
       "...         ...  ...   ...        ...            ...                  ...  \n",
       "122229        .    .     .          .              .                    .  \n",
       "122230        .    .     .          .              .                    .  \n",
       "122231        .    .     .          .              .                    .  \n",
       "122232        .    .     .          .              .                    .  \n",
       "\n",
       "[122233 rows x 270 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_coding_merged = human_derived_coding_merged.drop_duplicates(keep = 'first')\n",
    "human_derived_coding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### ClinVar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the Benign Coding Region for ClinVar:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100987606</td>\n",
       "      <td>100987606</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102065918</td>\n",
       "      <td>102065918</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102396271</td>\n",
       "      <td>102396271</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>0.018</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>102399466</td>\n",
       "      <td>102399466</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4655</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964016</td>\n",
       "      <td>85964016</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4656</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85978770</td>\n",
       "      <td>85978770</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0245874451416</td>\n",
       "      <td>0.248</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4657</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85978816</td>\n",
       "      <td>85978816</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>0.100</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4658</th>\n",
       "      <td>chrX</td>\n",
       "      <td>93671995</td>\n",
       "      <td>93671995</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0059993776503</td>\n",
       "      <td>0.060</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4659 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr      Start        End Ref Alt  ...             MCAP  REVEL  \\\n",
       "0     chr10  100987606  100987606   G   T  ...                .      .   \n",
       "1     chr10  102065918  102065918   C   G  ...                .      .   \n",
       "2     chr10  102396271  102396271   G   A  ...                .  0.018   \n",
       "3     chr10  102399466  102399466   C   T  ...                .      .   \n",
       "...     ...        ...        ...  ..  ..  ...              ...    ...   \n",
       "4655   chrX   85964016   85964016   T   C  ...                .      .   \n",
       "4656   chrX   85978770   85978770   G   C  ...  0.0245874451416  0.248   \n",
       "4657   chrX   85978816   85978816   T   A  ...                .  0.100   \n",
       "4658   chrX   93671995   93671995   G   C  ...  0.0059993776503  0.060   \n",
       "\n",
       "     regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0             .              .                    .  \n",
       "1             .              .                    .  \n",
       "2             .              .                    .  \n",
       "3             .              .                    .  \n",
       "...         ...            ...                  ...  \n",
       "4655          .              .                    .  \n",
       "4656          .              .                    .  \n",
       "4657          .              .                    .  \n",
       "4658          .              .                    .  \n",
       "\n",
       "[4659 rows x 149 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_coding_annotated_with_annovar = pd.read_csv('clinvar_benign_coding_for_annovar.hg38_multianno.csv')\n",
    "clinvar_benign_coding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>...</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>69.0</td>\n",
       "      <td>97.0</td>\n",
       "      <td>0.076262</td>\n",
       "      <td>2.519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9134</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.893695</td>\n",
       "      <td>10.880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9135</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.893695</td>\n",
       "      <td>10.880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9136</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.893695</td>\n",
       "      <td>10.880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9137</th>\n",
       "      <td>X</td>\n",
       "      <td>154929926</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.089486</td>\n",
       "      <td>2.669</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9138 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref   Alt Type  ...  dbscSNV-rf_score RemapOverlapTF  \\\n",
       "0         1    1014042   G     A  SNV  ...               NaN           69.0   \n",
       "1         1    1014042   G     A  SNV  ...               NaN           69.0   \n",
       "2         1    1014042   G     A  SNV  ...               NaN           69.0   \n",
       "3         1    1014042   G     A  SNV  ...               NaN           69.0   \n",
       "...     ...        ...  ..   ...  ...  ...               ...            ...   \n",
       "9134      X  154776813   C  CAAG  INS  ...               NaN            7.0   \n",
       "9135      X  154776813   C  CAAG  INS  ...               NaN            7.0   \n",
       "9136      X  154776813   C  CAAG  INS  ...               NaN            7.0   \n",
       "9137      X  154929926   T     G  SNV  ...               NaN            NaN   \n",
       "\n",
       "     RemapOverlapCL  RawScore   PHRED  \n",
       "0              97.0  0.076262   2.519  \n",
       "1              97.0  0.076262   2.519  \n",
       "2              97.0  0.076262   2.519  \n",
       "3              97.0  0.076262   2.519  \n",
       "...             ...       ...     ...  \n",
       "9134            7.0  0.893695  10.880  \n",
       "9135            7.0  0.893695  10.880  \n",
       "9136            7.0  0.893695  10.880  \n",
       "9137            NaN  0.089486   2.669  \n",
       "\n",
       "[9138 rows x 125 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_coding_annotated_with_cadd = pd.read_table('clinvar_coding_benign_cadd_annotations_noheader.tsv')\n",
    "clinvar_benign_coding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below are the steps used to combine the annotations for clinvar benign coding variants:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9134</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9135</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9136</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9137</th>\n",
       "      <td>X</td>\n",
       "      <td>154929926</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9138 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref   Alt Type  ...  MCAP REVEL regsnp_fpr  \\\n",
       "0         1    1014042   G     A  SNV  ...   NaN   NaN        NaN   \n",
       "1         1    1014042   G     A  SNV  ...   NaN   NaN        NaN   \n",
       "2         1    1014042   G     A  SNV  ...   NaN   NaN        NaN   \n",
       "3         1    1014042   G     A  SNV  ...   NaN   NaN        NaN   \n",
       "...     ...        ...  ..   ...  ...  ...   ...   ...        ...   \n",
       "9134      X  154776813   C  CAAG  INS  ...   NaN   NaN        NaN   \n",
       "9135      X  154776813   C  CAAG  INS  ...   NaN   NaN        NaN   \n",
       "9136      X  154776813   C  CAAG  INS  ...   NaN   NaN        NaN   \n",
       "9137      X  154929926   T     G  SNV  ...     .     .          .   \n",
       "\n",
       "      regsnp_disease regsnp_splicing_site  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "...              ...                  ...  \n",
       "9134             NaN                  NaN  \n",
       "9135             NaN                  NaN  \n",
       "9136             NaN                  NaN  \n",
       "9137               .                    .  \n",
       "\n",
       "[9138 rows x 270 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_coding_annotated_with_annovar['Chr'] = clinvar_benign_coding_annotated_with_annovar['Chr'].map(lambda x: x.lstrip('chr').rstrip('aAbBcC'))\n",
    "clinvar_benign_coding_annotated_with_annovar = clinvar_benign_coding_annotated_with_annovar.rename(columns = {'Chr': '#Chrom', 'Start': 'Pos'})\n",
    "clinvar_benign_coding_merged = pd.merge(clinvar_benign_coding_annotated_with_cadd, clinvar_benign_coding_annotated_with_annovar,\n",
    "                                         how = 'left', on = ['#Chrom', 'Pos', 'Ref', 'Alt'])\n",
    "clinvar_benign_coding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have dropped the duplicates of the merged dataframe:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9134</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9135</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9136</th>\n",
       "      <td>X</td>\n",
       "      <td>154776813</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAG</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9137</th>\n",
       "      <td>X</td>\n",
       "      <td>154929926</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9138 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref   Alt Type  ...  MCAP REVEL regsnp_fpr  \\\n",
       "0         1    1014042   G     A  SNV  ...   NaN   NaN        NaN   \n",
       "1         1    1014042   G     A  SNV  ...   NaN   NaN        NaN   \n",
       "2         1    1014042   G     A  SNV  ...   NaN   NaN        NaN   \n",
       "3         1    1014042   G     A  SNV  ...   NaN   NaN        NaN   \n",
       "...     ...        ...  ..   ...  ...  ...   ...   ...        ...   \n",
       "9134      X  154776813   C  CAAG  INS  ...   NaN   NaN        NaN   \n",
       "9135      X  154776813   C  CAAG  INS  ...   NaN   NaN        NaN   \n",
       "9136      X  154776813   C  CAAG  INS  ...   NaN   NaN        NaN   \n",
       "9137      X  154929926   T     G  SNV  ...     .     .          .   \n",
       "\n",
       "      regsnp_disease regsnp_splicing_site  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "...              ...                  ...  \n",
       "9134             NaN                  NaN  \n",
       "9135             NaN                  NaN  \n",
       "9136             NaN                  NaN  \n",
       "9137               .                    .  \n",
       "\n",
       "[9138 rows x 270 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_coding_merged = clinvar_benign_coding_merged.drop_duplicates(keep = 'first')\n",
    "clinvar_benign_coding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the final dataframe for all benign coding region variants:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122229</th>\n",
       "      <td>X</td>\n",
       "      <td>155260197</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122230</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122231</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122232</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>131371 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom        Pos Ref Alt Type  ...  MCAP REVEL regsnp_fpr  \\\n",
       "0           1    1014042   G   A  SNV  ...   NaN   NaN        NaN   \n",
       "1           1    1014042   G   A  SNV  ...   NaN   NaN        NaN   \n",
       "2           1    1014042   G   A  SNV  ...   NaN   NaN        NaN   \n",
       "3           1    1014042   G   A  SNV  ...   NaN   NaN        NaN   \n",
       "...       ...        ...  ..  ..  ...  ...   ...   ...        ...   \n",
       "122229      X  155260197   C   A  SNV  ...     .     .          .   \n",
       "122230      X  155612732   T   C  SNV  ...     .     .          .   \n",
       "122231      X  155612732   T   C  SNV  ...     .     .          .   \n",
       "122232      X  155612732   T   C  SNV  ...     .     .          .   \n",
       "\n",
       "        regsnp_disease regsnp_splicing_site  \n",
       "0                  NaN                  NaN  \n",
       "1                  NaN                  NaN  \n",
       "2                  NaN                  NaN  \n",
       "3                  NaN                  NaN  \n",
       "...                ...                  ...  \n",
       "122229               .                    .  \n",
       "122230               .                    .  \n",
       "122231               .                    .  \n",
       "122232               .                    .  \n",
       "\n",
       "[131371 rows x 270 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "benign_coding_region_variants = pd.concat([clinvar_benign_coding_merged, human_derived_coding_merged])\n",
    "benign_coding_region_variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We need to drop some columns that are unneccessary for our final model, shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122229</th>\n",
       "      <td>X</td>\n",
       "      <td>155260197</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122230</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122231</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122232</th>\n",
       "      <td>X</td>\n",
       "      <td>155612732</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>131371 rows × 241 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #Chrom        Pos Ref Alt Type  ...  MCAP REVEL regsnp_fpr  \\\n",
       "0           1    1014042   G   A  SNV  ...   NaN   NaN        NaN   \n",
       "1           1    1014042   G   A  SNV  ...   NaN   NaN        NaN   \n",
       "2           1    1014042   G   A  SNV  ...   NaN   NaN        NaN   \n",
       "3           1    1014042   G   A  SNV  ...   NaN   NaN        NaN   \n",
       "...       ...        ...  ..  ..  ...  ...   ...   ...        ...   \n",
       "122229      X  155260197   C   A  SNV  ...     .     .          .   \n",
       "122230      X  155612732   T   C  SNV  ...     .     .          .   \n",
       "122231      X  155612732   T   C  SNV  ...     .     .          .   \n",
       "122232      X  155612732   T   C  SNV  ...     .     .          .   \n",
       "\n",
       "        regsnp_disease regsnp_splicing_site  \n",
       "0                  NaN                  NaN  \n",
       "1                  NaN                  NaN  \n",
       "2                  NaN                  NaN  \n",
       "3                  NaN                  NaN  \n",
       "...                ...                  ...  \n",
       "122229               .                    .  \n",
       "122230               .                    .  \n",
       "122231               .                    .  \n",
       "122232               .                    .  \n",
       "\n",
       "[131371 rows x 241 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "benign_coding_region_variants = benign_coding_region_variants.drop(columns = ['PolyPhenVal', 'priPhCons', 'cHmm_E1', 'cHmm_E2', 'cHmm_E3', 'cHmm_E4', 'cHmm_E5', 'cHmm_E6', 'cHmm_E7', 'cHmm_E8', 'cHmm_E9', 'cHmm_E10', 'cHmm_E11', 'cHmm_E12', 'cHmm_E13', 'cHmm_E14', 'cHmm_E15', 'cHmm_E16', 'cHmm_E17', 'cHmm_E18', 'cHmm_E19', 'cHmm_E20', 'cHmm_E21', 'cHmm_E22', 'cHmm_E23', 'cHmm_E24', 'cHmm_E25', 'GerpN', 'GerpS'])\n",
    "benign_coding_region_variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following code changes all scores and other numerical outputs to floats from objects for both CADD and ANNOVAR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_coding_region_variants_with_nan = benign_coding_region_variants.replace('.', np.nan)\n",
    "benign_coding_region_variants_with_nan = benign_coding_region_variants_with_nan.astype({'SIFT_score': float, 'Polyphen2_HDIV_score' : float, \n",
    "                'Polyphen2_HVAR_score' : float, 'LRT_score' : float, \n",
    "                'MutationTaster_score' : float, 'MutationAssessor_score' : float, \n",
    "                'FATHMM_score' : float, 'RadialSVM_score' : float,\n",
    "               'LR_score' : float, 'VEST3_score' : float, 'CADD_raw' : float,\n",
    "               'CADD_phred' : float, 'GERP++_RS' : float, 'phyloP46way_placental' : float,\n",
    "               'phyloP100way_vertebrate' : float, 'SiPhy_29way_logOdds' : float, 'SIFT_score.1' : float,\n",
    "               'SIFT_converted_rankscore' : float, 'Polyphen2_HDIV_score.1' : float, 'Polyphen2_HDIV_rankscore' : float,\n",
    "               'Polyphen2_HVAR_score.1' : float, 'Polyphen2_HVAR_rankscore' : float, 'LRT_score.1' : float,\n",
    "               'LRT_converted_rankscore' : float, 'MutationTaster_score.1' : float, 'MutationTaster_converted_rankscore' : float,\n",
    "               'MutationAssessor_score.1' : float, 'MutationAssessor_score_rankscore' : float, 'FATHMM_score.1' : float,\n",
    "               'FATHMM_converted_rankscore' : float, 'PROVEAN_score' : float, 'PROVEAN_converted_rankscore' : float,\n",
    "               'VEST3_score.1' : float, 'VEST3_rankscore' : float, 'MetaSVM_score' : float, 'MetaSVM_rankscore' : float,\n",
    "               'MetaLR_score' : float, 'MetaLR_rankscore' : float, 'M-CAP_score' : float, 'M-CAP_rankscore' : float,\n",
    "               'CADD_raw.1' : float, 'CADD_raw_rankscore' : float, 'CADD_phred.1' : float, 'DANN_score' : float,\n",
    "               'DANN_rankscore' : float, 'fathmm-MKL_coding_score' : float, 'fathmm-MKL_coding_rankscore' : float,\n",
    "               'Eigen-raw' : float, 'Eigen-PC-raw' : float, 'GenoCanyon_score' : float, 'GenoCanyon_score_rankscore' : float,\n",
    "               'integrated_fitCons_score' : float, 'integrated_fitCons_score_rankscore' : float, 'integrated_confidence_value' : float,\n",
    "               'GERP++_RS.1' : float, 'GERP++_RS_rankscore' : float, 'phyloP100way_vertebrate.1' : float, \n",
    "                'phyloP100way_vertebrate_rankscore' : float, 'phyloP20way_mammalian' : float, 'phyloP20way_mammalian_rankscore' : float,\n",
    "               'phastCons100way_vertebrate' : float, 'phastCons100way_vertebrate_rankscore' : float, 'phastCons20way_mammalian' : float,\n",
    "               'phastCons20way_mammalian_rankscore' : float, 'SiPhy_29way_logOdds.1' : float, 'SiPhy_29way_logOdds_rankscore' : float,\n",
    "               'dbscSNV_ADA_SCORE' : float, 'dbscSNV_RF_SCORE' : float, 'PVS1' : float, 'PS1' : float, 'PS2' : float, 'PS3' : float,\n",
    "               'PS4' : float, 'PM1' : float, 'PM2' : float, 'PM3' : float, 'PM4' : float, 'PM5' : float, 'PM6' : float, 'PP1' : float, 'PP2' : float,\n",
    "               'PP3' : float, 'PP4' : float, 'PP5' : float, 'BA1' : float, 'BS1' : float, 'BS2' : float, 'BS3' : float, 'BS4' : float, 'BP1' : float, 'BP2' : float,\n",
    "               'BP3' : float, 'BP4' : float, 'BP5' : float, 'BP6' : float, 'BP7' : float, 'MCAP' : float, 'REVEL' : float, 'regsnp_fpr' : float,\n",
    "                'dbscSNV-rf_score' : float})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have saved the dataframe for use in other Notebooks:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_coding_region_variants_with_nan.to_csv('benign_coding_region_variants.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Benign Noncoding Region:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Before running the next cell, you must execute the following commands in terminal:\n",
    "1. pip install dask==1.0.0\n",
    "2. pip install toolz\n",
    "3. pip install dask[dataframe]\n",
    "\n",
    "This gives us the ability to use dask, which we will need to read in our large annotated CADD files for the human derived data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import dask.dataframe as dd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import dask.dataframe as dd\n",
    "import matplotlib.pyplot as plt\n",
    "pd.options.display.max_rows = 8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have read in the ANNOVAR annotations using a chunking method, which takes a small sample of files at a time and then groups them together, using less memory than if read all at once using pandas:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000013</td>\n",
       "      <td>1000013</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100020652</td>\n",
       "      <td>100020652</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>DNMBP;CPN1</td>\n",
       "      <td>dist=10699;dist=21656</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000297</td>\n",
       "      <td>1000297</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>1000555</td>\n",
       "      <td>1000555</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GTPBP4</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.482876712329</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966378</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931817</td>\n",
       "      <td>9931817</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SHROOM2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966379</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931818</td>\n",
       "      <td>9931818</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SHROOM2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966380</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9931993</td>\n",
       "      <td>9931993</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SHROOM2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.438356164384</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>966381</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9932000</td>\n",
       "      <td>9932000</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>SHROOM2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.794520547945</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>966382 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       chr10    1000013    1000013   G   A     intronic       GTPBP4   \n",
       "1       chr10  100020652  100020652   G   A   intergenic   DNMBP;CPN1   \n",
       "2       chr10    1000297    1000297   T   G     intronic       GTPBP4   \n",
       "3       chr10    1000555    1000555   A   T     intronic       GTPBP4   \n",
       "...       ...        ...        ...  ..  ..          ...          ...   \n",
       "966378   chrX    9931817    9931817   T   C     intronic      SHROOM2   \n",
       "966379   chrX    9931818    9931818   G   A     intronic      SHROOM2   \n",
       "966380   chrX    9931993    9931993   T   C     intronic      SHROOM2   \n",
       "966381   chrX    9932000    9932000   C   T     intronic      SHROOM2   \n",
       "\n",
       "           GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  ... BP4  \\\n",
       "0                           .                  .                .  ...   .   \n",
       "1       dist=10699;dist=21656                  .                .  ...   .   \n",
       "2                           .                  .                .  ...   .   \n",
       "3                           .                  .                .  ...   .   \n",
       "...                       ...                ...              ...  ...  ..   \n",
       "966378                      .                  .                .  ...   .   \n",
       "966379                      .                  .                .  ...   .   \n",
       "966380                      .                  .                .  ...   .   \n",
       "966381                      .                  .                .  ...   .   \n",
       "\n",
       "       BP5 BP6 BP7 cosmic70 MCAP REVEL      regsnp_fpr regsnp_disease  \\\n",
       "0        .   .   .        .    .     .               .              .   \n",
       "1        .   .   .        .    .     .               .              .   \n",
       "2        .   .   .        .    .     .               .              .   \n",
       "3        .   .   .        .    .     .  0.482876712329              B   \n",
       "...     ..  ..  ..      ...  ...   ...             ...            ...   \n",
       "966378   .   .   .        .    .     .               .              .   \n",
       "966379   .   .   .        .    .     .               .              .   \n",
       "966380   .   .   .        .    .     .  0.438356164384              B   \n",
       "966381   .   .   .        .    .     .  0.794520547945              B   \n",
       "\n",
       "       regsnp_splicing_site  \n",
       "0                         .  \n",
       "1                         .  \n",
       "2                         .  \n",
       "3                       off  \n",
       "...                     ...  \n",
       "966378                    .  \n",
       "966379                    .  \n",
       "966380                  off  \n",
       "966381                  off  \n",
       "\n",
       "[966382 rows x 149 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_list = []\n",
    "for chunk in pd.read_csv('human_derived_noncoding_annotated.hg38_multianno.csv', chunksize = 5000, low_memory = False):\n",
    "    my_list.append(chunk)\n",
    "human_derived_noncoding_annotated_with_annovar = pd.concat(my_list, axis = 0)\n",
    "del my_list\n",
    "human_derived_noncoding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have cut out 'chr' from the chromosome values and changed the column names of the ANNOVAR annotation dataframe so the two annotation files can be merged:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated_with_annovar['Chr'] = human_derived_noncoding_annotated_with_annovar['Chr'].map(lambda x: x.lstrip('chr').rstrip('aAbBcC'))\n",
    "human_derived_noncoding_annotated_with_annovar = human_derived_noncoding_annotated_with_annovar.rename(columns = {'Chr': '#Chrom', 'Start': 'Pos'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have pickled the above dataframe so that I do not have to call the above functions again:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated_with_annovar.to_pickle('human_derived_noncoding_annotated_with_annovar.pkl')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I have already changed the columns of the ANNOVAR annotation dataframe (Start -> Pos, Chr -> #Chrom) and cut out the string 'chr' from the dataframe, like I had to do with all the other dataframes. I then saved the resulting dataframe and have imported it as 'human_derived_noncoding_annotated_with_annovar.pkl' seen below. In order to merge the CADD and ANNOVAR annotations together, I also needed to change the values of the dataframe to strings, seen below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_noncoding_annotated_with_annovar = pd.read_pickle('human_derived_noncoding_annotated_with_annovar.pkl')\n",
    "human_derived_noncoding_annotated_with_annovar = human_derived_noncoding_annotated_with_annovar.astype(str)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the method used to create a single csv file with all the annotations from CADD. I have read the files into a dataframe and then sent them all to the same csv file using the extension \"mode = 'a'\" which appends the next annotation file to the same output. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = pd.read_csv('GRCh38-v1.5_anno_0265e2d84faa1b92f64be0ad04982d58_noheader.tsv', sep = '\\t')\n",
    "df1.to_csv('annnotationforhumanderivedcadd.csv', index=False)\n",
    "\n",
    "df2 = pd.read_csv('GRCh38-v1.5_anno_3c0e5f59d9c7d35f1d1311158fac67f3_noheader.tsv', sep = '\\t')\n",
    "df2.to_csv('annnotationforhumanderivedcadd.csv', mode = 'a', index=False)\n",
    "\n",
    "df3= pd.read_csv('GRCh38-v1.5_anno_4b0464a325358ba5cace1e505fbda034_noheader.tsv', sep = '\\t')\n",
    "df3.to_csv('annnotationforhumanderivedcadd.csv', mode = 'a', index=False)\n",
    "\n",
    "df4 = pd.read_csv('GRCh38-v1.5_anno_76ecb0e4f43d81dd38abdf89fd16850f_noheader.tsv', sep = '\\t')\n",
    "df4.to_csv('annnotationforhumanderivedcadd.csv', mode = 'a', index=False)\n",
    "\n",
    "df5 = pd.read_csv('GRCh38-v1.5_anno_85ec7a65fce6a2cf5c60035e2cdc5a14_noheader.tsv', sep = '\\t')\n",
    "df5.to_csv('annnotationforhumanderivedcadd.csv', mode = 'a', index=False)\n",
    "\n",
    "df6 = pd.read_csv('GRCh38-v1.5_anno_8f9e07c972f3b44de92c82a49704f297_noheader.tsv', sep = '\\t')\n",
    "df6.to_csv('annnotationforhumanderivedcadd.csv', mode = 'a', index=False)\n",
    "\n",
    "df7 = pd.read_csv('GRCh38-v1.5_anno_ab4b1db57087c3537ff7101b7cbd4eaf_noheader.tsv', sep = '\\t')\n",
    "df7.to_csv('annnotationforhumanderivedcadd.csv', mode = 'a', index=False)\n",
    "\n",
    "df8 = pd.read_csv('GRCh38-v1.5_anno_bb2bc990f3ac17f338ddfaeac5a643dd_noheader.tsv', sep = '\\t')\n",
    "df8.to_csv('annnotationforhumanderivedcadd.csv', mode = 'a', index=False)\n",
    "\n",
    "df9 = pd.read_csv('GRCh38-v1.5_anno_bc3fb6e7bed9d35196a24000cbba11d3_noheader.tsv', sep = '\\t')\n",
    "df9.to_csv('annnotationforhumanderivedcadd.csv', mode = 'a', index=False)\n",
    "\n",
    "df10 = pd.read_csv('GRCh38-v1.5_anno_caed4a68cd64fff0da36fe81fa3b5e85_noheader.tsv', sep = '\\t')\n",
    "df10.to_csv('annnotationforhumanderivedcadd.csv', mode = 'a', index=False)\n",
    "\n",
    "df11 = pd.read_csv('GRCh38-v1.5_anno_d334b44948c3cbc6067e6a0146556550_noheader.tsv', sep = '\\t')\n",
    "df11.to_csv('annnotationforhumanderivedcadd.csv', mode = 'a', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since we are using very large dataframes, we must use dask dataframe, which we called in above. The below code gives us an output of a new csv file which is the merged dataframe of the two annotation files. The 'converters' part of the code below changes the values of the CADD annotation to strings just as we did for the ANNOVAR values above. The output file is given at the end and is 1.3 GB large."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "MemoryError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/opt/miniconda3/lib/python3.7/site-packages/dask/base.py\u001b[0m in \u001b[0;36mnormalize_array\u001b[0;34m(x)\u001b[0m\n\u001b[1;32m    730\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 731\u001b[0;31m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhash_buffer_hex\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'-'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflat\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencode\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'utf-8'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    732\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: sequence item 966382: expected str instance, int found",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mMemoryError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-8-4bc8f5a5688f>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"annnotationforhumanderivedcadd.csv\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mconverters\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mstr\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m250\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhuman_derived_noncoding_annotated_with_annovar\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'left'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'#Chrom'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'Pos'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'Ref'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'Alt'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompute\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'human_derived_noncoding_merged.csv'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/opt/miniconda3/lib/python3.7/site-packages/dask/dataframe/multi.py\u001b[0m in \u001b[0;36mmerge\u001b[0;34m(left, right, how, on, left_on, right_on, left_index, right_index, suffixes, indicator, npartitions, shuffle, max_branch)\u001b[0m\n\u001b[1;32m    365\u001b[0m             \u001b[0mright_on\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    366\u001b[0m             \u001b[0mright_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 367\u001b[0;31m         \u001b[0mright\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfrom_pandas\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mright\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnpartitions\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# turn into DataFrame\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    368\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    369\u001b[0m     \u001b[0;31m# Both sides are now dd.DataFrame or dd.Series objects\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/lib/python3.7/site-packages/dask/dataframe/io/io.py\u001b[0m in \u001b[0;36mfrom_pandas\u001b[0;34m(data, npartitions, chunksize, sort, name)\u001b[0m\n\u001b[1;32m    179\u001b[0m         \u001b[0mnpartitions\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mceil\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnrows\u001b[0m \u001b[0;34m/\u001b[0m \u001b[0mchunksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    180\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 181\u001b[0;31m     \u001b[0mname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0;34m'from_pandas-'\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mtokenize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mchunksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    182\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    183\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mnrows\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/lib/python3.7/site-packages/dask/base.py\u001b[0m in \u001b[0;36mtokenize\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    598\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mkwargs\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    599\u001b[0m         \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0margs\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 600\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mmd5\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnormalize_token\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencode\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhexdigest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    601\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    602\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/lib/python3.7/site-packages/dask/utils.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, arg, *args, **kwargs)\u001b[0m\n\u001b[1;32m    411\u001b[0m         \"\"\"\n\u001b[1;32m    412\u001b[0m         \u001b[0mmeth\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdispatch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 413\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mmeth\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    414\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    415\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mproperty\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/lib/python3.7/site-packages/dask/base.py\u001b[0m in \u001b[0;36mnormalize_dataframe\u001b[0;34m(df)\u001b[0m\n\u001b[1;32m    708\u001b[0m         \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mblock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mblock\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mblocks\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    709\u001b[0m         \u001b[0mdata\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 710\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnormalize_token\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    711\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    712\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/lib/python3.7/site-packages/dask/utils.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, arg, *args, **kwargs)\u001b[0m\n\u001b[1;32m    411\u001b[0m         \"\"\"\n\u001b[1;32m    412\u001b[0m         \u001b[0mmeth\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdispatch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 413\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mmeth\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    414\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    415\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mproperty\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/miniconda3/lib/python3.7/site-packages/dask/base.py\u001b[0m in \u001b[0;36mnormalize_array\u001b[0;34m(x)\u001b[0m\n\u001b[1;32m    732\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    733\u001b[0m                 data = hash_buffer_hex(b'-'.join([unicode(item).encode('utf-8') for item in\n\u001b[0;32m--> 734\u001b[0;31m                                                   x.flat]))\n\u001b[0m\u001b[1;32m    735\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    736\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mMemoryError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "dd.merge(dd.read_csv(\"annnotationforhumanderivedcadd.csv\", converters={i: str for i in range(250)}), human_derived_noncoding_annotated_with_annovar, how='left', on=['#Chrom', 'Pos', 'Ref','Alt']).compute().to_csv('human_derived_noncoding_merged.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have called in the csv file and had to specify the dtype of all the values so that they could be read by dask and dropped the duplicate rows from the dask dataframe:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><strong>Dask DataFrame Structure:</strong></div>\n",
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEName</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>motifEScoreChng</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>FeatureID</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>Intron</th>\n",
       "      <th>Exon</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>relcDNApos</th>\n",
       "      <th>CDSpos</th>\n",
       "      <th>relCDSpos</th>\n",
       "      <th>protPos</th>\n",
       "      <th>relProtPos</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Dst2Splice</th>\n",
       "      <th>Dst2SplType</th>\n",
       "      <th>minDistTSS</th>\n",
       "      <th>minDistTSE</th>\n",
       "      <th>SIFTcat</th>\n",
       "      <th>SIFTval</th>\n",
       "      <th>PolyPhenCat</th>\n",
       "      <th>PolyPhenVal</th>\n",
       "      <th>priPhCons</th>\n",
       "      <th>mamPhCons</th>\n",
       "      <th>verPhCons</th>\n",
       "      <th>priPhyloP</th>\n",
       "      <th>mamPhyloP</th>\n",
       "      <th>verPhyloP</th>\n",
       "      <th>bStatistic</th>\n",
       "      <th>targetScan</th>\n",
       "      <th>mirSVR-Score</th>\n",
       "      <th>mirSVR-E</th>\n",
       "      <th>mirSVR-Aln</th>\n",
       "      <th>cHmm_E1</th>\n",
       "      <th>cHmm_E2</th>\n",
       "      <th>cHmm_E3</th>\n",
       "      <th>cHmm_E4</th>\n",
       "      <th>cHmm_E5</th>\n",
       "      <th>cHmm_E6</th>\n",
       "      <th>cHmm_E7</th>\n",
       "      <th>cHmm_E8</th>\n",
       "      <th>cHmm_E9</th>\n",
       "      <th>cHmm_E10</th>\n",
       "      <th>cHmm_E11</th>\n",
       "      <th>cHmm_E12</th>\n",
       "      <th>cHmm_E13</th>\n",
       "      <th>cHmm_E14</th>\n",
       "      <th>cHmm_E15</th>\n",
       "      <th>cHmm_E16</th>\n",
       "      <th>cHmm_E17</th>\n",
       "      <th>cHmm_E18</th>\n",
       "      <th>cHmm_E19</th>\n",
       "      <th>cHmm_E20</th>\n",
       "      <th>cHmm_E21</th>\n",
       "      <th>cHmm_E22</th>\n",
       "      <th>cHmm_E23</th>\n",
       "      <th>cHmm_E24</th>\n",
       "      <th>cHmm_E25</th>\n",
       "      <th>GerpRS</th>\n",
       "      <th>GerpRSpval</th>\n",
       "      <th>GerpN</th>\n",
       "      <th>GerpS</th>\n",
       "      <th>tOverlapMotifs</th>\n",
       "      <th>motifDist</th>\n",
       "      <th>EncodeH3K4me1-sum</th>\n",
       "      <th>EncodeH3K4me1-max</th>\n",
       "      <th>EncodeH3K4me2-sum</th>\n",
       "      <th>EncodeH3K4me2-max</th>\n",
       "      <th>EncodeH3K4me3-sum</th>\n",
       "      <th>EncodeH3K4me3-max</th>\n",
       "      <th>EncodeH3K9ac-sum</th>\n",
       "      <th>EncodeH3K9ac-max</th>\n",
       "      <th>EncodeH3K9me3-sum</th>\n",
       "      <th>EncodeH3K9me3-max</th>\n",
       "      <th>EncodeH3K27ac-sum</th>\n",
       "      <th>EncodeH3K27ac-max</th>\n",
       "      <th>EncodeH3K27me3-sum</th>\n",
       "      <th>EncodeH3K27me3-max</th>\n",
       "      <th>EncodeH3K36me3-sum</th>\n",
       "      <th>EncodeH3K36me3-max</th>\n",
       "      <th>EncodeH3K79me2-sum</th>\n",
       "      <th>EncodeH3K79me2-max</th>\n",
       "      <th>EncodeH4K20me1-sum</th>\n",
       "      <th>EncodeH4K20me1-max</th>\n",
       "      <th>EncodeH2AFZ-sum</th>\n",
       "      <th>EncodeH2AFZ-max</th>\n",
       "      <th>EncodeDNase-sum</th>\n",
       "      <th>EncodeDNase-max</th>\n",
       "      <th>EncodetotalRNA-sum</th>\n",
       "      <th>EncodetotalRNA-max</th>\n",
       "      <th>Grantham</th>\n",
       "      <th>Dist2Mutation</th>\n",
       "      <th>Freq100bp</th>\n",
       "      <th>Rare100bp</th>\n",
       "      <th>Sngl100bp</th>\n",
       "      <th>Freq1000bp</th>\n",
       "      <th>Rare1000bp</th>\n",
       "      <th>Sngl1000bp</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "      <th>End</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>Func.knownGene</th>\n",
       "      <th>Gene.knownGene</th>\n",
       "      <th>GeneDetail.knownGene</th>\n",
       "      <th>ExonicFunc.knownGene</th>\n",
       "      <th>AAChange.knownGene</th>\n",
       "      <th>Func.ensGene</th>\n",
       "      <th>Gene.ensGene</th>\n",
       "      <th>GeneDetail.ensGene</th>\n",
       "      <th>ExonicFunc.ensGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>SIFT_pred</th>\n",
       "      <th>Polyphen2_HDIV_score</th>\n",
       "      <th>Polyphen2_HDIV_pred</th>\n",
       "      <th>Polyphen2_HVAR_score</th>\n",
       "      <th>Polyphen2_HVAR_pred</th>\n",
       "      <th>LRT_score</th>\n",
       "      <th>LRT_pred</th>\n",
       "      <th>MutationTaster_score</th>\n",
       "      <th>MutationTaster_pred</th>\n",
       "      <th>MutationAssessor_score</th>\n",
       "      <th>MutationAssessor_pred</th>\n",
       "      <th>FATHMM_score</th>\n",
       "      <th>FATHMM_pred</th>\n",
       "      <th>RadialSVM_score</th>\n",
       "      <th>RadialSVM_pred</th>\n",
       "      <th>LR_score</th>\n",
       "      <th>LR_pred</th>\n",
       "      <th>VEST3_score</th>\n",
       "      <th>CADD_raw</th>\n",
       "      <th>CADD_phred</th>\n",
       "      <th>GERP++_RS</th>\n",
       "      <th>phyloP46way_placental</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>Interpro_domain</th>\n",
       "      <th>SIFT_score.1</th>\n",
       "      <th>SIFT_converted_rankscore</th>\n",
       "      <th>SIFT_pred.1</th>\n",
       "      <th>Polyphen2_HDIV_score.1</th>\n",
       "      <th>Polyphen2_HDIV_rankscore</th>\n",
       "      <th>Polyphen2_HDIV_pred.1</th>\n",
       "      <th>Polyphen2_HVAR_score.1</th>\n",
       "      <th>Polyphen2_HVAR_rankscore</th>\n",
       "      <th>Polyphen2_HVAR_pred.1</th>\n",
       "      <th>LRT_score.1</th>\n",
       "      <th>LRT_converted_rankscore</th>\n",
       "      <th>LRT_pred.1</th>\n",
       "      <th>MutationTaster_score.1</th>\n",
       "      <th>MutationTaster_converted_rankscore</th>\n",
       "      <th>MutationTaster_pred.1</th>\n",
       "      <th>MutationAssessor_score.1</th>\n",
       "      <th>MutationAssessor_score_rankscore</th>\n",
       "      <th>MutationAssessor_pred.1</th>\n",
       "      <th>FATHMM_score.1</th>\n",
       "      <th>FATHMM_converted_rankscore</th>\n",
       "      <th>FATHMM_pred.1</th>\n",
       "      <th>PROVEAN_score</th>\n",
       "      <th>PROVEAN_converted_rankscore</th>\n",
       "      <th>PROVEAN_pred</th>\n",
       "      <th>VEST3_score.1</th>\n",
       "      <th>VEST3_rankscore</th>\n",
       "      <th>MetaSVM_score</th>\n",
       "      <th>MetaSVM_rankscore</th>\n",
       "      <th>MetaSVM_pred</th>\n",
       "      <th>MetaLR_score</th>\n",
       "      <th>MetaLR_rankscore</th>\n",
       "      <th>MetaLR_pred</th>\n",
       "      <th>M-CAP_score</th>\n",
       "      <th>M-CAP_rankscore</th>\n",
       "      <th>M-CAP_pred</th>\n",
       "      <th>CADD_raw.1</th>\n",
       "      <th>CADD_raw_rankscore</th>\n",
       "      <th>CADD_phred.1</th>\n",
       "      <th>DANN_score</th>\n",
       "      <th>DANN_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_score</th>\n",
       "      <th>fathmm-MKL_coding_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_pred</th>\n",
       "      <th>Eigen_coding_or_noncoding</th>\n",
       "      <th>Eigen-raw</th>\n",
       "      <th>Eigen-PC-raw</th>\n",
       "      <th>GenoCanyon_score</th>\n",
       "      <th>GenoCanyon_score_rankscore</th>\n",
       "      <th>integrated_fitCons_score</th>\n",
       "      <th>integrated_fitCons_score_rankscore</th>\n",
       "      <th>integrated_confidence_value</th>\n",
       "      <th>GERP++_RS.1</th>\n",
       "      <th>GERP++_RS_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate.1</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP20way_mammalian</th>\n",
       "      <th>phyloP20way_mammalian_rankscore</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons20way_mammalian</th>\n",
       "      <th>phastCons20way_mammalian_rankscore</th>\n",
       "      <th>SiPhy_29way_logOdds.1</th>\n",
       "      <th>SiPhy_29way_logOdds_rankscore</th>\n",
       "      <th>Interpro_domain.1</th>\n",
       "      <th>GTEx_V6_gene</th>\n",
       "      <th>GTEx_V6_tissue</th>\n",
       "      <th>dbscSNV_ADA_SCORE</th>\n",
       "      <th>dbscSNV_RF_SCORE</th>\n",
       "      <th>InterVar_automated</th>\n",
       "      <th>PVS1</th>\n",
       "      <th>PS1</th>\n",
       "      <th>PS2</th>\n",
       "      <th>PS3</th>\n",
       "      <th>PS4</th>\n",
       "      <th>PM1</th>\n",
       "      <th>PM2</th>\n",
       "      <th>PM3</th>\n",
       "      <th>PM4</th>\n",
       "      <th>PM5</th>\n",
       "      <th>PM6</th>\n",
       "      <th>PP1</th>\n",
       "      <th>PP2</th>\n",
       "      <th>PP3</th>\n",
       "      <th>PP4</th>\n",
       "      <th>PP5</th>\n",
       "      <th>BA1</th>\n",
       "      <th>BS1</th>\n",
       "      <th>BS2</th>\n",
       "      <th>BS3</th>\n",
       "      <th>BS4</th>\n",
       "      <th>BP1</th>\n",
       "      <th>BP2</th>\n",
       "      <th>BP3</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>npartitions=1</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <td>int64</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>float64</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "<div>Dask Name: drop-duplicates-agg, 162 tasks</div>"
      ],
      "text/plain": [
       "Dask DataFrame Structure:\n",
       "              Unnamed: 0  #Chrom     Pos     Ref     Alt    Type  Length AnnoType Consequence ConsScore ConsDetail      GC     CpG motifECount motifEName motifEHIPos motifEScoreChng     oAA     nAA  GeneID FeatureID GeneName    CCDS  Intron    Exon cDNApos relcDNApos  CDSpos relCDSpos protPos relProtPos  Domain Dst2Splice Dst2SplType minDistTSS minDistTSE SIFTcat SIFTval PolyPhenCat PolyPhenVal priPhCons mamPhCons verPhCons priPhyloP mamPhyloP verPhyloP bStatistic targetScan mirSVR-Score mirSVR-E mirSVR-Aln cHmm_E1 cHmm_E2 cHmm_E3 cHmm_E4 cHmm_E5 cHmm_E6 cHmm_E7 cHmm_E8 cHmm_E9 cHmm_E10 cHmm_E11 cHmm_E12 cHmm_E13 cHmm_E14 cHmm_E15 cHmm_E16 cHmm_E17 cHmm_E18 cHmm_E19 cHmm_E20 cHmm_E21 cHmm_E22 cHmm_E23 cHmm_E24 cHmm_E25  GerpRS GerpRSpval   GerpN   GerpS tOverlapMotifs motifDist EncodeH3K4me1-sum EncodeH3K4me1-max EncodeH3K4me2-sum EncodeH3K4me2-max EncodeH3K4me3-sum EncodeH3K4me3-max EncodeH3K9ac-sum EncodeH3K9ac-max EncodeH3K9me3-sum EncodeH3K9me3-max EncodeH3K27ac-sum EncodeH3K27ac-max EncodeH3K27me3-sum EncodeH3K27me3-max EncodeH3K36me3-sum EncodeH3K36me3-max EncodeH3K79me2-sum EncodeH3K79me2-max EncodeH4K20me1-sum EncodeH4K20me1-max EncodeH2AFZ-sum EncodeH2AFZ-max EncodeDNase-sum EncodeDNase-max EncodetotalRNA-sum EncodetotalRNA-max Grantham Dist2Mutation Freq100bp Rare100bp Sngl100bp Freq1000bp Rare1000bp Sngl1000bp Freq10000bp Rare10000bp Sngl10000bp EnsembleRegulatoryFeature dbscSNV-ada_score dbscSNV-rf_score RemapOverlapTF RemapOverlapCL RawScore   PHRED      End Func.refGene Gene.refGene GeneDetail.refGene ExonicFunc.refGene AAChange.refGene Func.knownGene Gene.knownGene GeneDetail.knownGene ExonicFunc.knownGene AAChange.knownGene Func.ensGene Gene.ensGene GeneDetail.ensGene ExonicFunc.ensGene AAChange.ensGene SIFT_score SIFT_pred Polyphen2_HDIV_score Polyphen2_HDIV_pred Polyphen2_HVAR_score Polyphen2_HVAR_pred LRT_score LRT_pred MutationTaster_score MutationTaster_pred MutationAssessor_score MutationAssessor_pred FATHMM_score FATHMM_pred RadialSVM_score RadialSVM_pred LR_score LR_pred VEST3_score CADD_raw CADD_phred GERP++_RS phyloP46way_placental phyloP100way_vertebrate SiPhy_29way_logOdds Interpro_domain SIFT_score.1 SIFT_converted_rankscore SIFT_pred.1 Polyphen2_HDIV_score.1 Polyphen2_HDIV_rankscore Polyphen2_HDIV_pred.1 Polyphen2_HVAR_score.1 Polyphen2_HVAR_rankscore Polyphen2_HVAR_pred.1 LRT_score.1 LRT_converted_rankscore LRT_pred.1 MutationTaster_score.1 MutationTaster_converted_rankscore MutationTaster_pred.1 MutationAssessor_score.1 MutationAssessor_score_rankscore MutationAssessor_pred.1 FATHMM_score.1 FATHMM_converted_rankscore FATHMM_pred.1 PROVEAN_score PROVEAN_converted_rankscore PROVEAN_pred VEST3_score.1 VEST3_rankscore MetaSVM_score MetaSVM_rankscore MetaSVM_pred MetaLR_score MetaLR_rankscore MetaLR_pred M-CAP_score M-CAP_rankscore M-CAP_pred CADD_raw.1 CADD_raw_rankscore CADD_phred.1 DANN_score DANN_rankscore fathmm-MKL_coding_score fathmm-MKL_coding_rankscore fathmm-MKL_coding_pred Eigen_coding_or_noncoding Eigen-raw Eigen-PC-raw GenoCanyon_score GenoCanyon_score_rankscore integrated_fitCons_score integrated_fitCons_score_rankscore integrated_confidence_value GERP++_RS.1 GERP++_RS_rankscore phyloP100way_vertebrate.1 phyloP100way_vertebrate_rankscore phyloP20way_mammalian phyloP20way_mammalian_rankscore phastCons100way_vertebrate phastCons100way_vertebrate_rankscore phastCons20way_mammalian phastCons20way_mammalian_rankscore SiPhy_29way_logOdds.1 SiPhy_29way_logOdds_rankscore Interpro_domain.1 GTEx_V6_gene GTEx_V6_tissue dbscSNV_ADA_SCORE dbscSNV_RF_SCORE InterVar_automated    PVS1     PS1     PS2     PS3     PS4     PM1     PM2     PM3     PM4     PM5     PM6     PP1     PP2     PP3     PP4     PP5     BA1     BS1     BS2     BS3     BS4     BP1     BP2     BP3     BP4     BP5     BP6     BP7 cosmic70    MCAP   REVEL regsnp_fpr regsnp_disease regsnp_splicing_site\n",
       "npartitions=1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "                   int64  object  object  object  object  object  object   object      object    object     object  object  object      object     object      object          object  object  object  object    object   object  object  object  object  object     object  object    object  object     object  object     object      object     object     object  object  object      object      object    object    object    object    object    object    object     object     object       object   object     object  object  object  object  object  object  object  object  object  object   object   object   object   object   object   object   object   object   object   object   object   object   object   object   object   object  object     object  object  object         object    object            object            object            object            object            object            object           object           object            object            object            object            object             object             object             object             object             object             object             object             object          object          object          object          object             object             object   object        object    object    object    object     object     object     object      object      object      object                    object            object           object         object         object   object  object  float64       object       object             object             object           object         object         object               object               object             object       object       object             object             object           object     object    object               object              object               object              object    object   object               object              object                 object                object       object      object          object         object   object  object      object   object     object    object                object                  object              object          object       object                   object      object                 object                   object                object                 object                   object                object      object                  object     object                 object                             object                object                   object                           object                  object         object                     object        object        object                      object       object        object          object        object            object       object       object           object      object      object          object     object     object             object       object     object         object                  object                      object                 object                    object    object       object           object                     object                   object                             object                      object      object              object                    object                            object                object                          object                     object                               object                   object                             object                object                        object            object       object         object            object           object             object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object  object   object  object  object     object         object               object\n",
       "                     ...     ...     ...     ...     ...     ...     ...      ...         ...       ...        ...     ...     ...         ...        ...         ...             ...     ...     ...     ...       ...      ...     ...     ...     ...     ...        ...     ...       ...     ...        ...     ...        ...         ...        ...        ...     ...     ...         ...         ...       ...       ...       ...       ...       ...       ...        ...        ...          ...      ...        ...     ...     ...     ...     ...     ...     ...     ...     ...     ...      ...      ...      ...      ...      ...      ...      ...      ...      ...      ...      ...      ...      ...      ...      ...      ...     ...        ...     ...     ...            ...       ...               ...               ...               ...               ...               ...               ...              ...              ...               ...               ...               ...               ...                ...                ...                ...                ...                ...                ...                ...                ...             ...             ...             ...             ...                ...                ...      ...           ...       ...       ...       ...        ...        ...        ...         ...         ...         ...                       ...               ...              ...            ...            ...      ...     ...      ...          ...          ...                ...                ...              ...            ...            ...                  ...                  ...                ...          ...          ...                ...                ...              ...        ...       ...                  ...                 ...                  ...                 ...       ...      ...                  ...                 ...                    ...                   ...          ...         ...             ...            ...      ...     ...         ...      ...        ...       ...                   ...                     ...                 ...             ...          ...                      ...         ...                    ...                      ...                   ...                    ...                      ...                   ...         ...                     ...        ...                    ...                                ...                   ...                      ...                              ...                     ...            ...                        ...           ...           ...                         ...          ...           ...             ...           ...               ...          ...          ...              ...         ...         ...             ...        ...        ...                ...          ...        ...            ...                     ...                         ...                    ...                       ...       ...          ...              ...                        ...                      ...                                ...                         ...         ...                 ...                       ...                               ...                   ...                             ...                        ...                                  ...                      ...                                ...                   ...                           ...               ...          ...            ...               ...              ...                ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...     ...      ...     ...     ...        ...            ...                  ...\n",
       "Dask Name: drop-duplicates-agg, 162 tasks"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_derived_noncoding_merged = dd.read_csv('human_derived_noncoding_merged.csv', dtype={'#Chrom': 'object',\n",
    "       'CDSpos': 'object',\n",
    "       'ConsScore': 'object',\n",
    "       'CpG': 'object',\n",
    "       'Dist2Mutation': 'object',\n",
    "       'Domain': 'object',\n",
    "       'Dst2Splice': 'object',\n",
    "       'EncodeDNase-max': 'object',\n",
    "       'EncodeDNase-sum': 'object',\n",
    "       'EncodeH2AFZ-max': 'object',\n",
    "       'EncodeH2AFZ-sum': 'object',\n",
    "       'EncodeH3K27ac-max': 'object',\n",
    "       'EncodeH3K27ac-sum': 'object',\n",
    "       'EncodeH3K27me3-max': 'object',\n",
    "       'EncodeH3K27me3-sum': 'object',\n",
    "       'EncodeH3K36me3-max': 'object',\n",
    "       'EncodeH3K36me3-sum': 'object',\n",
    "       'EncodeH3K4me1-max': 'object',\n",
    "       'EncodeH3K4me1-sum': 'object',\n",
    "       'EncodeH3K4me2-max': 'object',\n",
    "       'EncodeH3K4me2-sum': 'object',\n",
    "       'EncodeH3K4me3-max': 'object',\n",
    "       'EncodeH3K4me3-sum': 'object',\n",
    "       'EncodeH3K79me2-max': 'object',\n",
    "       'EncodeH3K79me2-sum': 'object',\n",
    "       'EncodeH3K9ac-max': 'object',\n",
    "       'EncodeH3K9ac-sum': 'object',\n",
    "       'EncodeH3K9me3-max': 'object',\n",
    "       'EncodeH3K9me3-sum': 'object',\n",
    "       'EncodeH4K20me1-max': 'object',\n",
    "       'EncodeH4K20me1-sum': 'object',\n",
    "       'EncodetotalRNA-max': 'object',\n",
    "       'EncodetotalRNA-sum': 'object',\n",
    "       'End': 'float64',\n",
    "       'Exon': 'object',\n",
    "       'Freq10000bp': 'object',\n",
    "       'Freq1000bp': 'object',\n",
    "       'Freq100bp': 'object',\n",
    "       'GC': 'object',\n",
    "       'GerpN': 'object',\n",
    "       'GerpRS': 'object',\n",
    "       'GerpRSpval': 'object',\n",
    "       'GerpS': 'object',\n",
    "       'Grantham': 'object',\n",
    "       'Length': 'object',\n",
    "       'PHRED': 'object',\n",
    "       'PolyPhenCat': 'object',\n",
    "       'PolyPhenVal': 'object',\n",
    "       'Pos': 'object',\n",
    "       'Rare10000bp': 'object',\n",
    "       'Rare1000bp': 'object',\n",
    "       'Rare100bp': 'object',\n",
    "       'RawScore': 'object',\n",
    "       'RemapOverlapCL': 'object',\n",
    "       'RemapOverlapTF': 'object',\n",
    "       'SIFTcat': 'object',\n",
    "       'SIFTval': 'object',\n",
    "       'Sngl10000bp': 'object',\n",
    "       'Sngl1000bp': 'object',\n",
    "       'Sngl100bp': 'object',\n",
    "       'bStatistic': 'object',\n",
    "       'cDNApos': 'object',\n",
    "       'cHmm_E1': 'object',\n",
    "       'cHmm_E10': 'object',\n",
    "       'cHmm_E11': 'object',\n",
    "       'cHmm_E12': 'object',\n",
    "       'cHmm_E13': 'object',\n",
    "       'cHmm_E14': 'object',\n",
    "       'cHmm_E15': 'object',\n",
    "       'cHmm_E16': 'object',\n",
    "       'cHmm_E17': 'object',\n",
    "       'cHmm_E18': 'object',\n",
    "       'cHmm_E19': 'object',\n",
    "       'cHmm_E2': 'object',\n",
    "       'cHmm_E20': 'object',\n",
    "       'cHmm_E21': 'object',\n",
    "       'cHmm_E22': 'object',\n",
    "       'cHmm_E23': 'object',\n",
    "       'cHmm_E24': 'object',\n",
    "       'cHmm_E25': 'object',\n",
    "       'cHmm_E3': 'object',\n",
    "       'cHmm_E4': 'object',\n",
    "       'cHmm_E5': 'object',\n",
    "       'cHmm_E6': 'object',\n",
    "       'cHmm_E7': 'object',\n",
    "       'cHmm_E8': 'object',\n",
    "       'cHmm_E9': 'object',\n",
    "       'dbscSNV-ada_score': 'object',\n",
    "       'dbscSNV-rf_score': 'object',\n",
    "       'mamPhCons': 'object',\n",
    "       'mamPhyloP': 'object',\n",
    "       'minDistTSE': 'object',\n",
    "       'minDistTSS': 'object',\n",
    "       'mirSVR-Aln': 'object',\n",
    "       'mirSVR-E': 'object',\n",
    "       'mirSVR-Score': 'object',\n",
    "       'motifDist': 'object',\n",
    "       'motifECount': 'object',\n",
    "       'motifEHIPos': 'object',\n",
    "       'motifEScoreChng': 'object',\n",
    "       'nAA': 'object',\n",
    "       'oAA': 'object',\n",
    "       'priPhCons': 'object',\n",
    "       'priPhyloP': 'object',\n",
    "       'protPos': 'object',\n",
    "       'relCDSpos': 'object',\n",
    "       'relProtPos': 'object',\n",
    "       'relcDNApos': 'object',\n",
    "       'tOverlapMotifs': 'object',\n",
    "       'targetScan': 'object',\n",
    "       'verPhCons': 'object',\n",
    "       'verPhyloP': 'object'})\n",
    "human_derived_noncoding_merged.drop_duplicates()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### ClinVar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have called in the clinvar annotation files for CADD and ANNOVAR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989312</td>\n",
       "      <td>100989312</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.G1102A:p.V368I,TWNK:...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.353</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100990864</td>\n",
       "      <td>100990864</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.624215246637</td>\n",
       "      <td>B</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100990866</td>\n",
       "      <td>100990866</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0717488789238</td>\n",
       "      <td>PD</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100991026</td>\n",
       "      <td>100991026</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>NM_001163812:c.*1C&gt;A;NM_001163814:c.*1C&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1556</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132768</td>\n",
       "      <td>71132792</td>\n",
       "      <td>CTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1557</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132768</td>\n",
       "      <td>71132772</td>\n",
       "      <td>CTCTT</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1558</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71132768</td>\n",
       "      <td>71132797</td>\n",
       "      <td>CTCTTCTCTTCTCTTCTCTTCTCTTCTCTT</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>MED12</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1559</th>\n",
       "      <td>chrX</td>\n",
       "      <td>78118027</td>\n",
       "      <td>78118027</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PGK1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.61301369863</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1560 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr      Start        End                             Ref Alt  \\\n",
       "0     chr10  100989312  100989312                               G   A   \n",
       "1     chr10  100990864  100990864                               C   T   \n",
       "2     chr10  100990866  100990866                               T   C   \n",
       "3     chr10  100991026  100991026                               C   A   \n",
       "...     ...        ...        ...                             ...  ..   \n",
       "1556   chrX   71132768   71132792       CTCTTCTCTTCTCTTCTCTTCTCTT   -   \n",
       "1557   chrX   71132768   71132772                           CTCTT   -   \n",
       "1558   chrX   71132768   71132797  CTCTTCTCTTCTCTTCTCTTCTCTTCTCTT   -   \n",
       "1559   chrX   78118027   78118027                               C   T   \n",
       "\n",
       "     Func.refGene Gene.refGene                         GeneDetail.refGene  \\\n",
       "0          exonic         TWNK                                          .   \n",
       "1        intronic         TWNK                                          .   \n",
       "2        intronic         TWNK                                          .   \n",
       "3            UTR3         TWNK  NM_001163812:c.*1C>A;NM_001163814:c.*1C>A   \n",
       "...           ...          ...                                        ...   \n",
       "1556     intronic        MED12                                          .   \n",
       "1557     intronic        MED12                                          .   \n",
       "1558     intronic        MED12                                          .   \n",
       "1559     intronic         PGK1                                          .   \n",
       "\n",
       "     ExonicFunc.refGene                                   AAChange.refGene  \\\n",
       "0     nonsynonymous SNV  TWNK:NM_001163812:exon1:c.G1102A:p.V368I,TWNK:...   \n",
       "1                     .                                                  .   \n",
       "2                     .                                                  .   \n",
       "3                     .                                                  .   \n",
       "...                 ...                                                ...   \n",
       "1556                  .                                                  .   \n",
       "1557                  .                                                  .   \n",
       "1558                  .                                                  .   \n",
       "1559                  .                                                  .   \n",
       "\n",
       "      ... BP4 BP5 BP6 BP7 cosmic70 MCAP  REVEL       regsnp_fpr  \\\n",
       "0     ...   0   0   1   0        .    .  0.353                .   \n",
       "1     ...   .   .   .   .        .    .      .   0.624215246637   \n",
       "2     ...   .   .   .   .        .    .      .  0.0717488789238   \n",
       "3     ...   .   .   .   .        .    .      .                .   \n",
       "...   ...  ..  ..  ..  ..      ...  ...    ...              ...   \n",
       "1556  ...   .   .   .   .        .    .      .                .   \n",
       "1557  ...   .   .   .   .        .    .      .                .   \n",
       "1558  ...   .   .   .   .        .    .      .                .   \n",
       "1559  ...   .   .   .   .        .    .      .    0.61301369863   \n",
       "\n",
       "     regsnp_disease regsnp_splicing_site  \n",
       "0                 .                    .  \n",
       "1                 B                   on  \n",
       "2                PD                   on  \n",
       "3                 .                    .  \n",
       "...             ...                  ...  \n",
       "1556              .                    .  \n",
       "1557              .                    .  \n",
       "1558              .                    .  \n",
       "1559              B                  off  \n",
       "\n",
       "[1560 rows x 149 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_noncoding_annotated_with_annovar = pd.read_csv('clinvar_benign_noncoding.hg38_multianno.csv')\n",
    "clinvar_benign_noncoding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1041950</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>100</td>\n",
       "      <td>1931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>0.002</td>\n",
       "      <td>10.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.092155</td>\n",
       "      <td>2.700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1042190</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>109</td>\n",
       "      <td>1955</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>-0.401276</td>\n",
       "      <td>0.118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>117</td>\n",
       "      <td>2004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>-0.012669</td>\n",
       "      <td>1.640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1045707</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>131</td>\n",
       "      <td>2126</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>-0.057159</td>\n",
       "      <td>1.287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3200</th>\n",
       "      <td>X</td>\n",
       "      <td>154419685</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>-0.033909</td>\n",
       "      <td>1.464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>X</td>\n",
       "      <td>154420108</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>31</td>\n",
       "      <td>826</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>-0.549155</td>\n",
       "      <td>0.036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3202</th>\n",
       "      <td>X</td>\n",
       "      <td>154420108</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>31</td>\n",
       "      <td>826</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>-0.549155</td>\n",
       "      <td>0.036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3203</th>\n",
       "      <td>X</td>\n",
       "      <td>154961190</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>45</td>\n",
       "      <td>823</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.351934</td>\n",
       "      <td>5.879</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3204 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos  Ref Alt Type  Length    AnnoType  Consequence  \\\n",
       "0         1    1041950    T   C  SNV       0  Transcript  SPLICE_SITE   \n",
       "1         1    1042190    G   A  SNV       0  Transcript     INTRONIC   \n",
       "2         1    1043223  CCT   C  DEL       2  Transcript     INTRONIC   \n",
       "3         1    1045707    A   G  SNV       0  Transcript     INTRONIC   \n",
       "...     ...        ...  ...  ..  ...     ...         ...          ...   \n",
       "3200      X  154419685    C   T  SNV       0  Transcript     INTRONIC   \n",
       "3201      X  154420108    C   T  SNV       0  Intergenic   DOWNSTREAM   \n",
       "3202      X  154420108    C   T  SNV       0  Transcript     INTRONIC   \n",
       "3203      X  154961190    A   G  SNV       0  Transcript     INTRONIC   \n",
       "\n",
       "      ConsScore     ConsDetail  ...  Freq10000bp  Rare10000bp  Sngl10000bp  \\\n",
       "0             5  splice,intron  ...           23          100         1931   \n",
       "1             2         intron  ...           25          109         1955   \n",
       "2             2         intron  ...           24          117         2004   \n",
       "3             2         intron  ...           22          131         2126   \n",
       "...         ...            ...  ...          ...          ...          ...   \n",
       "3200          2         intron  ...           11           29          820   \n",
       "3201          1     downstream  ...           12           31          826   \n",
       "3202          2         intron  ...           12           31          826   \n",
       "3203          2         intron  ...            7           45          823   \n",
       "\n",
       "     EnsembleRegulatoryFeature  dbscSNV-ada_score  dbscSNV-rf_score  \\\n",
       "0                          NaN            0.00001             0.002   \n",
       "1                          NaN                NaN               NaN   \n",
       "2                          NaN                NaN               NaN   \n",
       "3                          NaN                NaN               NaN   \n",
       "...                        ...                ...               ...   \n",
       "3200                       NaN                NaN               NaN   \n",
       "3201                       NaN                NaN               NaN   \n",
       "3202                       NaN                NaN               NaN   \n",
       "3203                       NaN                NaN               NaN   \n",
       "\n",
       "     RemapOverlapTF RemapOverlapCL  RawScore  PHRED  \n",
       "0              10.0           13.0  0.092155  2.700  \n",
       "1              12.0           14.0 -0.401276  0.118  \n",
       "2               8.0            9.0 -0.012669  1.640  \n",
       "3               9.0           10.0 -0.057159  1.287  \n",
       "...             ...            ...       ...    ...  \n",
       "3200            4.0            4.0 -0.033909  1.464  \n",
       "3201           36.0           47.0 -0.549155  0.036  \n",
       "3202           36.0           47.0 -0.549155  0.036  \n",
       "3203            1.0            1.0  0.351934  5.879  \n",
       "\n",
       "[3204 rows x 125 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_noncoding_annotated_with_cadd = pd.read_table('clinvar_noncoding_benign_cadd_annotations_noheader.tsv')\n",
    "clinvar_benign_noncoding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Just as with the other regions, I had to manipulate the ANNOVAR data to look like the CADD data in order to merge it:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1041950</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.794618834081</td>\n",
       "      <td>B</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1042190</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0445205479452</td>\n",
       "      <td>D</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1045707</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.441780821918</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3200</th>\n",
       "      <td>X</td>\n",
       "      <td>154419685</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>X</td>\n",
       "      <td>154420108</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3202</th>\n",
       "      <td>X</td>\n",
       "      <td>154420108</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3203</th>\n",
       "      <td>X</td>\n",
       "      <td>154961190</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.825342465753</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3204 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos  Ref Alt Type  Length    AnnoType  Consequence  \\\n",
       "0         1    1041950    T   C  SNV       0  Transcript  SPLICE_SITE   \n",
       "1         1    1042190    G   A  SNV       0  Transcript     INTRONIC   \n",
       "2         1    1043223  CCT   C  DEL       2  Transcript     INTRONIC   \n",
       "3         1    1045707    A   G  SNV       0  Transcript     INTRONIC   \n",
       "...     ...        ...  ...  ..  ...     ...         ...          ...   \n",
       "3200      X  154419685    C   T  SNV       0  Transcript     INTRONIC   \n",
       "3201      X  154420108    C   T  SNV       0  Intergenic   DOWNSTREAM   \n",
       "3202      X  154420108    C   T  SNV       0  Transcript     INTRONIC   \n",
       "3203      X  154961190    A   G  SNV       0  Transcript     INTRONIC   \n",
       "\n",
       "      ConsScore     ConsDetail  ...  BP4  BP5  BP6  BP7  cosmic70  MCAP REVEL  \\\n",
       "0             5  splice,intron  ...    .    .    .    .         .     .     .   \n",
       "1             2         intron  ...    .    .    .    .         .     .     .   \n",
       "2             2         intron  ...  NaN  NaN  NaN  NaN       NaN   NaN   NaN   \n",
       "3             2         intron  ...    .    .    .    .         .     .     .   \n",
       "...         ...            ...  ...  ...  ...  ...  ...       ...   ...   ...   \n",
       "3200          2         intron  ...    .    .    .    .         .     .     .   \n",
       "3201          1     downstream  ...    .    .    .    .         .     .     .   \n",
       "3202          2         intron  ...    .    .    .    .         .     .     .   \n",
       "3203          2         intron  ...    .    .    .    .         .     .     .   \n",
       "\n",
       "           regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0      0.794618834081              B                   on  \n",
       "1     0.0445205479452              D                  off  \n",
       "2                 NaN            NaN                  NaN  \n",
       "3      0.441780821918              B                  off  \n",
       "...               ...            ...                  ...  \n",
       "3200                .              .                    .  \n",
       "3201                .              .                    .  \n",
       "3202                .              .                    .  \n",
       "3203   0.825342465753              B                  off  \n",
       "\n",
       "[3204 rows x 270 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_noncoding_annotated_with_annovar['Chr'] = clinvar_benign_noncoding_annotated_with_annovar['Chr'].map(lambda x: x.lstrip('chr').rstrip('aAbBcC'))\n",
    "clinvar_benign_noncoding_annotated_with_annovar = clinvar_benign_noncoding_annotated_with_annovar.rename(columns = {'Chr': '#Chrom', 'Start': 'Pos'})\n",
    "clinvar_benign_noncoding_merged = pd.merge(clinvar_benign_noncoding_annotated_with_cadd, clinvar_benign_noncoding_annotated_with_annovar,\n",
    "                                         how = 'left', on = ['#Chrom', 'Pos', 'Ref', 'Alt'])\n",
    "clinvar_benign_noncoding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have dropped duplicates:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1041950</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.794618834081</td>\n",
       "      <td>B</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1042190</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0445205479452</td>\n",
       "      <td>D</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1043223</td>\n",
       "      <td>CCT</td>\n",
       "      <td>C</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1045707</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.441780821918</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3200</th>\n",
       "      <td>X</td>\n",
       "      <td>154419685</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3201</th>\n",
       "      <td>X</td>\n",
       "      <td>154420108</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3202</th>\n",
       "      <td>X</td>\n",
       "      <td>154420108</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3203</th>\n",
       "      <td>X</td>\n",
       "      <td>154961190</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.825342465753</td>\n",
       "      <td>B</td>\n",
       "      <td>off</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3204 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos  Ref Alt Type  Length    AnnoType  Consequence  \\\n",
       "0         1    1041950    T   C  SNV       0  Transcript  SPLICE_SITE   \n",
       "1         1    1042190    G   A  SNV       0  Transcript     INTRONIC   \n",
       "2         1    1043223  CCT   C  DEL       2  Transcript     INTRONIC   \n",
       "3         1    1045707    A   G  SNV       0  Transcript     INTRONIC   \n",
       "...     ...        ...  ...  ..  ...     ...         ...          ...   \n",
       "3200      X  154419685    C   T  SNV       0  Transcript     INTRONIC   \n",
       "3201      X  154420108    C   T  SNV       0  Intergenic   DOWNSTREAM   \n",
       "3202      X  154420108    C   T  SNV       0  Transcript     INTRONIC   \n",
       "3203      X  154961190    A   G  SNV       0  Transcript     INTRONIC   \n",
       "\n",
       "      ConsScore     ConsDetail  ...  BP4  BP5  BP6  BP7  cosmic70  MCAP REVEL  \\\n",
       "0             5  splice,intron  ...    .    .    .    .         .     .     .   \n",
       "1             2         intron  ...    .    .    .    .         .     .     .   \n",
       "2             2         intron  ...  NaN  NaN  NaN  NaN       NaN   NaN   NaN   \n",
       "3             2         intron  ...    .    .    .    .         .     .     .   \n",
       "...         ...            ...  ...  ...  ...  ...  ...       ...   ...   ...   \n",
       "3200          2         intron  ...    .    .    .    .         .     .     .   \n",
       "3201          1     downstream  ...    .    .    .    .         .     .     .   \n",
       "3202          2         intron  ...    .    .    .    .         .     .     .   \n",
       "3203          2         intron  ...    .    .    .    .         .     .     .   \n",
       "\n",
       "           regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0      0.794618834081              B                   on  \n",
       "1     0.0445205479452              D                  off  \n",
       "2                 NaN            NaN                  NaN  \n",
       "3      0.441780821918              B                  off  \n",
       "...               ...            ...                  ...  \n",
       "3200                .              .                    .  \n",
       "3201                .              .                    .  \n",
       "3202                .              .                    .  \n",
       "3203   0.825342465753              B                  off  \n",
       "\n",
       "[3204 rows x 270 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_benign_noncoding_merged = clinvar_benign_noncoding_merged.drop_duplicates(keep = 'first')\n",
    "clinvar_benign_noncoding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the final merged dataframe for the benign noncoding region in which the output is given as a csv. It contains both the human derived and clinvar annotated variants and does not contain the columns which are unneccessary for our model:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/dask/local.py:252: DtypeWarning: Columns (33) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  args2 = [_execute_task(a, cache) for a in args]\n",
      "/opt/miniconda3/lib/python3.7/site-packages/dask/local.py:252: DtypeWarning: Columns (14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  args2 = [_execute_task(a, cache) for a in args]\n",
      "/opt/miniconda3/lib/python3.7/site-packages/dask/local.py:252: DtypeWarning: Columns (14,33) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  args2 = [_execute_task(a, cache) for a in args]\n",
      "/opt/miniconda3/lib/python3.7/site-packages/dask/local.py:252: DtypeWarning: Columns (22) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  args2 = [_execute_task(a, cache) for a in args]\n",
      "/opt/miniconda3/lib/python3.7/site-packages/dask/local.py:252: DtypeWarning: Columns (22,33) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  args2 = [_execute_task(a, cache) for a in args]\n",
      "/opt/miniconda3/lib/python3.7/site-packages/dask/local.py:252: DtypeWarning: Columns (14,22,33) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  args2 = [_execute_task(a, cache) for a in args]\n",
      "/opt/miniconda3/lib/python3.7/site-packages/pandas/core/frame.py:6692: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  sort=sort)\n"
     ]
    }
   ],
   "source": [
    "human_derived_noncoding_merged.compute().append(clinvar_benign_noncoding_merged).drop(columns = ['PolyPhenVal', 'priPhCons', 'cHmm_E1', 'cHmm_E2', 'cHmm_E3', 'cHmm_E4', 'cHmm_E5', 'cHmm_E6', 'cHmm_E7', 'cHmm_E8', 'cHmm_E9', 'cHmm_E10', 'cHmm_E11', 'cHmm_E12', 'cHmm_E13', 'cHmm_E14', 'cHmm_E15', 'cHmm_E16', 'cHmm_E17', 'cHmm_E18', 'cHmm_E19', 'cHmm_E20', 'cHmm_E21', 'cHmm_E22', 'cHmm_E23', 'cHmm_E24', 'cHmm_E25', 'GerpN', 'GerpS']).to_csv('final_benign_noncoding_dataframe.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the code for the final dataframe created in the cell above. We must change all the columns' dtypes to object in order for dask dataframe to read them correctly:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/dask/local.py:252: DtypeWarning: Columns (34) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  args2 = [_execute_task(a, cache) for a in args]\n",
      "/opt/miniconda3/lib/python3.7/site-packages/dask/local.py:252: DtypeWarning: Columns (217) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  args2 = [_execute_task(a, cache) for a in args]\n",
      "/opt/miniconda3/lib/python3.7/site-packages/dask/local.py:252: DtypeWarning: Columns (34,217) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  args2 = [_execute_task(a, cache) for a in args]\n",
      "/opt/miniconda3/lib/python3.7/site-packages/dask/local.py:252: DtypeWarning: Columns (24) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  args2 = [_execute_task(a, cache) for a in args]\n"
     ]
    }
   ],
   "source": [
    "human_derived_final_dataframe = dd.read_csv('final_benign_noncoding_dataframe.csv', dtype={'#Chrom': 'object',\n",
    "       'ConsScore': 'object',\n",
    "       'CpG': 'object',\n",
    "       'Dist2Mutation': 'object',\n",
    "       'Domain': 'object',\n",
    "       'Dst2Splice': 'object',\n",
    "       'EncodeDNase-max': 'object',\n",
    "       'EncodeDNase-sum': 'object',\n",
    "       'EncodeH2AFZ-max': 'object',\n",
    "       'EncodeH2AFZ-sum': 'object',\n",
    "       'EncodeH3K27ac-max': 'object',\n",
    "       'EncodeH3K27ac-sum': 'object',\n",
    "       'EncodeH3K27me3-max': 'object',\n",
    "       'EncodeH3K27me3-sum': 'object',\n",
    "       'EncodeH3K36me3-max': 'object',\n",
    "       'EncodeH3K36me3-sum': 'object',\n",
    "       'EncodeH3K4me1-max': 'object',\n",
    "       'EncodeH3K4me1-sum': 'object',\n",
    "       'EncodeH3K4me2-max': 'object',\n",
    "       'EncodeH3K4me2-sum': 'object',\n",
    "       'EncodeH3K4me3-max': 'object',\n",
    "       'EncodeH3K4me3-sum': 'object',\n",
    "       'EncodeH3K79me2-max': 'object',\n",
    "       'EncodeH3K79me2-sum': 'object',\n",
    "       'EncodeH3K9ac-max': 'object',\n",
    "       'EncodeH3K9ac-sum': 'object',\n",
    "       'EncodeH3K9me3-max': 'object',\n",
    "       'EncodeH3K9me3-sum': 'object',\n",
    "       'EncodeH4K20me1-max': 'object',\n",
    "       'EncodeH4K20me1-sum': 'object',\n",
    "       'EncodetotalRNA-max': 'object',\n",
    "       'EncodetotalRNA-sum': 'object',\n",
    "       'Exon': 'object',\n",
    "       'Freq10000bp': 'object',\n",
    "       'Freq1000bp': 'object',\n",
    "       'Freq100bp': 'object',\n",
    "       'GC': 'object',\n",
    "       'GerpRS': 'object',\n",
    "       'GerpRSpval': 'object',\n",
    "       'Grantham': 'object',\n",
    "       'Length': 'object',\n",
    "       'PHRED': 'object',\n",
    "       'PolyPhenCat': 'object',\n",
    "       'Pos': 'object',\n",
    "       'Rare10000bp': 'object',\n",
    "       'Rare1000bp': 'object',\n",
    "       'Rare100bp': 'object',\n",
    "       'RawScore': 'object',\n",
    "       'RemapOverlapCL': 'object',\n",
    "       'RemapOverlapTF': 'object',\n",
    "       'SIFTcat': 'object',\n",
    "       'SIFTval': 'object',\n",
    "       'Sngl10000bp': 'object',\n",
    "       'Sngl1000bp': 'object',\n",
    "       'Sngl100bp': 'object',\n",
    "       'bStatistic': 'object',\n",
    "       'cDNApos': 'object',\n",
    "       'dbscSNV-ada_score': 'object',\n",
    "       'dbscSNV-rf_score': 'object',\n",
    "       'mamPhCons': 'object',\n",
    "       'mamPhyloP': 'object',\n",
    "       'minDistTSE': 'object',\n",
    "       'minDistTSS': 'object',\n",
    "       'mirSVR-Aln': 'object',\n",
    "       'mirSVR-E': 'object',\n",
    "       'mirSVR-Score': 'object',\n",
    "       'motifDist': 'object',\n",
    "       'motifECount': 'object',\n",
    "       'motifEHIPos': 'object',\n",
    "       'motifEScoreChng': 'object',\n",
    "       'nAA': 'object',\n",
    "       'oAA': 'object',\n",
    "       'priPhyloP': 'object',\n",
    "       'protPos': 'object',\n",
    "       'relCDSpos': 'object',\n",
    "       'relProtPos': 'object',\n",
    "       'relcDNApos': 'object',\n",
    "       'tOverlapMotifs': 'object',\n",
    "       'targetScan': 'object',\n",
    "       'verPhCons': 'object',\n",
    "       'verPhyloP': 'object',\n",
    "        'CDSpos': 'object'}).compute()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We notice that there are two more columns which we do not want and therefore we drop these columns from the dataframe and once again save the dataframe as the same csv:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_derived_final_dataframe = human_derived_final_dataframe.drop(['Unnamed: 0', 'Unnamed: 0.1'], axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For our model, we want to have NaN values in place of the string '.' in the dataframe. The reason for this is that it is very easy to change NaN values and manipulate them to our will in Python. In order to do so, we must break the dataframe into multiple sections and replace values this way. Otherwise we would receive a Memory Error. We do this with the following code:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_noncoding_final_dataframe_1 = benign_noncoding_final_dataframe[:600000]\n",
    "benign_noncoding_final_dataframe_1_with_nan = benign_noncoding_final_dataframe_1.replace('.', np.nan)\n",
    "benign_noncoding_final_dataframe_2 = benign_noncoding_final_dataframe[600000:1100000]\n",
    "benign_noncoding_final_dataframe_2_with_nan = benign_noncoding_final_dataframe_2.replace('.', np.nan)\n",
    "benign_noncoding_final_dataframe_3 = benign_noncoding_final_dataframe[1100000:1538959]\n",
    "benign_noncoding_final_dataframe_3_with_nan = benign_noncoding_final_dataframe_3.replace('.', np.nan)\n",
    "benign_noncoding_final_dataframe_1_with_nan.to_csv('benignnoncodingfinaldataframe_withNaN.csv', index=False)\n",
    "benign_noncoding_final_dataframe_2_with_nan.to_csv('benignnoncodingfinaldataframe_withNaN.csv', mode = 'a', index=False)\n",
    "benign_noncoding_final_dataframe_3_with_nan.to_csv('benignnoncodingfinaldataframe_withNaN.csv', mode = 'a', index=False)\n",
    "del benign_noncoding_final_dataframe_1_with_nan, benign_noncoding_final_dataframe_2_with_nan, benign_noncoding_final_dataframe_3_with_nan"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Another thing that our method to create the resulting dataframe has unfortunately given us is extra rows in the dataframe that contain the names of the columns. This is because when we append dataframes using dask, a row is created for the header of the dataframes being added on. We need to drop these rows. To do this, we specify that only the duplicates with all values in common will be dropped, shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_noncoding_final_dataframe_with_nan = benign_noncoding_final_dataframe_with_nan.drop_duplicates(subset = ['#Chrom', 'AAChange.ensGene', 'AAChange.knownGene', 'AAChange.refGene',\n",
    " 'Alt',\n",
    " 'AnnoType',\n",
    " 'BA1',\n",
    " 'BP1',\n",
    " 'BP2',\n",
    " 'BP3',\n",
    " 'BP4',\n",
    " 'BP5',\n",
    " 'BP6',\n",
    " 'BP7',\n",
    " 'BS1',\n",
    " 'BS2',\n",
    " 'BS3',\n",
    " 'BS4',\n",
    " 'CADD_phred',\n",
    " 'CADD_phred.1',\n",
    " 'CADD_raw',\n",
    " 'CADD_raw.1',\n",
    " 'CADD_raw_rankscore',\n",
    " 'CCDS',\n",
    " 'CDSpos',\n",
    " 'ConsDetail',\n",
    " 'ConsScore',\n",
    " 'Consequence',\n",
    " 'CpG',\n",
    " 'DANN_rankscore',\n",
    " 'DANN_score',\n",
    " 'Dist2Mutation',\n",
    " 'Domain',\n",
    " 'Dst2SplType',\n",
    " 'Dst2Splice',\n",
    " 'Eigen-PC-raw',\n",
    " 'Eigen-raw',\n",
    " 'Eigen_coding_or_noncoding',\n",
    " 'EncodeDNase-max',\n",
    " 'EncodeDNase-sum',\n",
    " 'EncodeH2AFZ-max',\n",
    " 'EncodeH2AFZ-sum',\n",
    " 'EncodeH3K27ac-max',\n",
    " 'EncodeH3K27ac-sum',\n",
    " 'EncodeH3K27me3-max',\n",
    " 'EncodeH3K27me3-sum',\n",
    " 'EncodeH3K36me3-max',\n",
    " 'EncodeH3K36me3-sum',\n",
    " 'EncodeH3K4me1-max',\n",
    " 'EncodeH3K4me1-sum',\n",
    " 'EncodeH3K4me2-max',\n",
    " 'EncodeH3K4me2-sum',\n",
    " 'EncodeH3K4me3-max',\n",
    " 'EncodeH3K4me3-sum',\n",
    " 'EncodeH3K79me2-max',\n",
    " 'EncodeH3K79me2-sum',\n",
    " 'EncodeH3K9ac-max',\n",
    " 'EncodeH3K9ac-sum',\n",
    " 'EncodeH3K9me3-max',\n",
    " 'EncodeH3K9me3-sum',\n",
    " 'EncodeH4K20me1-max',\n",
    " 'EncodeH4K20me1-sum',\n",
    " 'EncodetotalRNA-max',\n",
    " 'EncodetotalRNA-sum',\n",
    " 'End',\n",
    " 'EnsembleRegulatoryFeature',\n",
    " 'Exon',\n",
    " 'ExonicFunc.ensGene',\n",
    " 'ExonicFunc.knownGene',\n",
    " 'ExonicFunc.refGene',\n",
    " 'FATHMM_converted_rankscore',\n",
    " 'FATHMM_pred',\n",
    " 'FATHMM_pred.1',\n",
    " 'FATHMM_score',\n",
    " 'FATHMM_score.1',\n",
    " 'FeatureID',\n",
    " 'Freq10000bp',\n",
    " 'Freq1000bp',\n",
    " 'Freq100bp',\n",
    " 'Func.ensGene',\n",
    " 'Func.knownGene',\n",
    " 'Func.refGene',\n",
    " 'GC',\n",
    " 'GERP++_RS',\n",
    " 'GERP++_RS.1',\n",
    " 'GERP++_RS_rankscore',\n",
    " 'GTEx_V6_gene',\n",
    " 'GTEx_V6_tissue',\n",
    " 'Gene.ensGene',\n",
    " 'Gene.knownGene',\n",
    " 'Gene.refGene',\n",
    " 'GeneDetail.ensGene',\n",
    " 'GeneDetail.knownGene',\n",
    " 'GeneDetail.refGene',\n",
    " 'GeneID',\n",
    " 'GeneName',\n",
    " 'GenoCanyon_score',\n",
    " 'GenoCanyon_score_rankscore',\n",
    " 'GerpRS',\n",
    " 'GerpRSpval',\n",
    " 'Grantham',\n",
    " 'InterVar_automated',\n",
    " 'Interpro_domain',\n",
    " 'Interpro_domain.1',\n",
    " 'Intron',\n",
    " 'LRT_converted_rankscore',\n",
    " 'LRT_pred',\n",
    " 'LRT_pred.1',\n",
    " 'LRT_score',\n",
    " 'LRT_score.1',\n",
    " 'LR_pred',\n",
    " 'LR_score',\n",
    " 'Length',\n",
    " 'M-CAP_pred',\n",
    " 'M-CAP_rankscore',\n",
    " 'M-CAP_score',\n",
    " 'MCAP',\n",
    " 'MetaLR_pred',\n",
    " 'MetaLR_rankscore',\n",
    " 'MetaLR_score',\n",
    " 'MetaSVM_pred',\n",
    " 'MetaSVM_rankscore',\n",
    " 'MetaSVM_score',\n",
    " 'MutationAssessor_pred',\n",
    " 'MutationAssessor_pred.1',\n",
    " 'MutationAssessor_score',\n",
    " 'MutationAssessor_score.1',\n",
    " 'MutationAssessor_score_rankscore',\n",
    " 'MutationTaster_converted_rankscore',\n",
    " 'MutationTaster_pred',\n",
    " 'MutationTaster_pred.1',\n",
    " 'MutationTaster_score',\n",
    " 'MutationTaster_score.1',\n",
    " 'PHRED',\n",
    " 'PM1',\n",
    " 'PM2',\n",
    " 'PM3',\n",
    " 'PM4',\n",
    " 'PM5',\n",
    " 'PM6',\n",
    " 'PP1',\n",
    " 'PP2',\n",
    " 'PP3',\n",
    " 'PP4',\n",
    " 'PP5',\n",
    " 'PROVEAN_converted_rankscore',\n",
    " 'PROVEAN_pred',\n",
    " 'PROVEAN_score',\n",
    " 'PS1',\n",
    " 'PS2',\n",
    " 'PS3',\n",
    " 'PS4',\n",
    " 'PVS1',\n",
    " 'PolyPhenCat',\n",
    " 'Polyphen2_HDIV_pred',\n",
    " 'Polyphen2_HDIV_pred.1',\n",
    " 'Polyphen2_HDIV_rankscore',\n",
    " 'Polyphen2_HDIV_score',\n",
    " 'Polyphen2_HDIV_score.1',\n",
    " 'Polyphen2_HVAR_pred',\n",
    " 'Polyphen2_HVAR_pred.1',\n",
    " 'Polyphen2_HVAR_rankscore',\n",
    " 'Polyphen2_HVAR_score',\n",
    " 'Polyphen2_HVAR_score.1',\n",
    " 'Pos',\n",
    " 'REVEL',\n",
    " 'RadialSVM_pred',\n",
    " 'RadialSVM_score',\n",
    " 'Rare10000bp',\n",
    " 'Rare1000bp',\n",
    " 'Rare100bp',\n",
    " 'RawScore',\n",
    " 'Ref',\n",
    " 'RemapOverlapCL',\n",
    " 'RemapOverlapTF',\n",
    " 'SIFT_converted_rankscore',\n",
    " 'SIFT_pred',\n",
    " 'SIFT_pred.1',\n",
    " 'SIFT_score',\n",
    " 'SIFT_score.1',\n",
    " 'SIFTcat',\n",
    " 'SIFTval',\n",
    " 'SiPhy_29way_logOdds',\n",
    " 'SiPhy_29way_logOdds.1',\n",
    " 'SiPhy_29way_logOdds_rankscore',\n",
    " 'Sngl10000bp',\n",
    " 'Sngl1000bp',\n",
    " 'Sngl100bp',\n",
    " 'Type',\n",
    " 'VEST3_rankscore',\n",
    " 'VEST3_score',\n",
    " 'VEST3_score.1',\n",
    " 'bStatistic',\n",
    " 'cDNApos',\n",
    " 'cosmic70',\n",
    " 'dbscSNV-ada_score',\n",
    " 'dbscSNV-rf_score',\n",
    " 'dbscSNV_ADA_SCORE',\n",
    " 'dbscSNV_RF_SCORE',\n",
    " 'fathmm-MKL_coding_pred',\n",
    " 'fathmm-MKL_coding_rankscore',\n",
    " 'fathmm-MKL_coding_score',\n",
    " 'integrated_confidence_value',\n",
    " 'integrated_fitCons_score',\n",
    " 'integrated_fitCons_score_rankscore',\n",
    " 'mamPhCons',\n",
    " 'mamPhyloP',\n",
    " 'minDistTSE',\n",
    " 'minDistTSS',\n",
    " 'mirSVR-Aln',\n",
    " 'mirSVR-E',\n",
    " 'mirSVR-Score',\n",
    " 'motifDist',\n",
    " 'motifECount',\n",
    " 'motifEHIPos',\n",
    " 'motifEName',\n",
    " 'motifEScoreChng',\n",
    " 'nAA',\n",
    " 'oAA',\n",
    " 'phastCons100way_vertebrate',\n",
    " 'phastCons100way_vertebrate_rankscore',\n",
    " 'phastCons20way_mammalian',\n",
    " 'phastCons20way_mammalian_rankscore',\n",
    " 'phyloP100way_vertebrate',\n",
    " 'phyloP100way_vertebrate.1',\n",
    " 'phyloP100way_vertebrate_rankscore',\n",
    " 'phyloP20way_mammalian',\n",
    " 'phyloP20way_mammalian_rankscore',\n",
    " 'phyloP46way_placental',\n",
    " 'priPhyloP',\n",
    " 'protPos',\n",
    " 'regsnp_disease',\n",
    " 'regsnp_fpr',\n",
    " 'regsnp_splicing_site',\n",
    " 'relCDSpos',\n",
    " 'relProtPos',\n",
    " 'relcDNApos',\n",
    " 'tOverlapMotifs',\n",
    " 'targetScan',\n",
    " 'verPhCons',\n",
    " 'verPhyloP'] , keep = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we must change the dtypes of the numerical values for use in the future from string to float. I have created the following code using the Python tool to_numeric in order to change the data types to floats for columns with numerical values:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_noncoding_final_dataframe_with_nan[\"verPhyloP\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"verPhyloP\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"verPhCons\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"verPhCons\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"targetScan\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"targetScan\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"tOverlapMotifs\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"tOverlapMotifs\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"relcDNApos\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"relcDNApos\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"relProtPos\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"relProtPos\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"relCDSpos\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"relCDSpos\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"regsnp_fpr\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"regsnp_fpr\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"protPos\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"protPos\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"priPhyloP\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"priPhyloP\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"phyloP46way_placental\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"phyloP46way_placental\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"phyloP20way_mammalian_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"phyloP20way_mammalian_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"phyloP20way_mammalian\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"phyloP20way_mammalian\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"phyloP100way_vertebrate_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"phyloP100way_vertebrate_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"phyloP100way_vertebrate.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"phyloP100way_vertebrate.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"phyloP100way_vertebrate\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"phyloP100way_vertebrate\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"phastCons20way_mammalian_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"phastCons20way_mammalian_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"phastCons20way_mammalian\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"phastCons20way_mammalian\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"phastCons100way_vertebrate_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"phastCons100way_vertebrate_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"phastCons100way_vertebrate\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"phastCons100way_vertebrate\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"motifEScoreChng\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"motifEScoreChng\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"motifDist\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"motifDist\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"mirSVR-Score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"mirSVR-Score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"mirSVR-E\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"mirSVR-E\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"mirSVR-Aln\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"mirSVR-Aln\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"minDistTSS\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"minDistTSS\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"minDistTSE\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"minDistTSE\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"mamPhyloP\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"mamPhyloP\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"mamPhCons\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"mamPhCons\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"integrated_fitCons_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"integrated_fitCons_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"integrated_confidence_value\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"integrated_confidence_value\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"fathmm-MKL_coding_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"fathmm-MKL_coding_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"fathmm-MKL_coding_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"fathmm-MKL_coding_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"dbscSNV_RF_SCORE\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"dbscSNV_RF_SCORE\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"dbscSNV_ADA_SCORE\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"dbscSNV_ADA_SCORE\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"dbscSNV-rf_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"dbscSNV-rf_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"dbscSNV-ada_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"dbscSNV-ada_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"cDNApos\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"cDNApos\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"bStatistic\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"bStatistic\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"VEST3_score.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"VEST3_score.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"VEST3_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"VEST3_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"VEST3_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"VEST3_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Sngl100bp\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Sngl100bp\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Sngl1000bp\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Sngl1000bp\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Sngl10000bp\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Sngl10000bp\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"SiPhy_29way_logOdds_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"SiPhy_29way_logOdds_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"SiPhy_29way_logOdds.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"SiPhy_29way_logOdds.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"SiPhy_29way_logOdds\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"SiPhy_29way_logOdds\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"SIFTval\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"SIFTval\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"SIFT_score.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"SIFT_score.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"SIFT_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"SIFT_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"SIFT_converted_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"SIFT_converted_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"RemapOverlapTF\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"RemapOverlapTF\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"RemapOverlapCL\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"RemapOverlapCL\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"RawScore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"RawScore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Rare100bp\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Rare100bp\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Rare1000bp\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Rare1000bp\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Rare10000bp\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Rare10000bp\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"RadialSVM_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"RadialSVM_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"REVEL\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"REVEL\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HVAR_score.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HVAR_score.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HVAR_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HVAR_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HVAR_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HVAR_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HDIV_score.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HDIV_score.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HDIV_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HDIV_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HDIV_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Polyphen2_HDIV_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PVS1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PVS1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PS4\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PS4\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PS3\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PS3\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PS2\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PS2\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PS1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PS1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PROVEAN_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PROVEAN_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PROVEAN_converted_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PROVEAN_converted_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PP5\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PP5\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PP4\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PP4\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PP3\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PP3\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PP2\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PP2\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PP1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PP1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PM6\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PM6\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PM5\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PM5\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PM4\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PM4\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PM3\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PM3\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PM2\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PM2\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PM1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PM1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"PHRED\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"PHRED\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MutationTaster_score.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MutationTaster_score.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MutationTaster_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MutationTaster_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MutationTaster_converted_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MutationTaster_converted_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MutationAssessor_score_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MutationAssessor_score_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MutationAssessor_score.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MutationAssessor_score.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MutationAssessor_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MutationAssessor_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MetaSVM_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MetaSVM_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MetaSVM_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MetaSVM_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MetaLR_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MetaLR_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MetaLR_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MetaLR_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"MCAP\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"MCAP\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"M-CAP_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"M-CAP_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"M-CAP_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"M-CAP_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Length\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Length\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"LR_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"LR_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"LRT_score.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"LRT_score.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"LRT_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"LRT_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"LRT_converted_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"LRT_converted_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Grantham\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Grantham\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"GerpRSpval\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"GerpRSpval\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"GerpRS\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"GerpRS\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"GenoCanyon_score_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"GenoCanyon_score_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"GenoCanyon_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"GenoCanyon_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"GERP++_RS_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"GERP++_RS_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"GERP++_RS.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"GERP++_RS.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"GERP++_RS\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"GERP++_RS\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"GC\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"GC\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Freq100bp\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Freq100bp\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Freq1000bp\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Freq1000bp\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Freq10000bp\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Freq10000bp\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"FATHMM_score.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"FATHMM_score.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"FATHMM_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"FATHMM_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"FATHMM_converted_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"FATHMM_converted_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"End\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"End\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodetotalRNA-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodetotalRNA-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodetotalRNA-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodetotalRNA-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH4K20me1-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH4K20me1-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH4K20me1-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH4K20me1-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K9me3-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K9me3-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K9me3-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K9me3-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K9ac-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K9ac-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K9ac-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K9ac-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K79me2-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K79me2-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K79me2-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K79me2-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me3-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me3-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me3-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me3-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me2-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me2-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me2-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me2-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me1-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me1-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me1-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K4me1-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K36me3-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K36me3-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K36me3-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K36me3-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K27me3-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K27me3-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K27me3-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K27me3-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K27ac-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K27ac-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH3K27ac-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH3K27ac-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH2AFZ-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH2AFZ-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeH2AFZ-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeH2AFZ-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeDNase-sum\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeDNase-sum\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"EncodeDNase-max\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"EncodeDNase-max\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Eigen-raw\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Eigen-raw\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Eigen-PC-raw\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Eigen-PC-raw\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Dst2Splice\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Dst2Splice\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"Dist2Mutation\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"Dist2Mutation\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"DANN_score\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"DANN_score\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"DANN_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"DANN_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"CpG\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"CpG\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"ConsScore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"ConsScore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"CDSpos\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"CDSpos\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"CADD_raw_rankscore\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"CADD_raw_rankscore\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"CADD_raw.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"CADD_raw.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"CADD_raw\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"CADD_raw\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"CADD_phred.1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"CADD_phred.1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"CADD_phred\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"CADD_phred\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BS4\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BS4\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BS3\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BS3\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BS2\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BS2\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BS1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BS1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BP7\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BP7\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BP6\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BP6\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BP5\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BP5\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BP4\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BP4\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BP3\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BP3\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BP2\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BP2\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BP1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BP1\"])\n",
    "benign_noncoding_final_dataframe_with_nan[\"BA1\"] = pd.to_numeric(benign_noncoding_final_dataframe_with_nan[\"BA1\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below we have saved the above dataframe for future use:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_noncoding_final_dataframe_with_nan.to_csv('benign_noncoding_final_dataframe_with_nan_and_floats.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to call in the dataframe correctly using dask dataframe, we must run the following code which specifies the data type for all annotations not included in our code. This gives us the final benign noncoding variant dataframe:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_noncoding_final_dataframe_with_nan_and_floats = dd.read_csv('benign_noncoding_final_dataframe_with_nan_and_floats.csv', dtype={'AAChange.ensGene': 'object',\n",
    "       'AAChange.knownGene': 'object',\n",
    "       'AAChange.refGene': 'object',\n",
    "       'Domain': 'object',\n",
    "       'Eigen_coding_or_noncoding': 'object',\n",
    "       'Exon': 'object',\n",
    "       'ExonicFunc.ensGene': 'object',\n",
    "       'ExonicFunc.knownGene': 'object',\n",
    "       'ExonicFunc.refGene': 'object',\n",
    "       'FATHMM_pred': 'object',\n",
    "       'FATHMM_pred.1': 'object',\n",
    "       'InterVar_automated': 'object',\n",
    "       'Interpro_domain': 'object',\n",
    "       'Interpro_domain.1': 'object',\n",
    "       'LRT_pred': 'object',\n",
    "       'LRT_pred.1': 'object',\n",
    "       'LR_pred': 'object',\n",
    "       'M-CAP_pred': 'object',\n",
    "       'MetaLR_pred': 'object',\n",
    "       'MetaSVM_pred': 'object',\n",
    "       'MutationAssessor_pred': 'object',\n",
    "       'MutationAssessor_pred.1': 'object',\n",
    "       'MutationTaster_pred': 'object',\n",
    "       'MutationTaster_pred.1': 'object',\n",
    "       'PROVEAN_pred': 'object',\n",
    "       'PolyPhenCat': 'object',\n",
    "       'Polyphen2_HDIV_pred': 'object',\n",
    "       'Polyphen2_HDIV_pred.1': 'object',\n",
    "       'Polyphen2_HVAR_pred': 'object',\n",
    "       'Polyphen2_HVAR_pred.1': 'object',\n",
    "       'RadialSVM_pred': 'object',\n",
    "       'SIFT_pred': 'object',\n",
    "       'SIFT_pred.1': 'object',\n",
    "       'SIFTcat': 'object',\n",
    "       'cosmic70': 'object',\n",
    "       'fathmm-MKL_coding_pred': 'object',\n",
    "       'nAA': 'object',\n",
    "       'oAA': 'object', \n",
    "        'GTEx_V6_gene': 'object',\n",
    "       'GTEx_V6_tissue': 'object', '#Chrom': 'object'}).compute()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pathogenic Coding Region:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### ClinVar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the Pathogenic Coding Region for ClinVar:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>100154922</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ERLIN1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>ERLIN1:NM_001347858:exon9:c.C283T:p.R95X,ERLIN...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>100183802</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ERLIN1</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>ERLIN1:NM_006459:exon2:c.G149T:p.G50V,ERLIN1:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5619750829</td>\n",
       "      <td>0.984</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100246865</td>\n",
       "      <td>100246865</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>CWF19L1:NM_001303406:exon5:c.368delA:p.D123Vfs...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253422</td>\n",
       "      <td>100253422</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>CWF19L1:NM_001303406:exon3:c.C211T:p.R71X,CWF1...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16771</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85981796</td>\n",
       "      <td>85981796</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>CHM:NM_000390:exon3:c.G130T:p.G44X,CHM:NM_0011...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16772</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759332</td>\n",
       "      <td>9759332</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GPR143:NM_000273:exon3:c.G455A:p.S152N</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.676591847234</td>\n",
       "      <td>0.695</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16773</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>9759390</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GPR143:NM_000273:exon3:c.T397C:p.W133R</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.790856144973</td>\n",
       "      <td>0.400</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16774</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9759390</td>\n",
       "      <td>9759390</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GPR143:NM_000273:exon3:c.T397A:p.W133R</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.790215722738</td>\n",
       "      <td>0.386</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16775 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0      chr10  100154922  100154922   G   A       exonic       ERLIN1   \n",
       "1      chr10  100183802  100183802   C   A       exonic       ERLIN1   \n",
       "2      chr10  100246865  100246865   T   -       exonic      CWF19L1   \n",
       "3      chr10  100253422  100253422   G   A       exonic      CWF19L1   \n",
       "...      ...        ...        ...  ..  ..          ...          ...   \n",
       "16771   chrX   85981796   85981796   C   A       exonic          CHM   \n",
       "16772   chrX    9759332    9759332   C   T       exonic       GPR143   \n",
       "16773   chrX    9759390    9759390   A   G       exonic       GPR143   \n",
       "16774   chrX    9759390    9759390   A   T       exonic       GPR143   \n",
       "\n",
       "      GeneDetail.refGene   ExonicFunc.refGene  \\\n",
       "0                      .             stopgain   \n",
       "1                      .    nonsynonymous SNV   \n",
       "2                      .  frameshift deletion   \n",
       "3                      .             stopgain   \n",
       "...                  ...                  ...   \n",
       "16771                  .             stopgain   \n",
       "16772                  .    nonsynonymous SNV   \n",
       "16773                  .    nonsynonymous SNV   \n",
       "16774                  .    nonsynonymous SNV   \n",
       "\n",
       "                                        AAChange.refGene  ... BP4 BP5 BP6 BP7  \\\n",
       "0      ERLIN1:NM_001347858:exon9:c.C283T:p.R95X,ERLIN...  ...   0   0   0   0   \n",
       "1      ERLIN1:NM_006459:exon2:c.G149T:p.G50V,ERLIN1:N...  ...   .   .   .   .   \n",
       "2      CWF19L1:NM_001303406:exon5:c.368delA:p.D123Vfs...  ...   .   .   .   .   \n",
       "3      CWF19L1:NM_001303406:exon3:c.C211T:p.R71X,CWF1...  ...   0   0   0   0   \n",
       "...                                                  ...  ...  ..  ..  ..  ..   \n",
       "16771  CHM:NM_000390:exon3:c.G130T:p.G44X,CHM:NM_0011...  ...   0   0   0   0   \n",
       "16772             GPR143:NM_000273:exon3:c.G455A:p.S152N  ...   0   0   0   0   \n",
       "16773             GPR143:NM_000273:exon3:c.T397C:p.W133R  ...   0   0   0   0   \n",
       "16774             GPR143:NM_000273:exon3:c.T397A:p.W133R  ...   0   0   0   0   \n",
       "\n",
       "      cosmic70            MCAP  REVEL regsnp_fpr regsnp_disease  \\\n",
       "0            .               .      .          .              .   \n",
       "1            .    0.5619750829  0.984          .              .   \n",
       "2            .               .      .          .              .   \n",
       "3            .               .      .          .              .   \n",
       "...        ...             ...    ...        ...            ...   \n",
       "16771        .               .      .          .              .   \n",
       "16772        .  0.676591847234  0.695          .              .   \n",
       "16773        .  0.790856144973  0.400          .              .   \n",
       "16774        .  0.790215722738  0.386          .              .   \n",
       "\n",
       "      regsnp_splicing_site  \n",
       "0                        .  \n",
       "1                        .  \n",
       "2                        .  \n",
       "3                        .  \n",
       "...                    ...  \n",
       "16771                    .  \n",
       "16772                    .  \n",
       "16773                    .  \n",
       "16774                    .  \n",
       "\n",
       "[16775 rows x 149 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_coding_annotated_with_annovar = pd.read_csv('clinvar_pathogenic_coding_from_cadd.csv')\n",
    "clinvar_pathogenic_coding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8.0</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1.0</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31074</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7.0</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>33</td>\n",
       "      <td>827</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.198603</td>\n",
       "      <td>13.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31075</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>33</td>\n",
       "      <td>827</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.198603</td>\n",
       "      <td>13.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31076</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7.0</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>39</td>\n",
       "      <td>652</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.018377</td>\n",
       "      <td>25.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31077</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>39</td>\n",
       "      <td>652</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.018377</td>\n",
       "      <td>25.80</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31078 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos  Ref Alt Type  Length           AnnoType  \\\n",
       "0          1    1014143    C   T  SNV       0   CodingTranscript   \n",
       "1          1    1014143    C   T  SNV       0         Intergenic   \n",
       "2          1    1014143    C   T  SNV       0  RegulatoryFeature   \n",
       "3          1    1014143    C   T  SNV       0  RegulatoryFeature   \n",
       "...      ...        ...  ...  ..  ...     ...                ...   \n",
       "31074      X  155280059    G   C  SNV       0   CodingTranscript   \n",
       "31075      X  155280059    G   C  SNV       0  RegulatoryFeature   \n",
       "31076      X  155506930  GAT   G  DEL       2   CodingTranscript   \n",
       "31077      X  155506930  GAT   G  DEL       2         Transcript   \n",
       "\n",
       "          Consequence  ConsScore         ConsDetail  ...  Freq10000bp  \\\n",
       "0         STOP_GAINED        8.0        stop_gained  ...           22   \n",
       "1            UPSTREAM        1.0           upstream  ...           22   \n",
       "2          REGULATORY        4.0         regulatory  ...           22   \n",
       "3          REGULATORY        4.0         regulatory  ...           22   \n",
       "...               ...        ...                ...  ...          ...   \n",
       "31074  NON_SYNONYMOUS        7.0           missense  ...           11   \n",
       "31075      REGULATORY        4.0         regulatory  ...           11   \n",
       "31076     FRAME_SHIFT        7.0         frameshift  ...            6   \n",
       "31077        INTRONIC        2.0  intron,non_coding  ...            6   \n",
       "\n",
       "       Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0               87         1622                  Promoter                NaN   \n",
       "1               87         1622                  Promoter                NaN   \n",
       "2               87         1622                  Promoter                NaN   \n",
       "3               87         1622                  Promoter                NaN   \n",
       "...            ...          ...                       ...                ...   \n",
       "31074           33          827  Promoter Flanking Region                NaN   \n",
       "31075           33          827  Promoter Flanking Region                NaN   \n",
       "31076           39          652                       NaN                NaN   \n",
       "31077           39          652                       NaN                NaN   \n",
       "\n",
       "       dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore  PHRED  \n",
       "0                   NaN           75.0          107.0  5.410558  35.00  \n",
       "1                   NaN           75.0          107.0  5.410558  35.00  \n",
       "2                   NaN           75.0          107.0  5.410558  35.00  \n",
       "3                   NaN           75.0          107.0  5.410558  35.00  \n",
       "...                 ...            ...            ...       ...    ...  \n",
       "31074               NaN            NaN            NaN  1.198603  13.85  \n",
       "31075               NaN            NaN            NaN  1.198603  13.85  \n",
       "31076               NaN            1.0            1.0  4.018377  25.80  \n",
       "31077               NaN            1.0            1.0  4.018377  25.80  \n",
       "\n",
       "[31078 rows x 125 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_coding_annotated_with_cadd = pd.read_table('clinvar_coding_pathogenic_cadd_annotations_noheader.tsv')\n",
    "clinvar_pathogenic_coding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below are the steps used to combine the annotations for clinvar pathogenic coding variants:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8.0</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1.0</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31074</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7.0</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.138062872375</td>\n",
       "      <td>0.222</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31075</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.138062872375</td>\n",
       "      <td>0.222</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31076</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7.0</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31077</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31078 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos  Ref Alt Type  Length           AnnoType  \\\n",
       "0          1    1014143    C   T  SNV       0   CodingTranscript   \n",
       "1          1    1014143    C   T  SNV       0         Intergenic   \n",
       "2          1    1014143    C   T  SNV       0  RegulatoryFeature   \n",
       "3          1    1014143    C   T  SNV       0  RegulatoryFeature   \n",
       "...      ...        ...  ...  ..  ...     ...                ...   \n",
       "31074      X  155280059    G   C  SNV       0   CodingTranscript   \n",
       "31075      X  155280059    G   C  SNV       0  RegulatoryFeature   \n",
       "31076      X  155506930  GAT   G  DEL       2   CodingTranscript   \n",
       "31077      X  155506930  GAT   G  DEL       2         Transcript   \n",
       "\n",
       "          Consequence  ConsScore         ConsDetail  ...  BP4  BP5  BP6  BP7  \\\n",
       "0         STOP_GAINED        8.0        stop_gained  ...  NaN  NaN  NaN  NaN   \n",
       "1            UPSTREAM        1.0           upstream  ...  NaN  NaN  NaN  NaN   \n",
       "2          REGULATORY        4.0         regulatory  ...  NaN  NaN  NaN  NaN   \n",
       "3          REGULATORY        4.0         regulatory  ...  NaN  NaN  NaN  NaN   \n",
       "...               ...        ...                ...  ...  ...  ...  ...  ...   \n",
       "31074  NON_SYNONYMOUS        7.0           missense  ...    1    0    0    0   \n",
       "31075      REGULATORY        4.0         regulatory  ...    1    0    0    0   \n",
       "31076     FRAME_SHIFT        7.0         frameshift  ...  NaN  NaN  NaN  NaN   \n",
       "31077        INTRONIC        2.0  intron,non_coding  ...  NaN  NaN  NaN  NaN   \n",
       "\n",
       "       cosmic70            MCAP  REVEL regsnp_fpr regsnp_disease  \\\n",
       "0           NaN             NaN    NaN        NaN            NaN   \n",
       "1           NaN             NaN    NaN        NaN            NaN   \n",
       "2           NaN             NaN    NaN        NaN            NaN   \n",
       "3           NaN             NaN    NaN        NaN            NaN   \n",
       "...         ...             ...    ...        ...            ...   \n",
       "31074         .  0.138062872375  0.222          .              .   \n",
       "31075         .  0.138062872375  0.222          .              .   \n",
       "31076       NaN             NaN    NaN        NaN            NaN   \n",
       "31077       NaN             NaN    NaN        NaN            NaN   \n",
       "\n",
       "      regsnp_splicing_site  \n",
       "0                      NaN  \n",
       "1                      NaN  \n",
       "2                      NaN  \n",
       "3                      NaN  \n",
       "...                    ...  \n",
       "31074                    .  \n",
       "31075                    .  \n",
       "31076                  NaN  \n",
       "31077                  NaN  \n",
       "\n",
       "[31078 rows x 270 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_coding_annotated_with_annovar['Chr'] = clinvar_pathogenic_coding_annotated_with_annovar['Chr'].map(lambda x: x.lstrip('chr').rstrip('aAbBcC'))\n",
    "clinvar_pathogenic_coding_annotated_with_annovar = clinvar_pathogenic_coding_annotated_with_annovar.rename(columns = {'Chr': '#Chrom', 'Start': 'Pos'})\n",
    "clinvar_pathogenic_coding_merged = pd.merge(clinvar_pathogenic_coding_annotated_with_cadd, clinvar_pathogenic_coding_annotated_with_annovar,\n",
    "                                         how = 'left', on = ['#Chrom', 'Pos', 'Ref', 'Alt'])\n",
    "clinvar_pathogenic_coding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below we drop the duplicate rows:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8.0</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1.0</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31074</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7.0</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.138062872375</td>\n",
       "      <td>0.222</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31075</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4.0</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.138062872375</td>\n",
       "      <td>0.222</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31076</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7.0</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31077</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31057 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos  Ref Alt Type  Length           AnnoType  \\\n",
       "0          1    1014143    C   T  SNV       0   CodingTranscript   \n",
       "1          1    1014143    C   T  SNV       0         Intergenic   \n",
       "2          1    1014143    C   T  SNV       0  RegulatoryFeature   \n",
       "3          1    1014143    C   T  SNV       0  RegulatoryFeature   \n",
       "...      ...        ...  ...  ..  ...     ...                ...   \n",
       "31074      X  155280059    G   C  SNV       0   CodingTranscript   \n",
       "31075      X  155280059    G   C  SNV       0  RegulatoryFeature   \n",
       "31076      X  155506930  GAT   G  DEL       2   CodingTranscript   \n",
       "31077      X  155506930  GAT   G  DEL       2         Transcript   \n",
       "\n",
       "          Consequence  ConsScore         ConsDetail  ...  BP4  BP5  BP6  BP7  \\\n",
       "0         STOP_GAINED        8.0        stop_gained  ...  NaN  NaN  NaN  NaN   \n",
       "1            UPSTREAM        1.0           upstream  ...  NaN  NaN  NaN  NaN   \n",
       "2          REGULATORY        4.0         regulatory  ...  NaN  NaN  NaN  NaN   \n",
       "3          REGULATORY        4.0         regulatory  ...  NaN  NaN  NaN  NaN   \n",
       "...               ...        ...                ...  ...  ...  ...  ...  ...   \n",
       "31074  NON_SYNONYMOUS        7.0           missense  ...    1    0    0    0   \n",
       "31075      REGULATORY        4.0         regulatory  ...    1    0    0    0   \n",
       "31076     FRAME_SHIFT        7.0         frameshift  ...  NaN  NaN  NaN  NaN   \n",
       "31077        INTRONIC        2.0  intron,non_coding  ...  NaN  NaN  NaN  NaN   \n",
       "\n",
       "       cosmic70            MCAP  REVEL regsnp_fpr regsnp_disease  \\\n",
       "0           NaN             NaN    NaN        NaN            NaN   \n",
       "1           NaN             NaN    NaN        NaN            NaN   \n",
       "2           NaN             NaN    NaN        NaN            NaN   \n",
       "3           NaN             NaN    NaN        NaN            NaN   \n",
       "...         ...             ...    ...        ...            ...   \n",
       "31074         .  0.138062872375  0.222          .              .   \n",
       "31075         .  0.138062872375  0.222          .              .   \n",
       "31076       NaN             NaN    NaN        NaN            NaN   \n",
       "31077       NaN             NaN    NaN        NaN            NaN   \n",
       "\n",
       "      regsnp_splicing_site  \n",
       "0                      NaN  \n",
       "1                      NaN  \n",
       "2                      NaN  \n",
       "3                      NaN  \n",
       "...                    ...  \n",
       "31074                    .  \n",
       "31075                    .  \n",
       "31076                  NaN  \n",
       "31077                  NaN  \n",
       "\n",
       "[31057 rows x 270 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_coding_merged = clinvar_pathogenic_coding_merged.drop_duplicates(keep = 'first')\n",
    "clinvar_pathogenic_coding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### HGMD"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Again we repeat the same processes as above:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100154922</td>\n",
       "      <td>100154922</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ERLIN1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>ERLIN1:NM_001347858:exon9:c.C283T:p.R95X,ERLIN...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100183802</td>\n",
       "      <td>100183802</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ERLIN1</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>ERLIN1:NM_006459:exon2:c.G149T:p.G50V,ERLIN1:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5619750829</td>\n",
       "      <td>0.984</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100253438</td>\n",
       "      <td>100253438</td>\n",
       "      <td>-</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>CWF19L1:NM_001303406:exon3:c.194dupA:p.Y65fs*0...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100256299</td>\n",
       "      <td>100256299</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>exonic</td>\n",
       "      <td>CWF19L1</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>CWF19L1:NM_001303406:exon2:c.56delC:p.P19Hfs*3...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47641</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760731</td>\n",
       "      <td>9760731</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GPR143:NM_000273:exon2:c.T346C:p.C116R</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.172141444647</td>\n",
       "      <td>0.295</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47642</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760732</td>\n",
       "      <td>9760732</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift insertion</td>\n",
       "      <td>GPR143:NM_000273:exon2:c.344dupT:p.C116Lfs*69</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47643</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760737</td>\n",
       "      <td>9760737</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>GPR143:NM_000273:exon2:c.340delG:p.A114Lfs*30</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47644</th>\n",
       "      <td>chrX</td>\n",
       "      <td>9760741</td>\n",
       "      <td>9760741</td>\n",
       "      <td>-</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GPR143</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift insertion</td>\n",
       "      <td>GPR143:NM_000273:exon2:c.335dupC:p.A113Cfs*72</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47645 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0      chr10  100154922  100154922   G   A       exonic       ERLIN1   \n",
       "1      chr10  100183802  100183802   C   A       exonic       ERLIN1   \n",
       "2      chr10  100253438  100253438   -   T       exonic      CWF19L1   \n",
       "3      chr10  100256299  100256299   G   -       exonic      CWF19L1   \n",
       "...      ...        ...        ...  ..  ..          ...          ...   \n",
       "47641   chrX    9760731    9760731   A   G       exonic       GPR143   \n",
       "47642   chrX    9760732    9760732   -   A       exonic       GPR143   \n",
       "47643   chrX    9760737    9760737   C   -       exonic       GPR143   \n",
       "47644   chrX    9760741    9760741   -   G       exonic       GPR143   \n",
       "\n",
       "      GeneDetail.refGene    ExonicFunc.refGene  \\\n",
       "0                      .              stopgain   \n",
       "1                      .     nonsynonymous SNV   \n",
       "2                      .              stopgain   \n",
       "3                      .   frameshift deletion   \n",
       "...                  ...                   ...   \n",
       "47641                  .     nonsynonymous SNV   \n",
       "47642                  .  frameshift insertion   \n",
       "47643                  .   frameshift deletion   \n",
       "47644                  .  frameshift insertion   \n",
       "\n",
       "                                        AAChange.refGene  ... BP4 BP5 BP6 BP7  \\\n",
       "0      ERLIN1:NM_001347858:exon9:c.C283T:p.R95X,ERLIN...  ...   0   0   0   0   \n",
       "1      ERLIN1:NM_006459:exon2:c.G149T:p.G50V,ERLIN1:N...  ...   .   .   .   .   \n",
       "2      CWF19L1:NM_001303406:exon3:c.194dupA:p.Y65fs*0...  ...   .   .   .   .   \n",
       "3      CWF19L1:NM_001303406:exon2:c.56delC:p.P19Hfs*3...  ...   .   .   .   .   \n",
       "...                                                  ...  ...  ..  ..  ..  ..   \n",
       "47641             GPR143:NM_000273:exon2:c.T346C:p.C116R  ...   0   0   0   0   \n",
       "47642      GPR143:NM_000273:exon2:c.344dupT:p.C116Lfs*69  ...   .   .   .   .   \n",
       "47643      GPR143:NM_000273:exon2:c.340delG:p.A114Lfs*30  ...   .   .   .   .   \n",
       "47644      GPR143:NM_000273:exon2:c.335dupC:p.A113Cfs*72  ...   .   .   .   .   \n",
       "\n",
       "      cosmic70            MCAP  REVEL regsnp_fpr regsnp_disease  \\\n",
       "0            .               .      .          .              .   \n",
       "1            .    0.5619750829  0.984          .              .   \n",
       "2            .               .      .          .              .   \n",
       "3            .               .      .          .              .   \n",
       "...        ...             ...    ...        ...            ...   \n",
       "47641        .  0.172141444647  0.295          .              .   \n",
       "47642        .               .      .          .              .   \n",
       "47643        .               .      .          .              .   \n",
       "47644        .               .      .          .              .   \n",
       "\n",
       "      regsnp_splicing_site  \n",
       "0                        .  \n",
       "1                        .  \n",
       "2                        .  \n",
       "3                        .  \n",
       "...                    ...  \n",
       "47641                    .  \n",
       "47642                    .  \n",
       "47643                    .  \n",
       "47644                    .  \n",
       "\n",
       "[47645 rows x 149 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_annotated_with_annovar = pd.read_csv('hgmd_coding_annotated.hg38_multianno.csv')\n",
    "hgmd_coding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (0,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>87</td>\n",
       "      <td>1622</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>107.0</td>\n",
       "      <td>5.410558</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89254</th>\n",
       "      <td>X</td>\n",
       "      <td>154966616</td>\n",
       "      <td>CTTCA</td>\n",
       "      <td>C</td>\n",
       "      <td>DEL</td>\n",
       "      <td>4</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>35</td>\n",
       "      <td>840</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.413846</td>\n",
       "      <td>21.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89255</th>\n",
       "      <td>X</td>\n",
       "      <td>154966617</td>\n",
       "      <td>TTC</td>\n",
       "      <td>T</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>35</td>\n",
       "      <td>840</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.392576</td>\n",
       "      <td>21.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89256</th>\n",
       "      <td>X</td>\n",
       "      <td>155524585</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>26</td>\n",
       "      <td>682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.231926</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89257</th>\n",
       "      <td>X</td>\n",
       "      <td>155524585</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>26</td>\n",
       "      <td>682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.231926</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89258 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos    Ref Alt Type  Length           AnnoType  \\\n",
       "0          1    1014143      C   T  SNV       0   CodingTranscript   \n",
       "1          1    1014143      C   T  SNV       0         Intergenic   \n",
       "2          1    1014143      C   T  SNV       0  RegulatoryFeature   \n",
       "3          1    1014143      C   T  SNV       0  RegulatoryFeature   \n",
       "...      ...        ...    ...  ..  ...     ...                ...   \n",
       "89254      X  154966616  CTTCA   C  DEL       4   CodingTranscript   \n",
       "89255      X  154966617    TTC   T  DEL       2   CodingTranscript   \n",
       "89256      X  155524585      G   A  SNV       0   CodingTranscript   \n",
       "89257      X  155524585      G   A  SNV       0         Transcript   \n",
       "\n",
       "       Consequence  ConsScore         ConsDetail  ...  Freq10000bp  \\\n",
       "0      STOP_GAINED          8        stop_gained  ...           22   \n",
       "1         UPSTREAM          1           upstream  ...           22   \n",
       "2       REGULATORY          4         regulatory  ...           22   \n",
       "3       REGULATORY          4         regulatory  ...           22   \n",
       "...            ...        ...                ...  ...          ...   \n",
       "89254  FRAME_SHIFT          7         frameshift  ...            7   \n",
       "89255  FRAME_SHIFT          7         frameshift  ...            7   \n",
       "89256  STOP_GAINED          8        stop_gained  ...            9   \n",
       "89257     INTRONIC          2  intron,non_coding  ...            9   \n",
       "\n",
       "       Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0               87         1622                  Promoter                NaN   \n",
       "1               87         1622                  Promoter                NaN   \n",
       "2               87         1622                  Promoter                NaN   \n",
       "3               87         1622                  Promoter                NaN   \n",
       "...            ...          ...                       ...                ...   \n",
       "89254           35          840                       NaN                NaN   \n",
       "89255           35          840                       NaN                NaN   \n",
       "89256           26          682                       NaN                NaN   \n",
       "89257           26          682                       NaN                NaN   \n",
       "\n",
       "       dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore PHRED  \n",
       "0                   NaN           75.0          107.0  5.410558  35.0  \n",
       "1                   NaN           75.0          107.0  5.410558  35.0  \n",
       "2                   NaN           75.0          107.0  5.410558  35.0  \n",
       "3                   NaN           75.0          107.0  5.410558  35.0  \n",
       "...                 ...            ...            ...       ...   ...  \n",
       "89254               NaN            NaN            NaN  2.413846  21.5  \n",
       "89255               NaN            NaN            NaN  2.392576  21.3  \n",
       "89256               NaN            4.0            4.0  6.231926  35.0  \n",
       "89257               NaN            4.0            4.0  6.231926  35.0  \n",
       "\n",
       "[89258 rows x 125 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_annotated_with_cadd = pd.read_table('hgmd_coding_cadd_annotations_noheader.tsv')\n",
    "hgmd_coding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below are the steps used to combine the annotations for hgmd pathogenic coding variants:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89288</th>\n",
       "      <td>X</td>\n",
       "      <td>154966616</td>\n",
       "      <td>CTTCA</td>\n",
       "      <td>C</td>\n",
       "      <td>DEL</td>\n",
       "      <td>4</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89289</th>\n",
       "      <td>X</td>\n",
       "      <td>154966617</td>\n",
       "      <td>TTC</td>\n",
       "      <td>T</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89290</th>\n",
       "      <td>X</td>\n",
       "      <td>155524585</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=COSM3424643;OCCURENCE=1(large_intestine)</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89291</th>\n",
       "      <td>X</td>\n",
       "      <td>155524585</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=COSM3424643;OCCURENCE=1(large_intestine)</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89292 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos    Ref Alt Type  Length           AnnoType  \\\n",
       "0          1    1014143      C   T  SNV       0   CodingTranscript   \n",
       "1          1    1014143      C   T  SNV       0         Intergenic   \n",
       "2          1    1014143      C   T  SNV       0  RegulatoryFeature   \n",
       "3          1    1014143      C   T  SNV       0  RegulatoryFeature   \n",
       "...      ...        ...    ...  ..  ...     ...                ...   \n",
       "89288      X  154966616  CTTCA   C  DEL       4   CodingTranscript   \n",
       "89289      X  154966617    TTC   T  DEL       2   CodingTranscript   \n",
       "89290      X  155524585      G   A  SNV       0   CodingTranscript   \n",
       "89291      X  155524585      G   A  SNV       0         Transcript   \n",
       "\n",
       "       Consequence  ConsScore         ConsDetail  ...  BP4  BP5  BP6  BP7  \\\n",
       "0      STOP_GAINED          8        stop_gained  ...  NaN  NaN  NaN  NaN   \n",
       "1         UPSTREAM          1           upstream  ...  NaN  NaN  NaN  NaN   \n",
       "2       REGULATORY          4         regulatory  ...  NaN  NaN  NaN  NaN   \n",
       "3       REGULATORY          4         regulatory  ...  NaN  NaN  NaN  NaN   \n",
       "...            ...        ...                ...  ...  ...  ...  ...  ...   \n",
       "89288  FRAME_SHIFT          7         frameshift  ...  NaN  NaN  NaN  NaN   \n",
       "89289  FRAME_SHIFT          7         frameshift  ...  NaN  NaN  NaN  NaN   \n",
       "89290  STOP_GAINED          8        stop_gained  ...    0    0    0    0   \n",
       "89291     INTRONIC          2  intron,non_coding  ...    0    0    0    0   \n",
       "\n",
       "                                          cosmic70  MCAP REVEL regsnp_fpr  \\\n",
       "0                                              NaN   NaN   NaN        NaN   \n",
       "1                                              NaN   NaN   NaN        NaN   \n",
       "2                                              NaN   NaN   NaN        NaN   \n",
       "3                                              NaN   NaN   NaN        NaN   \n",
       "...                                            ...   ...   ...        ...   \n",
       "89288                                          NaN   NaN   NaN        NaN   \n",
       "89289                                          NaN   NaN   NaN        NaN   \n",
       "89290  ID=COSM3424643;OCCURENCE=1(large_intestine)     .     .          .   \n",
       "89291  ID=COSM3424643;OCCURENCE=1(large_intestine)     .     .          .   \n",
       "\n",
       "      regsnp_disease regsnp_splicing_site  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "...              ...                  ...  \n",
       "89288            NaN                  NaN  \n",
       "89289            NaN                  NaN  \n",
       "89290              .                    .  \n",
       "89291              .                    .  \n",
       "\n",
       "[89292 rows x 270 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_annotated_with_annovar['Chr'] = hgmd_coding_annotated_with_annovar['Chr'].map(lambda x: x.lstrip('chr').rstrip('aAbBcC'))\n",
    "hgmd_coding_annotated_with_annovar = hgmd_coding_annotated_with_annovar.rename(columns = {'Chr': '#Chrom', 'Start': 'Pos'})\n",
    "hgmd_coding_merged = pd.merge(hgmd_coding_annotated_with_cadd, hgmd_coding_annotated_with_annovar,\n",
    "                                         how = 'left', on = ['#Chrom', 'Pos', 'Ref', 'Alt'])\n",
    "hgmd_coding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have dropped the duplicates which contain the same values for all annotations as another row, which means they are the exact same and therefore should be removed. I have kept one of them, as to not disregard the variant completely."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89288</th>\n",
       "      <td>X</td>\n",
       "      <td>154966616</td>\n",
       "      <td>CTTCA</td>\n",
       "      <td>C</td>\n",
       "      <td>DEL</td>\n",
       "      <td>4</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89289</th>\n",
       "      <td>X</td>\n",
       "      <td>154966617</td>\n",
       "      <td>TTC</td>\n",
       "      <td>T</td>\n",
       "      <td>DEL</td>\n",
       "      <td>2</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89290</th>\n",
       "      <td>X</td>\n",
       "      <td>155524585</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=COSM3424643;OCCURENCE=1(large_intestine)</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89291</th>\n",
       "      <td>X</td>\n",
       "      <td>155524585</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=COSM3424643;OCCURENCE=1(large_intestine)</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>89236 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos    Ref Alt Type  Length           AnnoType  \\\n",
       "0          1    1014143      C   T  SNV       0   CodingTranscript   \n",
       "1          1    1014143      C   T  SNV       0         Intergenic   \n",
       "2          1    1014143      C   T  SNV       0  RegulatoryFeature   \n",
       "3          1    1014143      C   T  SNV       0  RegulatoryFeature   \n",
       "...      ...        ...    ...  ..  ...     ...                ...   \n",
       "89288      X  154966616  CTTCA   C  DEL       4   CodingTranscript   \n",
       "89289      X  154966617    TTC   T  DEL       2   CodingTranscript   \n",
       "89290      X  155524585      G   A  SNV       0   CodingTranscript   \n",
       "89291      X  155524585      G   A  SNV       0         Transcript   \n",
       "\n",
       "       Consequence  ConsScore         ConsDetail  ...  BP4  BP5  BP6  BP7  \\\n",
       "0      STOP_GAINED          8        stop_gained  ...  NaN  NaN  NaN  NaN   \n",
       "1         UPSTREAM          1           upstream  ...  NaN  NaN  NaN  NaN   \n",
       "2       REGULATORY          4         regulatory  ...  NaN  NaN  NaN  NaN   \n",
       "3       REGULATORY          4         regulatory  ...  NaN  NaN  NaN  NaN   \n",
       "...            ...        ...                ...  ...  ...  ...  ...  ...   \n",
       "89288  FRAME_SHIFT          7         frameshift  ...  NaN  NaN  NaN  NaN   \n",
       "89289  FRAME_SHIFT          7         frameshift  ...  NaN  NaN  NaN  NaN   \n",
       "89290  STOP_GAINED          8        stop_gained  ...    0    0    0    0   \n",
       "89291     INTRONIC          2  intron,non_coding  ...    0    0    0    0   \n",
       "\n",
       "                                          cosmic70  MCAP REVEL regsnp_fpr  \\\n",
       "0                                              NaN   NaN   NaN        NaN   \n",
       "1                                              NaN   NaN   NaN        NaN   \n",
       "2                                              NaN   NaN   NaN        NaN   \n",
       "3                                              NaN   NaN   NaN        NaN   \n",
       "...                                            ...   ...   ...        ...   \n",
       "89288                                          NaN   NaN   NaN        NaN   \n",
       "89289                                          NaN   NaN   NaN        NaN   \n",
       "89290  ID=COSM3424643;OCCURENCE=1(large_intestine)     .     .          .   \n",
       "89291  ID=COSM3424643;OCCURENCE=1(large_intestine)     .     .          .   \n",
       "\n",
       "      regsnp_disease regsnp_splicing_site  \n",
       "0                NaN                  NaN  \n",
       "1                NaN                  NaN  \n",
       "2                NaN                  NaN  \n",
       "3                NaN                  NaN  \n",
       "...              ...                  ...  \n",
       "89288            NaN                  NaN  \n",
       "89289            NaN                  NaN  \n",
       "89290              .                    .  \n",
       "89291              .                    .  \n",
       "\n",
       "[89236 rows x 270 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_coding_merged = hgmd_coding_merged.drop_duplicates(keep = 'first')\n",
    "hgmd_coding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is the final pathogenic coding region dataframe:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31074</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.138062872375</td>\n",
       "      <td>0.222</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31075</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.138062872375</td>\n",
       "      <td>0.222</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31076</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31077</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>120293 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos  Ref Alt Type  ...            MCAP  REVEL regsnp_fpr  \\\n",
       "0          1    1014143    C   T  SNV  ...             NaN    NaN        NaN   \n",
       "1          1    1014143    C   T  SNV  ...             NaN    NaN        NaN   \n",
       "2          1    1014143    C   T  SNV  ...             NaN    NaN        NaN   \n",
       "3          1    1014143    C   T  SNV  ...             NaN    NaN        NaN   \n",
       "...      ...        ...  ...  ..  ...  ...             ...    ...        ...   \n",
       "31074      X  155280059    G   C  SNV  ...  0.138062872375  0.222          .   \n",
       "31075      X  155280059    G   C  SNV  ...  0.138062872375  0.222          .   \n",
       "31076      X  155506930  GAT   G  DEL  ...             NaN    NaN        NaN   \n",
       "31077      X  155506930  GAT   G  DEL  ...             NaN    NaN        NaN   \n",
       "\n",
       "       regsnp_disease regsnp_splicing_site  \n",
       "0                 NaN                  NaN  \n",
       "1                 NaN                  NaN  \n",
       "2                 NaN                  NaN  \n",
       "3                 NaN                  NaN  \n",
       "...               ...                  ...  \n",
       "31074               .                    .  \n",
       "31075               .                    .  \n",
       "31076             NaN                  NaN  \n",
       "31077             NaN                  NaN  \n",
       "\n",
       "[120293 rows x 270 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_coding_region_variants = pd.concat([hgmd_coding_merged, clinvar_pathogenic_coding_merged])\n",
    "pathogenic_coding_region_variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below we have dropped unneccessary columns we do not need for the model:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31074</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.138062872375</td>\n",
       "      <td>0.222</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31075</th>\n",
       "      <td>X</td>\n",
       "      <td>155280059</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.138062872375</td>\n",
       "      <td>0.222</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31076</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31077</th>\n",
       "      <td>X</td>\n",
       "      <td>155506930</td>\n",
       "      <td>GAT</td>\n",
       "      <td>G</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>120293 rows × 241 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Chrom        Pos  Ref Alt Type  ...            MCAP  REVEL regsnp_fpr  \\\n",
       "0          1    1014143    C   T  SNV  ...             NaN    NaN        NaN   \n",
       "1          1    1014143    C   T  SNV  ...             NaN    NaN        NaN   \n",
       "2          1    1014143    C   T  SNV  ...             NaN    NaN        NaN   \n",
       "3          1    1014143    C   T  SNV  ...             NaN    NaN        NaN   \n",
       "...      ...        ...  ...  ..  ...  ...             ...    ...        ...   \n",
       "31074      X  155280059    G   C  SNV  ...  0.138062872375  0.222          .   \n",
       "31075      X  155280059    G   C  SNV  ...  0.138062872375  0.222          .   \n",
       "31076      X  155506930  GAT   G  DEL  ...             NaN    NaN        NaN   \n",
       "31077      X  155506930  GAT   G  DEL  ...             NaN    NaN        NaN   \n",
       "\n",
       "       regsnp_disease regsnp_splicing_site  \n",
       "0                 NaN                  NaN  \n",
       "1                 NaN                  NaN  \n",
       "2                 NaN                  NaN  \n",
       "3                 NaN                  NaN  \n",
       "...               ...                  ...  \n",
       "31074               .                    .  \n",
       "31075               .                    .  \n",
       "31076             NaN                  NaN  \n",
       "31077             NaN                  NaN  \n",
       "\n",
       "[120293 rows x 241 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_coding_region_variants = pathogenic_coding_region_variants.drop(columns = ['PolyPhenVal', 'priPhCons', 'cHmm_E1', 'cHmm_E2', 'cHmm_E3', 'cHmm_E4', 'cHmm_E5', 'cHmm_E6', 'cHmm_E7', 'cHmm_E8', 'cHmm_E9', 'cHmm_E10', 'cHmm_E11', 'cHmm_E12', 'cHmm_E13', 'cHmm_E14', 'cHmm_E15', 'cHmm_E16', 'cHmm_E17', 'cHmm_E18', 'cHmm_E19', 'cHmm_E20', 'cHmm_E21', 'cHmm_E22', 'cHmm_E23', 'cHmm_E24', 'cHmm_E25', 'GerpN', 'GerpS'])\n",
    "pathogenic_coding_region_variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I have replaced the values specified by '.' by NaN values as done before and changed all dtypes of numerical values to float, shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic_coding_region_variants_with_nan = pathogenic_coding_region_variants.replace('.', np.nan)\n",
    "pathogenic_coding_region_variants_with_nan = pathogenic_coding_region_variants_with_nan.astype({'SIFT_score': float, 'Polyphen2_HDIV_score' : float, \n",
    "                'Polyphen2_HVAR_score' : float, 'LRT_score' : float, \n",
    "                'MutationTaster_score' : float, 'MutationAssessor_score' : float, \n",
    "                'FATHMM_score' : float, 'RadialSVM_score' : float,\n",
    "               'LR_score' : float, 'VEST3_score' : float, 'CADD_raw' : float,\n",
    "               'CADD_phred' : float, 'GERP++_RS' : float, 'phyloP46way_placental' : float,\n",
    "               'phyloP100way_vertebrate' : float, 'SiPhy_29way_logOdds' : float, 'SIFT_score.1' : float,\n",
    "               'SIFT_converted_rankscore' : float, 'Polyphen2_HDIV_score.1' : float, 'Polyphen2_HDIV_rankscore' : float,\n",
    "               'Polyphen2_HVAR_score.1' : float, 'Polyphen2_HVAR_rankscore' : float, 'LRT_score.1' : float,\n",
    "               'LRT_converted_rankscore' : float, 'MutationTaster_score.1' : float, 'MutationTaster_converted_rankscore' : float,\n",
    "               'MutationAssessor_score.1' : float, 'MutationAssessor_score_rankscore' : float, 'FATHMM_score.1' : float,\n",
    "               'FATHMM_converted_rankscore' : float, 'PROVEAN_score' : float, 'PROVEAN_converted_rankscore' : float,\n",
    "               'VEST3_score.1' : float, 'VEST3_rankscore' : float, 'MetaSVM_score' : float, 'MetaSVM_rankscore' : float,\n",
    "               'MetaLR_score' : float, 'MetaLR_rankscore' : float, 'M-CAP_score' : float, 'M-CAP_rankscore' : float,\n",
    "               'CADD_raw.1' : float, 'CADD_raw_rankscore' : float, 'CADD_phred.1' : float, 'DANN_score' : float,\n",
    "               'DANN_rankscore' : float, 'fathmm-MKL_coding_score' : float, 'fathmm-MKL_coding_rankscore' : float,\n",
    "               'Eigen-raw' : float, 'Eigen-PC-raw' : float, 'GenoCanyon_score' : float, 'GenoCanyon_score_rankscore' : float,\n",
    "               'integrated_fitCons_score' : float, 'integrated_fitCons_score_rankscore' : float, 'integrated_confidence_value' : float,\n",
    "               'GERP++_RS.1' : float, 'GERP++_RS_rankscore' : float, 'phyloP100way_vertebrate.1' : float, \n",
    "                'phyloP100way_vertebrate_rankscore' : float, 'phyloP20way_mammalian' : float, 'phyloP20way_mammalian_rankscore' : float,\n",
    "               'phastCons100way_vertebrate' : float, 'phastCons100way_vertebrate_rankscore' : float, 'phastCons20way_mammalian' : float,\n",
    "               'phastCons20way_mammalian_rankscore' : float, 'SiPhy_29way_logOdds.1' : float, 'SiPhy_29way_logOdds_rankscore' : float,\n",
    "               'dbscSNV_ADA_SCORE' : float, 'dbscSNV_RF_SCORE' : float, 'PVS1' : float, 'PS1' : float, 'PS2' : float, 'PS3' : float,\n",
    "               'PS4' : float, 'PM1' : float, 'PM2' : float, 'PM3' : float, 'PM4' : float, 'PM5' : float, 'PM6' : float, 'PP1' : float, 'PP2' : float,\n",
    "               'PP3' : float, 'PP4' : float, 'PP5' : float, 'BA1' : float, 'BS1' : float, 'BS2' : float, 'BS3' : float, 'BS4' : float, 'BP1' : float, 'BP2' : float,\n",
    "               'BP3' : float, 'BP4' : float, 'BP5' : float, 'BP6' : float, 'BP7' : float, 'MCAP' : float, 'REVEL' : float, 'regsnp_fpr' : float,\n",
    "                'dbscSNV-rf_score' : float})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have saved the resulting file for use in other notebooks:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic_coding_region_variants_with_nan.to_csv('pathogenic_coding_region_variants.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pathogenic NonCoding"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### HGMD"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Following the pipeline again:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988295</td>\n",
       "      <td>100988295</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.C85T:p.R29X,TWNK:NM_...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988415</td>\n",
       "      <td>100988415</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.A205T:p.I69F,TWNK:NM...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0927775052736</td>\n",
       "      <td>0.592</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988457</td>\n",
       "      <td>100988457</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.C247T:p.P83S,TWNK:NM...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.381929214361</td>\n",
       "      <td>0.584</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988526</td>\n",
       "      <td>100988526</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.A316G:p.K106E,TWNK:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.207740715004</td>\n",
       "      <td>0.787</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3803</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964053</td>\n",
       "      <td>85964053</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>splicing</td>\n",
       "      <td>CHM</td>\n",
       "      <td>NM_000390:exon5:c.315-1G&gt;C;NM_001320959:exon5:...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.019730941704</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3804</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85964054</td>\n",
       "      <td>85964054</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>splicing</td>\n",
       "      <td>CHM</td>\n",
       "      <td>NM_000390:exon5:c.315-2A&gt;G;NM_001320959:exon5:...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0206278026906</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3805</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85965588</td>\n",
       "      <td>85965588</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3806</th>\n",
       "      <td>chrX</td>\n",
       "      <td>85968639</td>\n",
       "      <td>85968639</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>CHM</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3807 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0     chr10  100988295  100988295   C   T       exonic         TWNK   \n",
       "1     chr10  100988415  100988415   A   T       exonic         TWNK   \n",
       "2     chr10  100988457  100988457   C   T       exonic         TWNK   \n",
       "3     chr10  100988526  100988526   A   G       exonic         TWNK   \n",
       "...     ...        ...        ...  ..  ..          ...          ...   \n",
       "3803   chrX   85964053   85964053   C   G     splicing          CHM   \n",
       "3804   chrX   85964054   85964054   T   C     splicing          CHM   \n",
       "3805   chrX   85965588   85965588   T   C     intronic          CHM   \n",
       "3806   chrX   85968639   85968639   A   T     intronic          CHM   \n",
       "\n",
       "                                     GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                                                     .           stopgain   \n",
       "1                                                     .  nonsynonymous SNV   \n",
       "2                                                     .  nonsynonymous SNV   \n",
       "3                                                     .  nonsynonymous SNV   \n",
       "...                                                 ...                ...   \n",
       "3803  NM_000390:exon5:c.315-1G>C;NM_001320959:exon5:...                  .   \n",
       "3804  NM_000390:exon5:c.315-2A>G;NM_001320959:exon5:...                  .   \n",
       "3805                                                  .                  .   \n",
       "3806                                                  .                  .   \n",
       "\n",
       "                                       AAChange.refGene  ... BP4 BP5 BP6 BP7  \\\n",
       "0     TWNK:NM_001163812:exon1:c.C85T:p.R29X,TWNK:NM_...  ...   0   0   0   0   \n",
       "1     TWNK:NM_001163812:exon1:c.A205T:p.I69F,TWNK:NM...  ...   0   0   0   0   \n",
       "2     TWNK:NM_001163812:exon1:c.C247T:p.P83S,TWNK:NM...  ...   0   0   0   0   \n",
       "3     TWNK:NM_001163812:exon1:c.A316G:p.K106E,TWNK:N...  ...   0   0   0   0   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "3803                                                  .  ...   .   .   .   .   \n",
       "3804                                                  .  ...   .   .   .   .   \n",
       "3805                                                  .  ...   .   .   .   .   \n",
       "3806                                                  .  ...   .   .   .   .   \n",
       "\n",
       "     cosmic70             MCAP  REVEL       regsnp_fpr regsnp_disease  \\\n",
       "0           .                .      .                .              .   \n",
       "1           .  0.0927775052736  0.592                .              .   \n",
       "2           .   0.381929214361  0.584                .              .   \n",
       "3           .   0.207740715004  0.787                .              .   \n",
       "...       ...              ...    ...              ...            ...   \n",
       "3803        .                .      .   0.019730941704              D   \n",
       "3804        .                .      .  0.0206278026906              D   \n",
       "3805        .                .      .                .              .   \n",
       "3806        .                .      .                .              .   \n",
       "\n",
       "     regsnp_splicing_site  \n",
       "0                       .  \n",
       "1                       .  \n",
       "2                       .  \n",
       "3                       .  \n",
       "...                   ...  \n",
       "3803                   on  \n",
       "3804                   on  \n",
       "3805                    .  \n",
       "3806                    .  \n",
       "\n",
       "[3807 rows x 149 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_annotated_with_annovar = pd.read_csv('hgmd_noncoding_annotated.hg38_multianno.csv')\n",
    "hgmd_noncoding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/opt/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (13,120) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>26</td>\n",
       "      <td>48</td>\n",
       "      <td>1451</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>208.0</td>\n",
       "      <td>475.0</td>\n",
       "      <td>1.043084</td>\n",
       "      <td>12.600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>26</td>\n",
       "      <td>48</td>\n",
       "      <td>1451</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>208.0</td>\n",
       "      <td>475.0</td>\n",
       "      <td>1.043084</td>\n",
       "      <td>12.600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>74</td>\n",
       "      <td>1406</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.014</td>\n",
       "      <td>12.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-0.385227</td>\n",
       "      <td>0.135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>74</td>\n",
       "      <td>1406</td>\n",
       "      <td>CTCF Binding Site</td>\n",
       "      <td>0.00003</td>\n",
       "      <td>0.014</td>\n",
       "      <td>12.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>-0.385227</td>\n",
       "      <td>0.135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8609</th>\n",
       "      <td>X</td>\n",
       "      <td>154965965</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>36</td>\n",
       "      <td>843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99521</td>\n",
       "      <td>0.924</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.271896</td>\n",
       "      <td>20.600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8610</th>\n",
       "      <td>X</td>\n",
       "      <td>154965965</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>36</td>\n",
       "      <td>843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98832</td>\n",
       "      <td>0.926</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.302688</td>\n",
       "      <td>20.800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8611</th>\n",
       "      <td>X</td>\n",
       "      <td>155492384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.463914</td>\n",
       "      <td>21.700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8612</th>\n",
       "      <td>X</td>\n",
       "      <td>155492384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.463914</td>\n",
       "      <td>21.700</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8613 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0         1    7961859   C   G  SNV       0  RegulatoryFeature   \n",
       "1         1    7961859   C   G  SNV       0         Transcript   \n",
       "2         1    9720021   G   A  SNV       0  RegulatoryFeature   \n",
       "3         1    9720021   G   A  SNV       0         Transcript   \n",
       "...     ...        ...  ..  ..  ...     ...                ...   \n",
       "8609      X  154965965   C   G  SNV       0         Transcript   \n",
       "8610      X  154965965   C   T  SNV       0         Transcript   \n",
       "8611      X  155492384   C   A  SNV       0   CodingTranscript   \n",
       "8612      X  155492384   C   A  SNV       0         Transcript   \n",
       "\n",
       "         Consequence  ConsScore         ConsDetail  ...  Freq10000bp  \\\n",
       "0         REGULATORY          4         regulatory  ...           26   \n",
       "1           INTRONIC          2             intron  ...           26   \n",
       "2         REGULATORY          4         regulatory  ...            9   \n",
       "3        SPLICE_SITE          5      splice,intron  ...            9   \n",
       "...              ...        ...                ...  ...          ...   \n",
       "8609     SPLICE_SITE          5      splice,intron  ...            7   \n",
       "8610     SPLICE_SITE          5      splice,intron  ...            7   \n",
       "8611  NON_SYNONYMOUS          7           missense  ...            3   \n",
       "8612        INTRONIC          2  intron,non_coding  ...            3   \n",
       "\n",
       "      Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              48         1451                  Promoter                NaN   \n",
       "1              48         1451                  Promoter                NaN   \n",
       "2              74         1406         CTCF Binding Site            0.00003   \n",
       "3              74         1406         CTCF Binding Site            0.00003   \n",
       "...           ...          ...                       ...                ...   \n",
       "8609           36          843                       NaN            0.99521   \n",
       "8610           36          843                       NaN            0.98832   \n",
       "8611            3           19                       NaN                NaN   \n",
       "8612            3           19                       NaN                NaN   \n",
       "\n",
       "      dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore   PHRED  \n",
       "0                  NaN          208.0          475.0  1.043084  12.600  \n",
       "1                  NaN          208.0          475.0  1.043084  12.600  \n",
       "2                0.014           12.0           15.0 -0.385227   0.135  \n",
       "3                0.014           12.0           15.0 -0.385227   0.135  \n",
       "...                ...            ...            ...       ...     ...  \n",
       "8609             0.924            NaN            NaN  2.271896  20.600  \n",
       "8610             0.926            NaN            NaN  2.302688  20.800  \n",
       "8611               NaN            NaN            NaN  2.463914  21.700  \n",
       "8612               NaN            NaN            NaN  2.463914  21.700  \n",
       "\n",
       "[8613 rows x 125 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_annotated_with_cadd = pd.read_table('hgmd_noncoding_cadd_annotations_noheader.tsv')\n",
    "hgmd_noncoding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8674</th>\n",
       "      <td>X</td>\n",
       "      <td>154965965</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8675</th>\n",
       "      <td>X</td>\n",
       "      <td>154965965</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8676</th>\n",
       "      <td>X</td>\n",
       "      <td>155492384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.651</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8677</th>\n",
       "      <td>X</td>\n",
       "      <td>155492384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.651</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8678 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0         1    7961859   C   G  SNV       0  RegulatoryFeature   \n",
       "1         1    7961859   C   G  SNV       0         Transcript   \n",
       "2         1    9720021   G   A  SNV       0  RegulatoryFeature   \n",
       "3         1    9720021   G   A  SNV       0         Transcript   \n",
       "...     ...        ...  ..  ..  ...     ...                ...   \n",
       "8674      X  154965965   C   G  SNV       0         Transcript   \n",
       "8675      X  154965965   C   T  SNV       0         Transcript   \n",
       "8676      X  155492384   C   A  SNV       0   CodingTranscript   \n",
       "8677      X  155492384   C   A  SNV       0         Transcript   \n",
       "\n",
       "         Consequence  ConsScore         ConsDetail  ...  BP4  BP5  BP6 BP7  \\\n",
       "0         REGULATORY          4         regulatory  ...    .    .    .   .   \n",
       "1           INTRONIC          2             intron  ...    .    .    .   .   \n",
       "2         REGULATORY          4         regulatory  ...    .    .    .   .   \n",
       "3        SPLICE_SITE          5      splice,intron  ...    .    .    .   .   \n",
       "...              ...        ...                ...  ...  ...  ...  ...  ..   \n",
       "8674     SPLICE_SITE          5      splice,intron  ...    .    .    .   .   \n",
       "8675     SPLICE_SITE          5      splice,intron  ...    .    .    .   .   \n",
       "8676  NON_SYNONYMOUS          7           missense  ...    0    0    0   0   \n",
       "8677        INTRONIC          2  intron,non_coding  ...    0    0    0   0   \n",
       "\n",
       "      cosmic70  MCAP  REVEL regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0            .     .      .          .              .                    .  \n",
       "1            .     .      .          .              .                    .  \n",
       "2            .     .      .          .              .                    .  \n",
       "3            .     .      .          .              .                    .  \n",
       "...        ...   ...    ...        ...            ...                  ...  \n",
       "8674         .     .      .        0.0              D                   on  \n",
       "8675         .     .      .        0.0              D                   on  \n",
       "8676         .     .  0.651          .              .                    .  \n",
       "8677         .     .  0.651          .              .                    .  \n",
       "\n",
       "[8678 rows x 270 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_annotated_with_annovar['Chr'] = hgmd_noncoding_annotated_with_annovar['Chr'].map(lambda x: x.lstrip('chr').rstrip('aAbBcC'))\n",
    "hgmd_noncoding_annotated_with_annovar = hgmd_noncoding_annotated_with_annovar.rename(columns = {'Chr': '#Chrom', 'Start': 'Pos'})\n",
    "hgmd_noncoding_merged = pd.merge(hgmd_noncoding_annotated_with_cadd, hgmd_noncoding_annotated_with_annovar,\n",
    "                                         how = 'left', on = ['#Chrom', 'Pos', 'Ref', 'Alt'])\n",
    "hgmd_noncoding_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8674</th>\n",
       "      <td>X</td>\n",
       "      <td>154965965</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8675</th>\n",
       "      <td>X</td>\n",
       "      <td>154965965</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>SPLICE_SITE</td>\n",
       "      <td>5</td>\n",
       "      <td>splice,intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8676</th>\n",
       "      <td>X</td>\n",
       "      <td>155492384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.651</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8677</th>\n",
       "      <td>X</td>\n",
       "      <td>155492384</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron,non_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.651</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8613 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0         1    7961859   C   G  SNV       0  RegulatoryFeature   \n",
       "1         1    7961859   C   G  SNV       0         Transcript   \n",
       "2         1    9720021   G   A  SNV       0  RegulatoryFeature   \n",
       "3         1    9720021   G   A  SNV       0         Transcript   \n",
       "...     ...        ...  ..  ..  ...     ...                ...   \n",
       "8674      X  154965965   C   G  SNV       0         Transcript   \n",
       "8675      X  154965965   C   T  SNV       0         Transcript   \n",
       "8676      X  155492384   C   A  SNV       0   CodingTranscript   \n",
       "8677      X  155492384   C   A  SNV       0         Transcript   \n",
       "\n",
       "         Consequence  ConsScore         ConsDetail  ...  BP4  BP5  BP6 BP7  \\\n",
       "0         REGULATORY          4         regulatory  ...    .    .    .   .   \n",
       "1           INTRONIC          2             intron  ...    .    .    .   .   \n",
       "2         REGULATORY          4         regulatory  ...    .    .    .   .   \n",
       "3        SPLICE_SITE          5      splice,intron  ...    .    .    .   .   \n",
       "...              ...        ...                ...  ...  ...  ...  ...  ..   \n",
       "8674     SPLICE_SITE          5      splice,intron  ...    .    .    .   .   \n",
       "8675     SPLICE_SITE          5      splice,intron  ...    .    .    .   .   \n",
       "8676  NON_SYNONYMOUS          7           missense  ...    0    0    0   0   \n",
       "8677        INTRONIC          2  intron,non_coding  ...    0    0    0   0   \n",
       "\n",
       "      cosmic70  MCAP  REVEL regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0            .     .      .          .              .                    .  \n",
       "1            .     .      .          .              .                    .  \n",
       "2            .     .      .          .              .                    .  \n",
       "3            .     .      .          .              .                    .  \n",
       "...        ...   ...    ...        ...            ...                  ...  \n",
       "8674         .     .      .        0.0              D                   on  \n",
       "8675         .     .      .        0.0              D                   on  \n",
       "8676         .     .  0.651          .              .                    .  \n",
       "8677         .     .  0.651          .              .                    .  \n",
       "\n",
       "[8613 rows x 270 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgmd_noncoding_merged = hgmd_noncoding_merged.drop_duplicates(keep = 'first')\n",
    "hgmd_noncoding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### ClinVar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100988541</td>\n",
       "      <td>100988541</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift deletion</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.331delT:p.L112Sfs*2,...</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989084</td>\n",
       "      <td>100989084</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.C874A:p.P292T,TWNK:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.153055471878</td>\n",
       "      <td>0.729</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989118</td>\n",
       "      <td>100989118</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.G908A:p.R303Q,TWNK:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0960831061338</td>\n",
       "      <td>0.654</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100989154</td>\n",
       "      <td>100989154</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TWNK</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TWNK:NM_001163812:exon1:c.G944T:p.W315L,TWNK:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.328561992494</td>\n",
       "      <td>0.803</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1110</th>\n",
       "      <td>chrX</td>\n",
       "      <td>74529428</td>\n",
       "      <td>74529428</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SLC16A2</td>\n",
       "      <td>.</td>\n",
       "      <td>frameshift insertion</td>\n",
       "      <td>SLC16A2:NM_006517:exon5:c.1387dupC:p.I465Hfs*51</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1111</th>\n",
       "      <td>chrX</td>\n",
       "      <td>77902641</td>\n",
       "      <td>77902641</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>splicing</td>\n",
       "      <td>COX7B</td>\n",
       "      <td>NM_001866:exon2:c.41-2A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00358744394619</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1112</th>\n",
       "      <td>chrX</td>\n",
       "      <td>78003237</td>\n",
       "      <td>78003237</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>splicing</td>\n",
       "      <td>ATP7A</td>\n",
       "      <td>NM_001282224:exon6:c.1707+1G&gt;A;NM_000052:exon6...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00179372197309</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1113</th>\n",
       "      <td>chrX</td>\n",
       "      <td>78122954</td>\n",
       "      <td>78122954</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PGK1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>D</td>\n",
       "      <td>on</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1114 rows × 149 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0     chr10  100988541  100988541   T   -       exonic         TWNK   \n",
       "1     chr10  100989084  100989084   C   A       exonic         TWNK   \n",
       "2     chr10  100989118  100989118   G   A       exonic         TWNK   \n",
       "3     chr10  100989154  100989154   G   T       exonic         TWNK   \n",
       "...     ...        ...        ...  ..  ..          ...          ...   \n",
       "1110   chrX   74529428   74529428   -   C       exonic      SLC16A2   \n",
       "1111   chrX   77902641   77902641   A   G     splicing        COX7B   \n",
       "1112   chrX   78003237   78003237   G   A     splicing        ATP7A   \n",
       "1113   chrX   78122954   78122954   G   A     intronic         PGK1   \n",
       "\n",
       "                                     GeneDetail.refGene    ExonicFunc.refGene  \\\n",
       "0                                                     .   frameshift deletion   \n",
       "1                                                     .     nonsynonymous SNV   \n",
       "2                                                     .     nonsynonymous SNV   \n",
       "3                                                     .     nonsynonymous SNV   \n",
       "...                                                 ...                   ...   \n",
       "1110                                                  .  frameshift insertion   \n",
       "1111                          NM_001866:exon2:c.41-2A>G                     .   \n",
       "1112  NM_001282224:exon6:c.1707+1G>A;NM_000052:exon6...                     .   \n",
       "1113                                                  .                     .   \n",
       "\n",
       "                                       AAChange.refGene  ... BP4 BP5 BP6 BP7  \\\n",
       "0     TWNK:NM_001163812:exon1:c.331delT:p.L112Sfs*2,...  ...   .   .   .   .   \n",
       "1     TWNK:NM_001163812:exon1:c.C874A:p.P292T,TWNK:N...  ...   0   0   0   0   \n",
       "2     TWNK:NM_001163812:exon1:c.G908A:p.R303Q,TWNK:N...  ...   0   0   0   0   \n",
       "3     TWNK:NM_001163812:exon1:c.G944T:p.W315L,TWNK:N...  ...   0   0   0   0   \n",
       "...                                                 ...  ...  ..  ..  ..  ..   \n",
       "1110    SLC16A2:NM_006517:exon5:c.1387dupC:p.I465Hfs*51  ...   .   .   .   .   \n",
       "1111                                                  .  ...   .   .   .   .   \n",
       "1112                                                  .  ...   .   .   .   .   \n",
       "1113                                                  .  ...   .   .   .   .   \n",
       "\n",
       "     cosmic70             MCAP  REVEL        regsnp_fpr regsnp_disease  \\\n",
       "0           .                .      .                 .              .   \n",
       "1           .   0.153055471878  0.729                 .              .   \n",
       "2           .  0.0960831061338  0.654                 .              .   \n",
       "3           .   0.328561992494  0.803                 .              .   \n",
       "...       ...              ...    ...               ...            ...   \n",
       "1110        .                .      .                 .              .   \n",
       "1111        .                .      .  0.00358744394619              D   \n",
       "1112        .                .      .  0.00179372197309              D   \n",
       "1113        .                .      .               0.0              D   \n",
       "\n",
       "     regsnp_splicing_site  \n",
       "0                       .  \n",
       "1                       .  \n",
       "2                       .  \n",
       "3                       .  \n",
       "...                   ...  \n",
       "1110                    .  \n",
       "1111                   on  \n",
       "1112                   on  \n",
       "1113                   on  \n",
       "\n",
       "[1114 rows x 149 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_noncoding_annotated_with_annovar = pd.read_csv('clinvar.hg38_multianno.csv')\n",
    "clinvar_pathogenic_noncoding_annotated_with_annovar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq10000bp</th>\n",
       "      <th>Rare10000bp</th>\n",
       "      <th>Sngl10000bp</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>RemapOverlapCL</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>11960768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>90</td>\n",
       "      <td>1434</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>0.99998</td>\n",
       "      <td>0.942</td>\n",
       "      <td>27.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>5.268674</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11960768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_donor</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>90</td>\n",
       "      <td>1434</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>0.99998</td>\n",
       "      <td>0.942</td>\n",
       "      <td>27.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>5.268674</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>11964787</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_donor</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>99</td>\n",
       "      <td>1332</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99987</td>\n",
       "      <td>0.886</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>4.686023</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>11972977</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>95</td>\n",
       "      <td>1304</td>\n",
       "      <td>Promoter Flanking Region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>7.497634</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2641</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>splice,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>7.476349</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2642</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>7.476349</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2643</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "      <td>828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>0.944</td>\n",
       "      <td>38.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>4.640280</td>\n",
       "      <td>29.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2644</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_acceptor</td>\n",
       "      <td>...</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "      <td>828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99999</td>\n",
       "      <td>0.944</td>\n",
       "      <td>38.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>4.640280</td>\n",
       "      <td>29.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2645 rows × 125 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0         1   11960768   G   A  SNV       0  RegulatoryFeature   \n",
       "1         1   11960768   G   A  SNV       0         Transcript   \n",
       "2         1   11964787   T   C  SNV       0         Transcript   \n",
       "3         1   11972977   C   T  SNV       0   CodingTranscript   \n",
       "...     ...        ...  ..  ..  ...     ...                ...   \n",
       "2641      X  154419746   G   T  SNV       0   CodingTranscript   \n",
       "2642      X  154419746   G   T  SNV       0         Intergenic   \n",
       "2643      X  154420211   G   C  SNV       0         Intergenic   \n",
       "2644      X  154420211   G   C  SNV       0         Transcript   \n",
       "\n",
       "           Consequence  ConsScore          ConsDetail  ...  Freq10000bp  \\\n",
       "0           REGULATORY          4          regulatory  ...           21   \n",
       "1     CANONICAL_SPLICE          6        splice_donor  ...           21   \n",
       "2     CANONICAL_SPLICE          6        splice_donor  ...           19   \n",
       "3          STOP_GAINED          8         stop_gained  ...           20   \n",
       "...                ...        ...                 ...  ...          ...   \n",
       "2641       STOP_GAINED          8  splice,stop_gained  ...           11   \n",
       "2642        DOWNSTREAM          1          downstream  ...           11   \n",
       "2643        DOWNSTREAM          1          downstream  ...           12   \n",
       "2644  CANONICAL_SPLICE          6     splice_acceptor  ...           12   \n",
       "\n",
       "      Rare10000bp  Sngl10000bp EnsembleRegulatoryFeature  dbscSNV-ada_score  \\\n",
       "0              90         1434  Promoter Flanking Region            0.99998   \n",
       "1              90         1434  Promoter Flanking Region            0.99998   \n",
       "2              99         1332                       NaN            0.99987   \n",
       "3              95         1304  Promoter Flanking Region                NaN   \n",
       "...           ...          ...                       ...                ...   \n",
       "2641           29          822                       NaN            0.99999   \n",
       "2642           29          822                       NaN            0.99999   \n",
       "2643           32          828                       NaN            0.99999   \n",
       "2644           32          828                       NaN            0.99999   \n",
       "\n",
       "      dbscSNV-rf_score RemapOverlapTF RemapOverlapCL  RawScore PHRED  \n",
       "0                0.942           27.0           36.0  5.268674  35.0  \n",
       "1                0.942           27.0           36.0  5.268674  35.0  \n",
       "2                0.886           10.0           10.0  4.686023  31.0  \n",
       "3                  NaN           19.0           23.0  7.497634  38.0  \n",
       "...                ...            ...            ...       ...   ...  \n",
       "2641               1.0            9.0           10.0  7.476349  38.0  \n",
       "2642               1.0            9.0           10.0  7.476349  38.0  \n",
       "2643             0.944           38.0           46.0  4.640280  29.8  \n",
       "2644             0.944           38.0           46.0  4.640280  29.8  \n",
       "\n",
       "[2645 rows x 125 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_noncoding_annotated_with_cadd = pd.read_table('clinvar_noncoding_pathogenic_cadd_annotations_noheader.tsv')\n",
    "clinvar_pathogenic_noncoding_annotated_with_cadd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>11960768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11960768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_donor</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>11964787</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_donor</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>11972977</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2641</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>splice,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2642</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2643</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.853367625825</td>\n",
       "      <td>0.869</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2644</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_acceptor</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.853367625825</td>\n",
       "      <td>0.869</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2645 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0         1   11960768   G   A  SNV       0  RegulatoryFeature   \n",
       "1         1   11960768   G   A  SNV       0         Transcript   \n",
       "2         1   11964787   T   C  SNV       0         Transcript   \n",
       "3         1   11972977   C   T  SNV       0   CodingTranscript   \n",
       "...     ...        ...  ..  ..  ...     ...                ...   \n",
       "2641      X  154419746   G   T  SNV       0   CodingTranscript   \n",
       "2642      X  154419746   G   T  SNV       0         Intergenic   \n",
       "2643      X  154420211   G   C  SNV       0         Intergenic   \n",
       "2644      X  154420211   G   C  SNV       0         Transcript   \n",
       "\n",
       "           Consequence  ConsScore          ConsDetail  ...  BP4  BP5  BP6 BP7  \\\n",
       "0           REGULATORY          4          regulatory  ...    .    .    .   .   \n",
       "1     CANONICAL_SPLICE          6        splice_donor  ...    .    .    .   .   \n",
       "2     CANONICAL_SPLICE          6        splice_donor  ...    .    .    .   .   \n",
       "3          STOP_GAINED          8         stop_gained  ...    0    0    0   0   \n",
       "...                ...        ...                 ...  ...  ...  ...  ...  ..   \n",
       "2641       STOP_GAINED          8  splice,stop_gained  ...    0    0    0   0   \n",
       "2642        DOWNSTREAM          1          downstream  ...    0    0    0   0   \n",
       "2643        DOWNSTREAM          1          downstream  ...    .    .    .   .   \n",
       "2644  CANONICAL_SPLICE          6     splice_acceptor  ...    .    .    .   .   \n",
       "\n",
       "      cosmic70            MCAP  REVEL regsnp_fpr regsnp_disease  \\\n",
       "0            .               .      .          .              .   \n",
       "1            .               .      .          .              .   \n",
       "2            .               .      .          .              .   \n",
       "3            .               .      .          .              .   \n",
       "...        ...             ...    ...        ...            ...   \n",
       "2641         .               .      .          .              .   \n",
       "2642         .               .      .          .              .   \n",
       "2643         .  0.853367625825  0.869          .              .   \n",
       "2644         .  0.853367625825  0.869          .              .   \n",
       "\n",
       "     regsnp_splicing_site  \n",
       "0                       .  \n",
       "1                       .  \n",
       "2                       .  \n",
       "3                       .  \n",
       "...                   ...  \n",
       "2641                    .  \n",
       "2642                    .  \n",
       "2643                    .  \n",
       "2644                    .  \n",
       "\n",
       "[2645 rows x 270 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_noncoding_annotated_with_annovar['Chr'] = clinvar_pathogenic_noncoding_annotated_with_annovar['Chr'].map(lambda x: x.lstrip('chr').rstrip('aAbBcC'))\n",
    "clinvar_pathogenic_noncoding_annotated_with_annovar = clinvar_pathogenic_noncoding_annotated_with_annovar.rename(columns = {'Chr': '#Chrom', 'Start': 'Pos'})\n",
    "clinvar_pathogenic_noncoding_merged = pd.merge(clinvar_pathogenic_noncoding_annotated_with_cadd, clinvar_pathogenic_noncoding_annotated_with_annovar,\n",
    "                                         how = 'left', on = ['#Chrom', 'Pos', 'Ref', 'Alt'])\n",
    "clinvar_pathogenic_noncoding_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>11960768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11960768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_donor</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>11964787</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_donor</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>11972977</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2641</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>splice,stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2642</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2643</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.853367625825</td>\n",
       "      <td>0.869</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2644</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>CANONICAL_SPLICE</td>\n",
       "      <td>6</td>\n",
       "      <td>splice_acceptor</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.853367625825</td>\n",
       "      <td>0.869</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2645 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  Length           AnnoType  \\\n",
       "0         1   11960768   G   A  SNV       0  RegulatoryFeature   \n",
       "1         1   11960768   G   A  SNV       0         Transcript   \n",
       "2         1   11964787   T   C  SNV       0         Transcript   \n",
       "3         1   11972977   C   T  SNV       0   CodingTranscript   \n",
       "...     ...        ...  ..  ..  ...     ...                ...   \n",
       "2641      X  154419746   G   T  SNV       0   CodingTranscript   \n",
       "2642      X  154419746   G   T  SNV       0         Intergenic   \n",
       "2643      X  154420211   G   C  SNV       0         Intergenic   \n",
       "2644      X  154420211   G   C  SNV       0         Transcript   \n",
       "\n",
       "           Consequence  ConsScore          ConsDetail  ...  BP4  BP5  BP6 BP7  \\\n",
       "0           REGULATORY          4          regulatory  ...    .    .    .   .   \n",
       "1     CANONICAL_SPLICE          6        splice_donor  ...    .    .    .   .   \n",
       "2     CANONICAL_SPLICE          6        splice_donor  ...    .    .    .   .   \n",
       "3          STOP_GAINED          8         stop_gained  ...    0    0    0   0   \n",
       "...                ...        ...                 ...  ...  ...  ...  ...  ..   \n",
       "2641       STOP_GAINED          8  splice,stop_gained  ...    0    0    0   0   \n",
       "2642        DOWNSTREAM          1          downstream  ...    0    0    0   0   \n",
       "2643        DOWNSTREAM          1          downstream  ...    .    .    .   .   \n",
       "2644  CANONICAL_SPLICE          6     splice_acceptor  ...    .    .    .   .   \n",
       "\n",
       "      cosmic70            MCAP  REVEL regsnp_fpr regsnp_disease  \\\n",
       "0            .               .      .          .              .   \n",
       "1            .               .      .          .              .   \n",
       "2            .               .      .          .              .   \n",
       "3            .               .      .          .              .   \n",
       "...        ...             ...    ...        ...            ...   \n",
       "2641         .               .      .          .              .   \n",
       "2642         .               .      .          .              .   \n",
       "2643         .  0.853367625825  0.869          .              .   \n",
       "2644         .  0.853367625825  0.869          .              .   \n",
       "\n",
       "     regsnp_splicing_site  \n",
       "0                       .  \n",
       "1                       .  \n",
       "2                       .  \n",
       "3                       .  \n",
       "...                   ...  \n",
       "2641                    .  \n",
       "2642                    .  \n",
       "2643                    .  \n",
       "2644                    .  \n",
       "\n",
       "[2645 rows x 270 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_pathogenic_noncoding_merged = clinvar_pathogenic_noncoding_merged.drop_duplicates(keep = 'first')\n",
    "clinvar_pathogenic_noncoding_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The final dataframe for the pathogenic noncoding region variants is shown below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2641</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2642</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2643</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.853367625825</td>\n",
       "      <td>0.869</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2644</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.853367625825</td>\n",
       "      <td>0.869</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11258 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  ...            MCAP  REVEL regsnp_fpr  \\\n",
       "0         1    7961859   C   G  SNV  ...               .      .          .   \n",
       "1         1    7961859   C   G  SNV  ...               .      .          .   \n",
       "2         1    9720021   G   A  SNV  ...               .      .          .   \n",
       "3         1    9720021   G   A  SNV  ...               .      .          .   \n",
       "...     ...        ...  ..  ..  ...  ...             ...    ...        ...   \n",
       "2641      X  154419746   G   T  SNV  ...               .      .          .   \n",
       "2642      X  154419746   G   T  SNV  ...               .      .          .   \n",
       "2643      X  154420211   G   C  SNV  ...  0.853367625825  0.869          .   \n",
       "2644      X  154420211   G   C  SNV  ...  0.853367625825  0.869          .   \n",
       "\n",
       "      regsnp_disease regsnp_splicing_site  \n",
       "0                  .                    .  \n",
       "1                  .                    .  \n",
       "2                  .                    .  \n",
       "3                  .                    .  \n",
       "...              ...                  ...  \n",
       "2641               .                    .  \n",
       "2642               .                    .  \n",
       "2643               .                    .  \n",
       "2644               .                    .  \n",
       "\n",
       "[11258 rows x 270 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_noncoding_region_variants = pd.concat([hgmd_noncoding_merged, clinvar_pathogenic_noncoding_merged])\n",
    "pathogenic_noncoding_region_variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below we have dropped unneccessary columns:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>7961859</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>9720021</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2641</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2642</th>\n",
       "      <td>X</td>\n",
       "      <td>154419746</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2643</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.853367625825</td>\n",
       "      <td>0.869</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2644</th>\n",
       "      <td>X</td>\n",
       "      <td>154420211</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>...</td>\n",
       "      <td>0.853367625825</td>\n",
       "      <td>0.869</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11258 rows × 241 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #Chrom        Pos Ref Alt Type  ...            MCAP  REVEL regsnp_fpr  \\\n",
       "0         1    7961859   C   G  SNV  ...               .      .          .   \n",
       "1         1    7961859   C   G  SNV  ...               .      .          .   \n",
       "2         1    9720021   G   A  SNV  ...               .      .          .   \n",
       "3         1    9720021   G   A  SNV  ...               .      .          .   \n",
       "...     ...        ...  ..  ..  ...  ...             ...    ...        ...   \n",
       "2641      X  154419746   G   T  SNV  ...               .      .          .   \n",
       "2642      X  154419746   G   T  SNV  ...               .      .          .   \n",
       "2643      X  154420211   G   C  SNV  ...  0.853367625825  0.869          .   \n",
       "2644      X  154420211   G   C  SNV  ...  0.853367625825  0.869          .   \n",
       "\n",
       "      regsnp_disease regsnp_splicing_site  \n",
       "0                  .                    .  \n",
       "1                  .                    .  \n",
       "2                  .                    .  \n",
       "3                  .                    .  \n",
       "...              ...                  ...  \n",
       "2641               .                    .  \n",
       "2642               .                    .  \n",
       "2643               .                    .  \n",
       "2644               .                    .  \n",
       "\n",
       "[11258 rows x 241 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_noncoding_region_variants = pathogenic_noncoding_region_variants.drop(columns = ['PolyPhenVal', 'priPhCons', 'cHmm_E1', 'cHmm_E2', 'cHmm_E3', 'cHmm_E4', 'cHmm_E5', 'cHmm_E6', 'cHmm_E7', 'cHmm_E8', 'cHmm_E9', 'cHmm_E10', 'cHmm_E11', 'cHmm_E12', 'cHmm_E13', 'cHmm_E14', 'cHmm_E15', 'cHmm_E16', 'cHmm_E17', 'cHmm_E18', 'cHmm_E19', 'cHmm_E20', 'cHmm_E21', 'cHmm_E22', 'cHmm_E23', 'cHmm_E24', 'cHmm_E25', 'GerpN', 'GerpS'])\n",
    "pathogenic_noncoding_region_variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have changed the dtype like before and replaced the empty values with NaN values like before: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic_noncoding_region_variants_with_nan = pathogenic_noncoding_region_variants.replace('.', np.nan)\n",
    "pathogenic_noncoding_region_variants_with_nan = pathogenic_noncoding_region_variants_with_nan.astype({'SIFT_score': float, 'Polyphen2_HDIV_score' : float, \n",
    "                'Polyphen2_HVAR_score' : float, 'LRT_score' : float, \n",
    "                'MutationTaster_score' : float, 'MutationAssessor_score' : float, \n",
    "                'FATHMM_score' : float, 'RadialSVM_score' : float,\n",
    "               'LR_score' : float, 'VEST3_score' : float, 'CADD_raw' : float,\n",
    "               'CADD_phred' : float, 'GERP++_RS' : float, 'phyloP46way_placental' : float,\n",
    "               'phyloP100way_vertebrate' : float, 'SiPhy_29way_logOdds' : float, 'SIFT_score.1' : float,\n",
    "               'SIFT_converted_rankscore' : float, 'Polyphen2_HDIV_score.1' : float, 'Polyphen2_HDIV_rankscore' : float,\n",
    "               'Polyphen2_HVAR_score.1' : float, 'Polyphen2_HVAR_rankscore' : float, 'LRT_score.1' : float,\n",
    "               'LRT_converted_rankscore' : float, 'MutationTaster_score.1' : float, 'MutationTaster_converted_rankscore' : float,\n",
    "               'MutationAssessor_score.1' : float, 'MutationAssessor_score_rankscore' : float, 'FATHMM_score.1' : float,\n",
    "               'FATHMM_converted_rankscore' : float, 'PROVEAN_score' : float, 'PROVEAN_converted_rankscore' : float,\n",
    "               'VEST3_score.1' : float, 'VEST3_rankscore' : float, 'MetaSVM_score' : float, 'MetaSVM_rankscore' : float,\n",
    "               'MetaLR_score' : float, 'MetaLR_rankscore' : float, 'M-CAP_score' : float, 'M-CAP_rankscore' : float,\n",
    "               'CADD_raw.1' : float, 'CADD_raw_rankscore' : float, 'CADD_phred.1' : float, 'DANN_score' : float,\n",
    "               'DANN_rankscore' : float, 'fathmm-MKL_coding_score' : float, 'fathmm-MKL_coding_rankscore' : float,\n",
    "               'Eigen-raw' : float, 'Eigen-PC-raw' : float, 'GenoCanyon_score' : float, 'GenoCanyon_score_rankscore' : float,\n",
    "               'integrated_fitCons_score' : float, 'integrated_fitCons_score_rankscore' : float, 'integrated_confidence_value' : float,\n",
    "               'GERP++_RS.1' : float, 'GERP++_RS_rankscore' : float, 'phyloP100way_vertebrate.1' : float, \n",
    "                'phyloP100way_vertebrate_rankscore' : float, 'phyloP20way_mammalian' : float, 'phyloP20way_mammalian_rankscore' : float,\n",
    "               'phastCons100way_vertebrate' : float, 'phastCons100way_vertebrate_rankscore' : float, 'phastCons20way_mammalian' : float,\n",
    "               'phastCons20way_mammalian_rankscore' : float, 'SiPhy_29way_logOdds.1' : float, 'SiPhy_29way_logOdds_rankscore' : float,\n",
    "               'dbscSNV_ADA_SCORE' : float, 'dbscSNV_RF_SCORE' : float, 'PVS1' : float, 'PS1' : float, 'PS2' : float, 'PS3' : float,\n",
    "               'PS4' : float, 'PM1' : float, 'PM2' : float, 'PM3' : float, 'PM4' : float, 'PM5' : float, 'PM6' : float, 'PP1' : float, 'PP2' : float,\n",
    "               'PP3' : float, 'PP4' : float, 'PP5' : float, 'BA1' : float, 'BS1' : float, 'BS2' : float, 'BS3' : float, 'BS4' : float, 'BP1' : float, 'BP2' : float,\n",
    "               'BP3' : float, 'BP4' : float, 'BP5' : float, 'BP6' : float, 'BP7' : float, 'MCAP' : float, 'REVEL' : float, 'regsnp_fpr' : float,\n",
    "                'dbscSNV-rf_score' : float})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I have saved the above dataframe for use in other Notebooks:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic_noncoding_region_variants_with_nan.to_csv('pathogenic_noncoding_region_variants.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
